# Genomic analysis of a parasite invasion: colonization of the New World by the blood fluke, *Schistosoma mansoni* 

Roy Nelson Platt II*<sup>1</sup>, Frédéric D. Chevalier*<sup>1</sup>, Winka Le Clec'h<sup>1>/sup>, Marina McDew-White<sup>1</sup>, Philip T. LoVerde<sup>2</sup>, Rafael Ramiro de Assis<sup>3</sup>, Guilherme 
Oliveira<sup>3</sup>, Safari Kinunghi<sup>4</sup>, Anouk Gouvras<sup>5</sup>, Bonnie Webster<sup>5</sup>, Joanne Webster<sup>6</sup>, Aidan Emery<sup>5</sup>, David Rollinson<sup>5</sup>, Timothy J. Anderson<sup>1</sup>

<sup>1</sup>Texas Biomedical Research Institute, San Antonio, San Antonio, TX, United States
<sup>2</sup>University of Texas Health Science Center, San Antonio, San Antonio, TX, United States
<suo>3</sup>Centro de Pesquisas René Rachou - Fiocruz/MG, Belo Horizonte, Brazil
<sup>4</sup>National Institute for Medical Research, Mwanza, Tanzania.
<sup>5</sup>Natural History Museum, London, United Kingdom
<sup>6</sup>Imperial College, London, United Kingdom

*Schistosoma mansoni*, a snail-vectored, blood fluke that infects human, was introduced into the New World from Africa during the Trans-Atlantic slave trade. As this parasite shows strong specificity to the snail intermediate host, we expected that adaptation to S. American *Biomphalaria* spp. snails would result in population bottlenecks and strong signatures of selection. We scored 475,081 single nucleotide variants (SNVs) in 143 *S. mansoni* from the New World (Brazil, Guadeloupe, and Puerto Rico) and Old World (Cameroon, Niger, Senegal, Tanzania, and Uganda), and used these data to ask: (i) Was there a population bottleneck during colonization? (ii) Can we identify signatures of selection associated with colonization? And (iii) what were the source populations for colonizing parasites?  We found a 2.4-2.9-fold reduction in diversity and much slower decay in linkage disequilibrium (LD) in parasites from East to West Africa. However, we observed similar nuclear diversity and LD in West Africa and Brazil, suggesting no strong bottlenecks and limited barriers to colonization. We identified five genome regions showing selection in the new World, compared with 3 in West Africa and none in E. Africa, which we speculate may reflect adaptation during colonization. Finally, we infer that unsampled African populations between Benin and Angola are likely the major source for Brazilian S. mansoni, with contributions from Niger. The absence of a bottleneck suggests that this is a rare case of a serendipitous invasion, where parasites were preadapted to the New World and were able to establish with relative ease. 
 

---

### Reproducibility
The analyses in the manuscript can be reproduced within a Jupyter Notebook. To begin, complete the following steps in order to replicate the software environment and directory structure. All analyses were run using the 
`sch_man_nwinvasion-jupyter` conda environment.  Then within the notebook use `ipykernel` to activate the notebook specific environment.  

