{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-11-23T19:41:58.789485Z",
     "start_time": "2020-11-23T19:41:58.784359Z"
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import shutil\n",
    "import vcf\n",
    "import re\n",
    "import gzip\n",
    "import pandas as pd\n",
    "import pysam\n",
    "\n",
    "from IPython.display import Image\n",
    "from Bio import SeqIO\n",
    "\n",
    "\n",
    "## import rpy2.ipython\n",
    "\n",
    "\n",
    "## %load_ext rpy2.ipython\n",
    "\n",
    "os.chdir(\"/master/nplatt/sch_man_nwinvasion\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Calculate Descriptive Stats\n",
    "\n",
    "Calculate basic descriptive stats including read count, sample coverage, probe depth etc."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#make directory\n",
    "if [ -d results/probe_coverage_and_read_counts/ ]; then\n",
    "    rm -r  results/probe_coverage_and_read_counts\n",
    "fi\n",
    "\n",
    "mkdir -p results/probe_coverage_and_read_counts/mosdepth\n",
    "mkdir -p results/probe_coverage_and_read_counts/flagstat\n",
    "\n",
    "\n",
    "echo -e \"SAMPLE,NUM_RAW_READS,NUM_FILTERED_READS,NUM_MAPPED_READS,AVG_COV_PER_PROBE,NUM_PROBES_LT_12X_COV,NUM_PROBES_W_0X_COV,MISSING_ALL_SITES,MISSING_FILT_SITES\" \\\n",
    "    >results/probe_coverage_and_read_counts/stats.csv\n",
    "\n",
    "for R1_FQ in $(ls data/exomes/*1.fastq.gz data/sra/*1.fastq.gz); do\n",
    "    SAMPLE=$(basename $R1_FQ _R1.fastq.gz)\n",
    "    \n",
    "    #NUM_RAW_READS  \n",
    "    R2_FQ=data/exomes/\"$SAMPLE\"_R2.fastq.gz \n",
    " \n",
    "    NUM_RAW_LINES=$(zcat $R1_FQ $R2_FQ| wc -l)\n",
    "    NUM_RAW_READS=$(( $NUM_RAW_LINES / 4 ))\n",
    "\n",
    "    #NUM_FILTERED_READS\n",
    "    FILT_R1=results/filtered_reads/\"$SAMPLE\"_filtered_R1.fastq.gz \n",
    "    FILT_R2=results/filtered_reads/\"$SAMPLE\"_filtered_R2.fastq.gz \n",
    "\n",
    "    NUM_FILT_LINES=$(zcat $FILT_R1 $FILT_R2| wc -l)\n",
    "    NUM_FILT_READS=$(( $NUM_FILT_LINES / 4 ))\n",
    "   \n",
    "    #NUM_MAPPED_READS\n",
    "    BAM_FILE=results/mapped_reads/\"$SAMPLE\"_processed.bam\n",
    "    FLAG_FILE=results/probe_coverage_and_read_counts/flagstat/\"$SAMPLE\"_processed.bam.flagstat\n",
    "    \n",
    "    samtools flagstat $BAM_FILE >$FLAG_FILE\n",
    "    NUM_MAPPED_READS=$(grep mapped $FLAG_FILE | cut -f1 -d\" \" | head -n 1)\n",
    "\n",
    "    #COVERAGE\n",
    "    mosdepth \\\n",
    "        -t 10 \\\n",
    "        --by data/renamed-sma_agilent_baits.v7.0.chr_reorderd.bed  \\\n",
    "        results/probe_coverage_and_read_counts/mosdepth/$SAMPLE \\\n",
    "        --no-per-base \\\n",
    "        $BAM_FILE\n",
    "\n",
    "    MOSDEPTH_BED=results/probe_coverage_and_read_counts/mosdepth/$SAMPLE.regions.bed.gz\n",
    "\n",
    "    AVG_COV=$(zcat $MOSDEPTH_BED  | awk '{sum += $4} END {print sum/NR}')\n",
    "    LT12_COV=$(zcat $MOSDEPTH_BED | awk '{if ($4<12) print $0}' | wc -l)\n",
    "    NO_COV=$(zcat $MOSDEPTH_BED   | awk '{if ($4==0) print $0}' | wc -l)\n",
    "\n",
    "    #gt rate at all gtd sites\n",
    "    MISSING_ALL=$(grep $SAMPLE results/variant_filtration/indiv_gt_rate.tbl | cut -f5)\n",
    "\n",
    "\n",
    "    #gt rate at filtered sites\n",
    "    MISSING_FILT=$(grep $SAMPLE results/variant_filtration/gt_rate_per_indiv_at_filtered_sites.tbl | cut -f5)\n",
    "    \n",
    "    #print to file\n",
    "    echo \"$SAMPLE,$NUM_RAW_READS,$NUM_FILT_READS,$NUM_MAPPED_READS,$AVG_COV,$LT12_COV,$NO_COV,$MISSING_ALL,$MISSING_FILT\" >>results/probe_coverage_and_read_counts/stats.csv\n",
    "    echo -e \"$SAMPLE\\t$NUM_RAW_READS\\t$NUM_FILT_READS\\t$NUM_MAPPED_READS\\t$AVG_COV\\t$LT12_COV\\t$NO_COV\\t$MISSING_ALL\\t$MISSING_FILT\"\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAMAAABKCk6nAAAC3FBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxeXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5wcHBycnJzc3N0dHR2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+RkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+xsbGysrKzs7O0tLS1tbW2tra3t7e4uLi6urq7u7u8vLy+vr6/v7/BwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7///8yhuQkAAASgklEQVR4nO2diX+UxRnHNwkQIDREIAYFCTfIpYJgvRASqlaQoGilUI+21optPSgtaosVEAUVKILQ4lUCiIKoURRRjsohWES8uENABUGuHPMPdI8Am919Z2bfc56H3/fD593N+77OzO7Xefd5Z96ZCQnAmlDQBQDeAsHMgWDmQDBzIJg5EMwcCGYOBDMHgpkDwcyBYOZAMHMgmDkQzBwIZg4EMweCmQPBzIFg5kAwcyCYORDMHAhmDgQzB4KZA8HMgWDmQDBzIJg5EMwcCGYOBDMHgpkDwcyBYOZAMHMgmDkQzBwIZg4EMweCmQPBzIFg5kAwcyCYORDMHAhmDgQzB4KZA8HMgWDmQDBzIJg5EMwcCGYOBDMHgpnjQPD+UmAAC096JXj+iJkgeK74wjPB0+z/t8A17oJg3kAwc1wQfHzr+m2Vybsh2AgcC95zc3ZBu/zGt3+XeACCjcCx4OIxB8PbffcMTTwAwUbgWPA5VdGXE80TD0CwETgW3PX96EtZj8QDEGwEjgUva15034P3DswvSzwAwUbgPIo+OH/82McW/ZC0H4KNwK374PIViXsg2AjcEvxSwem3qydEGfp7+6ViyCcT6nDvI3X+nPSjV/l60JK1oyzKTSX2SsSU8rI6/OLpOn++k6KlyB1cE1yeuOOPt6ZfmrOHMR/5lJFrgrMTd0CwjDGrfMrIseD9tfgoeENxHS7pW/fvb7zK1012HPcpI8eCM7JiJJ3pWw1+ttSnjEjiWPADj8Zeg7tEkxS8OqlzxiMcC64cuj76Gpzg2a/7lJGb0AuyDiTuQJAlg57gJCBYBp0o2hLfBJet9SkjN6ETRVuCIMsIIDgY6ETRliCKloEgizkQTJ6X2/aW0K677Gjz5Ocn7MJAsKFR9JyH3rXNle79QjMQbGiQBcFuAcEyGAg2NIqGYOZAMHMg2C0QRctgIBhBlgwI9goIdgtE0TIYCDYUCGYOBLsFomgZDAQjyJIBwV4x59YnbHMxBMdhaBT9fMg+zSHYfCCYOfgNdgtE0TIYCDY1yIJgbdZIn1/r0EX6dNtXrhUjPSBYn7I77X9Xwz5xrRjpAcH6QLB9INgrIFgfCLYPBHsFBOsDwfaBYK+AYH0g2D4Q7BUQrA8E2weCvQKC9YFg+0CwV0CwPhBsHwj2CgjWp6x1Z9tcAMEyDBGcY//5tTwIlmGI4P5/sk0/CJZhiGD8BtsGgr2itKX9wKH5YdeK4Y7g6p3J6/6c7YINwbHgzwcUjPyiU4MW7ycegGAjcCx4wJRNf2u9QLzVN/EABMuYsMmnjBwL7ihEzTmx17pAsAw6s812+0Z8XO8bUdE58QAEy6AjeH5e95YvtBtRODHxAATLmL3Np4ycR9G73jskNk55O2k/BBuBBwtE//hllDtusV+qBCDYPh4sEL1keJROxfZLlQBDwXSiaEtwiZZBJ8gS4vjW9dtSLGANwTLoCN5zc3ZBu/zGtyc1j0OwDDpRdPGYg+HtvnuGJh6AYCNwLPicqujLieaJByDYCBwL7hrrZSjrkXgAgmXQiaKXNS+678F7B+aXJR6AYBl0gixxcP74sY8tSl6qC4JlEBJsBQTLoBNFWwLBRgDBzIHgYKATRVsCwTIQZMUDwfaB4GBAFB0PQ8G+AcHMgeBgQBQdT9l1siVKHv277OiVZgpGkBVP+QQZQ38pOzoxuZXcBCBYn227fMrITRBFA3dgILjcvdHSDGEgeJqZq67IQRStj6HL6shBkKUPBMtgIBhRtAwGgoEMBoIRRctgIBhRtAwGghFkyYDgYIBgfRBFy2AgGMhgIBhRtAwGghFFy2AgGEGWDAgOBgjWB1G0DAaCgQwGghFFy2AgGFG0DAaCEWTJgOBggGB9EEXLYCAYyGAgGFG0DAaCEUXLYCAYQZYMCA6EjFBGPX9yYiCYRhRdVhxHRigz/C9ux7WeDXJlIJgejcN+M+o7+G7TQFfwyT3ppowo2pLMUCgjI2SS4H0l9fPE6JVppYwo2pKcUFZGRj2TBA+YfLhQrOmTVsoIsiwxrwa3F6IwxfqTpzlwMHkfBFvSKPIbnGWS4J4bw4K3dE91xrpR4tPuDepf+nniAUTRlphXg5c0G5QzNH9pqjN6zhUDnqqqmtI/8QCiaEuyQ2G9mSYJ3rFr5sR55atTnZFTKQpratcQrgOiaEsiNThkVA3uFtkczkt1xjWTa+56V4hZfRIPIIq2xLQaPLcgIztM5qBUZ2y/6oIBjS4q7JzUeI4gy5JwDQ4ZVYNrSvaHSREpR/lq8dxXP65O2g3BljQwrAaf4kHJiXHrB9eCKNqSLNNqsNhx96Cion7nSU6MWz94Yaz9vM1AF4rHk/qRGpxhkuCrf/NSp38Xp4yiLUEUbYl5NbidEH3FtxZLenu/frAcelF0vYheo2pwp+3ikh9El1Rn+LF+sBx6QZZ5NXhW9tFxvQZfluoMP9YPlkNPsHk1WFSI6gXPVKQ6w4/1g+UgipahJ7jI+gw/1g/mRpZxNfi20uSGjFr8WD9YDtEo2ijBF+c0Pr9Vq1YpT/Fh/WA59KJo8y7R6zZFSStlBFmWGBhk2QCCLTGvBtsBUbQlqMHMQQ1OC3pRtHmdDYUdInTs84cD+ikjirbEuA5/MeP6RasW/3za8juTGiStQZBliXkd/r1OhDfH+4maNvopQ7Al5tXgNpGRSRWtxOYO+ikjirbEtIfuhBjftOTXtzS/v7rJLP2UEUVbYt5js2LT9MeeWR3pVNIHUbQlDcN+zRq6YgdE0ZaYN3TljcvatLLqbLACQZYljY0bfNZ1wQZ0NriHeTU45ZgGBYiiLflppAYbNQD8L++knzKiaEsWGFeDe2U2Nfc3mF4UvadZuAY3zPclLz3B603u8KcXRYurev7nibznfcmKwW1SdkOfMnKRv19y9bv+5KQjuMP+DjHSStk7wQdK44h2njeI2/HqYa/yJYmO4LWVa2OklbJ3gndOnHCGUGaYUNyOSWl0aZ4F6F6iK7buTzNlfy7RjWMdMw/7kRdJ9AS/1zkzJ3Thh2ml7I/gpqF6vjUKkURP8EXLKsWxeYVppeyP4IaxW0rUYCs0Z7qLbjullbI/ggt97DsniZ7g0W8KUTP//rRS9kdwNmqwHM1plEJ5hU3qdUzrTskfwf19fECRJHqC16w/hX7KvtXgrEzUYGs026If/zLtlP0R/DBqsBzNuSrvzL908vb0UvatBmeiBkvQbeioeu++lpdPO5RGyv4IvtDHYT4k0RV8cM71uUPuLlymnzKiaCPQE7xgWKNrnvtOiE+Ne/A9H/fBcvQEFzy5O7xdLcQY/ZT9EdwINViO4+mELfFHcC5qsBzH0wlb4o/gJuhNkuPCdMIW+HaJhmAZulH0oW3pLr7mj+BmuETL0RO8pW9mk8z+X6eVMm6TjEBzOuHpx8TRJ038DUZ3oQLNoStxW11Qg41AT3D3beHN50nTUUpBd6ER6AkubTrkjiF5r6WVsj+CG4RCWVmowdZoRtHbZz8+J80xXqjBRkB9ZAN+gxVQHwCOKFoB9QHgqMEKXBoAviR5F7oLjcDxAPAtUVps2ZJ4AG3RRuB4AHiooFuYet26JR5Ab5IROB4Avvyih48JUZB8wNf+YAi2Qvc26eQeq1OOP9K9LDjBjWKTxkGwFXqC95XUzxOjV1qctHXgyBbJexFkGYHm4LPJhwvFmj6Wp/2rJHkfbpOMQE9weyEKhegoOTGo9YPxXLQCPcE9N4YFb+kuOTFu/eCNM6MUDXaheEowskGB5tCVZoNyhuYv1Utya2wylBtudFg0LTA2SYFmFL1r5sR55RbnBLp+MEYXKtASfFCI56ZMeTvlGQGvH4zuQgU6gre2/EoUjPpt7n9TnRHw+sGIohXoCB72ZyEK9oqnh6U6I+D1g9FdqEBHcIvyqOADKVozAl8/GLPsKNAR3OCIEIdqRGXKOSEDXj8YD74r0BHcZXn0/Zupn6oMdv1gzHSnQEfw5M4bw9sV5/8zrZT9EZwDwXJ0BNf8tXHnK9vlTkovZb+mMsQlWopeQ0fFOy8u/z7NlDEA3AioPzZ7HmqwHOqC0dChgLpgdBcqoC4Y3YUKqAtGd6EC6oJRgxVQF9wfv8FyqAtGFK2AumDcByugLhjdhQqoC0ZbtALqgtGbpIC6YPQHK6AuGE90KKAuGEGWAuqC8VSlAuqC0dChgLpgjGxQQF0wxiYpoC4Y3YUKqAtGd6EC6oLxyI4C6oIRRSugLhiz7CigLhjzZCmgLhgrnymgLhhzVSqgLhizzSqgLhjdhQqoC8ZtkgLqgtFdqIC6YNRgBdQFo7tQAXXBWPlMAXXBqMEKqAvGb7AC6oIRRSugLhg1WAF1weguVEBdMNqiFVAXjN4kBdQFoz9YAXXBeKJDAXXBCLIUUBeM2yQFrgk+kLgDz0UbgWPBWwa0vm13+DU78QBGNhiBY8FXjFs3scO2wARjbJICx4J/Ui3Em132BliDMbpQhmPBXT4Ib17q8XVAgtFdqMCx4FdzF0S2rTMSDyCKNgLnUfTOXZHtt3MS96O70Ajcuk2KWyC68rso92CWHRNwS3DcAtGLiqO0KbJfKn3w4LsC6i1ZmOlOgQuCA10gGnNVKnAsOOAFojHbrALHggNeIBornylwLDjgBaIx47sCx4IDXiAaDR0KHAsOeIFodBcqcB5FB7tANLoLFVC/D0Z3oQLqglGDFVAXjN9gBdQFI4pWQF0w7oMVUBeMliwF1AWjLVoBdcHoTVJAXTD6gxVQF4wnOhRQF4xnshRQF4ynKhVQF4yGDgXUBWNkgwLqgjE2SQF1weguVEBdMLoLFVAXjO5CBdQFI4pWQF0wZtlRQF0w5slSQF1wLjob5FAXjLkqFVAXjNlmFVAXjCBLAXXBuE1SQF0wugsVUBeMGqyAumB0FyqgLhgrnymgLhg1WAF1wfgNVkBdMKJoBdQFowYroC4YD74roC4YbdEKqAtGb5IC6oKx8pkC6oIxAFwBdcHoLlRAXTBukxRQF4znohVQF1w7suE1P/IiCXXBGJukgLrg2tGFq/3IiyTUBc9DDZZDXXBtFL1CfeZZCvVVVx7CfbAc6quu5MZq8Do/8iIJ9VVXhqEGy6G+6krPmOBtfuRFEuqrrnSJCd7sR14kob7qykBcouVQX3WldirDlX7kRRIP1g+uxR/BJajBcjxYP/i14VE6/8x+qfTJDmWsHJcRmuVHXiTxoCXrZGwF8NlT7ZUoPcaiu1COdy1Z86fZK1GahELZ9UMZvmRFEu9asnwSPCvS2eBLTjTxriXLJ8FAjnctWRBsBN61ZEGwEXjXkgXBRuBdSxYEG4F3T3RAsBFAMHMgmDkQzBzvBL/Vq9gfOuY2o0fOpT59O112eyXYN54tDboENhjzUdAliAHBHgHB+kCwAyDYIyBYHwh2AAR7BATrM2NR0CWwwVhDBrxSEHzsZNAlsMGRmqBLEIOCYOAACGYOBDMHgpkDwcyBYOZAMHMgmDnmCb68Z3izvkPkX4TN153bom+ZWBv+a26wBYujMpQVptvaWCmtymXEJzBO8ObrB38YL7jztOqa0pxvK/eLmgLFf+oflaGd0Zf9kVJalsuIT2Cc4AfmvPy7OMEnQ3vD28+qwv//l2R027GkR/uiCnFiZPu2I44GWMhawbEafKZc6y4eWSxib8U/Wvcab8InME1wVftDP7Y5HleDB/d+cY+Ifpf7s8WuZhvF5BKxYFBN9f0fBFjKOoLPlGtTziun3v4vb0/1CBM+gWmC37hViFEL4wQfm3p1dq8FtV/PzEFCHK5/ckWrpccCLWVlKL+goGDkacG15drUqPrU2xk3CvGWCZ/ANMHDc5o2zRkSJzjM0RdzVsW+nok5hYWFeXtEaf/cO44EWMrK0Lq9e/d+f1pwbbk2nS9OvX3sV0KsMeETGCb4+/wT4a+voOK04O2vR7bXzoh9PS+cHqR8oGhSUGUUyZfo2nJtaiVOvZ1eIsQSEz6BYYKnR2fuGTX1tODPchdW1SxvtiH89RzMOlJ+7laxZrSY+khNze1PBFjMOoLPlCsiuPbthrzdlTeZ8AkME9zvlch2ce8zl+iyK5q16Ls4ehc5KG/V0h4de68QFTdc0Hb44QCLWUfwmXJFBIvYWzGuZden2hrwCQwTDNwGgpkDwcyBYOZAMHMgmDkQzBwIZg4EMweCmQPBzIFg5kAwcyCYORDMHAhmDgQzB4KZA8HMgWDmQDBzIJg5EMwcCGYOBDMHgpkDwcyBYOZAMHMgmDkQzBwIZg4EMweCmQPBzIFg5kAwcyCYORDMHAhmDgQzB4KZA8HMgWDmQDBzIJg5EMwcCGbO/wFYuZOPJGQPewAAAABJRU5ErkJggg==\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "#gt_rate_all_sites_tbl<-read.table(\"results/variant_filtration/gt_rate_per_site.tbl\", header=TRUE)\n",
    "#gt_rate_flt_sites_tbl<-read.table(\"results/variant_filtration/gt_rate_per_site_at_filtered_sites.tbl\", header=TRUE)\n",
    "\n",
    "#head(gt_rate_flt_sites_tbl)\n",
    "\n",
    "boxplot(1-gt_rate_all_sites_tbl$F_MISS, \n",
    "        1-gt_rate_flt_sites_tbl$F_MISS,\n",
    "        notch=FALSE,\n",
    "        ylab=\"Genotyping rate\",\n",
    "        col=c(\"grey\", \"grey\"),\n",
    "        names=c(\"All Sites\", \"Filtered Sites\"),\n",
    "        boxwex=0.33\n",
    "       )\n",
    "\n",
    "\n",
    "\n",
    "#hist(1-gt_rate_flt_sites_tbl$F_MISS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-11-23T19:42:12.422832Z",
     "start_time": "2020-11-23T19:42:12.400695Z"
    }
   },
   "outputs": [],
   "source": [
    "#read vcf to get all samples and store based on genome or exome\n",
    "vcf_reader = vcf.Reader(open('results/variant_filtration/smv7_ex_snps.vcf', 'r'))\n",
    "\n",
    "samples={}\n",
    "samples['genome'] = [x for x in vcf_reader.samples if re.search(\"^ER\", x)]     \n",
    "samples['exome'] = [x for x in vcf_reader.samples if not re.search(\"^ER\", x)]     \n",
    "\n",
    "#create empty df to store info\n",
    "df = pd.DataFrame(index=[samples['exome'] + samples['genome']])\n",
    "\n",
    "\n",
    "#fxn to count num seqs in fastq.gz file\n",
    "def num_seqs(in_fastq_gz):\n",
    "    n=0\n",
    "    with gzip.open(in_fastq_gz, \"rt\") as handle:\n",
    "        for record in SeqIO.parse(handle, \"fastq\"):\n",
    "            n=n+1\n",
    "    return n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-11-23T19:49:08.123446Z",
     "start_time": "2020-11-23T19:49:08.016723Z"
    }
   },
   "outputs": [],
   "source": [
    "import pysam\n",
    "samfile = pysam.AlignmentFile(\"results/mapped_reads/Sm.BR_PdV.0447.1_processed.bam\", \"rb\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-11-23T19:49:27.414972Z",
     "start_time": "2020-11-23T19:49:27.409660Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "18207430"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "samfile.mapped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2020-11-23T19:50:55.409Z"
    }
   },
   "outputs": [],
   "source": [
    "#get all read count info\n",
    "\n",
    "num_raws  = []\n",
    "num_filts = []\n",
    "num_maps  = []\n",
    "seq_types = []\n",
    "#count num of raw and filtered reads\n",
    "for sample in samples['exome'] + samples['genome']:\n",
    "    seq_type = \"\"\n",
    "    \n",
    "    if sample in samples['exome']:\n",
    "        raw_file = \"data/exomes/{}_R1.fastq.gz\".format(sample)\n",
    "        seq_type = \"exome\"\n",
    "    if sample in samples['genome']:\n",
    "        raw_file = \"data/sra/{}_1.fastq.gz\".format(sample)\n",
    "        seq_type = \"genome\"\n",
    "        \n",
    "    filtered_file = \"results/filtered_reads/{}_filtered_R1.fastq.gz\".format(sample)\n",
    "    bam_file = pysam.AlignmentFile(\"results/mapped_reads/{}_processed.bam\".format(sample), \"rb\")\n",
    "    \n",
    "    #raw reads\n",
    "    num_raw = num_seqs(raw_file)\n",
    "    \n",
    "    #filtered reads\n",
    "    num_filt = num_seqs(filtered_file)\n",
    "\n",
    "\n",
    "    #count num mapped reads\n",
    "    num_map = samfile.mapped\n",
    "\n",
    "    seq_types.append(seq_type)\n",
    "    num_raws.append(num_raw)\n",
    "    num_filts.append(num_filt)\n",
    "    num_maps.append(num_map)\n",
    "    \n",
    "    print(\"{}\\t{}\\t{}\\t{}\\t{}\".format(sample, seq_type, num_raw, num_filt, num_map))\n",
    "    \n",
    "df.insert(0,'seq_type', seq_types)\n",
    "df.insert(1,'num_raw', num_raws)\n",
    "df.insert(2,'num_filt', num_filts)\n",
    "df.insert(3,'mapped_reads', num_maps)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2020-11-23T19:50:59.151Z"
    }
   },
   "outputs": [],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#now get gt info\n",
    "\n",
    "for sample in samples['exome'] + samples['genome']:\n",
    "    \n",
    "    #count num gtd\n",
    "    raw_vcf = 'results/genotype/cohort_raw.vcf'\n",
    "    \n",
    "    \n",
    "    #count num in final vcf file\n",
    "\n",
    "    #count depth of final vcf per site"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
