{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-11-23T20:24:41.786828Z",
     "start_time": "2020-11-23T20:24:41.274144Z"
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import shutil\n",
    "import vcf\n",
    "import re\n",
    "import gzip\n",
    "import pandas as pd\n",
    "import pysam\n",
    "import pybedtools\n",
    "import subprocess\n",
    "import numpy as np\n",
    "\n",
    "from IPython.display import Image\n",
    "from Bio import SeqIO\n",
    "\n",
    "\n",
    "## import rpy2.ipython\n",
    "\n",
    "\n",
    "## %load_ext rpy2.ipython\n",
    "\n",
    "os.chdir(\"/master/nplatt/sch_man_nwinvasion\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Calculate Descriptive Stats\n",
    "\n",
    "Calculate basic descriptive stats including read count, sample coverage, probe depth etc."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-11-23T20:36:21.444617Z",
     "start_time": "2020-11-23T20:36:20.839149Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/master/nplatt/miniconda3/envs/sch_man_nwinvasion-jupyter/lib/python3.8/subprocess.py:849: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<BedTool(results/probe_coverage_and_read_counts/merged_probes.bed)>"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#get merged probes\n",
    "probe_bed = pybedtools.BedTool(\"data/renamed-sma_agilent_baits.v7.0.chr_reorderd.bed\")\n",
    "merged_probe_bed = probe_bed.merge(d=0)\n",
    "\n",
    "merged_probe_bed.saveas('results/probe_coverage_and_read_counts/merged_probes.bed')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-11-23T21:10:50.396819Z",
     "start_time": "2020-11-23T21:10:50.365187Z"
    }
   },
   "outputs": [],
   "source": [
    "#read vcf to get all samples and store based on genome or exome\n",
    "vcf_reader = vcf.Reader(open('results/variant_filtration/smv7_ex_snps.vcf', 'r'))\n",
    "\n",
    "samples={}\n",
    "samples['genome'] = [x for x in vcf_reader.samples if re.search(\"^ER\", x)]     \n",
    "samples['exome'] = [x for x in vcf_reader.samples if not re.search(\"^ER\", x)]     \n",
    "\n",
    "#fxn to count num seqs in fastq.gz file\n",
    "def num_seqs(in_fastq_gz):\n",
    "    n=0\n",
    "    with gzip.open(in_fastq_gz, \"rt\") as handle:\n",
    "        for record in SeqIO.parse(handle, \"fastq\"):\n",
    "            n=n+1\n",
    "    return n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-11-25T05:34:35.658727Z",
     "start_time": "2020-11-24T16:21:22.072845Z"
    }
   },
   "outputs": [],
   "source": [
    "#create empty df to store info\n",
    "df = pd.DataFrame(index=[samples['exome'] + samples['genome']])\n",
    "\n",
    "#get all read count info\n",
    "merged_probes_bed = 'results/probe_coverage_and_read_counts/merged_probes.bed'\n",
    "\n",
    "num_raws  = []\n",
    "num_filts = []\n",
    "num_maps  = []\n",
    "seq_types = []\n",
    "avgs_covs       = []\n",
    "perc_cov_gt_10s = []\n",
    "perc_cov_lt_1s  = []\n",
    "\n",
    "#count num of raw and filtered reads\n",
    "i=0\n",
    "for sample in samples['exome'] + samples['genome']:\n",
    "    i+=1\n",
    "    print(\"{} of {}\".format(i, len(df)))\n",
    "    seq_type = \"\"\n",
    "    \n",
    "    if sample in samples['exome']:\n",
    "        raw_file = \"data/exomes/{}_R1.fastq.gz\".format(sample)\n",
    "        seq_type = \"exome\"\n",
    "    if sample in samples['genome']:\n",
    "        raw_file = \"data/sra/{}_1.fastq.gz\".format(sample)\n",
    "        seq_type = \"genome\"\n",
    "        \n",
    "    #raw reads\n",
    "    num_raw = num_seqs(raw_file)\n",
    "    \n",
    "    #num mapped reads\n",
    "    bam_file = pysam.AlignmentFile(\"results/mapped_reads/{}_processed.bam\".format(sample), \"rb\")\n",
    "    num_map = bam_file.mapped\n",
    "\n",
    "    #get coverage\n",
    "    bam_file = \"results/mapped_reads/{}_processed.bam\".format(sample)\n",
    "    out_prfx = \"results/probe_coverage_and_read_counts/mosdepth/{}\".format(sample)\n",
    "    \n",
    "    mos_cmd = \"mosdepth -t 4 --by {} {} --no-per-base {}\".format(merged_probes_bed, out_prfx, bam_file)\n",
    "    mos_cmd=mos_cmd.split(\" \")\n",
    "\n",
    "    #run mosdepth\n",
    "    process = subprocess.run(mos_cmd, \n",
    "                         stdout=subprocess.PIPE, \n",
    "                         universal_newlines=True)\n",
    "\n",
    "    #calculate cov stats\n",
    "    covs=np.array([])\n",
    "    with gzip.open(\"results/probe_coverage_and_read_counts/mosdepth/{}.regions.bed.gz\".format(sample), 'rb') as mos_infile:\n",
    "        for cov_entry in mos_infile:\n",
    "            chrom, start, stop, cov = cov_entry.decode(\"utf-8\") .rstrip().split(\"\\t\") \n",
    "            cov=float(cov)\n",
    "            covs=np.append(covs, cov)\n",
    "        \n",
    "    avg_cov = np.mean(covs)\n",
    "    perc_cov_gt_10 = np.sum(covs>10)/len(covs)\n",
    "    perc_cov_lt_1 = np.sum(covs<1)/len(covs)\n",
    "    \n",
    "    #build arrays for df import\n",
    "    seq_types.append(seq_type)\n",
    "    num_raws.append(num_raw)\n",
    "    num_maps.append(num_map)\n",
    "    avgs_covs.append(avg_cov)\n",
    "    perc_cov_gt_10s.append(perc_cov_gt_10)\n",
    "    perc_cov_lt_1s.append(perc_cov_lt_1)\n",
    "\n",
    "df.insert(0,'seq_type', seq_types)\n",
    "df.insert(1,'num_raw', num_raws)\n",
    "df.insert(2,'mapped_reads', num_maps)\n",
    "df.insert(3,'avg_probe_cov', avgs_covs)\n",
    "df.insert(4,'perc_probes_gt10x', perc_cov_gt_10s)\n",
    "df.insert(5,'perc_probes_failed', perc_cov_lt_1s)\n",
    "\n",
    "csv_file = \"results/probe_coverage_and_read_counts/samples_stats.csv\"\n",
    "df.to_csv(csv_file, index=True, header=True, mode='w')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-11-25T15:06:03.039134Z",
     "start_time": "2020-11-25T15:06:03.009749Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_type</th>\n",
       "      <th>num_raw</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>avg_probe_cov</th>\n",
       "      <th>perc_probes_gt10x</th>\n",
       "      <th>perc_probes_failed</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Sro_female_1.1_CCATCCTC</th>\n",
       "      <td>exome</td>\n",
       "      <td>10892425</td>\n",
       "      <td>18856533</td>\n",
       "      <td>49.880128</td>\n",
       "      <td>0.907384</td>\n",
       "      <td>0.014554</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sro_female_1.2_CCGACAAC</th>\n",
       "      <td>exome</td>\n",
       "      <td>10015111</td>\n",
       "      <td>17291766</td>\n",
       "      <td>47.860285</td>\n",
       "      <td>0.905216</td>\n",
       "      <td>0.015754</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sro_female_2.1_CCTAATCC</th>\n",
       "      <td>exome</td>\n",
       "      <td>9864727</td>\n",
       "      <td>17165447</td>\n",
       "      <td>47.281340</td>\n",
       "      <td>0.900583</td>\n",
       "      <td>0.015803</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sro_female_2.2_CCTCTATC</th>\n",
       "      <td>exome</td>\n",
       "      <td>9563172</td>\n",
       "      <td>16729520</td>\n",
       "      <td>46.810542</td>\n",
       "      <td>0.899072</td>\n",
       "      <td>0.016476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sro_male_1.1_ATCATTCC</th>\n",
       "      <td>exome</td>\n",
       "      <td>10833101</td>\n",
       "      <td>18912188</td>\n",
       "      <td>51.479108</td>\n",
       "      <td>0.915630</td>\n",
       "      <td>0.015097</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERR103050</th>\n",
       "      <td>genome</td>\n",
       "      <td>92179813</td>\n",
       "      <td>180540948</td>\n",
       "      <td>44.107285</td>\n",
       "      <td>0.995745</td>\n",
       "      <td>0.001298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERR103049</th>\n",
       "      <td>genome</td>\n",
       "      <td>124690537</td>\n",
       "      <td>241099490</td>\n",
       "      <td>64.530226</td>\n",
       "      <td>0.996337</td>\n",
       "      <td>0.001610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERR119614</th>\n",
       "      <td>genome</td>\n",
       "      <td>124372222</td>\n",
       "      <td>244862320</td>\n",
       "      <td>60.752446</td>\n",
       "      <td>0.996090</td>\n",
       "      <td>0.001593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERR119615</th>\n",
       "      <td>genome</td>\n",
       "      <td>118853923</td>\n",
       "      <td>232077360</td>\n",
       "      <td>59.624398</td>\n",
       "      <td>0.996386</td>\n",
       "      <td>0.001298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERR997461</th>\n",
       "      <td>genome</td>\n",
       "      <td>118729715</td>\n",
       "      <td>228077620</td>\n",
       "      <td>65.478743</td>\n",
       "      <td>0.995302</td>\n",
       "      <td>0.001725</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>156 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        seq_type    num_raw  mapped_reads  avg_probe_cov  \\\n",
       "Sro_female_1.1_CCATCCTC    exome   10892425      18856533      49.880128   \n",
       "Sro_female_1.2_CCGACAAC    exome   10015111      17291766      47.860285   \n",
       "Sro_female_2.1_CCTAATCC    exome    9864727      17165447      47.281340   \n",
       "Sro_female_2.2_CCTCTATC    exome    9563172      16729520      46.810542   \n",
       "Sro_male_1.1_ATCATTCC      exome   10833101      18912188      51.479108   \n",
       "...                          ...        ...           ...            ...   \n",
       "ERR103050                 genome   92179813     180540948      44.107285   \n",
       "ERR103049                 genome  124690537     241099490      64.530226   \n",
       "ERR119614                 genome  124372222     244862320      60.752446   \n",
       "ERR119615                 genome  118853923     232077360      59.624398   \n",
       "ERR997461                 genome  118729715     228077620      65.478743   \n",
       "\n",
       "                         perc_probes_gt10x  perc_probes_failed  \n",
       "Sro_female_1.1_CCATCCTC           0.907384            0.014554  \n",
       "Sro_female_1.2_CCGACAAC           0.905216            0.015754  \n",
       "Sro_female_2.1_CCTAATCC           0.900583            0.015803  \n",
       "Sro_female_2.2_CCTCTATC           0.899072            0.016476  \n",
       "Sro_male_1.1_ATCATTCC             0.915630            0.015097  \n",
       "...                                    ...                 ...  \n",
       "ERR103050                         0.995745            0.001298  \n",
       "ERR103049                         0.996337            0.001610  \n",
       "ERR119614                         0.996090            0.001593  \n",
       "ERR119615                         0.996386            0.001298  \n",
       "ERR997461                         0.995302            0.001725  \n",
       "\n",
       "[156 rows x 6 columns]"
      ]
     },
     "execution_count": 140,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.1"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
