{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The rpy2.ipython extension is already loaded. To reload it, use:\n",
      "  %reload_ext rpy2.ipython\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import shutil\n",
    "import vcf\n",
    "import re\n",
    "import gzip\n",
    "import pandas as pd\n",
    "\n",
    "from IPython.display import Image\n",
    "from Bio import SeqIO\n",
    "\n",
    "\n",
    "import rpy2.ipython\n",
    "\n",
    "\n",
    "%load_ext rpy2.ipython\n",
    "\n",
    "os.chdir(\"/master/nplatt/sch_man_nwinvasion\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Genotyping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#run snakemake\n",
    "snakemake \\\n",
    "    --printshellcmds \\\n",
    "    --use-conda \\\n",
    "    --cluster 'qsub -V -cwd -S /bin/bash -pe smp {threads} -o {log}.log -j y' \\\n",
    "    --jobs 1000 \\\n",
    "    --latency-wait 200 \\\n",
    "    --keep-going"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Filtering\n",
    "\n",
    "Multiple rounds of filtering were used to get rid of low quality sites and poorly genotyped sites/individuals"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "First low quality/coverage were removed and only bi-allelic sites were retained"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/hard_filtered.vcf\n",
      "\t--recode-INFO-all\n",
      "\t--max-alleles 2\n",
      "\t--min-alleles 2\n",
      "\t--minDP 12\n",
      "\t--minGQ 25\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "After filtering, kept 172 out of 172 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 671303 out of a possible 703368 Sites\n",
      "Run Time = 1100.00 seconds\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "vcftools \\\n",
    "    --vcf results/variant_filtration/hard_filtered.vcf \\\n",
    "    --minDP 12 \\\n",
    "    --minGQ 25 \\\n",
    "    --min-alleles 2 \\\n",
    "    --max-alleles 2 \\\n",
    "    --recode \\\n",
    "    --recode-INFO-all \\\n",
    "    --stdout \\\n",
    "    >results/variant_filtration/gq_dp.vcf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "sites that were genotyped less than 50% of the time were removed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/gq_dp.vcf\n",
      "\t--recode-INFO-all\n",
      "\t--positions results/variant_filtration/gt_rate_ge_50p.list\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "After filtering, kept 172 out of 172 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 631588 out of a possible 671303 Sites\n",
      "Run Time = 857.00 seconds\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "#find sites with gt50% missing data\n",
    "vcftools \\\n",
    "    --vcf results/variant_filtration/gq_dp.vcf \\\n",
    "    --missing-site \\\n",
    "    --stdout \\\n",
    "    >results/variant_filtration/gt_rate_per_site.tbl\n",
    "\n",
    "awk '{if ($6<0.5) print $1\"\\t\"$2}' \\\n",
    "    results/variant_filtration/gt_rate_per_site.tbl \\\n",
    "    >results/variant_filtration/gt_rate_ge_50p.list\n",
    "\n",
    "vcftools \\\n",
    "    --vcf results/variant_filtration/gq_dp.vcf \\\n",
    "    --positions results/variant_filtration/gt_rate_ge_50p.list \\\n",
    "    --recode \\\n",
    "    --recode-INFO-all \\\n",
    "    --stdout \\\n",
    "    >results/variant_filtration/gt_rate_ge_50p.vcf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "individuals there were genotyped at less than 50% of sites were removed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/gt_rate_ge_50p.vcf\n",
      "\t--missing-indv\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "After filtering, kept 172 out of 172 Individuals\n",
      "Outputting Individual Missingness\n",
      "After filtering, kept 631588 out of a possible 631588 Sites\n",
      "Run Time = 180.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/gt_rate_ge_50p.vcf\n",
      "\t--remove results/variant_filtration/indiv_gt_rate_lt_50p.list\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Excluding individuals in 'exclude' list\n",
      "After filtering, kept 156 out of 172 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 631588 out of a possible 631588 Sites\n",
      "Run Time = 834.00 seconds\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "#find idivs with lt 50p data\n",
    "vcftools \\\n",
    "    --vcf results/variant_filtration/gt_rate_ge_50p.vcf \\\n",
    "    --missing-indv \\\n",
    "    --stdout \\\n",
    "    >results/variant_filtration/indiv_gt_rate.tbl\n",
    "\n",
    "cat results/variant_filtration/indiv_gt_rate.tbl \\\n",
    "    | awk '$5>0.50 {print $1}' \\\n",
    "    | sed 1d \\\n",
    "    >results/variant_filtration/indiv_gt_rate_lt_50p.list\n",
    "\n",
    "vcftools \\\n",
    "    --vcf results/variant_filtration/gt_rate_ge_50p.vcf \\\n",
    "    --remove results/variant_filtration/indiv_gt_rate_lt_50p.list \\\n",
    "    --recode \\\n",
    "    --recode-INFO-all \\\n",
    "    --stdout \\\n",
    "    >results/variant_filtration/50p_site_50p_indiv_filtered.vcf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "added an id to each site (chrom:pos)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "bcftools annotate \\\n",
    "    --set-id +'%CHROM\\:%POS' \\\n",
    "    results/variant_filtration/50p_site_50p_indiv_filtered.vcf \\\n",
    "    >results/variant_filtration/50p_site_50p_indiv_filtered_annotated.vcf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "to make processing easier, i reorganized the order of individuals in \"geographic\" order rather than alphabetical order of sample names (the ```re-organized_header.vcf``` file was manually generated)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "grep \"#\" results/variant_filtration/50p_site_50p_indiv_filtered_annotated.vcf >results/variant_filtration/header.vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#manually re-arrange samples into desired order and save as results/variant_filtration/re-organized_header.vcf\n",
    "gzip results/variant_filtration/50p_site_50p_indiv_filtered_annotated.vcf\n",
    "gzip results/variant_filtration/re-organized_header.vcf\n",
    "\n",
    "vcf-shuffle-cols -t results/variant_filtration/re-organized_header.vcf.gz \\\n",
    "    results/variant_filtration/50p_site_50p_indiv_filtered_annotated.vcf.gz \\\n",
    "    >results/variant_filtration/smv7_ex_snps.vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#get gt rate of final filtered dataset\n",
    "vcftools \\\n",
    "    --vcf results/variant_filtration/smv7_ex_snps.vcf \\\n",
    "    --missing-indv \\\n",
    "    --stdout \\\n",
    "    >results/variant_filtration/gt_rate_per_indiv_at_filtered_sites.tbl\n",
    "    \n",
    "vcftools \\\n",
    "    --vcf results/variant_filtration/smv7_ex_snps.vcf \\\n",
    "    --missing-site \\\n",
    "    --stdout \\\n",
    "    >results/variant_filtration/gt_rate_per_site_at_filtered_sites.tbl"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "split the filtered file into autosomal, sex_chr, and mitochondrial variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_snps.vcf\n",
      "\t--chr SM_V7_1\n",
      "\t--chr SM_V7_2\n",
      "\t--chr SM_V7_3\n",
      "\t--chr SM_V7_4\n",
      "\t--chr SM_V7_5\n",
      "\t--chr SM_V7_6\n",
      "\t--chr SM_V7_7\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "After filtering, kept 156 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 475081 out of a possible 631588 Sites\n",
      "Run Time = 657.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_snps.vcf\n",
      "\t--chr SM_V7_ZW\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "After filtering, kept 156 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 155410 out of a possible 631588 Sites\n",
      "Run Time = 205.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_snps.vcf\n",
      "\t--chr SM_V7_MITO\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "After filtering, kept 156 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 815 out of a possible 631588 Sites\n",
      "Run Time = 23.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_snps.vcf\n",
      "\t--not-chr SM_V7_1\n",
      "\t--not-chr SM_V7_2\n",
      "\t--not-chr SM_V7_3\n",
      "\t--not-chr SM_V7_4\n",
      "\t--not-chr SM_V7_5\n",
      "\t--not-chr SM_V7_6\n",
      "\t--not-chr SM_V7_7\n",
      "\t--not-chr SM_V7_MITO\n",
      "\t--not-chr SM_V7_ZW\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "After filtering, kept 156 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 282 out of a possible 631588 Sites\n",
      "Run Time = 20.00 seconds\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "vcftools \\\n",
    "    --vcf results/variant_filtration/smv7_ex_snps.vcf \\\n",
    "    --chr SM_V7_1 \\\n",
    "    --chr SM_V7_2 \\\n",
    "    --chr SM_V7_3 \\\n",
    "    --chr SM_V7_4 \\\n",
    "    --chr SM_V7_5 \\\n",
    "    --chr SM_V7_6 \\\n",
    "    --chr SM_V7_7 \\\n",
    "    --recode \\\n",
    "    --recode-INFO-all \\\n",
    "    --stdout \\\n",
    "    >results/variant_filtration/smv7_ex_autosomes.vcf\n",
    "    \n",
    "vcftools \\\n",
    "    --vcf results/variant_filtration/smv7_ex_snps.vcf \\\n",
    "    --chr SM_V7_ZW \\\n",
    "    --recode \\\n",
    "    --recode-INFO-all \\\n",
    "    --stdout \\\n",
    "    >results/variant_filtration/smv7_ex_zw.vcf\n",
    "    \n",
    "vcftools \\\n",
    "    --vcf results/variant_filtration/smv7_ex_snps.vcf \\\n",
    "    --chr SM_V7_MITO \\\n",
    "    --recode \\\n",
    "    --recode-INFO-all \\\n",
    "    --stdout \\\n",
    "    >results/variant_filtration/smv7_ex_mito.vcf\n",
    "    \n",
    "vcftools \\\n",
    "    --vcf results/variant_filtration/smv7_ex_snps.vcf \\\n",
    "    --not-chr SM_V7_1 \\\n",
    "    --not-chr SM_V7_2 \\\n",
    "    --not-chr SM_V7_3 \\\n",
    "    --not-chr SM_V7_4 \\\n",
    "    --not-chr SM_V7_5 \\\n",
    "    --not-chr SM_V7_6 \\\n",
    "    --not-chr SM_V7_7 \\\n",
    "    --not-chr SM_V7_MITO \\\n",
    "    --not-chr SM_V7_Z \\\n",
    "    --recode \\\n",
    "    --recode-INFO-all \\\n",
    "    --stdout \\\n",
    "    >results/variant_filtration/smv7_ex_other.vcf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "filter sites based on ld"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to results/variant_filtration/smv7_ex_autosomes_ld.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --indep-pairwise 250kb 1 0.20\n",
      "  --out results/variant_filtration/smv7_ex_autosomes_ld\n",
      "  --vcf results/variant_filtration/smv7_ex_autosomes.vcf\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "\r",
      "--vcf: 1k variants complete.\r",
      "--vcf: 2k variants complete.\r",
      "--vcf: 3k variants complete.\r",
      "--vcf: 4k variants complete.\r",
      "--vcf: 5k variants complete.\r",
      "--vcf: 6k variants complete.\r",
      "--vcf: 7k variants complete.\r",
      "--vcf: 8k variants complete.\r",
      "--vcf: 9k variants complete.\r",
      "--vcf: 10k variants complete.\r",
      "--vcf: 11k variants complete.\r",
      "--vcf: 12k variants complete.\r",
      "--vcf: 13k variants complete.\r",
      "--vcf: 14k variants complete.\r",
      "--vcf: 15k variants complete.\r",
      "--vcf: 16k variants complete.\r",
      "--vcf: 17k variants complete.\r",
      "--vcf: 18k variants complete.\r",
      "--vcf: 19k variants complete.\r",
      "--vcf: 20k variants complete.\r",
      "--vcf: 21k variants complete.\r",
      "--vcf: 22k variants complete.\r",
      "--vcf: 23k variants complete.\r",
      "--vcf: 24k variants complete.\r",
      "--vcf: 25k variants complete.\r",
      "--vcf: 26k variants complete.\r",
      "--vcf: 27k variants complete.\r",
      "--vcf: 28k variants complete.\r",
      "--vcf: 29k variants complete.\r",
      "--vcf: 30k variants complete.\r",
      "--vcf: 31k variants complete.\r",
      "--vcf: 32k variants complete.\r",
      "--vcf: 33k variants complete.\r",
      "--vcf: 34k variants complete.\r",
      "--vcf: 35k variants complete.\r",
      "--vcf: 36k variants complete.\r",
      "--vcf: 37k variants complete.\r",
      "--vcf: 38k variants complete.\r",
      "--vcf: 39k variants complete.\r",
      "--vcf: 40k variants complete.\r",
      "--vcf: 41k variants complete.\r",
      "--vcf: 42k variants complete.\r",
      "--vcf: 43k variants complete.\r",
      "--vcf: 44k variants complete.\r",
      "--vcf: 45k variants complete.\r",
      "--vcf: 46k variants complete.\r",
      "--vcf: 47k variants complete.\r",
      "--vcf: 48k variants complete.\r",
      "--vcf: 49k variants complete.\r",
      "--vcf: 50k variants complete.\r",
      "--vcf: 51k variants complete.\r",
      "--vcf: 52k variants complete.\r",
      "--vcf: 53k variants complete.\r",
      "--vcf: 54k variants complete.\r",
      "--vcf: 55k variants complete.\r",
      "--vcf: 56k variants complete.\r",
      "--vcf: 57k variants complete.\r",
      "--vcf: 58k variants complete.\r",
      "--vcf: 59k variants complete.\r",
      "--vcf: 60k variants complete.\r",
      "--vcf: 61k variants complete.\r",
      "--vcf: 62k variants complete.\r",
      "--vcf: 63k variants complete.\r",
      "--vcf: 64k variants complete.\r",
      "--vcf: 65k variants complete.\r",
      "--vcf: 66k variants complete.\r",
      "--vcf: 67k variants complete.\r",
      "--vcf: 68k variants complete.\r",
      "--vcf: 69k variants complete.\r",
      "--vcf: 70k variants complete.\r",
      "--vcf: 71k variants complete.\r",
      "--vcf: 72k variants complete.\r",
      "--vcf: 73k variants complete.\r",
      "--vcf: 74k variants complete.\r",
      "--vcf: 75k variants complete.\r",
      "--vcf: 76k variants complete.\r",
      "--vcf: 77k variants complete.\r",
      "--vcf: 78k variants complete.\r",
      "--vcf: 79k variants complete.\r",
      "--vcf: 80k variants complete.\r",
      "--vcf: 81k variants complete.\r",
      "--vcf: 82k variants complete.\r",
      "--vcf: 83k variants complete.\r",
      "--vcf: 84k variants complete.\r",
      "--vcf: 85k variants complete.\r",
      "--vcf: 86k variants complete.\r",
      "--vcf: 87k variants complete.\r",
      "--vcf: 88k variants complete.\r",
      "--vcf: 89k variants complete.\r",
      "--vcf: 90k variants complete.\r",
      "--vcf: 91k variants complete.\r",
      "--vcf: 92k variants complete.\r",
      "--vcf: 93k variants complete.\r",
      "--vcf: 94k variants complete.\r",
      "--vcf: 95k variants complete.\r",
      "--vcf: 96k variants complete.\r",
      "--vcf: 97k variants complete.\r",
      "--vcf: 98k variants complete.\r",
      "--vcf: 99k variants complete.\r",
      "--vcf: 100k variants complete.\r",
      "--vcf: 101k variants complete.\r",
      "--vcf: 102k variants complete.\r",
      "--vcf: 103k variants complete.\r",
      "--vcf: 104k variants complete.\r",
      "--vcf: 105k variants complete.\r",
      "--vcf: 106k variants complete.\r",
      "--vcf: 107k variants complete.\r",
      "--vcf: 108k variants complete.\r",
      "--vcf: 109k variants complete.\r",
      "--vcf: 110k variants complete.\r",
      "--vcf: 111k variants complete.\r",
      "--vcf: 112k variants complete.\r",
      "--vcf: 113k variants complete.\r",
      "--vcf: 114k variants complete.\r",
      "--vcf: 115k variants complete.\r",
      "--vcf: 116k variants complete.\r",
      "--vcf: 117k variants complete.\r",
      "--vcf: 118k variants complete.\r",
      "--vcf: 119k variants complete.\r",
      "--vcf: 120k variants complete.\r",
      "--vcf: 121k variants complete.\r",
      "--vcf: 122k variants complete.\r",
      "--vcf: 123k variants complete.\r",
      "--vcf: 124k variants complete.\r",
      "--vcf: 125k variants complete.\r",
      "--vcf: 126k variants complete.\r",
      "--vcf: 127k variants complete.\r",
      "--vcf: 128k variants complete.\r",
      "--vcf: 129k variants complete.\r",
      "--vcf: 130k variants complete.\r",
      "--vcf: 131k variants complete.\r",
      "--vcf: 132k variants complete.\r",
      "--vcf: 133k variants complete.\r",
      "--vcf: 134k variants complete.\r",
      "--vcf: 135k variants complete.\r",
      "--vcf: 136k variants complete.\r",
      "--vcf: 137k variants complete.\r",
      "--vcf: 138k variants complete.\r",
      "--vcf: 139k variants complete.\r",
      "--vcf: 140k variants complete.\r",
      "--vcf: 141k variants complete.\r",
      "--vcf: 142k variants complete.\r",
      "--vcf: 143k variants complete.\r",
      "--vcf: 144k variants complete.\r",
      "--vcf: 145k variants complete.\r",
      "--vcf: 146k variants complete.\r",
      "--vcf: 147k variants complete.\r",
      "--vcf: 148k variants complete.\r",
      "--vcf: 149k variants complete.\r",
      "--vcf: 150k variants complete.\r",
      "--vcf: 151k variants complete.\r",
      "--vcf: 152k variants complete.\r",
      "--vcf: 153k variants complete.\r",
      "--vcf: 154k variants complete.\r",
      "--vcf: 155k variants complete.\r",
      "--vcf: 156k variants complete.\r",
      "--vcf: 157k variants complete.\r",
      "--vcf: 158k variants complete.\r",
      "--vcf: 159k variants complete.\r",
      "--vcf: 160k variants complete.\r",
      "--vcf: 161k variants complete.\r",
      "--vcf: 162k variants complete.\r",
      "--vcf: 163k variants complete.\r",
      "--vcf: 164k variants complete.\r",
      "--vcf: 165k variants complete.\r",
      "--vcf: 166k variants complete.\r",
      "--vcf: 167k variants complete.\r",
      "--vcf: 168k variants complete.\r",
      "--vcf: 169k variants complete.\r",
      "--vcf: 170k variants complete.\r",
      "--vcf: 171k variants complete.\r",
      "--vcf: 172k variants complete.\r",
      "--vcf: 173k variants complete.\r",
      "--vcf: 174k variants complete.\r",
      "--vcf: 175k variants complete.\r",
      "--vcf: 176k variants complete.\r",
      "--vcf: 177k variants complete.\r",
      "--vcf: 178k variants complete.\r",
      "--vcf: 179k variants complete.\r",
      "--vcf: 180k variants complete.\r",
      "--vcf: 181k variants complete.\r",
      "--vcf: 182k variants complete.\r",
      "--vcf: 183k variants complete.\r",
      "--vcf: 184k variants complete.\r",
      "--vcf: 185k variants complete.\r",
      "--vcf: 186k variants complete.\r",
      "--vcf: 187k variants complete.\r",
      "--vcf: 188k variants complete.\r",
      "--vcf: 189k variants complete.\r",
      "--vcf: 190k variants complete.\r",
      "--vcf: 191k variants complete.\r",
      "--vcf: 192k variants complete.\r",
      "--vcf: 193k variants complete.\r",
      "--vcf: 194k variants complete.\r",
      "--vcf: 195k variants complete.\r",
      "--vcf: 196k variants complete.\r",
      "--vcf: 197k variants complete.\r",
      "--vcf: 198k variants complete.\r",
      "--vcf: 199k variants complete.\r",
      "--vcf: 200k variants complete.\r",
      "--vcf: 201k variants complete.\r",
      "--vcf: 202k variants complete.\r",
      "--vcf: 203k variants complete.\r",
      "--vcf: 204k variants complete.\r",
      "--vcf: 205k variants complete.\r",
      "--vcf: 206k variants complete.\r",
      "--vcf: 207k variants complete.\r",
      "--vcf: 208k variants complete.\r",
      "--vcf: 209k variants complete.\r",
      "--vcf: 210k variants complete.\r",
      "--vcf: 211k variants complete.\r",
      "--vcf: 212k variants complete.\r",
      "--vcf: 213k variants complete.\r",
      "--vcf: 214k variants complete.\r",
      "--vcf: 215k variants complete.\r",
      "--vcf: 216k variants complete.\r",
      "--vcf: 217k variants complete.\r",
      "--vcf: 218k variants complete.\r",
      "--vcf: 219k variants complete.\r",
      "--vcf: 220k variants complete.\r",
      "--vcf: 221k variants complete.\r",
      "--vcf: 222k variants complete.\r",
      "--vcf: 223k variants complete.\r",
      "--vcf: 224k variants complete.\r",
      "--vcf: 225k variants complete.\r",
      "--vcf: 226k variants complete.\r",
      "--vcf: 227k variants complete.\r",
      "--vcf: 228k variants complete.\r",
      "--vcf: 229k variants complete.\r",
      "--vcf: 230k variants complete.\r",
      "--vcf: 231k variants complete.\r",
      "--vcf: 232k variants complete.\r",
      "--vcf: 233k variants complete.\r",
      "--vcf: 234k variants complete.\r",
      "--vcf: 235k variants complete.\r",
      "--vcf: 236k variants complete.\r",
      "--vcf: 237k variants complete.\r",
      "--vcf: 238k variants complete.\r",
      "--vcf: 239k variants complete.\r",
      "--vcf: 240k variants complete.\r",
      "--vcf: 241k variants complete.\r",
      "--vcf: 242k variants complete.\r",
      "--vcf: 243k variants complete.\r",
      "--vcf: 244k variants complete.\r",
      "--vcf: 245k variants complete.\r",
      "--vcf: 246k variants complete.\r",
      "--vcf: 247k variants complete.\r",
      "--vcf: 248k variants complete.\r",
      "--vcf: 249k variants complete.\r",
      "--vcf: 250k variants complete.\r",
      "--vcf: 251k variants complete.\r",
      "--vcf: 252k variants complete.\r",
      "--vcf: 253k variants complete.\r",
      "--vcf: 254k variants complete.\r",
      "--vcf: 255k variants complete.\r",
      "--vcf: 256k variants complete.\r",
      "--vcf: 257k variants complete.\r",
      "--vcf: 258k variants complete.\r",
      "--vcf: 259k variants complete.\r",
      "--vcf: 260k variants complete.\r",
      "--vcf: 261k variants complete.\r",
      "--vcf: 262k variants complete.\r",
      "--vcf: 263k variants complete.\r",
      "--vcf: 264k variants complete.\r",
      "--vcf: 265k variants complete.\r",
      "--vcf: 266k variants complete.\r",
      "--vcf: 267k variants complete.\r",
      "--vcf: 268k variants complete.\r",
      "--vcf: 269k variants complete.\r",
      "--vcf: 270k variants complete.\r",
      "--vcf: 271k variants complete.\r",
      "--vcf: 272k variants complete.\r",
      "--vcf: 273k variants complete.\r",
      "--vcf: 274k variants complete.\r",
      "--vcf: 275k variants complete.\r",
      "--vcf: 276k variants complete.\r",
      "--vcf: 277k variants complete.\r",
      "--vcf: 278k variants complete.\r",
      "--vcf: 279k variants complete.\r",
      "--vcf: 280k variants complete.\r",
      "--vcf: 281k variants complete.\r",
      "--vcf: 282k variants complete.\r",
      "--vcf: 283k variants complete.\r",
      "--vcf: 284k variants complete.\r",
      "--vcf: 285k variants complete.\r",
      "--vcf: 286k variants complete.\r",
      "--vcf: 287k variants complete.\r",
      "--vcf: 288k variants complete.\r",
      "--vcf: 289k variants complete.\r",
      "--vcf: 290k variants complete.\r",
      "--vcf: 291k variants complete.\r",
      "--vcf: 292k variants complete.\r",
      "--vcf: 293k variants complete.\r",
      "--vcf: 294k variants complete.\r",
      "--vcf: 295k variants complete.\r",
      "--vcf: 296k variants complete.\r",
      "--vcf: 297k variants complete.\r",
      "--vcf: 298k variants complete.\r",
      "--vcf: 299k variants complete.\r",
      "--vcf: 300k variants complete.\r",
      "--vcf: 301k variants complete.\r",
      "--vcf: 302k variants complete.\r",
      "--vcf: 303k variants complete.\r",
      "--vcf: 304k variants complete.\r",
      "--vcf: 305k variants complete.\r",
      "--vcf: 306k variants complete.\r",
      "--vcf: 307k variants complete.\r",
      "--vcf: 308k variants complete.\r",
      "--vcf: 309k variants complete.\r",
      "--vcf: 310k variants complete.\r",
      "--vcf: 311k variants complete.\r",
      "--vcf: 312k variants complete.\r",
      "--vcf: 313k variants complete.\r",
      "--vcf: 314k variants complete.\r",
      "--vcf: 315k variants complete.\r",
      "--vcf: 316k variants complete.\r",
      "--vcf: 317k variants complete.\r",
      "--vcf: 318k variants complete.\r",
      "--vcf: 319k variants complete.\r",
      "--vcf: 320k variants complete.\r",
      "--vcf: 321k variants complete.\r",
      "--vcf: 322k variants complete.\r",
      "--vcf: 323k variants complete.\r",
      "--vcf: 324k variants complete.\r",
      "--vcf: 325k variants complete.\r",
      "--vcf: 326k variants complete.\r",
      "--vcf: 327k variants complete.\r",
      "--vcf: 328k variants complete.\r",
      "--vcf: 329k variants complete.\r",
      "--vcf: 330k variants complete.\r",
      "--vcf: 331k variants complete.\r",
      "--vcf: 332k variants complete.\r",
      "--vcf: 333k variants complete.\r",
      "--vcf: 334k variants complete.\r",
      "--vcf: 335k variants complete.\r",
      "--vcf: 336k variants complete.\r",
      "--vcf: 337k variants complete.\r",
      "--vcf: 338k variants complete.\r",
      "--vcf: 339k variants complete.\r",
      "--vcf: 340k variants complete.\r",
      "--vcf: 341k variants complete.\r",
      "--vcf: 342k variants complete.\r",
      "--vcf: 343k variants complete.\r",
      "--vcf: 344k variants complete.\r",
      "--vcf: 345k variants complete.\r",
      "--vcf: 346k variants complete.\r",
      "--vcf: 347k variants complete.\r",
      "--vcf: 348k variants complete.\r",
      "--vcf: 349k variants complete.\r",
      "--vcf: 350k variants complete.\r",
      "--vcf: 351k variants complete.\r",
      "--vcf: 352k variants complete.\r",
      "--vcf: 353k variants complete.\r",
      "--vcf: 354k variants complete.\r",
      "--vcf: 355k variants complete.\r",
      "--vcf: 356k variants complete.\r",
      "--vcf: 357k variants complete.\r",
      "--vcf: 358k variants complete.\r",
      "--vcf: 359k variants complete.\r",
      "--vcf: 360k variants complete.\r",
      "--vcf: 361k variants complete.\r",
      "--vcf: 362k variants complete.\r",
      "--vcf: 363k variants complete.\r",
      "--vcf: 364k variants complete.\r",
      "--vcf: 365k variants complete.\r",
      "--vcf: 366k variants complete.\r",
      "--vcf: 367k variants complete.\r",
      "--vcf: 368k variants complete.\r",
      "--vcf: 369k variants complete.\r",
      "--vcf: 370k variants complete.\r",
      "--vcf: 371k variants complete.\r",
      "--vcf: 372k variants complete.\r",
      "--vcf: 373k variants complete.\r",
      "--vcf: 374k variants complete.\r",
      "--vcf: 375k variants complete.\r",
      "--vcf: 376k variants complete.\r",
      "--vcf: 377k variants complete.\r",
      "--vcf: 378k variants complete.\r",
      "--vcf: 379k variants complete.\r",
      "--vcf: 380k variants complete.\r",
      "--vcf: 381k variants complete.\r",
      "--vcf: 382k variants complete.\r",
      "--vcf: 383k variants complete.\r",
      "--vcf: 384k variants complete.\r",
      "--vcf: 385k variants complete.\r",
      "--vcf: 386k variants complete.\r",
      "--vcf: 387k variants complete.\r",
      "--vcf: 388k variants complete.\r",
      "--vcf: 389k variants complete.\r",
      "--vcf: 390k variants complete.\r",
      "--vcf: 391k variants complete.\r",
      "--vcf: 392k variants complete.\r",
      "--vcf: 393k variants complete.\r",
      "--vcf: 394k variants complete.\r",
      "--vcf: 395k variants complete.\r",
      "--vcf: 396k variants complete.\r",
      "--vcf: 397k variants complete.\r",
      "--vcf: 398k variants complete.\r",
      "--vcf: 399k variants complete.\r",
      "--vcf: 400k variants complete.\r",
      "--vcf: 401k variants complete.\r",
      "--vcf: 402k variants complete.\r",
      "--vcf: 403k variants complete.\r",
      "--vcf: 404k variants complete.\r",
      "--vcf: 405k variants complete.\r",
      "--vcf: 406k variants complete.\r",
      "--vcf: 407k variants complete.\r",
      "--vcf: 408k variants complete.\r",
      "--vcf: 409k variants complete.\r",
      "--vcf: 410k variants complete.\r",
      "--vcf: 411k variants complete.\r",
      "--vcf: 412k variants complete.\r",
      "--vcf: 413k variants complete.\r",
      "--vcf: 414k variants complete.\r",
      "--vcf: 415k variants complete.\r",
      "--vcf: 416k variants complete.\r",
      "--vcf: 417k variants complete.\r",
      "--vcf: 418k variants complete.\r",
      "--vcf: 419k variants complete.\r",
      "--vcf: 420k variants complete.\r",
      "--vcf: 421k variants complete.\r",
      "--vcf: 422k variants complete.\r",
      "--vcf: 423k variants complete.\r",
      "--vcf: 424k variants complete.\r",
      "--vcf: 425k variants complete.\r",
      "--vcf: 426k variants complete.\r",
      "--vcf: 427k variants complete.\r",
      "--vcf: 428k variants complete.\r",
      "--vcf: 429k variants complete.\r",
      "--vcf: 430k variants complete.\r",
      "--vcf: 431k variants complete.\r",
      "--vcf: 432k variants complete.\r",
      "--vcf: 433k variants complete.\r",
      "--vcf: 434k variants complete.\r",
      "--vcf: 435k variants complete.\r",
      "--vcf: 436k variants complete.\r",
      "--vcf: 437k variants complete.\r",
      "--vcf: 438k variants complete.\r",
      "--vcf: 439k variants complete.\r",
      "--vcf: 440k variants complete.\r",
      "--vcf: 441k variants complete.\r",
      "--vcf: 442k variants complete.\r",
      "--vcf: 443k variants complete.\r",
      "--vcf: 444k variants complete.\r",
      "--vcf: 445k variants complete.\r",
      "--vcf: 446k variants complete.\r",
      "--vcf: 447k variants complete.\r",
      "--vcf: 448k variants complete.\r",
      "--vcf: 449k variants complete.\r",
      "--vcf: 450k variants complete.\r",
      "--vcf: 451k variants complete.\r",
      "--vcf: 452k variants complete.\r",
      "--vcf: 453k variants complete.\r",
      "--vcf: 454k variants complete.\r",
      "--vcf: 455k variants complete.\r",
      "--vcf: 456k variants complete.\r",
      "--vcf: 457k variants complete.\r",
      "--vcf: 458k variants complete.\r",
      "--vcf: 459k variants complete.\r",
      "--vcf: 460k variants complete.\r",
      "--vcf: 461k variants complete.\r",
      "--vcf: 462k variants complete.\r",
      "--vcf: 463k variants complete.\r",
      "--vcf: 464k variants complete.\r",
      "--vcf: 465k variants complete.\r",
      "--vcf: 466k variants complete.\r",
      "--vcf: 467k variants complete.\r",
      "--vcf: 468k variants complete.\r",
      "--vcf: 469k variants complete.\r",
      "--vcf: 470k variants complete.\r",
      "--vcf: 471k variants complete.\r",
      "--vcf: 472k variants complete.\r",
      "--vcf: 473k variants complete.\r",
      "--vcf: 474k variants complete.\r",
      "--vcf: 475k variants complete.\r",
      "--vcf: results/variant_filtration/smv7_ex_autosomes_ld-temporary.bed +\n",
      "results/variant_filtration/smv7_ex_autosomes_ld-temporary.bim +\n",
      "results/variant_filtration/smv7_ex_autosomes_ld-temporary.fam written.\n",
      "475081 variants loaded from .bim file.\n",
      "156 people (0 males, 0 females, 156 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/variant_filtration/smv7_ex_autosomes_ld.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 156 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.879418.\n",
      "475081 variants and 156 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "\r",
      "0%\r",
      "1%\r",
      "1%\r",
      "2%\r",
      "2%\r",
      "3%\r",
      "3%\r",
      "4%\r",
      "4%\r",
      "5%\r",
      "5%\r",
      "6%\r",
      "6%\r",
      "7%\r",
      "7%\r",
      "8%\r",
      "8%\r",
      "9%\r",
      "9%\r",
      "10%\r",
      "10%\r",
      "11%\r",
      "11%\r",
      "12%\r",
      "12%\r",
      "13%\r",
      "13%\r",
      "14%\r",
      "14%\r",
      "15%\r",
      "15%\r",
      "16%\r",
      "16%\r",
      "17%\r",
      "17%\r",
      "18%\r",
      "18%\r",
      "19%\r",
      "19%\r",
      "20%\r",
      "20%\r",
      "21%\r",
      "21%\r",
      "22%\r",
      "22%\r",
      "23%\r",
      "23%\r",
      "24%\r",
      "24%\r",
      "25%\r",
      "25%\r",
      "26%\r",
      "26%\r",
      "27%\r",
      "27%\r",
      "28%\r",
      "28%\r",
      "29%\r",
      "29%\r",
      "30%\r",
      "30%\r",
      "31%\r",
      "31%\r",
      "32%\r",
      "32%\r",
      "33%\r",
      "33%\r",
      "34%\r",
      "34%\r",
      "35%\r",
      "35%\r",
      "36%\r",
      "36%\r",
      "37%\r",
      "37%\r",
      "38%\r",
      "38%\r",
      "39%\r",
      "39%\r",
      "40%\r",
      "40%\r",
      "41%\r",
      "41%\r",
      "42%\r",
      "42%\r",
      "43%\r",
      "43%\r",
      "44%\r",
      "44%\r",
      "45%\r",
      "45%\r",
      "46%\r",
      "46%\r",
      "47%\r",
      "47%\r",
      "48%\r",
      "48%\r",
      "49%\r",
      "49%\r",
      "50%\r",
      "50%\r",
      "51%\r",
      "51%\r",
      "52%\r",
      "52%\r",
      "53%\r",
      "53%\r",
      "54%\r",
      "54%\r",
      "55%\r",
      "55%\r",
      "56%\r",
      "56%\r",
      "57%\r",
      "57%\r",
      "58%\r",
      "58%\r",
      "59%\r",
      "59%\r",
      "60%\r",
      "60%\r",
      "61%\r",
      "61%\r",
      "62%\r",
      "62%\r",
      "63%\r",
      "63%\r",
      "64%\r",
      "64%\r",
      "65%\r",
      "65%\r",
      "66%\r",
      "66%\r",
      "67%\r",
      "67%\r",
      "68%\r",
      "68%\r",
      "69%\r",
      "69%\r",
      "70%\r",
      "70%\r",
      "71%\r",
      "71%\r",
      "72%\r",
      "72%\r",
      "73%\r",
      "73%\r",
      "74%\r",
      "74%\r",
      "75%\r",
      "75%\r",
      "76%\r",
      "76%\r",
      "77%\r",
      "77%\r",
      "78%\r",
      "78%\r",
      "79%\r",
      "79%\r",
      "80%\r",
      "80%\r",
      "81%\r",
      "81%\r",
      "82%\r",
      "82%\r",
      "83%\r",
      "83%\r",
      "84%\r",
      "84%\r",
      "85%\r",
      "85%\r",
      "86%\r",
      "86%\r",
      "87%\r",
      "87%\r",
      "88%\r",
      "88%\r",
      "89%\r",
      "89%\r",
      "90%\r",
      "90%\r",
      "91%\r",
      "91%\r",
      "92%\r",
      "92%\r",
      "93%\r",
      "93%\r",
      "94%\r",
      "94%\r",
      "95%\r",
      "95%\r",
      "96%\r",
      "96%\r",
      "97%\r",
      "97%\r",
      "98%\r",
      "98%\r",
      "99%\r",
      "Pruned 150151 variants from chromosome 27, leaving 12068.\n",
      "\r",
      "0%\r",
      "1%\r",
      "1%\r",
      "2%\r",
      "2%\r",
      "3%\r",
      "3%\r",
      "4%\r",
      "4%\r",
      "5%\r",
      "5%\r",
      "6%\r",
      "6%\r",
      "7%\r",
      "7%\r",
      "8%\r",
      "8%\r",
      "9%\r",
      "9%\r",
      "10%\r",
      "10%\r",
      "11%\r",
      "11%\r",
      "12%\r",
      "12%\r",
      "13%\r",
      "13%\r",
      "14%\r",
      "14%\r",
      "15%\r",
      "15%\r",
      "16%\r",
      "16%\r",
      "17%\r",
      "17%\r",
      "18%\r",
      "18%\r",
      "19%\r",
      "19%\r",
      "20%\r",
      "20%\r",
      "21%\r",
      "21%\r",
      "22%\r",
      "22%\r",
      "23%\r",
      "23%\r",
      "24%\r",
      "24%\r",
      "25%\r",
      "25%\r",
      "26%\r",
      "26%\r",
      "27%\r",
      "27%\r",
      "28%\r",
      "28%\r",
      "29%\r",
      "29%\r",
      "30%\r",
      "30%\r",
      "31%\r",
      "31%\r",
      "32%\r",
      "32%\r",
      "33%\r",
      "33%\r",
      "34%\r",
      "34%\r",
      "35%\r",
      "35%\r",
      "36%\r",
      "36%\r",
      "37%\r",
      "37%\r",
      "38%\r",
      "38%\r",
      "39%\r",
      "39%\r",
      "40%\r",
      "40%\r",
      "41%\r",
      "41%\r",
      "42%\r",
      "42%\r",
      "43%\r",
      "43%\r",
      "44%\r",
      "44%\r",
      "45%\r",
      "45%\r",
      "46%\r",
      "46%\r",
      "47%\r",
      "47%\r",
      "48%\r",
      "48%\r",
      "49%\r",
      "49%\r",
      "50%\r",
      "50%\r",
      "51%\r",
      "51%\r",
      "52%\r",
      "52%\r",
      "53%\r",
      "53%\r",
      "54%\r",
      "54%\r",
      "55%\r",
      "55%\r",
      "56%\r",
      "56%\r",
      "57%\r",
      "57%\r",
      "58%\r",
      "58%\r",
      "59%\r",
      "59%\r",
      "60%\r",
      "60%\r",
      "61%\r",
      "61%\r",
      "62%\r",
      "62%\r",
      "63%\r",
      "63%\r",
      "64%\r",
      "64%\r",
      "65%\r",
      "65%\r",
      "66%\r",
      "66%\r",
      "67%\r",
      "67%\r",
      "68%\r",
      "68%\r",
      "69%\r",
      "69%\r",
      "70%\r",
      "70%\r",
      "71%\r",
      "71%\r",
      "72%\r",
      "72%\r",
      "73%\r",
      "73%\r",
      "74%\r",
      "74%\r",
      "75%\r",
      "75%\r",
      "76%\r",
      "76%\r",
      "77%\r",
      "77%\r",
      "78%\r",
      "78%\r",
      "79%\r",
      "79%\r",
      "80%\r",
      "80%\r",
      "81%\r",
      "81%\r",
      "82%\r",
      "82%\r",
      "83%\r",
      "83%\r",
      "84%\r",
      "84%\r",
      "85%\r",
      "85%\r",
      "86%\r",
      "86%\r",
      "87%\r",
      "87%\r",
      "88%\r",
      "88%\r",
      "89%\r",
      "89%\r",
      "90%\r",
      "90%\r",
      "91%\r",
      "91%\r",
      "92%\r",
      "92%\r",
      "93%\r",
      "93%\r",
      "94%\r",
      "94%\r",
      "95%\r",
      "95%\r",
      "96%\r",
      "96%\r",
      "97%\r",
      "97%\r",
      "98%\r",
      "98%\r",
      "99%\r",
      "Pruned 74266 variants from chromosome 28, leaving 6215.\n",
      "\r",
      "0%\r",
      "1%\r",
      "1%\r",
      "2%\r",
      "2%\r",
      "3%\r",
      "3%\r",
      "4%\r",
      "4%\r",
      "5%\r",
      "5%\r",
      "6%\r",
      "6%\r",
      "7%\r",
      "7%\r",
      "8%\r",
      "8%\r",
      "9%\r",
      "9%\r",
      "10%\r",
      "10%\r",
      "11%\r",
      "11%\r",
      "12%\r",
      "12%\r",
      "13%\r",
      "13%\r",
      "14%\r",
      "14%\r",
      "15%\r",
      "15%\r",
      "16%\r",
      "16%\r",
      "17%\r",
      "17%\r",
      "18%\r",
      "18%\r",
      "19%\r",
      "19%\r",
      "20%\r",
      "20%\r",
      "21%\r",
      "21%\r",
      "22%\r",
      "22%\r",
      "23%\r",
      "23%\r",
      "24%\r",
      "24%\r",
      "25%\r",
      "25%\r",
      "26%\r",
      "26%\r",
      "27%\r",
      "27%\r",
      "28%\r",
      "28%\r",
      "29%\r",
      "29%\r",
      "30%\r",
      "30%\r",
      "31%\r",
      "31%\r",
      "32%\r",
      "32%\r",
      "33%\r",
      "33%\r",
      "34%\r",
      "34%\r",
      "35%\r",
      "35%\r",
      "36%\r",
      "36%\r",
      "37%\r",
      "37%\r",
      "38%\r",
      "38%\r",
      "39%\r",
      "39%\r",
      "40%\r",
      "40%\r",
      "41%\r",
      "41%\r",
      "42%\r",
      "42%\r",
      "43%\r",
      "43%\r",
      "44%\r",
      "44%\r",
      "45%\r",
      "45%\r",
      "46%\r",
      "46%\r",
      "47%\r",
      "47%\r",
      "48%\r",
      "48%\r",
      "49%\r",
      "50%\r",
      "50%\r",
      "51%\r",
      "51%\r",
      "52%\r",
      "52%\r",
      "53%\r",
      "53%\r",
      "54%\r",
      "54%\r",
      "55%\r",
      "55%\r",
      "56%\r",
      "56%\r",
      "57%\r",
      "57%\r",
      "58%\r",
      "58%\r",
      "59%\r",
      "59%\r",
      "60%\r",
      "60%\r",
      "61%\r",
      "61%\r",
      "62%\r",
      "62%\r",
      "63%\r",
      "63%\r",
      "64%\r",
      "64%\r",
      "65%\r",
      "65%\r",
      "66%\r",
      "66%\r",
      "67%\r",
      "67%\r",
      "68%\r",
      "68%\r",
      "69%\r",
      "69%\r",
      "70%\r",
      "70%\r",
      "71%\r",
      "71%\r",
      "72%\r",
      "72%\r",
      "73%\r",
      "73%\r",
      "74%\r",
      "74%\r",
      "75%\r",
      "75%\r",
      "76%\r",
      "76%\r",
      "77%\r",
      "77%\r",
      "78%\r",
      "78%\r",
      "79%\r",
      "79%\r",
      "80%\r",
      "80%\r",
      "81%\r",
      "81%\r",
      "82%\r",
      "82%\r",
      "83%\r",
      "83%\r",
      "84%\r",
      "84%\r",
      "85%\r",
      "85%\r",
      "86%\r",
      "86%\r",
      "87%\r",
      "87%\r",
      "88%\r",
      "88%\r",
      "89%\r",
      "89%\r",
      "90%\r",
      "90%\r",
      "91%\r",
      "91%\r",
      "92%\r",
      "92%\r",
      "93%\r",
      "93%\r",
      "94%\r",
      "94%\r",
      "95%\r",
      "95%\r",
      "96%\r",
      "96%\r",
      "97%\r",
      "97%\r",
      "98%\r",
      "98%\r",
      "99%\r",
      "Pruned 73043 variants from chromosome 29, leaving 5859.\n",
      "\r",
      "0%\r",
      "1%\r",
      "1%\r",
      "2%\r",
      "2%\r",
      "3%\r",
      "3%\r",
      "4%\r",
      "4%\r",
      "5%\r",
      "5%\r",
      "6%\r",
      "6%\r",
      "7%\r",
      "7%\r",
      "8%\r",
      "8%\r",
      "9%\r",
      "9%\r",
      "10%\r",
      "10%\r",
      "11%\r",
      "11%\r",
      "12%\r",
      "12%\r",
      "13%\r",
      "13%\r",
      "14%\r",
      "14%\r",
      "15%\r",
      "15%\r",
      "16%\r",
      "16%\r",
      "17%\r",
      "17%\r",
      "18%\r",
      "18%\r",
      "19%\r",
      "19%\r",
      "20%\r",
      "20%\r",
      "21%\r",
      "21%\r",
      "22%\r",
      "22%\r",
      "23%\r",
      "23%\r",
      "24%\r",
      "24%\r",
      "25%\r",
      "25%\r",
      "26%\r",
      "26%\r",
      "27%\r",
      "27%\r",
      "28%\r",
      "28%\r",
      "29%\r",
      "29%\r",
      "30%\r",
      "30%\r",
      "31%\r",
      "31%\r",
      "32%\r",
      "32%\r",
      "33%\r",
      "33%\r",
      "34%\r",
      "34%\r",
      "35%\r",
      "35%\r",
      "36%\r",
      "36%\r",
      "37%\r",
      "37%\r",
      "38%\r",
      "38%\r",
      "39%\r",
      "39%\r",
      "40%\r",
      "40%\r",
      "41%\r",
      "41%\r",
      "42%\r",
      "42%\r",
      "43%\r",
      "43%\r",
      "44%\r",
      "44%\r",
      "45%\r",
      "45%\r",
      "46%\r",
      "46%\r",
      "47%\r",
      "47%\r",
      "48%\r",
      "48%\r",
      "49%\r",
      "49%\r",
      "50%\r",
      "50%\r",
      "51%\r",
      "51%\r",
      "52%\r",
      "52%\r",
      "53%\r",
      "53%\r",
      "54%\r",
      "54%\r",
      "55%\r",
      "55%\r",
      "56%\r",
      "56%\r",
      "57%\r",
      "57%\r",
      "58%\r",
      "58%\r",
      "59%\r",
      "59%\r",
      "60%\r",
      "60%\r",
      "61%\r",
      "61%\r",
      "62%\r",
      "62%\r",
      "63%\r",
      "63%\r",
      "64%\r",
      "64%\r",
      "65%\r",
      "65%\r",
      "66%\r",
      "66%\r",
      "67%\r",
      "67%\r",
      "68%\r",
      "68%\r",
      "69%\r",
      "69%\r",
      "70%\r",
      "70%\r",
      "71%\r",
      "71%\r",
      "72%\r",
      "72%\r",
      "73%\r",
      "73%\r",
      "74%\r",
      "74%\r",
      "75%\r",
      "75%\r",
      "76%\r",
      "76%\r",
      "77%\r",
      "77%\r",
      "78%\r",
      "78%\r",
      "79%\r",
      "79%\r",
      "80%\r",
      "80%\r",
      "81%\r",
      "81%\r",
      "82%\r",
      "82%\r",
      "83%\r",
      "83%\r",
      "84%\r",
      "84%\r",
      "85%\r",
      "85%\r",
      "86%\r",
      "86%\r",
      "87%\r",
      "87%\r",
      "88%\r",
      "88%\r",
      "89%\r",
      "89%\r",
      "90%\r",
      "90%\r",
      "91%\r",
      "91%\r",
      "92%\r",
      "92%\r",
      "93%\r",
      "93%\r",
      "94%\r",
      "94%\r",
      "95%\r",
      "95%\r",
      "96%\r",
      "96%\r",
      "97%\r",
      "97%\r",
      "98%\r",
      "98%\r",
      "99%\r",
      "Pruned 69406 variants from chromosome 30, leaving 6175.\n",
      "\r",
      "0%\r",
      "1%\r",
      "1%\r",
      "2%\r",
      "2%\r",
      "3%\r",
      "3%\r",
      "4%\r",
      "4%\r",
      "5%\r",
      "5%\r",
      "6%\r",
      "6%\r",
      "7%\r",
      "7%\r",
      "8%\r",
      "8%\r",
      "9%\r",
      "9%\r",
      "10%\r",
      "10%\r",
      "11%\r",
      "11%\r",
      "12%\r",
      "12%\r",
      "13%\r",
      "13%\r",
      "14%\r",
      "14%\r",
      "15%\r",
      "15%\r",
      "16%\r",
      "16%\r",
      "17%\r",
      "17%\r",
      "18%\r",
      "18%\r",
      "19%\r",
      "19%\r",
      "20%\r",
      "20%\r",
      "21%\r",
      "21%\r",
      "22%\r",
      "22%\r",
      "23%\r",
      "23%\r",
      "24%\r",
      "24%\r",
      "25%\r",
      "25%\r",
      "26%\r",
      "26%\r",
      "27%\r",
      "27%\r",
      "28%\r",
      "28%\r",
      "29%\r",
      "29%\r",
      "30%\r",
      "30%\r",
      "31%\r",
      "31%\r",
      "32%\r",
      "32%\r",
      "33%\r",
      "33%\r",
      "34%\r",
      "34%\r",
      "35%\r",
      "35%\r",
      "36%\r",
      "36%\r",
      "37%\r",
      "37%\r",
      "38%\r",
      "38%\r",
      "39%\r",
      "39%\r",
      "40%\r",
      "40%\r",
      "41%\r",
      "41%\r",
      "42%\r",
      "42%\r",
      "43%\r",
      "43%\r",
      "44%\r",
      "44%\r",
      "45%\r",
      "45%\r",
      "46%\r",
      "46%\r",
      "47%\r",
      "47%\r",
      "48%\r",
      "48%\r",
      "49%\r",
      "49%\r",
      "50%\r",
      "50%\r",
      "51%\r",
      "51%\r",
      "52%\r",
      "52%\r",
      "53%\r",
      "53%\r",
      "54%\r",
      "54%\r",
      "55%\r",
      "55%\r",
      "56%\r",
      "56%\r",
      "57%\r",
      "57%\r",
      "58%\r",
      "58%\r",
      "59%\r",
      "59%\r",
      "60%\r",
      "60%\r",
      "61%\r",
      "61%\r",
      "62%\r",
      "62%\r",
      "63%\r",
      "63%\r",
      "64%\r",
      "64%\r",
      "65%\r",
      "65%\r",
      "66%\r",
      "66%\r",
      "67%\r",
      "67%\r",
      "68%\r",
      "68%\r",
      "69%\r",
      "69%\r",
      "70%\r",
      "70%\r",
      "71%\r",
      "71%\r",
      "72%\r",
      "72%\r",
      "73%\r",
      "73%\r",
      "74%\r",
      "74%\r",
      "75%\r",
      "75%\r",
      "76%\r",
      "76%\r",
      "77%\r",
      "77%\r",
      "78%\r",
      "78%\r",
      "79%\r",
      "79%\r",
      "80%\r",
      "80%\r",
      "81%\r",
      "81%\r",
      "82%\r",
      "82%\r",
      "83%\r",
      "83%\r",
      "84%\r",
      "84%\r",
      "85%\r",
      "85%\r",
      "86%\r",
      "86%\r",
      "87%\r",
      "87%\r",
      "88%\r",
      "88%\r",
      "89%\r",
      "89%\r",
      "90%\r",
      "90%\r",
      "91%\r",
      "91%\r",
      "92%\r",
      "92%\r",
      "93%\r",
      "93%\r",
      "94%\r",
      "94%\r",
      "95%\r",
      "95%\r",
      "96%\r",
      "96%\r",
      "97%\r",
      "97%\r",
      "98%\r",
      "98%\r",
      "99%\r",
      "Pruned 22160 variants from chromosome 31, leaving 2521.\n",
      "\r",
      "0%\r",
      "1%\r",
      "1%\r",
      "2%\r",
      "2%\r",
      "3%\r",
      "3%\r",
      "4%\r",
      "4%\r",
      "5%\r",
      "5%\r",
      "6%\r",
      "6%\r",
      "7%\r",
      "7%\r",
      "8%\r",
      "8%\r",
      "9%\r",
      "9%\r",
      "10%\r",
      "10%\r",
      "11%\r",
      "11%\r",
      "12%\r",
      "12%\r",
      "13%\r",
      "13%\r",
      "14%\r",
      "14%\r",
      "15%\r",
      "15%\r",
      "16%\r",
      "16%\r",
      "17%\r",
      "17%\r",
      "18%\r",
      "18%\r",
      "19%\r",
      "19%\r",
      "20%\r",
      "20%\r",
      "21%\r",
      "21%\r",
      "22%\r",
      "22%\r",
      "23%\r",
      "23%\r",
      "24%\r",
      "24%\r",
      "25%\r",
      "25%\r",
      "26%\r",
      "26%\r",
      "27%\r",
      "27%\r",
      "28%\r",
      "28%\r",
      "29%\r",
      "29%\r",
      "30%\r",
      "30%\r",
      "31%\r",
      "31%\r",
      "32%\r",
      "32%\r",
      "33%\r",
      "33%\r",
      "34%\r",
      "34%\r",
      "35%\r",
      "35%\r",
      "36%\r",
      "36%\r",
      "37%\r",
      "37%\r",
      "38%\r",
      "38%\r",
      "39%\r",
      "39%\r",
      "40%\r",
      "40%\r",
      "41%\r",
      "41%\r",
      "42%\r",
      "42%\r",
      "43%\r",
      "43%\r",
      "44%\r",
      "44%\r",
      "45%\r",
      "45%\r",
      "46%\r",
      "46%\r",
      "47%\r",
      "47%\r",
      "48%\r",
      "48%\r",
      "49%\r",
      "49%\r",
      "50%\r",
      "50%\r",
      "51%\r",
      "51%\r",
      "52%\r",
      "52%\r",
      "53%\r",
      "53%\r",
      "54%\r",
      "54%\r",
      "55%\r",
      "55%\r",
      "56%\r",
      "56%\r",
      "57%\r",
      "57%\r",
      "58%\r",
      "58%\r",
      "59%\r",
      "59%\r",
      "60%\r",
      "60%\r",
      "61%\r",
      "61%\r",
      "62%\r",
      "62%\r",
      "63%\r",
      "63%\r",
      "64%\r",
      "64%\r",
      "65%\r",
      "65%\r",
      "66%\r",
      "66%\r",
      "67%\r",
      "67%\r",
      "68%\r",
      "68%\r",
      "69%\r",
      "69%\r",
      "70%\r",
      "70%\r",
      "71%\r",
      "71%\r",
      "72%\r",
      "72%\r",
      "73%\r",
      "73%\r",
      "74%\r",
      "74%\r",
      "75%\r",
      "75%\r",
      "76%\r",
      "76%\r",
      "77%\r",
      "77%\r",
      "78%\r",
      "78%\r",
      "79%\r",
      "79%\r",
      "80%\r",
      "80%\r",
      "81%\r",
      "81%\r",
      "82%\r",
      "82%\r",
      "83%\r",
      "83%\r",
      "84%\r",
      "84%\r",
      "85%\r",
      "85%\r",
      "86%\r",
      "86%\r",
      "87%\r",
      "87%\r",
      "88%\r",
      "88%\r",
      "89%\r",
      "89%\r",
      "90%\r",
      "90%\r",
      "91%\r",
      "91%\r",
      "92%\r",
      "92%\r",
      "93%\r",
      "93%\r",
      "94%\r",
      "94%\r",
      "95%\r",
      "95%\r",
      "96%\r",
      "96%\r",
      "97%\r",
      "97%\r",
      "98%\r",
      "98%\r",
      "99%\r",
      "Pruned 28774 variants from chromosome 32, leaving 3103.\n",
      "\r",
      "0%\r",
      "1%\r",
      "1%\r",
      "2%\r",
      "2%\r",
      "3%\r",
      "3%\r",
      "4%\r",
      "5%\r",
      "5%\r",
      "6%\r",
      "6%\r",
      "7%\r",
      "7%\r",
      "8%\r",
      "8%\r",
      "9%\r",
      "10%\r",
      "10%\r",
      "11%\r",
      "11%\r",
      "12%\r",
      "12%\r",
      "13%\r",
      "13%\r",
      "14%\r",
      "15%\r",
      "15%\r",
      "16%\r",
      "16%\r",
      "17%\r",
      "17%\r",
      "18%\r",
      "18%\r",
      "19%\r",
      "20%\r",
      "20%\r",
      "21%\r",
      "21%\r",
      "22%\r",
      "22%\r",
      "23%\r",
      "23%\r",
      "24%\r",
      "25%\r",
      "25%\r",
      "26%\r",
      "26%\r",
      "27%\r",
      "27%\r",
      "28%\r",
      "28%\r",
      "29%\r",
      "30%\r",
      "30%\r",
      "31%\r",
      "31%\r",
      "32%\r",
      "32%\r",
      "33%\r",
      "33%\r",
      "34%\r",
      "35%\r",
      "35%\r",
      "36%\r",
      "36%\r",
      "37%\r",
      "37%\r",
      "38%\r",
      "38%\r",
      "39%\r",
      "40%\r",
      "40%\r",
      "41%\r",
      "41%\r",
      "42%\r",
      "42%\r",
      "43%\r",
      "43%\r",
      "44%\r",
      "45%\r",
      "45%\r",
      "46%\r",
      "46%\r",
      "47%\r",
      "47%\r",
      "48%\r",
      "48%\r",
      "49%\r",
      "50%\r",
      "50%\r",
      "51%\r",
      "51%\r",
      "52%\r",
      "52%\r",
      "53%\r",
      "53%\r",
      "54%\r",
      "55%\r",
      "55%\r",
      "56%\r",
      "56%\r",
      "57%\r",
      "57%\r",
      "58%\r",
      "58%\r",
      "59%\r",
      "60%\r",
      "60%\r",
      "61%\r",
      "61%\r",
      "62%\r",
      "62%\r",
      "63%\r",
      "63%\r",
      "64%\r",
      "65%\r",
      "65%\r",
      "66%\r",
      "66%\r",
      "67%\r",
      "67%\r",
      "68%\r",
      "68%\r",
      "69%\r",
      "70%\r",
      "70%\r",
      "71%\r",
      "71%\r",
      "72%\r",
      "72%\r",
      "73%\r",
      "73%\r",
      "74%\r",
      "75%\r",
      "75%\r",
      "76%\r",
      "76%\r",
      "77%\r",
      "77%\r",
      "78%\r",
      "78%\r",
      "79%\r",
      "80%\r",
      "80%\r",
      "81%\r",
      "81%\r",
      "82%\r",
      "82%\r",
      "83%\r",
      "83%\r",
      "84%\r",
      "85%\r",
      "85%\r",
      "86%\r",
      "86%\r",
      "87%\r",
      "87%\r",
      "88%\r",
      "88%\r",
      "89%\r",
      "90%\r",
      "90%\r",
      "91%\r",
      "91%\r",
      "92%\r",
      "92%\r",
      "93%\r",
      "93%\r",
      "94%\r",
      "95%\r",
      "95%\r",
      "96%\r",
      "96%\r",
      "97%\r",
      "97%\r",
      "98%\r",
      "98%\r",
      "99%\r",
      "Pruned 19084 variants from chromosome 33, leaving 2256.\n",
      "Pruning complete.  436884 of 475081 variants removed.\n",
      "Writing...\r",
      "Marker lists written to\n",
      "results/variant_filtration/smv7_ex_autosomes_ld.prune.in and\n",
      "results/variant_filtration/smv7_ex_autosomes_ld.prune.out .\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_autosomes.vcf\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--exclude results/variant_filtration/smv7_ex_autosomes_ld.prune.out\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "After filtering, kept 156 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 38197 out of a possible 475081 Sites\n",
      "Run Time = 43.00 seconds\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "#find SNPs in LD\n",
    "plink \\\n",
    "    --vcf results/variant_filtration/smv7_ex_autosomes.vcf \\\n",
    "    --double-id \\\n",
    "    --allow-extra-chr \\\n",
    "    --indep-pairwise 250kb 1 0.20 \\\n",
    "    --out results/variant_filtration/smv7_ex_autosomes_ld\n",
    "\n",
    "#extract SNPs in LD\n",
    "vcftools \\\n",
    "    --vcf results/variant_filtration/smv7_ex_autosomes.vcf \\\n",
    "    --exclude results/variant_filtration/smv7_ex_autosomes_ld.prune.out \\\n",
    "    --recode \\\n",
    "    --recode-INFO-all \\\n",
    "    --stdout \\\n",
    "    >results/variant_filtration/smv7_ex_autosomes_ld.vcf"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
