{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The rpy2.ipython extension is already loaded. To reload it, use:\n",
      "  %reload_ext rpy2.ipython\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import shutil\n",
    "import vcf\n",
    "import re\n",
    "import gzip\n",
    "import pandas as pd\n",
    "\n",
    "from IPython.display import Image\n",
    "from Bio import SeqIO\n",
    "\n",
    "\n",
    "import rpy2.ipython\n",
    "\n",
    "\n",
    "%load_ext rpy2.ipython\n",
    "\n",
    "os.chdir(\"/master/nplatt/sch_man_nwinvasion\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Linkage Disequilibrium"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [],
   "source": [
    "#make lists for each population\n",
    "if not os.path.isdir(\"results/ld\"):\n",
    "    os.mkdir(\"results/ld\")\n",
    "\n",
    "ld_pops=[\"east_africa\", \"west_africa\", \"new_world\"]\n",
    "\n",
    "for pop in ld_pops:\n",
    "    list_file=\"results/ld/\" + pop + \".list\"\n",
    "    with open(list_file, 'w') as filehandle:  \n",
    "        filehandle.writelines(\"%s\\n\" % sample for sample in pop_ids[pop])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Note: --recode12 flag deprecated.  Use 'recode 12 ...'.\n",
      "Logging to results/ld/smv7_ex_autosomes_east_africa_SM_V7_1.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --out results/ld/smv7_ex_autosomes_east_africa_SM_V7_1\n",
      "  --recode 12\n",
      "  --threads 6\n",
      "  --vcf results/ld/smv7_ex_autosomes_east_africa_SM_V7_1.vcf\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "\r",
      "--vcf: 1k variants complete.\r",
      "--vcf: 2k variants complete.\r",
      "--vcf: 3k variants complete.\r",
      "--vcf: 4k variants complete.\r",
      "--vcf: 5k variants complete.\r",
      "--vcf: 6k variants complete.\r",
      "--vcf: 7k variants complete.\r",
      "--vcf: 8k variants complete.\r",
      "--vcf: 9k variants complete.\r",
      "--vcf: 10k variants complete.\r",
      "--vcf: 11k variants complete.\r",
      "--vcf: 12k variants complete.\r",
      "--vcf: 13k variants complete.\r",
      "--vcf: 14k variants complete.\r",
      "--vcf: 15k variants complete.\r",
      "--vcf: 16k variants complete.\r",
      "--vcf: 17k variants complete.\r",
      "--vcf: 18k variants complete.\r",
      "--vcf: 19k variants complete.\r",
      "--vcf: 20k variants complete.\r",
      "--vcf: 21k variants complete.\r",
      "--vcf: 22k variants complete.\r",
      "--vcf: 23k variants complete.\r",
      "--vcf: 24k variants complete.\r",
      "--vcf: 25k variants complete.\r",
      "--vcf: 26k variants complete.\r",
      "--vcf: 27k variants complete.\r",
      "--vcf: 28k variants complete.\r",
      "--vcf: 29k variants complete.\r",
      "--vcf: 30k variants complete.\r",
      "--vcf: 31k variants complete.\r",
      "--vcf: 32k variants complete.\r",
      "--vcf: 33k variants complete.\r",
      "--vcf: 34k variants complete.\r",
      "--vcf: 35k variants complete.\r",
      "--vcf: 36k variants complete.\r",
      "--vcf: 37k variants complete.\r",
      "--vcf: 38k variants complete.\r",
      "--vcf: 39k variants complete.\r",
      "--vcf: 40k variants complete.\r",
      "--vcf: 41k variants complete.\r",
      "--vcf: 42k variants complete.\r",
      "--vcf: 43k variants complete.\r",
      "--vcf: 44k variants complete.\r",
      "--vcf: 45k variants complete.\r",
      "--vcf: 46k variants complete.\r",
      "--vcf: 47k variants complete.\r",
      "--vcf: 48k variants complete.\r",
      "--vcf: 49k variants complete.\r",
      "--vcf: 50k variants complete.\r",
      "--vcf: 51k variants complete.\r",
      "--vcf: 52k variants complete.\r",
      "--vcf: 53k variants complete.\r",
      "--vcf: 54k variants complete.\r",
      "--vcf: 55k variants complete.\r",
      "--vcf: 56k variants complete.\r",
      "--vcf: 57k variants complete.\r",
      "--vcf: 58k variants complete.\r",
      "--vcf: 59k variants complete.\r",
      "--vcf: 60k variants complete.\r",
      "--vcf: 61k variants complete.\r",
      "--vcf: 62k variants complete.\r",
      "--vcf: 63k variants complete.\r",
      "--vcf: 64k variants complete.\r",
      "--vcf: 65k variants complete.\r",
      "--vcf: 66k variants complete.\r",
      "--vcf: 67k variants complete.\r",
      "--vcf: 68k variants complete.\r",
      "--vcf: 69k variants complete.\r",
      "--vcf: 70k variants complete.\r",
      "--vcf: 71k variants complete.\r",
      "--vcf: 72k variants complete.\r",
      "--vcf: 73k variants complete.\r",
      "--vcf: 74k variants complete.\r",
      "--vcf: 75k variants complete.\r",
      "--vcf: 76k variants complete.\r",
      "--vcf: 77k variants complete.\r",
      "--vcf: 78k variants complete.\r",
      "--vcf: 79k variants complete.\r",
      "--vcf: 80k variants complete.\r",
      "--vcf: 81k variants complete.\r",
      "--vcf: 82k variants complete.\r",
      "--vcf: 83k variants complete.\r",
      "--vcf: 84k variants complete.\r",
      "--vcf: 85k variants complete.\r",
      "--vcf: 86k variants complete.\r",
      "--vcf: 87k variants complete.\r",
      "--vcf: 88k variants complete.\r",
      "--vcf: 89k variants complete.\r",
      "--vcf: 90k variants complete.\r",
      "--vcf: 91k variants complete.\r",
      "--vcf: 92k variants complete.\r",
      "--vcf: 93k variants complete.\r",
      "--vcf: 94k variants complete.\r",
      "--vcf: 95k variants complete.\r",
      "--vcf: 96k variants complete.\r",
      "--vcf: 97k variants complete.\r",
      "--vcf: 98k variants complete.\r",
      "--vcf: 99k variants complete.\r",
      "--vcf: 100k variants complete.\r",
      "--vcf: 101k variants complete.\r",
      "--vcf: 102k variants complete.\r",
      "--vcf: 103k variants complete.\r",
      "--vcf: 104k variants complete.\r",
      "--vcf: 105k variants complete.\r",
      "--vcf: 106k variants complete.\r",
      "--vcf: 107k variants complete.\r",
      "--vcf: 108k variants complete.\r",
      "--vcf: 109k variants complete.\r",
      "--vcf: 110k variants complete.\r",
      "--vcf: 111k variants complete.\r",
      "--vcf: 112k variants complete.\r",
      "--vcf: 113k variants complete.\r",
      "--vcf: 114k variants complete.\r",
      "--vcf: 115k variants complete.\r",
      "--vcf: 116k variants complete.\r",
      "--vcf: 117k variants complete.\r",
      "--vcf: 118k variants complete.\r",
      "--vcf: 119k variants complete.\r",
      "--vcf: 120k variants complete.\r",
      "--vcf: 121k variants complete.\r",
      "--vcf: 122k variants complete.\r",
      "--vcf: 123k variants complete.\r",
      "--vcf: 124k variants complete.\r",
      "--vcf: 125k variants complete.\r",
      "--vcf: 126k variants complete.\r",
      "--vcf: 127k variants complete.\r",
      "--vcf: 128k variants complete.\r",
      "--vcf: 129k variants complete.\r",
      "--vcf: 130k variants complete.\r",
      "--vcf: 131k variants complete.\r",
      "--vcf: 132k variants complete.\r",
      "--vcf: 133k variants complete.\r",
      "--vcf: 134k variants complete.\r",
      "--vcf: 135k variants complete.\r",
      "--vcf: 136k variants complete.\r",
      "--vcf: 137k variants complete.\r",
      "--vcf: 138k variants complete.\r",
      "--vcf: 139k variants complete.\r",
      "--vcf: 140k variants complete.\r",
      "--vcf: 141k variants complete.\r",
      "--vcf: 142k variants complete.\r",
      "--vcf: 143k variants complete.\r",
      "--vcf: 144k variants complete.\r",
      "--vcf: 145k variants complete.\r",
      "--vcf: 146k variants complete.\r",
      "--vcf: 147k variants complete.\r",
      "--vcf: 148k variants complete.\r",
      "--vcf: 149k variants complete.\r",
      "--vcf: 150k variants complete.\r",
      "--vcf: 151k variants complete.\r",
      "--vcf: 152k variants complete.\r",
      "--vcf: 153k variants complete.\r",
      "--vcf: 154k variants complete.\r",
      "--vcf: 155k variants complete.\r",
      "--vcf: 156k variants complete.\r",
      "--vcf: 157k variants complete.\r",
      "--vcf: 158k variants complete.\r",
      "--vcf: 159k variants complete.\r",
      "--vcf: 160k variants complete.\r",
      "--vcf: 161k variants complete.\r",
      "--vcf: 162k variants complete.\r",
      "--vcf: results/ld/smv7_ex_autosomes_east_africa_SM_V7_1-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_1-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_1-temporary.fam written.\n",
      "162219 variants loaded from .bim file.\n",
      "58 people (0 males, 0 females, 58 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_1.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 58 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.862235.\n",
      "162219 variants and 58 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode ped to results/ld/smv7_ex_autosomes_east_africa_SM_V7_1.ped +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_1.map ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to results/ld/smv7_ex_autosomes_east_africa_SM_V7_1.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --file results/ld/smv7_ex_autosomes_east_africa_SM_V7_1\n",
      "  --ld-window 1000000\n",
      "  --ld-window-kb 90000\n",
      "  --ld-window-r2 0.0\n",
      "  --out results/ld/smv7_ex_autosomes_east_africa_SM_V7_1\n",
      "  --r2\n",
      "  --threads 6\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "Scanning .ped file... 0%\b\b1%\b\b3%\b\b5%\b\b6%\b\b8%\b\b10%\b\b\b12%\b\b\b13%\b\b\b15%\b\b\b17%\b\b\b18%\b\b\b20%\b\b\b22%\b\b\b24%\b\b\b25%\b\b\b27%\b\b\b29%\b\b\b31%\b\b\b32%\b\b\b34%\b\b\b36%\b\b\b37%\b\b\b39%\b\b\b41%\b\b\b43%\b\b\b44%\b\b\b46%\b\b\b48%\b\b\b49%\b\b\b51%\b\b\b53%\b\b\b55%\b\b\b56%\b\b\b58%\b\b\b60%\b\b\b62%\b\b\b63%\b\b\b65%\b\b\b67%\b\b\b68%\b\b\b70%\b\b\b72%\b\b\b74%\b\b\b75%\b\b\b77%\b\b\b79%\b\b\b81%\b\b\b82%\b\b\b84%\b\b\b86%\b\b\b87%\b\b\b89%\b\b\b91%\b\b\b93%\b\b\b94%\b\b\b96%\b\b\b98%\b\b\b100%\r",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (162219 variants, 58 people).\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\r",
      "--file: results/ld/smv7_ex_autosomes_east_africa_SM_V7_1-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_1-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_1-temporary.fam written.\n",
      "162219 variants loaded from .bim file.\n",
      "58 people (0 males, 0 females, 58 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_1.nosex .\n",
      "Using up to 6 threads (change this with --threads).\n",
      "Before main variant filters, 58 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.862235.\n",
      "162219 variants and 58 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--r2 to results/ld/smv7_ex_autosomes_east_africa_SM_V7_1.ld ...\n",
      "0% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b3% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b6% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b9% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b12% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b15% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b18% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b21% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b24% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b27% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b30% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b33% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b36% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b39% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b42% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b45% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b48% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b51% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b54% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b57% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b60% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b63% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b66% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b69% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b72% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b75% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b78% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b81% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b84% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b87% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b90% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b93% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b96% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b99% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Note: --recode12 flag deprecated.  Use 'recode 12 ...'.\n",
      "Logging to results/ld/smv7_ex_autosomes_east_africa_SM_V7_2.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --out results/ld/smv7_ex_autosomes_east_africa_SM_V7_2\n",
      "  --recode 12\n",
      "  --threads 6\n",
      "  --vcf results/ld/smv7_ex_autosomes_east_africa_SM_V7_2.vcf\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "\r",
      "--vcf: 1k variants complete.\r",
      "--vcf: 2k variants complete.\r",
      "--vcf: 3k variants complete.\r",
      "--vcf: 4k variants complete.\r",
      "--vcf: 5k variants complete.\r",
      "--vcf: 6k variants complete.\r",
      "--vcf: 7k variants complete.\r",
      "--vcf: 8k variants complete.\r",
      "--vcf: 9k variants complete.\r",
      "--vcf: 10k variants complete.\r",
      "--vcf: 11k variants complete.\r",
      "--vcf: 12k variants complete.\r",
      "--vcf: 13k variants complete.\r",
      "--vcf: 14k variants complete.\r",
      "--vcf: 15k variants complete.\r",
      "--vcf: 16k variants complete.\r",
      "--vcf: 17k variants complete.\r",
      "--vcf: 18k variants complete.\r",
      "--vcf: 19k variants complete.\r",
      "--vcf: 20k variants complete.\r",
      "--vcf: 21k variants complete.\r",
      "--vcf: 22k variants complete.\r",
      "--vcf: 23k variants complete.\r",
      "--vcf: 24k variants complete.\r",
      "--vcf: 25k variants complete.\r",
      "--vcf: 26k variants complete.\r",
      "--vcf: 27k variants complete.\r",
      "--vcf: 28k variants complete.\r",
      "--vcf: 29k variants complete.\r",
      "--vcf: 30k variants complete.\r",
      "--vcf: 31k variants complete.\r",
      "--vcf: 32k variants complete.\r",
      "--vcf: 33k variants complete.\r",
      "--vcf: 34k variants complete.\r",
      "--vcf: 35k variants complete.\r",
      "--vcf: 36k variants complete.\r",
      "--vcf: 37k variants complete.\r",
      "--vcf: 38k variants complete.\r",
      "--vcf: 39k variants complete.\r",
      "--vcf: 40k variants complete.\r",
      "--vcf: 41k variants complete.\r",
      "--vcf: 42k variants complete.\r",
      "--vcf: 43k variants complete.\r",
      "--vcf: 44k variants complete.\r",
      "--vcf: 45k variants complete.\r",
      "--vcf: 46k variants complete.\r",
      "--vcf: 47k variants complete.\r",
      "--vcf: 48k variants complete.\r",
      "--vcf: 49k variants complete.\r",
      "--vcf: 50k variants complete.\r",
      "--vcf: 51k variants complete.\r",
      "--vcf: 52k variants complete.\r",
      "--vcf: 53k variants complete.\r",
      "--vcf: 54k variants complete.\r",
      "--vcf: 55k variants complete.\r",
      "--vcf: 56k variants complete.\r",
      "--vcf: 57k variants complete.\r",
      "--vcf: 58k variants complete.\r",
      "--vcf: 59k variants complete.\r",
      "--vcf: 60k variants complete.\r",
      "--vcf: 61k variants complete.\r",
      "--vcf: 62k variants complete.\r",
      "--vcf: 63k variants complete.\r",
      "--vcf: 64k variants complete.\r",
      "--vcf: 65k variants complete.\r",
      "--vcf: 66k variants complete.\r",
      "--vcf: 67k variants complete.\r",
      "--vcf: 68k variants complete.\r",
      "--vcf: 69k variants complete.\r",
      "--vcf: 70k variants complete.\r",
      "--vcf: 71k variants complete.\r",
      "--vcf: 72k variants complete.\r",
      "--vcf: 73k variants complete.\r",
      "--vcf: 74k variants complete.\r",
      "--vcf: 75k variants complete.\r",
      "--vcf: 76k variants complete.\r",
      "--vcf: 77k variants complete.\r",
      "--vcf: 78k variants complete.\r",
      "--vcf: results/ld/smv7_ex_autosomes_east_africa_SM_V7_2-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_2-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_2-temporary.fam written.\n",
      "78902 variants loaded from .bim file.\n",
      "58 people (0 males, 0 females, 58 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_2.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 58 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.842064.\n",
      "78902 variants and 58 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode ped to results/ld/smv7_ex_autosomes_east_africa_SM_V7_2.ped +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_2.map ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to results/ld/smv7_ex_autosomes_east_africa_SM_V7_2.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --file results/ld/smv7_ex_autosomes_east_africa_SM_V7_2\n",
      "  --ld-window 1000000\n",
      "  --ld-window-kb 90000\n",
      "  --ld-window-r2 0.0\n",
      "  --out results/ld/smv7_ex_autosomes_east_africa_SM_V7_2\n",
      "  --r2\n",
      "  --threads 6\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "Scanning .ped file... 0%\b\b1%\b\b3%\b\b5%\b\b6%\b\b8%\b\b10%\b\b\b12%\b\b\b13%\b\b\b15%\b\b\b17%\b\b\b18%\b\b\b20%\b\b\b22%\b\b\b24%\b\b\b25%\b\b\b27%\b\b\b29%\b\b\b31%\b\b\b32%\b\b\b34%\b\b\b36%\b\b\b37%\b\b\b39%\b\b\b41%\b\b\b43%\b\b\b44%\b\b\b46%\b\b\b48%\b\b\b49%\b\b\b51%\b\b\b53%\b\b\b55%\b\b\b56%\b\b\b58%\b\b\b60%\b\b\b62%\b\b\b63%\b\b\b65%\b\b\b67%\b\b\b68%\b\b\b70%\b\b\b72%\b\b\b74%\b\b\b75%\b\b\b77%\b\b\b79%\b\b\b81%\b\b\b82%\b\b\b84%\b\b\b86%\b\b\b87%\b\b\b89%\b\b\b91%\b\b\b93%\b\b\b94%\b\b\b96%\b\b\b98%\b\b\b100%\r",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (78902 variants, 58 people).\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\r",
      "--file: results/ld/smv7_ex_autosomes_east_africa_SM_V7_2-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_2-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_2-temporary.fam written.\n",
      "78902 variants loaded from .bim file.\n",
      "58 people (0 males, 0 females, 58 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_2.nosex .\n",
      "Using up to 6 threads (change this with --threads).\n",
      "Before main variant filters, 58 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.842064.\n",
      "78902 variants and 58 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--r2 to results/ld/smv7_ex_autosomes_east_africa_SM_V7_2.ld ...\n",
      "0% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b12% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b25% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b38% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b50% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b63% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b76% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b88% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Note: --recode12 flag deprecated.  Use 'recode 12 ...'.\n",
      "Logging to results/ld/smv7_ex_autosomes_east_africa_SM_V7_3.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --out results/ld/smv7_ex_autosomes_east_africa_SM_V7_3\n",
      "  --recode 12\n",
      "  --threads 6\n",
      "  --vcf results/ld/smv7_ex_autosomes_east_africa_SM_V7_3.vcf\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "\r",
      "--vcf: 1k variants complete.\r",
      "--vcf: 2k variants complete.\r",
      "--vcf: 3k variants complete.\r",
      "--vcf: 4k variants complete.\r",
      "--vcf: 5k variants complete.\r",
      "--vcf: 6k variants complete.\r",
      "--vcf: 7k variants complete.\r",
      "--vcf: 8k variants complete.\r",
      "--vcf: 9k variants complete.\r",
      "--vcf: 10k variants complete.\r",
      "--vcf: 11k variants complete.\r",
      "--vcf: 12k variants complete.\r",
      "--vcf: 13k variants complete.\r",
      "--vcf: 14k variants complete.\r",
      "--vcf: 15k variants complete.\r",
      "--vcf: 16k variants complete.\r",
      "--vcf: 17k variants complete.\r",
      "--vcf: 18k variants complete.\r",
      "--vcf: 19k variants complete.\r",
      "--vcf: 20k variants complete.\r",
      "--vcf: 21k variants complete.\r",
      "--vcf: 22k variants complete.\r",
      "--vcf: 23k variants complete.\r",
      "--vcf: 24k variants complete.\r",
      "--vcf: 25k variants complete.\r",
      "--vcf: 26k variants complete.\r",
      "--vcf: 27k variants complete.\r",
      "--vcf: 28k variants complete.\r",
      "--vcf: 29k variants complete.\r",
      "--vcf: 30k variants complete.\r",
      "--vcf: 31k variants complete.\r",
      "--vcf: 32k variants complete.\r",
      "--vcf: 33k variants complete.\r",
      "--vcf: 34k variants complete.\r",
      "--vcf: 35k variants complete.\r",
      "--vcf: 36k variants complete.\r",
      "--vcf: 37k variants complete.\r",
      "--vcf: 38k variants complete.\r",
      "--vcf: 39k variants complete.\r",
      "--vcf: 40k variants complete.\r",
      "--vcf: 41k variants complete.\r",
      "--vcf: 42k variants complete.\r",
      "--vcf: 43k variants complete.\r",
      "--vcf: 44k variants complete.\r",
      "--vcf: 45k variants complete.\r",
      "--vcf: 46k variants complete.\r",
      "--vcf: 47k variants complete.\r",
      "--vcf: 48k variants complete.\r",
      "--vcf: 49k variants complete.\r",
      "--vcf: 50k variants complete.\r",
      "--vcf: 51k variants complete.\r",
      "--vcf: 52k variants complete.\r",
      "--vcf: 53k variants complete.\r",
      "--vcf: 54k variants complete.\r",
      "--vcf: 55k variants complete.\r",
      "--vcf: 56k variants complete.\r",
      "--vcf: 57k variants complete.\r",
      "--vcf: 58k variants complete.\r",
      "--vcf: 59k variants complete.\r",
      "--vcf: 60k variants complete.\r",
      "--vcf: 61k variants complete.\r",
      "--vcf: 62k variants complete.\r",
      "--vcf: 63k variants complete.\r",
      "--vcf: 64k variants complete.\r",
      "--vcf: 65k variants complete.\r",
      "--vcf: 66k variants complete.\r",
      "--vcf: 67k variants complete.\r",
      "--vcf: 68k variants complete.\r",
      "--vcf: 69k variants complete.\r",
      "--vcf: 70k variants complete.\r",
      "--vcf: 71k variants complete.\r",
      "--vcf: 72k variants complete.\r",
      "--vcf: 73k variants complete.\r",
      "--vcf: 74k variants complete.\r",
      "--vcf: 75k variants complete.\r",
      "--vcf: 76k variants complete.\r",
      "--vcf: 77k variants complete.\r",
      "--vcf: 78k variants complete.\r",
      "--vcf: 79k variants complete.\r",
      "--vcf: 80k variants complete.\r",
      "--vcf: results/ld/smv7_ex_autosomes_east_africa_SM_V7_3-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_3-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_3-temporary.fam written.\n",
      "80481 variants loaded from .bim file.\n",
      "58 people (0 males, 0 females, 58 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_3.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 58 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.8454.\n",
      "80481 variants and 58 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode ped to results/ld/smv7_ex_autosomes_east_africa_SM_V7_3.ped +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_3.map ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to results/ld/smv7_ex_autosomes_east_africa_SM_V7_3.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --file results/ld/smv7_ex_autosomes_east_africa_SM_V7_3\n",
      "  --ld-window 1000000\n",
      "  --ld-window-kb 90000\n",
      "  --ld-window-r2 0.0\n",
      "  --out results/ld/smv7_ex_autosomes_east_africa_SM_V7_3\n",
      "  --r2\n",
      "  --threads 6\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "Scanning .ped file... 0%\b\b1%\b\b3%\b\b5%\b\b6%\b\b8%\b\b10%\b\b\b12%\b\b\b13%\b\b\b15%\b\b\b17%\b\b\b18%\b\b\b20%\b\b\b22%\b\b\b24%\b\b\b25%\b\b\b27%\b\b\b29%\b\b\b31%\b\b\b32%\b\b\b34%\b\b\b36%\b\b\b37%\b\b\b39%\b\b\b41%\b\b\b43%\b\b\b44%\b\b\b46%\b\b\b48%\b\b\b49%\b\b\b51%\b\b\b53%\b\b\b55%\b\b\b56%\b\b\b58%\b\b\b60%\b\b\b62%\b\b\b63%\b\b\b65%\b\b\b67%\b\b\b68%\b\b\b70%\b\b\b72%\b\b\b74%\b\b\b75%\b\b\b77%\b\b\b79%\b\b\b81%\b\b\b82%\b\b\b84%\b\b\b86%\b\b\b87%\b\b\b89%\b\b\b91%\b\b\b93%\b\b\b94%\b\b\b96%\b\b\b98%\b\b\b100%\r",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (80481 variants, 58 people).\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\r",
      "--file: results/ld/smv7_ex_autosomes_east_africa_SM_V7_3-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_3-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_3-temporary.fam written.\n",
      "80481 variants loaded from .bim file.\n",
      "58 people (0 males, 0 females, 58 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_3.nosex .\n",
      "Using up to 6 threads (change this with --threads).\n",
      "Before main variant filters, 58 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.8454.\n",
      "80481 variants and 58 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--r2 to results/ld/smv7_ex_autosomes_east_africa_SM_V7_3.ld ...\n",
      "0% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b12% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b24% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b36% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b48% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b61% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b73% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b85% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b97% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Note: --recode12 flag deprecated.  Use 'recode 12 ...'.\n",
      "Logging to results/ld/smv7_ex_autosomes_east_africa_SM_V7_4.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --out results/ld/smv7_ex_autosomes_east_africa_SM_V7_4\n",
      "  --recode 12\n",
      "  --threads 6\n",
      "  --vcf results/ld/smv7_ex_autosomes_east_africa_SM_V7_4.vcf\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "\r",
      "--vcf: 1k variants complete.\r",
      "--vcf: 2k variants complete.\r",
      "--vcf: 3k variants complete.\r",
      "--vcf: 4k variants complete.\r",
      "--vcf: 5k variants complete.\r",
      "--vcf: 6k variants complete.\r",
      "--vcf: 7k variants complete.\r",
      "--vcf: 8k variants complete.\r",
      "--vcf: 9k variants complete.\r",
      "--vcf: 10k variants complete.\r",
      "--vcf: 11k variants complete.\r",
      "--vcf: 12k variants complete.\r",
      "--vcf: 13k variants complete.\r",
      "--vcf: 14k variants complete.\r",
      "--vcf: 15k variants complete.\r",
      "--vcf: 16k variants complete.\r",
      "--vcf: 17k variants complete.\r",
      "--vcf: 18k variants complete.\r",
      "--vcf: 19k variants complete.\r",
      "--vcf: 20k variants complete.\r",
      "--vcf: 21k variants complete.\r",
      "--vcf: 22k variants complete.\r",
      "--vcf: 23k variants complete.\r",
      "--vcf: 24k variants complete.\r",
      "--vcf: 25k variants complete.\r",
      "--vcf: 26k variants complete.\r",
      "--vcf: 27k variants complete.\r",
      "--vcf: 28k variants complete.\r",
      "--vcf: 29k variants complete.\r",
      "--vcf: 30k variants complete.\r",
      "--vcf: 31k variants complete.\r",
      "--vcf: 32k variants complete.\r",
      "--vcf: 33k variants complete.\r",
      "--vcf: 34k variants complete.\r",
      "--vcf: 35k variants complete.\r",
      "--vcf: 36k variants complete.\r",
      "--vcf: 37k variants complete.\r",
      "--vcf: 38k variants complete.\r",
      "--vcf: 39k variants complete.\r",
      "--vcf: 40k variants complete.\r",
      "--vcf: 41k variants complete.\r",
      "--vcf: 42k variants complete.\r",
      "--vcf: 43k variants complete.\r",
      "--vcf: 44k variants complete.\r",
      "--vcf: 45k variants complete.\r",
      "--vcf: 46k variants complete.\r",
      "--vcf: 47k variants complete.\r",
      "--vcf: 48k variants complete.\r",
      "--vcf: 49k variants complete.\r",
      "--vcf: 50k variants complete.\r",
      "--vcf: 51k variants complete.\r",
      "--vcf: 52k variants complete.\r",
      "--vcf: 53k variants complete.\r",
      "--vcf: 54k variants complete.\r",
      "--vcf: 55k variants complete.\r",
      "--vcf: 56k variants complete.\r",
      "--vcf: 57k variants complete.\r",
      "--vcf: 58k variants complete.\r",
      "--vcf: 59k variants complete.\r",
      "--vcf: 60k variants complete.\r",
      "--vcf: 61k variants complete.\r",
      "--vcf: 62k variants complete.\r",
      "--vcf: 63k variants complete.\r",
      "--vcf: 64k variants complete.\r",
      "--vcf: 65k variants complete.\r",
      "--vcf: 66k variants complete.\r",
      "--vcf: 67k variants complete.\r",
      "--vcf: 68k variants complete.\r",
      "--vcf: 69k variants complete.\r",
      "--vcf: 70k variants complete.\r",
      "--vcf: 71k variants complete.\r",
      "--vcf: 72k variants complete.\r",
      "--vcf: 73k variants complete.\r",
      "--vcf: 74k variants complete.\r",
      "--vcf: 75k variants complete.\r",
      "--vcf: results/ld/smv7_ex_autosomes_east_africa_SM_V7_4-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_4-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_4-temporary.fam written.\n",
      "75581 variants loaded from .bim file.\n",
      "58 people (0 males, 0 females, 58 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_4.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 58 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.845241.\n",
      "75581 variants and 58 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode ped to results/ld/smv7_ex_autosomes_east_africa_SM_V7_4.ped +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_4.map ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to results/ld/smv7_ex_autosomes_east_africa_SM_V7_4.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --file results/ld/smv7_ex_autosomes_east_africa_SM_V7_4\n",
      "  --ld-window 1000000\n",
      "  --ld-window-kb 90000\n",
      "  --ld-window-r2 0.0\n",
      "  --out results/ld/smv7_ex_autosomes_east_africa_SM_V7_4\n",
      "  --r2\n",
      "  --threads 6\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "Scanning .ped file... 0%\b\b1%\b\b3%\b\b5%\b\b6%\b\b8%\b\b10%\b\b\b12%\b\b\b13%\b\b\b15%\b\b\b17%\b\b\b18%\b\b\b20%\b\b\b22%\b\b\b24%\b\b\b25%\b\b\b27%\b\b\b29%\b\b\b31%\b\b\b32%\b\b\b34%\b\b\b36%\b\b\b37%\b\b\b39%\b\b\b41%\b\b\b43%\b\b\b44%\b\b\b46%\b\b\b48%\b\b\b49%\b\b\b51%\b\b\b53%\b\b\b55%\b\b\b56%\b\b\b58%\b\b\b60%\b\b\b62%\b\b\b63%\b\b\b65%\b\b\b67%\b\b\b68%\b\b\b70%\b\b\b72%\b\b\b74%\b\b\b75%\b\b\b77%\b\b\b79%\b\b\b81%\b\b\b82%\b\b\b84%\b\b\b86%\b\b\b87%\b\b\b89%\b\b\b91%\b\b\b93%\b\b\b94%\b\b\b96%\b\b\b98%\b\b\b100%\r",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (75581 variants, 58 people).\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\r",
      "--file: results/ld/smv7_ex_autosomes_east_africa_SM_V7_4-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_4-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_4-temporary.fam written.\n",
      "75581 variants loaded from .bim file.\n",
      "58 people (0 males, 0 females, 58 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_4.nosex .\n",
      "Using up to 6 threads (change this with --threads).\n",
      "Before main variant filters, 58 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.845241.\n",
      "75581 variants and 58 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--r2 to results/ld/smv7_ex_autosomes_east_africa_SM_V7_4.ld ...\n",
      "0% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b13% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b27% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b41% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b55% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b69% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b83% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b96% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Note: --recode12 flag deprecated.  Use 'recode 12 ...'.\n",
      "Logging to results/ld/smv7_ex_autosomes_east_africa_SM_V7_5.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --out results/ld/smv7_ex_autosomes_east_africa_SM_V7_5\n",
      "  --recode 12\n",
      "  --threads 6\n",
      "  --vcf results/ld/smv7_ex_autosomes_east_africa_SM_V7_5.vcf\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "\r",
      "--vcf: 1k variants complete.\r",
      "--vcf: 2k variants complete.\r",
      "--vcf: 3k variants complete.\r",
      "--vcf: 4k variants complete.\r",
      "--vcf: 5k variants complete.\r",
      "--vcf: 6k variants complete.\r",
      "--vcf: 7k variants complete.\r",
      "--vcf: 8k variants complete.\r",
      "--vcf: 9k variants complete.\r",
      "--vcf: 10k variants complete.\r",
      "--vcf: 11k variants complete.\r",
      "--vcf: 12k variants complete.\r",
      "--vcf: 13k variants complete.\r",
      "--vcf: 14k variants complete.\r",
      "--vcf: 15k variants complete.\r",
      "--vcf: 16k variants complete.\r",
      "--vcf: 17k variants complete.\r",
      "--vcf: 18k variants complete.\r",
      "--vcf: 19k variants complete.\r",
      "--vcf: 20k variants complete.\r",
      "--vcf: 21k variants complete.\r",
      "--vcf: 22k variants complete.\r",
      "--vcf: 23k variants complete.\r",
      "--vcf: 24k variants complete.\r",
      "--vcf: results/ld/smv7_ex_autosomes_east_africa_SM_V7_5-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_5-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_5-temporary.fam written.\n",
      "24681 variants loaded from .bim file.\n",
      "58 people (0 males, 0 females, 58 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_5.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 58 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.817355.\n",
      "24681 variants and 58 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode ped to results/ld/smv7_ex_autosomes_east_africa_SM_V7_5.ped +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_5.map ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to results/ld/smv7_ex_autosomes_east_africa_SM_V7_5.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --file results/ld/smv7_ex_autosomes_east_africa_SM_V7_5\n",
      "  --ld-window 1000000\n",
      "  --ld-window-kb 90000\n",
      "  --ld-window-r2 0.0\n",
      "  --out results/ld/smv7_ex_autosomes_east_africa_SM_V7_5\n",
      "  --r2\n",
      "  --threads 6\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "Scanning .ped file... 0%\b\b1%\b\b3%\b\b5%\b\b6%\b\b8%\b\b10%\b\b\b12%\b\b\b13%\b\b\b15%\b\b\b17%\b\b\b18%\b\b\b20%\b\b\b22%\b\b\b24%\b\b\b25%\b\b\b27%\b\b\b29%\b\b\b31%\b\b\b32%\b\b\b34%\b\b\b36%\b\b\b37%\b\b\b39%\b\b\b41%\b\b\b43%\b\b\b44%\b\b\b46%\b\b\b48%\b\b\b49%\b\b\b51%\b\b\b53%\b\b\b55%\b\b\b56%\b\b\b58%\b\b\b60%\b\b\b62%\b\b\b63%\b\b\b65%\b\b\b67%\b\b\b68%\b\b\b70%\b\b\b72%\b\b\b74%\b\b\b75%\b\b\b77%\b\b\b79%\b\b\b81%\b\b\b82%\b\b\b84%\b\b\b86%\b\b\b87%\b\b\b89%\b\b\b91%\b\b\b93%\b\b\b94%\b\b\b96%\b\b\b98%\b\b\b100%\r",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (24681 variants, 58 people).\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\r",
      "--file: results/ld/smv7_ex_autosomes_east_africa_SM_V7_5-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_5-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_5-temporary.fam written.\n",
      "24681 variants loaded from .bim file.\n",
      "58 people (0 males, 0 females, 58 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_5.nosex .\n",
      "Using up to 6 threads (change this with --threads).\n",
      "Before main variant filters, 58 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.817355.\n",
      "24681 variants and 58 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--r2 to results/ld/smv7_ex_autosomes_east_africa_SM_V7_5.ld ...\n",
      "0% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Note: --recode12 flag deprecated.  Use 'recode 12 ...'.\n",
      "Logging to results/ld/smv7_ex_autosomes_east_africa_SM_V7_6.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --out results/ld/smv7_ex_autosomes_east_africa_SM_V7_6\n",
      "  --recode 12\n",
      "  --threads 6\n",
      "  --vcf results/ld/smv7_ex_autosomes_east_africa_SM_V7_6.vcf\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "\r",
      "--vcf: 1k variants complete.\r",
      "--vcf: 2k variants complete.\r",
      "--vcf: 3k variants complete.\r",
      "--vcf: 4k variants complete.\r",
      "--vcf: 5k variants complete.\r",
      "--vcf: 6k variants complete.\r",
      "--vcf: 7k variants complete.\r",
      "--vcf: 8k variants complete.\r",
      "--vcf: 9k variants complete.\r",
      "--vcf: 10k variants complete.\r",
      "--vcf: 11k variants complete.\r",
      "--vcf: 12k variants complete.\r",
      "--vcf: 13k variants complete.\r",
      "--vcf: 14k variants complete.\r",
      "--vcf: 15k variants complete.\r",
      "--vcf: 16k variants complete.\r",
      "--vcf: 17k variants complete.\r",
      "--vcf: 18k variants complete.\r",
      "--vcf: 19k variants complete.\r",
      "--vcf: 20k variants complete.\r",
      "--vcf: 21k variants complete.\r",
      "--vcf: 22k variants complete.\r",
      "--vcf: 23k variants complete.\r",
      "--vcf: 24k variants complete.\r",
      "--vcf: 25k variants complete.\r",
      "--vcf: 26k variants complete.\r",
      "--vcf: 27k variants complete.\r",
      "--vcf: 28k variants complete.\r",
      "--vcf: 29k variants complete.\r",
      "--vcf: 30k variants complete.\r",
      "--vcf: 31k variants complete.\r",
      "--vcf: results/ld/smv7_ex_autosomes_east_africa_SM_V7_6-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_6-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_6-temporary.fam written.\n",
      "31877 variants loaded from .bim file.\n",
      "58 people (0 males, 0 females, 58 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_6.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 58 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.825326.\n",
      "31877 variants and 58 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode ped to results/ld/smv7_ex_autosomes_east_africa_SM_V7_6.ped +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_6.map ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to results/ld/smv7_ex_autosomes_east_africa_SM_V7_6.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --file results/ld/smv7_ex_autosomes_east_africa_SM_V7_6\n",
      "  --ld-window 1000000\n",
      "  --ld-window-kb 90000\n",
      "  --ld-window-r2 0.0\n",
      "  --out results/ld/smv7_ex_autosomes_east_africa_SM_V7_6\n",
      "  --r2\n",
      "  --threads 6\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "Scanning .ped file... 0%\b\b1%\b\b3%\b\b5%\b\b6%\b\b8%\b\b10%\b\b\b12%\b\b\b13%\b\b\b15%\b\b\b17%\b\b\b18%\b\b\b20%\b\b\b22%\b\b\b24%\b\b\b25%\b\b\b27%\b\b\b29%\b\b\b31%\b\b\b32%\b\b\b34%\b\b\b36%\b\b\b37%\b\b\b39%\b\b\b41%\b\b\b43%\b\b\b44%\b\b\b46%\b\b\b48%\b\b\b49%\b\b\b51%\b\b\b53%\b\b\b55%\b\b\b56%\b\b\b58%\b\b\b60%\b\b\b62%\b\b\b63%\b\b\b65%\b\b\b67%\b\b\b68%\b\b\b70%\b\b\b72%\b\b\b74%\b\b\b75%\b\b\b77%\b\b\b79%\b\b\b81%\b\b\b82%\b\b\b84%\b\b\b86%\b\b\b87%\b\b\b89%\b\b\b91%\b\b\b93%\b\b\b94%\b\b\b96%\b\b\b98%\b\b\b100%\r",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (31877 variants, 58 people).\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\r",
      "--file: results/ld/smv7_ex_autosomes_east_africa_SM_V7_6-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_6-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_6-temporary.fam written.\n",
      "31877 variants loaded from .bim file.\n",
      "58 people (0 males, 0 females, 58 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_6.nosex .\n",
      "Using up to 6 threads (change this with --threads).\n",
      "Before main variant filters, 58 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.825326.\n",
      "31877 variants and 58 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--r2 to results/ld/smv7_ex_autosomes_east_africa_SM_V7_6.ld ...\n",
      "0% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b77% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Note: --recode12 flag deprecated.  Use 'recode 12 ...'.\n",
      "Logging to results/ld/smv7_ex_autosomes_east_africa_SM_V7_7.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --out results/ld/smv7_ex_autosomes_east_africa_SM_V7_7\n",
      "  --recode 12\n",
      "  --threads 6\n",
      "  --vcf results/ld/smv7_ex_autosomes_east_africa_SM_V7_7.vcf\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "\r",
      "--vcf: 1k variants complete.\r",
      "--vcf: 2k variants complete.\r",
      "--vcf: 3k variants complete.\r",
      "--vcf: 4k variants complete.\r",
      "--vcf: 5k variants complete.\r",
      "--vcf: 6k variants complete.\r",
      "--vcf: 7k variants complete.\r",
      "--vcf: 8k variants complete.\r",
      "--vcf: 9k variants complete.\r",
      "--vcf: 10k variants complete.\r",
      "--vcf: 11k variants complete.\r",
      "--vcf: 12k variants complete.\r",
      "--vcf: 13k variants complete.\r",
      "--vcf: 14k variants complete.\r",
      "--vcf: 15k variants complete.\r",
      "--vcf: 16k variants complete.\r",
      "--vcf: 17k variants complete.\r",
      "--vcf: 18k variants complete.\r",
      "--vcf: 19k variants complete.\r",
      "--vcf: 20k variants complete.\r",
      "--vcf: 21k variants complete.\r",
      "--vcf: results/ld/smv7_ex_autosomes_east_africa_SM_V7_7-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_7-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_7-temporary.fam written.\n",
      "21340 variants loaded from .bim file.\n",
      "58 people (0 males, 0 females, 58 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_7.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 58 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.827234.\n",
      "21340 variants and 58 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode ped to results/ld/smv7_ex_autosomes_east_africa_SM_V7_7.ped +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_7.map ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to results/ld/smv7_ex_autosomes_east_africa_SM_V7_7.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --file results/ld/smv7_ex_autosomes_east_africa_SM_V7_7\n",
      "  --ld-window 1000000\n",
      "  --ld-window-kb 90000\n",
      "  --ld-window-r2 0.0\n",
      "  --out results/ld/smv7_ex_autosomes_east_africa_SM_V7_7\n",
      "  --r2\n",
      "  --threads 6\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "Scanning .ped file... 0%\b\b1%\b\b3%\b\b5%\b\b6%\b\b8%\b\b10%\b\b\b12%\b\b\b13%\b\b\b15%\b\b\b17%\b\b\b18%\b\b\b20%\b\b\b22%\b\b\b24%\b\b\b25%\b\b\b27%\b\b\b29%\b\b\b31%\b\b\b32%\b\b\b34%\b\b\b36%\b\b\b37%\b\b\b39%\b\b\b41%\b\b\b43%\b\b\b44%\b\b\b46%\b\b\b48%\b\b\b49%\b\b\b51%\b\b\b53%\b\b\b55%\b\b\b56%\b\b\b58%\b\b\b60%\b\b\b62%\b\b\b63%\b\b\b65%\b\b\b67%\b\b\b68%\b\b\b70%\b\b\b72%\b\b\b74%\b\b\b75%\b\b\b77%\b\b\b79%\b\b\b81%\b\b\b82%\b\b\b84%\b\b\b86%\b\b\b87%\b\b\b89%\b\b\b91%\b\b\b93%\b\b\b94%\b\b\b96%\b\b\b98%\b\b\b100%\r",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (21340 variants, 58 people).\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\r",
      "--file: results/ld/smv7_ex_autosomes_east_africa_SM_V7_7-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_7-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_7-temporary.fam written.\n",
      "21340 variants loaded from .bim file.\n",
      "58 people (0 males, 0 females, 58 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_east_africa_SM_V7_7.nosex .\n",
      "Using up to 6 threads (change this with --threads).\n",
      "Before main variant filters, 58 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.827234.\n",
      "21340 variants and 58 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--r2 to results/ld/smv7_ex_autosomes_east_africa_SM_V7_7.ld ...\n",
      "0% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Note: --recode12 flag deprecated.  Use 'recode 12 ...'.\n",
      "Logging to results/ld/smv7_ex_autosomes_west_africa_SM_V7_1.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --out results/ld/smv7_ex_autosomes_west_africa_SM_V7_1\n",
      "  --recode 12\n",
      "  --threads 6\n",
      "  --vcf results/ld/smv7_ex_autosomes_west_africa_SM_V7_1.vcf\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "\r",
      "--vcf: 1k variants complete.\r",
      "--vcf: 2k variants complete.\r",
      "--vcf: 3k variants complete.\r",
      "--vcf: 4k variants complete.\r",
      "--vcf: 5k variants complete.\r",
      "--vcf: 6k variants complete.\r",
      "--vcf: 7k variants complete.\r",
      "--vcf: 8k variants complete.\r",
      "--vcf: 9k variants complete.\r",
      "--vcf: 10k variants complete.\r",
      "--vcf: 11k variants complete.\r",
      "--vcf: 12k variants complete.\r",
      "--vcf: 13k variants complete.\r",
      "--vcf: 14k variants complete.\r",
      "--vcf: 15k variants complete.\r",
      "--vcf: 16k variants complete.\r",
      "--vcf: 17k variants complete.\r",
      "--vcf: 18k variants complete.\r",
      "--vcf: 19k variants complete.\r",
      "--vcf: 20k variants complete.\r",
      "--vcf: 21k variants complete.\r",
      "--vcf: 22k variants complete.\r",
      "--vcf: 23k variants complete.\r",
      "--vcf: 24k variants complete.\r",
      "--vcf: 25k variants complete.\r",
      "--vcf: 26k variants complete.\r",
      "--vcf: 27k variants complete.\r",
      "--vcf: 28k variants complete.\r",
      "--vcf: 29k variants complete.\r",
      "--vcf: 30k variants complete.\r",
      "--vcf: 31k variants complete.\r",
      "--vcf: 32k variants complete.\r",
      "--vcf: 33k variants complete.\r",
      "--vcf: 34k variants complete.\r",
      "--vcf: 35k variants complete.\r",
      "--vcf: 36k variants complete.\r",
      "--vcf: 37k variants complete.\r",
      "--vcf: 38k variants complete.\r",
      "--vcf: 39k variants complete.\r",
      "--vcf: 40k variants complete.\r",
      "--vcf: 41k variants complete.\r",
      "--vcf: 42k variants complete.\r",
      "--vcf: 43k variants complete.\r",
      "--vcf: 44k variants complete.\r",
      "--vcf: 45k variants complete.\r",
      "--vcf: 46k variants complete.\r",
      "--vcf: 47k variants complete.\r",
      "--vcf: 48k variants complete.\r",
      "--vcf: 49k variants complete.\r",
      "--vcf: 50k variants complete.\r",
      "--vcf: 51k variants complete.\r",
      "--vcf: 52k variants complete.\r",
      "--vcf: 53k variants complete.\r",
      "--vcf: 54k variants complete.\r",
      "--vcf: 55k variants complete.\r",
      "--vcf: 56k variants complete.\r",
      "--vcf: 57k variants complete.\r",
      "--vcf: 58k variants complete.\r",
      "--vcf: 59k variants complete.\r",
      "--vcf: 60k variants complete.\r",
      "--vcf: 61k variants complete.\r",
      "--vcf: 62k variants complete.\r",
      "--vcf: 63k variants complete.\r",
      "--vcf: 64k variants complete.\r",
      "--vcf: 65k variants complete.\r",
      "--vcf: 66k variants complete.\r",
      "--vcf: 67k variants complete.\r",
      "--vcf: 68k variants complete.\r",
      "--vcf: 69k variants complete.\r",
      "--vcf: 70k variants complete.\r",
      "--vcf: 71k variants complete.\r",
      "--vcf: 72k variants complete.\r",
      "--vcf: 73k variants complete.\r",
      "--vcf: 74k variants complete.\r",
      "--vcf: 75k variants complete.\r",
      "--vcf: 76k variants complete.\r",
      "--vcf: 77k variants complete.\r",
      "--vcf: 78k variants complete.\r",
      "--vcf: 79k variants complete.\r",
      "--vcf: 80k variants complete.\r",
      "--vcf: 81k variants complete.\r",
      "--vcf: 82k variants complete.\r",
      "--vcf: 83k variants complete.\r",
      "--vcf: 84k variants complete.\r",
      "--vcf: 85k variants complete.\r",
      "--vcf: 86k variants complete.\r",
      "--vcf: 87k variants complete.\r",
      "--vcf: 88k variants complete.\r",
      "--vcf: 89k variants complete.\r",
      "--vcf: 90k variants complete.\r",
      "--vcf: 91k variants complete.\r",
      "--vcf: 92k variants complete.\r",
      "--vcf: 93k variants complete.\r",
      "--vcf: 94k variants complete.\r",
      "--vcf: 95k variants complete.\r",
      "--vcf: 96k variants complete.\r",
      "--vcf: 97k variants complete.\r",
      "--vcf: 98k variants complete.\r",
      "--vcf: 99k variants complete.\r",
      "--vcf: 100k variants complete.\r",
      "--vcf: 101k variants complete.\r",
      "--vcf: 102k variants complete.\r",
      "--vcf: 103k variants complete.\r",
      "--vcf: 104k variants complete.\r",
      "--vcf: 105k variants complete.\r",
      "--vcf: 106k variants complete.\r",
      "--vcf: 107k variants complete.\r",
      "--vcf: 108k variants complete.\r",
      "--vcf: 109k variants complete.\r",
      "--vcf: 110k variants complete.\r",
      "--vcf: 111k variants complete.\r",
      "--vcf: 112k variants complete.\r",
      "--vcf: 113k variants complete.\r",
      "--vcf: 114k variants complete.\r",
      "--vcf: 115k variants complete.\r",
      "--vcf: 116k variants complete.\r",
      "--vcf: 117k variants complete.\r",
      "--vcf: 118k variants complete.\r",
      "--vcf: 119k variants complete.\r",
      "--vcf: 120k variants complete.\r",
      "--vcf: 121k variants complete.\r",
      "--vcf: 122k variants complete.\r",
      "--vcf: 123k variants complete.\r",
      "--vcf: 124k variants complete.\r",
      "--vcf: 125k variants complete.\r",
      "--vcf: 126k variants complete.\r",
      "--vcf: 127k variants complete.\r",
      "--vcf: 128k variants complete.\r",
      "--vcf: 129k variants complete.\r",
      "--vcf: 130k variants complete.\r",
      "--vcf: 131k variants complete.\r",
      "--vcf: 132k variants complete.\r",
      "--vcf: 133k variants complete.\r",
      "--vcf: 134k variants complete.\r",
      "--vcf: 135k variants complete.\r",
      "--vcf: 136k variants complete.\r",
      "--vcf: 137k variants complete.\r",
      "--vcf: 138k variants complete.\r",
      "--vcf: 139k variants complete.\r",
      "--vcf: 140k variants complete.\r",
      "--vcf: 141k variants complete.\r",
      "--vcf: 142k variants complete.\r",
      "--vcf: 143k variants complete.\r",
      "--vcf: 144k variants complete.\r",
      "--vcf: 145k variants complete.\r",
      "--vcf: 146k variants complete.\r",
      "--vcf: 147k variants complete.\r",
      "--vcf: 148k variants complete.\r",
      "--vcf: 149k variants complete.\r",
      "--vcf: 150k variants complete.\r",
      "--vcf: 151k variants complete.\r",
      "--vcf: 152k variants complete.\r",
      "--vcf: 153k variants complete.\r",
      "--vcf: 154k variants complete.\r",
      "--vcf: 155k variants complete.\r",
      "--vcf: 156k variants complete.\r",
      "--vcf: 157k variants complete.\r",
      "--vcf: 158k variants complete.\r",
      "--vcf: 159k variants complete.\r",
      "--vcf: 160k variants complete.\r",
      "--vcf: 161k variants complete.\r",
      "--vcf: 162k variants complete.\r",
      "--vcf: results/ld/smv7_ex_autosomes_west_africa_SM_V7_1-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_1-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_1-temporary.fam written.\n",
      "162219 variants loaded from .bim file.\n",
      "37 people (0 males, 0 females, 37 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_1.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 37 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.893221.\n",
      "162219 variants and 37 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode ped to results/ld/smv7_ex_autosomes_west_africa_SM_V7_1.ped +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_1.map ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to results/ld/smv7_ex_autosomes_west_africa_SM_V7_1.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --file results/ld/smv7_ex_autosomes_west_africa_SM_V7_1\n",
      "  --ld-window 1000000\n",
      "  --ld-window-kb 90000\n",
      "  --ld-window-r2 0.0\n",
      "  --out results/ld/smv7_ex_autosomes_west_africa_SM_V7_1\n",
      "  --r2\n",
      "  --threads 6\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "Scanning .ped file... 0%\b\b2%\b\b5%\b\b8%\b\b10%\b\b\b13%\b\b\b16%\b\b\b18%\b\b\b21%\b\b\b24%\b\b\b27%\b\b\b29%\b\b\b32%\b\b\b35%\b\b\b37%\b\b\b40%\b\b\b43%\b\b\b45%\b\b\b48%\b\b\b51%\b\b\b54%\b\b\b56%\b\b\b59%\b\b\b62%\b\b\b64%\b\b\b67%\b\b\b70%\b\b\b72%\b\b\b75%\b\b\b78%\b\b\b81%\b\b\b83%\b\b\b86%\b\b\b89%\b\b\b91%\b\b\b94%\b\b\b97%\b\b\b100%\r",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (162219 variants, 37 people).\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\r",
      "--file: results/ld/smv7_ex_autosomes_west_africa_SM_V7_1-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_1-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_1-temporary.fam written.\n",
      "162219 variants loaded from .bim file.\n",
      "37 people (0 males, 0 females, 37 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_1.nosex .\n",
      "Using up to 6 threads (change this with --threads).\n",
      "Before main variant filters, 37 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.893221.\n",
      "162219 variants and 37 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--r2 to results/ld/smv7_ex_autosomes_west_africa_SM_V7_1.ld ...\n",
      "0% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b3% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b6% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b9% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b12% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b15% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b18% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b21% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b24% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b27% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b30% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b33% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b36% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b39% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b42% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b45% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b48% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b51% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b54% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b57% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b60% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b63% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b66% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b69% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b72% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b75% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b78% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b81% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b84% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b87% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b90% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b93% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b96% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b99% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Note: --recode12 flag deprecated.  Use 'recode 12 ...'.\n",
      "Logging to results/ld/smv7_ex_autosomes_west_africa_SM_V7_2.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --out results/ld/smv7_ex_autosomes_west_africa_SM_V7_2\n",
      "  --recode 12\n",
      "  --threads 6\n",
      "  --vcf results/ld/smv7_ex_autosomes_west_africa_SM_V7_2.vcf\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "\r",
      "--vcf: 1k variants complete.\r",
      "--vcf: 2k variants complete.\r",
      "--vcf: 3k variants complete.\r",
      "--vcf: 4k variants complete.\r",
      "--vcf: 5k variants complete.\r",
      "--vcf: 6k variants complete.\r",
      "--vcf: 7k variants complete.\r",
      "--vcf: 8k variants complete.\r",
      "--vcf: 9k variants complete.\r",
      "--vcf: 10k variants complete.\r",
      "--vcf: 11k variants complete.\r",
      "--vcf: 12k variants complete.\r",
      "--vcf: 13k variants complete.\r",
      "--vcf: 14k variants complete.\r",
      "--vcf: 15k variants complete.\r",
      "--vcf: 16k variants complete.\r",
      "--vcf: 17k variants complete.\r",
      "--vcf: 18k variants complete.\r",
      "--vcf: 19k variants complete.\r",
      "--vcf: 20k variants complete.\r",
      "--vcf: 21k variants complete.\r",
      "--vcf: 22k variants complete.\r",
      "--vcf: 23k variants complete.\r",
      "--vcf: 24k variants complete.\r",
      "--vcf: 25k variants complete.\r",
      "--vcf: 26k variants complete.\r",
      "--vcf: 27k variants complete.\r",
      "--vcf: 28k variants complete.\r",
      "--vcf: 29k variants complete.\r",
      "--vcf: 30k variants complete.\r",
      "--vcf: 31k variants complete.\r",
      "--vcf: 32k variants complete.\r",
      "--vcf: 33k variants complete.\r",
      "--vcf: 34k variants complete.\r",
      "--vcf: 35k variants complete.\r",
      "--vcf: 36k variants complete.\r",
      "--vcf: 37k variants complete.\r",
      "--vcf: 38k variants complete.\r",
      "--vcf: 39k variants complete.\r",
      "--vcf: 40k variants complete.\r",
      "--vcf: 41k variants complete.\r",
      "--vcf: 42k variants complete.\r",
      "--vcf: 43k variants complete.\r",
      "--vcf: 44k variants complete.\r",
      "--vcf: 45k variants complete.\r",
      "--vcf: 46k variants complete.\r",
      "--vcf: 47k variants complete.\r",
      "--vcf: 48k variants complete.\r",
      "--vcf: 49k variants complete.\r",
      "--vcf: 50k variants complete.\r",
      "--vcf: 51k variants complete.\r",
      "--vcf: 52k variants complete.\r",
      "--vcf: 53k variants complete.\r",
      "--vcf: 54k variants complete.\r",
      "--vcf: 55k variants complete.\r",
      "--vcf: 56k variants complete.\r",
      "--vcf: 57k variants complete.\r",
      "--vcf: 58k variants complete.\r",
      "--vcf: 59k variants complete.\r",
      "--vcf: 60k variants complete.\r",
      "--vcf: 61k variants complete.\r",
      "--vcf: 62k variants complete.\r",
      "--vcf: 63k variants complete.\r",
      "--vcf: 64k variants complete.\r",
      "--vcf: 65k variants complete.\r",
      "--vcf: 66k variants complete.\r",
      "--vcf: 67k variants complete.\r",
      "--vcf: 68k variants complete.\r",
      "--vcf: 69k variants complete.\r",
      "--vcf: 70k variants complete.\r",
      "--vcf: 71k variants complete.\r",
      "--vcf: 72k variants complete.\r",
      "--vcf: 73k variants complete.\r",
      "--vcf: 74k variants complete.\r",
      "--vcf: 75k variants complete.\r",
      "--vcf: 76k variants complete.\r",
      "--vcf: 77k variants complete.\r",
      "--vcf: 78k variants complete.\r",
      "--vcf: results/ld/smv7_ex_autosomes_west_africa_SM_V7_2-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_2-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_2-temporary.fam written.\n",
      "78902 variants loaded from .bim file.\n",
      "37 people (0 males, 0 females, 37 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_2.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 37 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.883423.\n",
      "78902 variants and 37 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode ped to results/ld/smv7_ex_autosomes_west_africa_SM_V7_2.ped +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_2.map ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to results/ld/smv7_ex_autosomes_west_africa_SM_V7_2.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --file results/ld/smv7_ex_autosomes_west_africa_SM_V7_2\n",
      "  --ld-window 1000000\n",
      "  --ld-window-kb 90000\n",
      "  --ld-window-r2 0.0\n",
      "  --out results/ld/smv7_ex_autosomes_west_africa_SM_V7_2\n",
      "  --r2\n",
      "  --threads 6\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "Scanning .ped file... 0%\b\b2%\b\b5%\b\b8%\b\b10%\b\b\b13%\b\b\b16%\b\b\b18%\b\b\b21%\b\b\b24%\b\b\b27%\b\b\b29%\b\b\b32%\b\b\b35%\b\b\b37%\b\b\b40%\b\b\b43%\b\b\b45%\b\b\b48%\b\b\b51%\b\b\b54%\b\b\b56%\b\b\b59%\b\b\b62%\b\b\b64%\b\b\b67%\b\b\b70%\b\b\b72%\b\b\b75%\b\b\b78%\b\b\b81%\b\b\b83%\b\b\b86%\b\b\b89%\b\b\b91%\b\b\b94%\b\b\b97%\b\b\b100%\r",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (78902 variants, 37 people).\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\r",
      "--file: results/ld/smv7_ex_autosomes_west_africa_SM_V7_2-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_2-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_2-temporary.fam written.\n",
      "78902 variants loaded from .bim file.\n",
      "37 people (0 males, 0 females, 37 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_2.nosex .\n",
      "Using up to 6 threads (change this with --threads).\n",
      "Before main variant filters, 37 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.883423.\n",
      "78902 variants and 37 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--r2 to results/ld/smv7_ex_autosomes_west_africa_SM_V7_2.ld ...\n",
      "0% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b12% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b25% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b38% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b50% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b63% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b76% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b88% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Note: --recode12 flag deprecated.  Use 'recode 12 ...'.\n",
      "Logging to results/ld/smv7_ex_autosomes_west_africa_SM_V7_3.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --out results/ld/smv7_ex_autosomes_west_africa_SM_V7_3\n",
      "  --recode 12\n",
      "  --threads 6\n",
      "  --vcf results/ld/smv7_ex_autosomes_west_africa_SM_V7_3.vcf\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "\r",
      "--vcf: 1k variants complete.\r",
      "--vcf: 2k variants complete.\r",
      "--vcf: 3k variants complete.\r",
      "--vcf: 4k variants complete.\r",
      "--vcf: 5k variants complete.\r",
      "--vcf: 6k variants complete.\r",
      "--vcf: 7k variants complete.\r",
      "--vcf: 8k variants complete.\r",
      "--vcf: 9k variants complete.\r",
      "--vcf: 10k variants complete.\r",
      "--vcf: 11k variants complete.\r",
      "--vcf: 12k variants complete.\r",
      "--vcf: 13k variants complete.\r",
      "--vcf: 14k variants complete.\r",
      "--vcf: 15k variants complete.\r",
      "--vcf: 16k variants complete.\r",
      "--vcf: 17k variants complete.\r",
      "--vcf: 18k variants complete.\r",
      "--vcf: 19k variants complete.\r",
      "--vcf: 20k variants complete.\r",
      "--vcf: 21k variants complete.\r",
      "--vcf: 22k variants complete.\r",
      "--vcf: 23k variants complete.\r",
      "--vcf: 24k variants complete.\r",
      "--vcf: 25k variants complete.\r",
      "--vcf: 26k variants complete.\r",
      "--vcf: 27k variants complete.\r",
      "--vcf: 28k variants complete.\r",
      "--vcf: 29k variants complete.\r",
      "--vcf: 30k variants complete.\r",
      "--vcf: 31k variants complete.\r",
      "--vcf: 32k variants complete.\r",
      "--vcf: 33k variants complete.\r",
      "--vcf: 34k variants complete.\r",
      "--vcf: 35k variants complete.\r",
      "--vcf: 36k variants complete.\r",
      "--vcf: 37k variants complete.\r",
      "--vcf: 38k variants complete.\r",
      "--vcf: 39k variants complete.\r",
      "--vcf: 40k variants complete.\r",
      "--vcf: 41k variants complete.\r",
      "--vcf: 42k variants complete.\r",
      "--vcf: 43k variants complete.\r",
      "--vcf: 44k variants complete.\r",
      "--vcf: 45k variants complete.\r",
      "--vcf: 46k variants complete.\r",
      "--vcf: 47k variants complete.\r",
      "--vcf: 48k variants complete.\r",
      "--vcf: 49k variants complete.\r",
      "--vcf: 50k variants complete.\r",
      "--vcf: 51k variants complete.\r",
      "--vcf: 52k variants complete.\r",
      "--vcf: 53k variants complete.\r",
      "--vcf: 54k variants complete.\r",
      "--vcf: 55k variants complete.\r",
      "--vcf: 56k variants complete.\r",
      "--vcf: 57k variants complete.\r",
      "--vcf: 58k variants complete.\r",
      "--vcf: 59k variants complete.\r",
      "--vcf: 60k variants complete.\r",
      "--vcf: 61k variants complete.\r",
      "--vcf: 62k variants complete.\r",
      "--vcf: 63k variants complete.\r",
      "--vcf: 64k variants complete.\r",
      "--vcf: 65k variants complete.\r",
      "--vcf: 66k variants complete.\r",
      "--vcf: 67k variants complete.\r",
      "--vcf: 68k variants complete.\r",
      "--vcf: 69k variants complete.\r",
      "--vcf: 70k variants complete.\r",
      "--vcf: 71k variants complete.\r",
      "--vcf: 72k variants complete.\r",
      "--vcf: 73k variants complete.\r",
      "--vcf: 74k variants complete.\r",
      "--vcf: 75k variants complete.\r",
      "--vcf: 76k variants complete.\r",
      "--vcf: 77k variants complete.\r",
      "--vcf: 78k variants complete.\r",
      "--vcf: 79k variants complete.\r",
      "--vcf: 80k variants complete.\r",
      "--vcf: results/ld/smv7_ex_autosomes_west_africa_SM_V7_3-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_3-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_3-temporary.fam written.\n",
      "80481 variants loaded from .bim file.\n",
      "37 people (0 males, 0 females, 37 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_3.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 37 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.886114.\n",
      "80481 variants and 37 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode ped to results/ld/smv7_ex_autosomes_west_africa_SM_V7_3.ped +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_3.map ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to results/ld/smv7_ex_autosomes_west_africa_SM_V7_3.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --file results/ld/smv7_ex_autosomes_west_africa_SM_V7_3\n",
      "  --ld-window 1000000\n",
      "  --ld-window-kb 90000\n",
      "  --ld-window-r2 0.0\n",
      "  --out results/ld/smv7_ex_autosomes_west_africa_SM_V7_3\n",
      "  --r2\n",
      "  --threads 6\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "Scanning .ped file... 0%\b\b2%\b\b5%\b\b8%\b\b10%\b\b\b13%\b\b\b16%\b\b\b18%\b\b\b21%\b\b\b24%\b\b\b27%\b\b\b29%\b\b\b32%\b\b\b35%\b\b\b37%\b\b\b40%\b\b\b43%\b\b\b45%\b\b\b48%\b\b\b51%\b\b\b54%\b\b\b56%\b\b\b59%\b\b\b62%\b\b\b64%\b\b\b67%\b\b\b70%\b\b\b72%\b\b\b75%\b\b\b78%\b\b\b81%\b\b\b83%\b\b\b86%\b\b\b89%\b\b\b91%\b\b\b94%\b\b\b97%\b\b\b100%\r",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (80481 variants, 37 people).\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\r",
      "--file: results/ld/smv7_ex_autosomes_west_africa_SM_V7_3-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_3-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_3-temporary.fam written.\n",
      "80481 variants loaded from .bim file.\n",
      "37 people (0 males, 0 females, 37 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_3.nosex .\n",
      "Using up to 6 threads (change this with --threads).\n",
      "Before main variant filters, 37 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.886114.\n",
      "80481 variants and 37 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--r2 to results/ld/smv7_ex_autosomes_west_africa_SM_V7_3.ld ...\n",
      "0% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b12% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b24% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b36% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b48% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b61% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b73% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b85% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b97% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Note: --recode12 flag deprecated.  Use 'recode 12 ...'.\n",
      "Logging to results/ld/smv7_ex_autosomes_west_africa_SM_V7_4.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --out results/ld/smv7_ex_autosomes_west_africa_SM_V7_4\n",
      "  --recode 12\n",
      "  --threads 6\n",
      "  --vcf results/ld/smv7_ex_autosomes_west_africa_SM_V7_4.vcf\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "\r",
      "--vcf: 1k variants complete.\r",
      "--vcf: 2k variants complete.\r",
      "--vcf: 3k variants complete.\r",
      "--vcf: 4k variants complete.\r",
      "--vcf: 5k variants complete.\r",
      "--vcf: 6k variants complete.\r",
      "--vcf: 7k variants complete.\r",
      "--vcf: 8k variants complete.\r",
      "--vcf: 9k variants complete.\r",
      "--vcf: 10k variants complete.\r",
      "--vcf: 11k variants complete.\r",
      "--vcf: 12k variants complete.\r",
      "--vcf: 13k variants complete.\r",
      "--vcf: 14k variants complete.\r",
      "--vcf: 15k variants complete.\r",
      "--vcf: 16k variants complete.\r",
      "--vcf: 17k variants complete.\r",
      "--vcf: 18k variants complete.\r",
      "--vcf: 19k variants complete.\r",
      "--vcf: 20k variants complete.\r",
      "--vcf: 21k variants complete.\r",
      "--vcf: 22k variants complete.\r",
      "--vcf: 23k variants complete.\r",
      "--vcf: 24k variants complete.\r",
      "--vcf: 25k variants complete.\r",
      "--vcf: 26k variants complete.\r",
      "--vcf: 27k variants complete.\r",
      "--vcf: 28k variants complete.\r",
      "--vcf: 29k variants complete.\r",
      "--vcf: 30k variants complete.\r",
      "--vcf: 31k variants complete.\r",
      "--vcf: 32k variants complete.\r",
      "--vcf: 33k variants complete.\r",
      "--vcf: 34k variants complete.\r",
      "--vcf: 35k variants complete.\r",
      "--vcf: 36k variants complete.\r",
      "--vcf: 37k variants complete.\r",
      "--vcf: 38k variants complete.\r",
      "--vcf: 39k variants complete.\r",
      "--vcf: 40k variants complete.\r",
      "--vcf: 41k variants complete.\r",
      "--vcf: 42k variants complete.\r",
      "--vcf: 43k variants complete.\r",
      "--vcf: 44k variants complete.\r",
      "--vcf: 45k variants complete.\r",
      "--vcf: 46k variants complete.\r",
      "--vcf: 47k variants complete.\r",
      "--vcf: 48k variants complete.\r",
      "--vcf: 49k variants complete.\r",
      "--vcf: 50k variants complete.\r",
      "--vcf: 51k variants complete.\r",
      "--vcf: 52k variants complete.\r",
      "--vcf: 53k variants complete.\r",
      "--vcf: 54k variants complete.\r",
      "--vcf: 55k variants complete.\r",
      "--vcf: 56k variants complete.\r",
      "--vcf: 57k variants complete.\r",
      "--vcf: 58k variants complete.\r",
      "--vcf: 59k variants complete.\r",
      "--vcf: 60k variants complete.\r",
      "--vcf: 61k variants complete.\r",
      "--vcf: 62k variants complete.\r",
      "--vcf: 63k variants complete.\r",
      "--vcf: 64k variants complete.\r",
      "--vcf: 65k variants complete.\r",
      "--vcf: 66k variants complete.\r",
      "--vcf: 67k variants complete.\r",
      "--vcf: 68k variants complete.\r",
      "--vcf: 69k variants complete.\r",
      "--vcf: 70k variants complete.\r",
      "--vcf: 71k variants complete.\r",
      "--vcf: 72k variants complete.\r",
      "--vcf: 73k variants complete.\r",
      "--vcf: 74k variants complete.\r",
      "--vcf: 75k variants complete.\r",
      "--vcf: results/ld/smv7_ex_autosomes_west_africa_SM_V7_4-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_4-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_4-temporary.fam written.\n",
      "75581 variants loaded from .bim file.\n",
      "37 people (0 males, 0 females, 37 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_4.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 37 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.885805.\n",
      "75581 variants and 37 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode ped to results/ld/smv7_ex_autosomes_west_africa_SM_V7_4.ped +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_4.map ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to results/ld/smv7_ex_autosomes_west_africa_SM_V7_4.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --file results/ld/smv7_ex_autosomes_west_africa_SM_V7_4\n",
      "  --ld-window 1000000\n",
      "  --ld-window-kb 90000\n",
      "  --ld-window-r2 0.0\n",
      "  --out results/ld/smv7_ex_autosomes_west_africa_SM_V7_4\n",
      "  --r2\n",
      "  --threads 6\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "Scanning .ped file... 0%\b\b2%\b\b5%\b\b8%\b\b10%\b\b\b13%\b\b\b16%\b\b\b18%\b\b\b21%\b\b\b24%\b\b\b27%\b\b\b29%\b\b\b32%\b\b\b35%\b\b\b37%\b\b\b40%\b\b\b43%\b\b\b45%\b\b\b48%\b\b\b51%\b\b\b54%\b\b\b56%\b\b\b59%\b\b\b62%\b\b\b64%\b\b\b67%\b\b\b70%\b\b\b72%\b\b\b75%\b\b\b78%\b\b\b81%\b\b\b83%\b\b\b86%\b\b\b89%\b\b\b91%\b\b\b94%\b\b\b97%\b\b\b100%\r",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (75581 variants, 37 people).\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\r",
      "--file: results/ld/smv7_ex_autosomes_west_africa_SM_V7_4-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_4-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_4-temporary.fam written.\n",
      "75581 variants loaded from .bim file.\n",
      "37 people (0 males, 0 females, 37 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_4.nosex .\n",
      "Using up to 6 threads (change this with --threads).\n",
      "Before main variant filters, 37 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.885805.\n",
      "75581 variants and 37 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--r2 to results/ld/smv7_ex_autosomes_west_africa_SM_V7_4.ld ...\n",
      "0% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b13% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b27% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b41% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b55% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b69% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b83% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b96% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Note: --recode12 flag deprecated.  Use 'recode 12 ...'.\n",
      "Logging to results/ld/smv7_ex_autosomes_west_africa_SM_V7_5.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --out results/ld/smv7_ex_autosomes_west_africa_SM_V7_5\n",
      "  --recode 12\n",
      "  --threads 6\n",
      "  --vcf results/ld/smv7_ex_autosomes_west_africa_SM_V7_5.vcf\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "\r",
      "--vcf: 1k variants complete.\r",
      "--vcf: 2k variants complete.\r",
      "--vcf: 3k variants complete.\r",
      "--vcf: 4k variants complete.\r",
      "--vcf: 5k variants complete.\r",
      "--vcf: 6k variants complete.\r",
      "--vcf: 7k variants complete.\r",
      "--vcf: 8k variants complete.\r",
      "--vcf: 9k variants complete.\r",
      "--vcf: 10k variants complete.\r",
      "--vcf: 11k variants complete.\r",
      "--vcf: 12k variants complete.\r",
      "--vcf: 13k variants complete.\r",
      "--vcf: 14k variants complete.\r",
      "--vcf: 15k variants complete.\r",
      "--vcf: 16k variants complete.\r",
      "--vcf: 17k variants complete.\r",
      "--vcf: 18k variants complete.\r",
      "--vcf: 19k variants complete.\r",
      "--vcf: 20k variants complete.\r",
      "--vcf: 21k variants complete.\r",
      "--vcf: 22k variants complete.\r",
      "--vcf: 23k variants complete.\r",
      "--vcf: 24k variants complete.\r",
      "--vcf: results/ld/smv7_ex_autosomes_west_africa_SM_V7_5-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_5-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_5-temporary.fam written.\n",
      "24681 variants loaded from .bim file.\n",
      "37 people (0 males, 0 females, 37 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_5.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 37 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.877173.\n",
      "24681 variants and 37 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode ped to results/ld/smv7_ex_autosomes_west_africa_SM_V7_5.ped +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_5.map ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to results/ld/smv7_ex_autosomes_west_africa_SM_V7_5.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --file results/ld/smv7_ex_autosomes_west_africa_SM_V7_5\n",
      "  --ld-window 1000000\n",
      "  --ld-window-kb 90000\n",
      "  --ld-window-r2 0.0\n",
      "  --out results/ld/smv7_ex_autosomes_west_africa_SM_V7_5\n",
      "  --r2\n",
      "  --threads 6\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "Scanning .ped file... 0%\b\b2%\b\b5%\b\b8%\b\b10%\b\b\b13%\b\b\b16%\b\b\b18%\b\b\b21%\b\b\b24%\b\b\b27%\b\b\b29%\b\b\b32%\b\b\b35%\b\b\b37%\b\b\b40%\b\b\b43%\b\b\b45%\b\b\b48%\b\b\b51%\b\b\b54%\b\b\b56%\b\b\b59%\b\b\b62%\b\b\b64%\b\b\b67%\b\b\b70%\b\b\b72%\b\b\b75%\b\b\b78%\b\b\b81%\b\b\b83%\b\b\b86%\b\b\b89%\b\b\b91%\b\b\b94%\b\b\b97%\b\b\b100%\r",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (24681 variants, 37 people).\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\r",
      "--file: results/ld/smv7_ex_autosomes_west_africa_SM_V7_5-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_5-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_5-temporary.fam written.\n",
      "24681 variants loaded from .bim file.\n",
      "37 people (0 males, 0 females, 37 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_5.nosex .\n",
      "Using up to 6 threads (change this with --threads).\n",
      "Before main variant filters, 37 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.877173.\n",
      "24681 variants and 37 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--r2 to results/ld/smv7_ex_autosomes_west_africa_SM_V7_5.ld ...\n",
      "0% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Note: --recode12 flag deprecated.  Use 'recode 12 ...'.\n",
      "Logging to results/ld/smv7_ex_autosomes_west_africa_SM_V7_6.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --out results/ld/smv7_ex_autosomes_west_africa_SM_V7_6\n",
      "  --recode 12\n",
      "  --threads 6\n",
      "  --vcf results/ld/smv7_ex_autosomes_west_africa_SM_V7_6.vcf\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "\r",
      "--vcf: 1k variants complete.\r",
      "--vcf: 2k variants complete.\r",
      "--vcf: 3k variants complete.\r",
      "--vcf: 4k variants complete.\r",
      "--vcf: 5k variants complete.\r",
      "--vcf: 6k variants complete.\r",
      "--vcf: 7k variants complete.\r",
      "--vcf: 8k variants complete.\r",
      "--vcf: 9k variants complete.\r",
      "--vcf: 10k variants complete.\r",
      "--vcf: 11k variants complete.\r",
      "--vcf: 12k variants complete.\r",
      "--vcf: 13k variants complete.\r",
      "--vcf: 14k variants complete.\r",
      "--vcf: 15k variants complete.\r",
      "--vcf: 16k variants complete.\r",
      "--vcf: 17k variants complete.\r",
      "--vcf: 18k variants complete.\r",
      "--vcf: 19k variants complete.\r",
      "--vcf: 20k variants complete.\r",
      "--vcf: 21k variants complete.\r",
      "--vcf: 22k variants complete.\r",
      "--vcf: 23k variants complete.\r",
      "--vcf: 24k variants complete.\r",
      "--vcf: 25k variants complete.\r",
      "--vcf: 26k variants complete.\r",
      "--vcf: 27k variants complete.\r",
      "--vcf: 28k variants complete.\r",
      "--vcf: 29k variants complete.\r",
      "--vcf: 30k variants complete.\r",
      "--vcf: 31k variants complete.\r",
      "--vcf: results/ld/smv7_ex_autosomes_west_africa_SM_V7_6-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_6-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_6-temporary.fam written.\n",
      "31877 variants loaded from .bim file.\n",
      "37 people (0 males, 0 females, 37 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_6.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 37 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.872009.\n",
      "31877 variants and 37 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode ped to results/ld/smv7_ex_autosomes_west_africa_SM_V7_6.ped +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_6.map ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to results/ld/smv7_ex_autosomes_west_africa_SM_V7_6.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --file results/ld/smv7_ex_autosomes_west_africa_SM_V7_6\n",
      "  --ld-window 1000000\n",
      "  --ld-window-kb 90000\n",
      "  --ld-window-r2 0.0\n",
      "  --out results/ld/smv7_ex_autosomes_west_africa_SM_V7_6\n",
      "  --r2\n",
      "  --threads 6\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "Scanning .ped file... 0%\b\b2%\b\b5%\b\b8%\b\b10%\b\b\b13%\b\b\b16%\b\b\b18%\b\b\b21%\b\b\b24%\b\b\b27%\b\b\b29%\b\b\b32%\b\b\b35%\b\b\b37%\b\b\b40%\b\b\b43%\b\b\b45%\b\b\b48%\b\b\b51%\b\b\b54%\b\b\b56%\b\b\b59%\b\b\b62%\b\b\b64%\b\b\b67%\b\b\b70%\b\b\b72%\b\b\b75%\b\b\b78%\b\b\b81%\b\b\b83%\b\b\b86%\b\b\b89%\b\b\b91%\b\b\b94%\b\b\b97%\b\b\b100%\r",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (31877 variants, 37 people).\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\r",
      "--file: results/ld/smv7_ex_autosomes_west_africa_SM_V7_6-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_6-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_6-temporary.fam written.\n",
      "31877 variants loaded from .bim file.\n",
      "37 people (0 males, 0 females, 37 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_6.nosex .\n",
      "Using up to 6 threads (change this with --threads).\n",
      "Before main variant filters, 37 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.872009.\n",
      "31877 variants and 37 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--r2 to results/ld/smv7_ex_autosomes_west_africa_SM_V7_6.ld ...\n",
      "0% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b77% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Note: --recode12 flag deprecated.  Use 'recode 12 ...'.\n",
      "Logging to results/ld/smv7_ex_autosomes_west_africa_SM_V7_7.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --out results/ld/smv7_ex_autosomes_west_africa_SM_V7_7\n",
      "  --recode 12\n",
      "  --threads 6\n",
      "  --vcf results/ld/smv7_ex_autosomes_west_africa_SM_V7_7.vcf\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "\r",
      "--vcf: 1k variants complete.\r",
      "--vcf: 2k variants complete.\r",
      "--vcf: 3k variants complete.\r",
      "--vcf: 4k variants complete.\r",
      "--vcf: 5k variants complete.\r",
      "--vcf: 6k variants complete.\r",
      "--vcf: 7k variants complete.\r",
      "--vcf: 8k variants complete.\r",
      "--vcf: 9k variants complete.\r",
      "--vcf: 10k variants complete.\r",
      "--vcf: 11k variants complete.\r",
      "--vcf: 12k variants complete.\r",
      "--vcf: 13k variants complete.\r",
      "--vcf: 14k variants complete.\r",
      "--vcf: 15k variants complete.\r",
      "--vcf: 16k variants complete.\r",
      "--vcf: 17k variants complete.\r",
      "--vcf: 18k variants complete.\r",
      "--vcf: 19k variants complete.\r",
      "--vcf: 20k variants complete.\r",
      "--vcf: 21k variants complete.\r",
      "--vcf: results/ld/smv7_ex_autosomes_west_africa_SM_V7_7-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_7-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_7-temporary.fam written.\n",
      "21340 variants loaded from .bim file.\n",
      "37 people (0 males, 0 females, 37 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_7.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 37 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.876709.\n",
      "21340 variants and 37 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode ped to results/ld/smv7_ex_autosomes_west_africa_SM_V7_7.ped +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_7.map ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to results/ld/smv7_ex_autosomes_west_africa_SM_V7_7.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --file results/ld/smv7_ex_autosomes_west_africa_SM_V7_7\n",
      "  --ld-window 1000000\n",
      "  --ld-window-kb 90000\n",
      "  --ld-window-r2 0.0\n",
      "  --out results/ld/smv7_ex_autosomes_west_africa_SM_V7_7\n",
      "  --r2\n",
      "  --threads 6\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "Scanning .ped file... 0%\b\b2%\b\b5%\b\b8%\b\b10%\b\b\b13%\b\b\b16%\b\b\b18%\b\b\b21%\b\b\b24%\b\b\b27%\b\b\b29%\b\b\b32%\b\b\b35%\b\b\b37%\b\b\b40%\b\b\b43%\b\b\b45%\b\b\b48%\b\b\b51%\b\b\b54%\b\b\b56%\b\b\b59%\b\b\b62%\b\b\b64%\b\b\b67%\b\b\b70%\b\b\b72%\b\b\b75%\b\b\b78%\b\b\b81%\b\b\b83%\b\b\b86%\b\b\b89%\b\b\b91%\b\b\b94%\b\b\b97%\b\b\b100%\r",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (21340 variants, 37 people).\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\r",
      "--file: results/ld/smv7_ex_autosomes_west_africa_SM_V7_7-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_7-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_7-temporary.fam written.\n",
      "21340 variants loaded from .bim file.\n",
      "37 people (0 males, 0 females, 37 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_west_africa_SM_V7_7.nosex .\n",
      "Using up to 6 threads (change this with --threads).\n",
      "Before main variant filters, 37 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.876709.\n",
      "21340 variants and 37 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--r2 to results/ld/smv7_ex_autosomes_west_africa_SM_V7_7.ld ...\n",
      "0% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Note: --recode12 flag deprecated.  Use 'recode 12 ...'.\n",
      "Logging to results/ld/smv7_ex_autosomes_new_world_SM_V7_1.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --out results/ld/smv7_ex_autosomes_new_world_SM_V7_1\n",
      "  --recode 12\n",
      "  --threads 6\n",
      "  --vcf results/ld/smv7_ex_autosomes_new_world_SM_V7_1.vcf\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "\r",
      "--vcf: 1k variants complete.\r",
      "--vcf: 2k variants complete.\r",
      "--vcf: 3k variants complete.\r",
      "--vcf: 4k variants complete.\r",
      "--vcf: 5k variants complete.\r",
      "--vcf: 6k variants complete.\r",
      "--vcf: 7k variants complete.\r",
      "--vcf: 8k variants complete.\r",
      "--vcf: 9k variants complete.\r",
      "--vcf: 10k variants complete.\r",
      "--vcf: 11k variants complete.\r",
      "--vcf: 12k variants complete.\r",
      "--vcf: 13k variants complete.\r",
      "--vcf: 14k variants complete.\r",
      "--vcf: 15k variants complete.\r",
      "--vcf: 16k variants complete.\r",
      "--vcf: 17k variants complete.\r",
      "--vcf: 18k variants complete.\r",
      "--vcf: 19k variants complete.\r",
      "--vcf: 20k variants complete.\r",
      "--vcf: 21k variants complete.\r",
      "--vcf: 22k variants complete.\r",
      "--vcf: 23k variants complete.\r",
      "--vcf: 24k variants complete.\r",
      "--vcf: 25k variants complete.\r",
      "--vcf: 26k variants complete.\r",
      "--vcf: 27k variants complete.\r",
      "--vcf: 28k variants complete.\r",
      "--vcf: 29k variants complete.\r",
      "--vcf: 30k variants complete.\r",
      "--vcf: 31k variants complete.\r",
      "--vcf: 32k variants complete.\r",
      "--vcf: 33k variants complete.\r",
      "--vcf: 34k variants complete.\r",
      "--vcf: 35k variants complete.\r",
      "--vcf: 36k variants complete.\r",
      "--vcf: 37k variants complete.\r",
      "--vcf: 38k variants complete.\r",
      "--vcf: 39k variants complete.\r",
      "--vcf: 40k variants complete.\r",
      "--vcf: 41k variants complete.\r",
      "--vcf: 42k variants complete.\r",
      "--vcf: 43k variants complete.\r",
      "--vcf: 44k variants complete.\r",
      "--vcf: 45k variants complete.\r",
      "--vcf: 46k variants complete.\r",
      "--vcf: 47k variants complete.\r",
      "--vcf: 48k variants complete.\r",
      "--vcf: 49k variants complete.\r",
      "--vcf: 50k variants complete.\r",
      "--vcf: 51k variants complete.\r",
      "--vcf: 52k variants complete.\r",
      "--vcf: 53k variants complete.\r",
      "--vcf: 54k variants complete.\r",
      "--vcf: 55k variants complete.\r",
      "--vcf: 56k variants complete.\r",
      "--vcf: 57k variants complete.\r",
      "--vcf: 58k variants complete.\r",
      "--vcf: 59k variants complete.\r",
      "--vcf: 60k variants complete.\r",
      "--vcf: 61k variants complete.\r",
      "--vcf: 62k variants complete.\r",
      "--vcf: 63k variants complete.\r",
      "--vcf: 64k variants complete.\r",
      "--vcf: 65k variants complete.\r",
      "--vcf: 66k variants complete.\r",
      "--vcf: 67k variants complete.\r",
      "--vcf: 68k variants complete.\r",
      "--vcf: 69k variants complete.\r",
      "--vcf: 70k variants complete.\r",
      "--vcf: 71k variants complete.\r",
      "--vcf: 72k variants complete.\r",
      "--vcf: 73k variants complete.\r",
      "--vcf: 74k variants complete.\r",
      "--vcf: 75k variants complete.\r",
      "--vcf: 76k variants complete.\r",
      "--vcf: 77k variants complete.\r",
      "--vcf: 78k variants complete.\r",
      "--vcf: 79k variants complete.\r",
      "--vcf: 80k variants complete.\r",
      "--vcf: 81k variants complete.\r",
      "--vcf: 82k variants complete.\r",
      "--vcf: 83k variants complete.\r",
      "--vcf: 84k variants complete.\r",
      "--vcf: 85k variants complete.\r",
      "--vcf: 86k variants complete.\r",
      "--vcf: 87k variants complete.\r",
      "--vcf: 88k variants complete.\r",
      "--vcf: 89k variants complete.\r",
      "--vcf: 90k variants complete.\r",
      "--vcf: 91k variants complete.\r",
      "--vcf: 92k variants complete.\r",
      "--vcf: 93k variants complete.\r",
      "--vcf: 94k variants complete.\r",
      "--vcf: 95k variants complete.\r",
      "--vcf: 96k variants complete.\r",
      "--vcf: 97k variants complete.\r",
      "--vcf: 98k variants complete.\r",
      "--vcf: 99k variants complete.\r",
      "--vcf: 100k variants complete.\r",
      "--vcf: 101k variants complete.\r",
      "--vcf: 102k variants complete.\r",
      "--vcf: 103k variants complete.\r",
      "--vcf: 104k variants complete.\r",
      "--vcf: 105k variants complete.\r",
      "--vcf: 106k variants complete.\r",
      "--vcf: 107k variants complete.\r",
      "--vcf: 108k variants complete.\r",
      "--vcf: 109k variants complete.\r",
      "--vcf: 110k variants complete.\r",
      "--vcf: 111k variants complete.\r",
      "--vcf: 112k variants complete.\r",
      "--vcf: 113k variants complete.\r",
      "--vcf: 114k variants complete.\r",
      "--vcf: 115k variants complete.\r",
      "--vcf: 116k variants complete.\r",
      "--vcf: 117k variants complete.\r",
      "--vcf: 118k variants complete.\r",
      "--vcf: 119k variants complete.\r",
      "--vcf: 120k variants complete.\r",
      "--vcf: 121k variants complete.\r",
      "--vcf: 122k variants complete.\r",
      "--vcf: 123k variants complete.\r",
      "--vcf: 124k variants complete.\r",
      "--vcf: 125k variants complete.\r",
      "--vcf: 126k variants complete.\r",
      "--vcf: 127k variants complete.\r",
      "--vcf: 128k variants complete.\r",
      "--vcf: 129k variants complete.\r",
      "--vcf: 130k variants complete.\r",
      "--vcf: 131k variants complete.\r",
      "--vcf: 132k variants complete.\r",
      "--vcf: 133k variants complete.\r",
      "--vcf: 134k variants complete.\r",
      "--vcf: 135k variants complete.\r",
      "--vcf: 136k variants complete.\r",
      "--vcf: 137k variants complete.\r",
      "--vcf: 138k variants complete.\r",
      "--vcf: 139k variants complete.\r",
      "--vcf: 140k variants complete.\r",
      "--vcf: 141k variants complete.\r",
      "--vcf: 142k variants complete.\r",
      "--vcf: 143k variants complete.\r",
      "--vcf: 144k variants complete.\r",
      "--vcf: 145k variants complete.\r",
      "--vcf: 146k variants complete.\r",
      "--vcf: 147k variants complete.\r",
      "--vcf: 148k variants complete.\r",
      "--vcf: 149k variants complete.\r",
      "--vcf: 150k variants complete.\r",
      "--vcf: 151k variants complete.\r",
      "--vcf: 152k variants complete.\r",
      "--vcf: 153k variants complete.\r",
      "--vcf: 154k variants complete.\r",
      "--vcf: 155k variants complete.\r",
      "--vcf: 156k variants complete.\r",
      "--vcf: 157k variants complete.\r",
      "--vcf: 158k variants complete.\r",
      "--vcf: 159k variants complete.\r",
      "--vcf: 160k variants complete.\r",
      "--vcf: 161k variants complete.\r",
      "--vcf: 162k variants complete.\r",
      "--vcf: results/ld/smv7_ex_autosomes_new_world_SM_V7_1-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_1-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_1-temporary.fam written.\n",
      "162219 variants loaded from .bim file.\n",
      "51 people (0 males, 0 females, 51 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_1.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 51 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.907102.\n",
      "162219 variants and 51 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode ped to results/ld/smv7_ex_autosomes_new_world_SM_V7_1.ped +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_1.map ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to results/ld/smv7_ex_autosomes_new_world_SM_V7_1.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --file results/ld/smv7_ex_autosomes_new_world_SM_V7_1\n",
      "  --ld-window 1000000\n",
      "  --ld-window-kb 90000\n",
      "  --ld-window-r2 0.0\n",
      "  --out results/ld/smv7_ex_autosomes_new_world_SM_V7_1\n",
      "  --r2\n",
      "  --threads 6\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "Scanning .ped file... 0%\b\b1%\b\b3%\b\b5%\b\b7%\b\b9%\b\b11%\b\b\b13%\b\b\b15%\b\b\b17%\b\b\b19%\b\b\b21%\b\b\b23%\b\b\b25%\b\b\b27%\b\b\b29%\b\b\b31%\b\b\b33%\b\b\b35%\b\b\b37%\b\b\b39%\b\b\b41%\b\b\b43%\b\b\b45%\b\b\b47%\b\b\b49%\b\b\b50%\b\b\b52%\b\b\b54%\b\b\b56%\b\b\b58%\b\b\b60%\b\b\b62%\b\b\b64%\b\b\b66%\b\b\b68%\b\b\b70%\b\b\b72%\b\b\b74%\b\b\b76%\b\b\b78%\b\b\b80%\b\b\b82%\b\b\b84%\b\b\b86%\b\b\b88%\b\b\b90%\b\b\b92%\b\b\b94%\b\b\b96%\b\b\b98%\b\b\b100%\r",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (162219 variants, 51 people).\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\r",
      "--file: results/ld/smv7_ex_autosomes_new_world_SM_V7_1-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_1-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_1-temporary.fam written.\n",
      "162219 variants loaded from .bim file.\n",
      "51 people (0 males, 0 females, 51 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_1.nosex .\n",
      "Using up to 6 threads (change this with --threads).\n",
      "Before main variant filters, 51 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.907102.\n",
      "162219 variants and 51 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--r2 to results/ld/smv7_ex_autosomes_new_world_SM_V7_1.ld ... 0% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b3% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b6% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b9% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b12% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b15% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b18% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b21% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b24% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b27% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b30% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b33% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b36% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b39% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b42% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b45% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b48% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b51% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b54% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b57% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b60% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b63% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b66% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b69% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b72% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b75% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b78% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b81% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b84% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b87% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b90% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b93% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b96% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b99% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Note: --recode12 flag deprecated.  Use 'recode 12 ...'.\n",
      "Logging to results/ld/smv7_ex_autosomes_new_world_SM_V7_2.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --out results/ld/smv7_ex_autosomes_new_world_SM_V7_2\n",
      "  --recode 12\n",
      "  --threads 6\n",
      "  --vcf results/ld/smv7_ex_autosomes_new_world_SM_V7_2.vcf\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "\r",
      "--vcf: 1k variants complete.\r",
      "--vcf: 2k variants complete.\r",
      "--vcf: 3k variants complete.\r",
      "--vcf: 4k variants complete.\r",
      "--vcf: 5k variants complete.\r",
      "--vcf: 6k variants complete.\r",
      "--vcf: 7k variants complete.\r",
      "--vcf: 8k variants complete.\r",
      "--vcf: 9k variants complete.\r",
      "--vcf: 10k variants complete.\r",
      "--vcf: 11k variants complete.\r",
      "--vcf: 12k variants complete.\r",
      "--vcf: 13k variants complete.\r",
      "--vcf: 14k variants complete.\r",
      "--vcf: 15k variants complete.\r",
      "--vcf: 16k variants complete.\r",
      "--vcf: 17k variants complete.\r",
      "--vcf: 18k variants complete.\r",
      "--vcf: 19k variants complete.\r",
      "--vcf: 20k variants complete.\r",
      "--vcf: 21k variants complete.\r",
      "--vcf: 22k variants complete.\r",
      "--vcf: 23k variants complete.\r",
      "--vcf: 24k variants complete.\r",
      "--vcf: 25k variants complete.\r",
      "--vcf: 26k variants complete.\r",
      "--vcf: 27k variants complete.\r",
      "--vcf: 28k variants complete.\r",
      "--vcf: 29k variants complete.\r",
      "--vcf: 30k variants complete.\r",
      "--vcf: 31k variants complete.\r",
      "--vcf: 32k variants complete.\r",
      "--vcf: 33k variants complete.\r",
      "--vcf: 34k variants complete.\r",
      "--vcf: 35k variants complete.\r",
      "--vcf: 36k variants complete.\r",
      "--vcf: 37k variants complete.\r",
      "--vcf: 38k variants complete.\r",
      "--vcf: 39k variants complete.\r",
      "--vcf: 40k variants complete.\r",
      "--vcf: 41k variants complete.\r",
      "--vcf: 42k variants complete.\r",
      "--vcf: 43k variants complete.\r",
      "--vcf: 44k variants complete.\r",
      "--vcf: 45k variants complete.\r",
      "--vcf: 46k variants complete.\r",
      "--vcf: 47k variants complete.\r",
      "--vcf: 48k variants complete.\r",
      "--vcf: 49k variants complete.\r",
      "--vcf: 50k variants complete.\r",
      "--vcf: 51k variants complete.\r",
      "--vcf: 52k variants complete.\r",
      "--vcf: 53k variants complete.\r",
      "--vcf: 54k variants complete.\r",
      "--vcf: 55k variants complete.\r",
      "--vcf: 56k variants complete.\r",
      "--vcf: 57k variants complete.\r",
      "--vcf: 58k variants complete.\r",
      "--vcf: 59k variants complete.\r",
      "--vcf: 60k variants complete.\r",
      "--vcf: 61k variants complete.\r",
      "--vcf: 62k variants complete.\r",
      "--vcf: 63k variants complete.\r",
      "--vcf: 64k variants complete.\r",
      "--vcf: 65k variants complete.\r",
      "--vcf: 66k variants complete.\r",
      "--vcf: 67k variants complete.\r",
      "--vcf: 68k variants complete.\r",
      "--vcf: 69k variants complete.\r",
      "--vcf: 70k variants complete.\r",
      "--vcf: 71k variants complete.\r",
      "--vcf: 72k variants complete.\r",
      "--vcf: 73k variants complete.\r",
      "--vcf: 74k variants complete.\r",
      "--vcf: 75k variants complete.\r",
      "--vcf: 76k variants complete.\r",
      "--vcf: 77k variants complete.\r",
      "--vcf: 78k variants complete.\r",
      "--vcf: results/ld/smv7_ex_autosomes_new_world_SM_V7_2-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_2-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_2-temporary.fam written.\n",
      "78902 variants loaded from .bim file.\n",
      "51 people (0 males, 0 females, 51 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_2.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 51 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.898286.\n",
      "78902 variants and 51 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode ped to results/ld/smv7_ex_autosomes_new_world_SM_V7_2.ped +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_2.map ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to results/ld/smv7_ex_autosomes_new_world_SM_V7_2.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --file results/ld/smv7_ex_autosomes_new_world_SM_V7_2\n",
      "  --ld-window 1000000\n",
      "  --ld-window-kb 90000\n",
      "  --ld-window-r2 0.0\n",
      "  --out results/ld/smv7_ex_autosomes_new_world_SM_V7_2\n",
      "  --r2\n",
      "  --threads 6\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "Scanning .ped file... 0%\b\b1%\b\b3%\b\b5%\b\b7%\b\b9%\b\b11%\b\b\b13%\b\b\b15%\b\b\b17%\b\b\b19%\b\b\b21%\b\b\b23%\b\b\b25%\b\b\b27%\b\b\b29%\b\b\b31%\b\b\b33%\b\b\b35%\b\b\b37%\b\b\b39%\b\b\b41%\b\b\b43%\b\b\b45%\b\b\b47%\b\b\b49%\b\b\b50%\b\b\b52%\b\b\b54%\b\b\b56%\b\b\b58%\b\b\b60%\b\b\b62%\b\b\b64%\b\b\b66%\b\b\b68%\b\b\b70%\b\b\b72%\b\b\b74%\b\b\b76%\b\b\b78%\b\b\b80%\b\b\b82%\b\b\b84%\b\b\b86%\b\b\b88%\b\b\b90%\b\b\b92%\b\b\b94%\b\b\b96%\b\b\b98%\b\b\b100%\r",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (78902 variants, 51 people).\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\r",
      "--file: results/ld/smv7_ex_autosomes_new_world_SM_V7_2-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_2-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_2-temporary.fam written.\n",
      "78902 variants loaded from .bim file.\n",
      "51 people (0 males, 0 females, 51 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_2.nosex .\n",
      "Using up to 6 threads (change this with --threads).\n",
      "Before main variant filters, 51 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.898286.\n",
      "78902 variants and 51 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--r2 to results/ld/smv7_ex_autosomes_new_world_SM_V7_2.ld ... 0% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b12% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b25% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b38% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b50% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b63% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b76% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b88% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Note: --recode12 flag deprecated.  Use 'recode 12 ...'.\n",
      "Logging to results/ld/smv7_ex_autosomes_new_world_SM_V7_3.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --out results/ld/smv7_ex_autosomes_new_world_SM_V7_3\n",
      "  --recode 12\n",
      "  --threads 6\n",
      "  --vcf results/ld/smv7_ex_autosomes_new_world_SM_V7_3.vcf\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "\r",
      "--vcf: 1k variants complete.\r",
      "--vcf: 2k variants complete.\r",
      "--vcf: 3k variants complete.\r",
      "--vcf: 4k variants complete.\r",
      "--vcf: 5k variants complete.\r",
      "--vcf: 6k variants complete.\r",
      "--vcf: 7k variants complete.\r",
      "--vcf: 8k variants complete.\r",
      "--vcf: 9k variants complete.\r",
      "--vcf: 10k variants complete.\r",
      "--vcf: 11k variants complete.\r",
      "--vcf: 12k variants complete.\r",
      "--vcf: 13k variants complete.\r",
      "--vcf: 14k variants complete.\r",
      "--vcf: 15k variants complete.\r",
      "--vcf: 16k variants complete.\r",
      "--vcf: 17k variants complete.\r",
      "--vcf: 18k variants complete.\r",
      "--vcf: 19k variants complete.\r",
      "--vcf: 20k variants complete.\r",
      "--vcf: 21k variants complete.\r",
      "--vcf: 22k variants complete.\r",
      "--vcf: 23k variants complete.\r",
      "--vcf: 24k variants complete.\r",
      "--vcf: 25k variants complete.\r",
      "--vcf: 26k variants complete.\r",
      "--vcf: 27k variants complete.\r",
      "--vcf: 28k variants complete.\r",
      "--vcf: 29k variants complete.\r",
      "--vcf: 30k variants complete.\r",
      "--vcf: 31k variants complete.\r",
      "--vcf: 32k variants complete.\r",
      "--vcf: 33k variants complete.\r",
      "--vcf: 34k variants complete.\r",
      "--vcf: 35k variants complete.\r",
      "--vcf: 36k variants complete.\r",
      "--vcf: 37k variants complete.\r",
      "--vcf: 38k variants complete.\r",
      "--vcf: 39k variants complete.\r",
      "--vcf: 40k variants complete.\r",
      "--vcf: 41k variants complete.\r",
      "--vcf: 42k variants complete.\r",
      "--vcf: 43k variants complete.\r",
      "--vcf: 44k variants complete.\r",
      "--vcf: 45k variants complete.\r",
      "--vcf: 46k variants complete.\r",
      "--vcf: 47k variants complete.\r",
      "--vcf: 48k variants complete.\r",
      "--vcf: 49k variants complete.\r",
      "--vcf: 50k variants complete.\r",
      "--vcf: 51k variants complete.\r",
      "--vcf: 52k variants complete.\r",
      "--vcf: 53k variants complete.\r",
      "--vcf: 54k variants complete.\r",
      "--vcf: 55k variants complete.\r",
      "--vcf: 56k variants complete.\r",
      "--vcf: 57k variants complete.\r",
      "--vcf: 58k variants complete.\r",
      "--vcf: 59k variants complete.\r",
      "--vcf: 60k variants complete.\r",
      "--vcf: 61k variants complete.\r",
      "--vcf: 62k variants complete.\r",
      "--vcf: 63k variants complete.\r",
      "--vcf: 64k variants complete.\r",
      "--vcf: 65k variants complete.\r",
      "--vcf: 66k variants complete.\r",
      "--vcf: 67k variants complete.\r",
      "--vcf: 68k variants complete.\r",
      "--vcf: 69k variants complete.\r",
      "--vcf: 70k variants complete.\r",
      "--vcf: 71k variants complete.\r",
      "--vcf: 72k variants complete.\r",
      "--vcf: 73k variants complete.\r",
      "--vcf: 74k variants complete.\r",
      "--vcf: 75k variants complete.\r",
      "--vcf: 76k variants complete.\r",
      "--vcf: 77k variants complete.\r",
      "--vcf: 78k variants complete.\r",
      "--vcf: 79k variants complete.\r",
      "--vcf: 80k variants complete.\r",
      "--vcf: results/ld/smv7_ex_autosomes_new_world_SM_V7_3-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_3-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_3-temporary.fam written.\n",
      "80481 variants loaded from .bim file.\n",
      "51 people (0 males, 0 females, 51 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_3.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 51 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.896862.\n",
      "80481 variants and 51 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode ped to results/ld/smv7_ex_autosomes_new_world_SM_V7_3.ped +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_3.map ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to results/ld/smv7_ex_autosomes_new_world_SM_V7_3.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --file results/ld/smv7_ex_autosomes_new_world_SM_V7_3\n",
      "  --ld-window 1000000\n",
      "  --ld-window-kb 90000\n",
      "  --ld-window-r2 0.0\n",
      "  --out results/ld/smv7_ex_autosomes_new_world_SM_V7_3\n",
      "  --r2\n",
      "  --threads 6\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "Scanning .ped file... 0%\b\b1%\b\b3%\b\b5%\b\b7%\b\b9%\b\b11%\b\b\b13%\b\b\b15%\b\b\b17%\b\b\b19%\b\b\b21%\b\b\b23%\b\b\b25%\b\b\b27%\b\b\b29%\b\b\b31%\b\b\b33%\b\b\b35%\b\b\b37%\b\b\b39%\b\b\b41%\b\b\b43%\b\b\b45%\b\b\b47%\b\b\b49%\b\b\b50%\b\b\b52%\b\b\b54%\b\b\b56%\b\b\b58%\b\b\b60%\b\b\b62%\b\b\b64%\b\b\b66%\b\b\b68%\b\b\b70%\b\b\b72%\b\b\b74%\b\b\b76%\b\b\b78%\b\b\b80%\b\b\b82%\b\b\b84%\b\b\b86%\b\b\b88%\b\b\b90%\b\b\b92%\b\b\b94%\b\b\b96%\b\b\b98%\b\b\b100%\r",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (80481 variants, 51 people).\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\r",
      "--file: results/ld/smv7_ex_autosomes_new_world_SM_V7_3-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_3-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_3-temporary.fam written.\n",
      "80481 variants loaded from .bim file.\n",
      "51 people (0 males, 0 females, 51 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_3.nosex .\n",
      "Using up to 6 threads (change this with --threads).\n",
      "Before main variant filters, 51 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.896862.\n",
      "80481 variants and 51 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--r2 to results/ld/smv7_ex_autosomes_new_world_SM_V7_3.ld ... 0% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b12% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b24% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b36% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b48% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b61% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b73% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b85% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b97% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Note: --recode12 flag deprecated.  Use 'recode 12 ...'.\n",
      "Logging to results/ld/smv7_ex_autosomes_new_world_SM_V7_4.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --out results/ld/smv7_ex_autosomes_new_world_SM_V7_4\n",
      "  --recode 12\n",
      "  --threads 6\n",
      "  --vcf results/ld/smv7_ex_autosomes_new_world_SM_V7_4.vcf\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "\r",
      "--vcf: 1k variants complete.\r",
      "--vcf: 2k variants complete.\r",
      "--vcf: 3k variants complete.\r",
      "--vcf: 4k variants complete.\r",
      "--vcf: 5k variants complete.\r",
      "--vcf: 6k variants complete.\r",
      "--vcf: 7k variants complete.\r",
      "--vcf: 8k variants complete.\r",
      "--vcf: 9k variants complete.\r",
      "--vcf: 10k variants complete.\r",
      "--vcf: 11k variants complete.\r",
      "--vcf: 12k variants complete.\r",
      "--vcf: 13k variants complete.\r",
      "--vcf: 14k variants complete.\r",
      "--vcf: 15k variants complete.\r",
      "--vcf: 16k variants complete.\r",
      "--vcf: 17k variants complete.\r",
      "--vcf: 18k variants complete.\r",
      "--vcf: 19k variants complete.\r",
      "--vcf: 20k variants complete.\r",
      "--vcf: 21k variants complete.\r",
      "--vcf: 22k variants complete.\r",
      "--vcf: 23k variants complete.\r",
      "--vcf: 24k variants complete.\r",
      "--vcf: 25k variants complete.\r",
      "--vcf: 26k variants complete.\r",
      "--vcf: 27k variants complete.\r",
      "--vcf: 28k variants complete.\r",
      "--vcf: 29k variants complete.\r",
      "--vcf: 30k variants complete.\r",
      "--vcf: 31k variants complete.\r",
      "--vcf: 32k variants complete.\r",
      "--vcf: 33k variants complete.\r",
      "--vcf: 34k variants complete.\r",
      "--vcf: 35k variants complete.\r",
      "--vcf: 36k variants complete.\r",
      "--vcf: 37k variants complete.\r",
      "--vcf: 38k variants complete.\r",
      "--vcf: 39k variants complete.\r",
      "--vcf: 40k variants complete.\r",
      "--vcf: 41k variants complete.\r",
      "--vcf: 42k variants complete.\r",
      "--vcf: 43k variants complete.\r",
      "--vcf: 44k variants complete.\r",
      "--vcf: 45k variants complete.\r",
      "--vcf: 46k variants complete.\r",
      "--vcf: 47k variants complete.\r",
      "--vcf: 48k variants complete.\r",
      "--vcf: 49k variants complete.\r",
      "--vcf: 50k variants complete.\r",
      "--vcf: 51k variants complete.\r",
      "--vcf: 52k variants complete.\r",
      "--vcf: 53k variants complete.\r",
      "--vcf: 54k variants complete.\r",
      "--vcf: 55k variants complete.\r",
      "--vcf: 56k variants complete.\r",
      "--vcf: 57k variants complete.\r",
      "--vcf: 58k variants complete.\r",
      "--vcf: 59k variants complete.\r",
      "--vcf: 60k variants complete.\r",
      "--vcf: 61k variants complete.\r",
      "--vcf: 62k variants complete.\r",
      "--vcf: 63k variants complete.\r",
      "--vcf: 64k variants complete.\r",
      "--vcf: 65k variants complete.\r",
      "--vcf: 66k variants complete.\r",
      "--vcf: 67k variants complete.\r",
      "--vcf: 68k variants complete.\r",
      "--vcf: 69k variants complete.\r",
      "--vcf: 70k variants complete.\r",
      "--vcf: 71k variants complete.\r",
      "--vcf: 72k variants complete.\r",
      "--vcf: 73k variants complete.\r",
      "--vcf: 74k variants complete.\r",
      "--vcf: 75k variants complete.\r",
      "--vcf: results/ld/smv7_ex_autosomes_new_world_SM_V7_4-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_4-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_4-temporary.fam written.\n",
      "75581 variants loaded from .bim file.\n",
      "51 people (0 males, 0 females, 51 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_4.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 51 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.89777.\n",
      "75581 variants and 51 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode ped to results/ld/smv7_ex_autosomes_new_world_SM_V7_4.ped +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_4.map ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to results/ld/smv7_ex_autosomes_new_world_SM_V7_4.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --file results/ld/smv7_ex_autosomes_new_world_SM_V7_4\n",
      "  --ld-window 1000000\n",
      "  --ld-window-kb 90000\n",
      "  --ld-window-r2 0.0\n",
      "  --out results/ld/smv7_ex_autosomes_new_world_SM_V7_4\n",
      "  --r2\n",
      "  --threads 6\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "Scanning .ped file... 0%\b\b1%\b\b3%\b\b5%\b\b7%\b\b9%\b\b11%\b\b\b13%\b\b\b15%\b\b\b17%\b\b\b19%\b\b\b21%\b\b\b23%\b\b\b25%\b\b\b27%\b\b\b29%\b\b\b31%\b\b\b33%\b\b\b35%\b\b\b37%\b\b\b39%\b\b\b41%\b\b\b43%\b\b\b45%\b\b\b47%\b\b\b49%\b\b\b50%\b\b\b52%\b\b\b54%\b\b\b56%\b\b\b58%\b\b\b60%\b\b\b62%\b\b\b64%\b\b\b66%\b\b\b68%\b\b\b70%\b\b\b72%\b\b\b74%\b\b\b76%\b\b\b78%\b\b\b80%\b\b\b82%\b\b\b84%\b\b\b86%\b\b\b88%\b\b\b90%\b\b\b92%\b\b\b94%\b\b\b96%\b\b\b98%\b\b\b100%\r",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (75581 variants, 51 people).\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\r",
      "--file: results/ld/smv7_ex_autosomes_new_world_SM_V7_4-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_4-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_4-temporary.fam written.\n",
      "75581 variants loaded from .bim file.\n",
      "51 people (0 males, 0 females, 51 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_4.nosex .\n",
      "Using up to 6 threads (change this with --threads).\n",
      "Before main variant filters, 51 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.89777.\n",
      "75581 variants and 51 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--r2 to results/ld/smv7_ex_autosomes_new_world_SM_V7_4.ld ... 0% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b13% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b27% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b41% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b55% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b69% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b83% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\b96% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Note: --recode12 flag deprecated.  Use 'recode 12 ...'.\n",
      "Logging to results/ld/smv7_ex_autosomes_new_world_SM_V7_5.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --out results/ld/smv7_ex_autosomes_new_world_SM_V7_5\n",
      "  --recode 12\n",
      "  --threads 6\n",
      "  --vcf results/ld/smv7_ex_autosomes_new_world_SM_V7_5.vcf\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "\r",
      "--vcf: 1k variants complete.\r",
      "--vcf: 2k variants complete.\r",
      "--vcf: 3k variants complete.\r",
      "--vcf: 4k variants complete.\r",
      "--vcf: 5k variants complete.\r",
      "--vcf: 6k variants complete.\r",
      "--vcf: 7k variants complete.\r",
      "--vcf: 8k variants complete.\r",
      "--vcf: 9k variants complete.\r",
      "--vcf: 10k variants complete.\r",
      "--vcf: 11k variants complete.\r",
      "--vcf: 12k variants complete.\r",
      "--vcf: 13k variants complete.\r",
      "--vcf: 14k variants complete.\r",
      "--vcf: 15k variants complete.\r",
      "--vcf: 16k variants complete.\r",
      "--vcf: 17k variants complete.\r",
      "--vcf: 18k variants complete.\r",
      "--vcf: 19k variants complete.\r",
      "--vcf: 20k variants complete.\r",
      "--vcf: 21k variants complete.\r",
      "--vcf: 22k variants complete.\r",
      "--vcf: 23k variants complete.\r",
      "--vcf: 24k variants complete.\r",
      "--vcf: results/ld/smv7_ex_autosomes_new_world_SM_V7_5-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_5-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_5-temporary.fam written.\n",
      "24681 variants loaded from .bim file.\n",
      "51 people (0 males, 0 females, 51 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_5.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 51 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.884476.\n",
      "24681 variants and 51 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode ped to results/ld/smv7_ex_autosomes_new_world_SM_V7_5.ped +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_5.map ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to results/ld/smv7_ex_autosomes_new_world_SM_V7_5.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --file results/ld/smv7_ex_autosomes_new_world_SM_V7_5\n",
      "  --ld-window 1000000\n",
      "  --ld-window-kb 90000\n",
      "  --ld-window-r2 0.0\n",
      "  --out results/ld/smv7_ex_autosomes_new_world_SM_V7_5\n",
      "  --r2\n",
      "  --threads 6\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "Scanning .ped file... 0%\b\b1%\b\b3%\b\b5%\b\b7%\b\b9%\b\b11%\b\b\b13%\b\b\b15%\b\b\b17%\b\b\b19%\b\b\b21%\b\b\b23%\b\b\b25%\b\b\b27%\b\b\b29%\b\b\b31%\b\b\b33%\b\b\b35%\b\b\b37%\b\b\b39%\b\b\b41%\b\b\b43%\b\b\b45%\b\b\b47%\b\b\b49%\b\b\b50%\b\b\b52%\b\b\b54%\b\b\b56%\b\b\b58%\b\b\b60%\b\b\b62%\b\b\b64%\b\b\b66%\b\b\b68%\b\b\b70%\b\b\b72%\b\b\b74%\b\b\b76%\b\b\b78%\b\b\b80%\b\b\b82%\b\b\b84%\b\b\b86%\b\b\b88%\b\b\b90%\b\b\b92%\b\b\b94%\b\b\b96%\b\b\b98%\b\b\b100%\r",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (24681 variants, 51 people).\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\r",
      "--file: results/ld/smv7_ex_autosomes_new_world_SM_V7_5-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_5-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_5-temporary.fam written.\n",
      "24681 variants loaded from .bim file.\n",
      "51 people (0 males, 0 females, 51 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_5.nosex .\n",
      "Using up to 6 threads (change this with --threads).\n",
      "Before main variant filters, 51 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.884476.\n",
      "24681 variants and 51 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--r2 to results/ld/smv7_ex_autosomes_new_world_SM_V7_5.ld ... 0% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Note: --recode12 flag deprecated.  Use 'recode 12 ...'.\n",
      "Logging to results/ld/smv7_ex_autosomes_new_world_SM_V7_6.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --out results/ld/smv7_ex_autosomes_new_world_SM_V7_6\n",
      "  --recode 12\n",
      "  --threads 6\n",
      "  --vcf results/ld/smv7_ex_autosomes_new_world_SM_V7_6.vcf\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "\r",
      "--vcf: 1k variants complete.\r",
      "--vcf: 2k variants complete.\r",
      "--vcf: 3k variants complete.\r",
      "--vcf: 4k variants complete.\r",
      "--vcf: 5k variants complete.\r",
      "--vcf: 6k variants complete.\r",
      "--vcf: 7k variants complete.\r",
      "--vcf: 8k variants complete.\r",
      "--vcf: 9k variants complete.\r",
      "--vcf: 10k variants complete.\r",
      "--vcf: 11k variants complete.\r",
      "--vcf: 12k variants complete.\r",
      "--vcf: 13k variants complete.\r",
      "--vcf: 14k variants complete.\r",
      "--vcf: 15k variants complete.\r",
      "--vcf: 16k variants complete.\r",
      "--vcf: 17k variants complete.\r",
      "--vcf: 18k variants complete.\r",
      "--vcf: 19k variants complete.\r",
      "--vcf: 20k variants complete.\r",
      "--vcf: 21k variants complete.\r",
      "--vcf: 22k variants complete.\r",
      "--vcf: 23k variants complete.\r",
      "--vcf: 24k variants complete.\r",
      "--vcf: 25k variants complete.\r",
      "--vcf: 26k variants complete.\r",
      "--vcf: 27k variants complete.\r",
      "--vcf: 28k variants complete.\r",
      "--vcf: 29k variants complete.\r",
      "--vcf: 30k variants complete.\r",
      "--vcf: 31k variants complete.\r",
      "--vcf: results/ld/smv7_ex_autosomes_new_world_SM_V7_6-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_6-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_6-temporary.fam written.\n",
      "31877 variants loaded from .bim file.\n",
      "51 people (0 males, 0 females, 51 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_6.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 51 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.890773.\n",
      "31877 variants and 51 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode ped to results/ld/smv7_ex_autosomes_new_world_SM_V7_6.ped +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_6.map ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to results/ld/smv7_ex_autosomes_new_world_SM_V7_6.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --file results/ld/smv7_ex_autosomes_new_world_SM_V7_6\n",
      "  --ld-window 1000000\n",
      "  --ld-window-kb 90000\n",
      "  --ld-window-r2 0.0\n",
      "  --out results/ld/smv7_ex_autosomes_new_world_SM_V7_6\n",
      "  --r2\n",
      "  --threads 6\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "Scanning .ped file... 0%\b\b1%\b\b3%\b\b5%\b\b7%\b\b9%\b\b11%\b\b\b13%\b\b\b15%\b\b\b17%\b\b\b19%\b\b\b21%\b\b\b23%\b\b\b25%\b\b\b27%\b\b\b29%\b\b\b31%\b\b\b33%\b\b\b35%\b\b\b37%\b\b\b39%\b\b\b41%\b\b\b43%\b\b\b45%\b\b\b47%\b\b\b49%\b\b\b50%\b\b\b52%\b\b\b54%\b\b\b56%\b\b\b58%\b\b\b60%\b\b\b62%\b\b\b64%\b\b\b66%\b\b\b68%\b\b\b70%\b\b\b72%\b\b\b74%\b\b\b76%\b\b\b78%\b\b\b80%\b\b\b82%\b\b\b84%\b\b\b86%\b\b\b88%\b\b\b90%\b\b\b92%\b\b\b94%\b\b\b96%\b\b\b98%\b\b\b100%\r",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (31877 variants, 51 people).\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\r",
      "--file: results/ld/smv7_ex_autosomes_new_world_SM_V7_6-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_6-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_6-temporary.fam written.\n",
      "31877 variants loaded from .bim file.\n",
      "51 people (0 males, 0 females, 51 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_6.nosex .\n",
      "Using up to 6 threads (change this with --threads).\n",
      "Before main variant filters, 51 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.890773.\n",
      "31877 variants and 51 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--r2 to results/ld/smv7_ex_autosomes_new_world_SM_V7_6.ld ... 0% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b77% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Note: --recode12 flag deprecated.  Use 'recode 12 ...'.\n",
      "Logging to results/ld/smv7_ex_autosomes_new_world_SM_V7_7.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --out results/ld/smv7_ex_autosomes_new_world_SM_V7_7\n",
      "  --recode 12\n",
      "  --threads 6\n",
      "  --vcf results/ld/smv7_ex_autosomes_new_world_SM_V7_7.vcf\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "\r",
      "--vcf: 1k variants complete.\r",
      "--vcf: 2k variants complete.\r",
      "--vcf: 3k variants complete.\r",
      "--vcf: 4k variants complete.\r",
      "--vcf: 5k variants complete.\r",
      "--vcf: 6k variants complete.\r",
      "--vcf: 7k variants complete.\r",
      "--vcf: 8k variants complete.\r",
      "--vcf: 9k variants complete.\r",
      "--vcf: 10k variants complete.\r",
      "--vcf: 11k variants complete.\r",
      "--vcf: 12k variants complete.\r",
      "--vcf: 13k variants complete.\r",
      "--vcf: 14k variants complete.\r",
      "--vcf: 15k variants complete.\r",
      "--vcf: 16k variants complete.\r",
      "--vcf: 17k variants complete.\r",
      "--vcf: 18k variants complete.\r",
      "--vcf: 19k variants complete.\r",
      "--vcf: 20k variants complete.\r",
      "--vcf: 21k variants complete.\r",
      "--vcf: results/ld/smv7_ex_autosomes_new_world_SM_V7_7-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_7-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_7-temporary.fam written.\n",
      "21340 variants loaded from .bim file.\n",
      "51 people (0 males, 0 females, 51 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_7.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 51 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.887649.\n",
      "21340 variants and 51 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode ped to results/ld/smv7_ex_autosomes_new_world_SM_V7_7.ped +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_7.map ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "PLINK v1.90b4 64-bit (20 Mar 2017)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to results/ld/smv7_ex_autosomes_new_world_SM_V7_7.log.\n",
      "Options in effect:\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --file results/ld/smv7_ex_autosomes_new_world_SM_V7_7\n",
      "  --ld-window 1000000\n",
      "  --ld-window-kb 90000\n",
      "  --ld-window-r2 0.0\n",
      "  --out results/ld/smv7_ex_autosomes_new_world_SM_V7_7\n",
      "  --r2\n",
      "  --threads 6\n",
      "\n",
      "24158 MB RAM detected; reserving 12079 MB for main workspace.\n",
      "Scanning .ped file... 0%\b\b1%\b\b3%\b\b5%\b\b7%\b\b9%\b\b11%\b\b\b13%\b\b\b15%\b\b\b17%\b\b\b19%\b\b\b21%\b\b\b23%\b\b\b25%\b\b\b27%\b\b\b29%\b\b\b31%\b\b\b33%\b\b\b35%\b\b\b37%\b\b\b39%\b\b\b41%\b\b\b43%\b\b\b45%\b\b\b47%\b\b\b49%\b\b\b50%\b\b\b52%\b\b\b54%\b\b\b56%\b\b\b58%\b\b\b60%\b\b\b62%\b\b\b64%\b\b\b66%\b\b\b68%\b\b\b70%\b\b\b72%\b\b\b74%\b\b\b76%\b\b\b78%\b\b\b80%\b\b\b82%\b\b\b84%\b\b\b86%\b\b\b88%\b\b\b90%\b\b\b92%\b\b\b94%\b\b\b96%\b\b\b98%\b\b\b100%\r",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (21340 variants, 51 people).\n",
      "0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\r",
      "--file: results/ld/smv7_ex_autosomes_new_world_SM_V7_7-temporary.bed +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_7-temporary.bim +\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_7-temporary.fam written.\n",
      "21340 variants loaded from .bim file.\n",
      "51 people (0 males, 0 females, 51 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "results/ld/smv7_ex_autosomes_new_world_SM_V7_7.nosex .\n",
      "Using up to 6 threads (change this with --threads).\n",
      "Before main variant filters, 51 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\b\b done.\n",
      "Total genotyping rate is 0.887649.\n",
      "21340 variants and 51 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--r2 to results/ld/smv7_ex_autosomes_new_world_SM_V7_7.ld ... 0% [processing]\b\b\b\b\b\b\b\b\b\b\bwriting]   \b\b\b\b\b\b\b\b\b\b\b\b\b          \b\b\b\b\b\b\b\b\b\b\b\bdone.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "rm: cannot remove `results/ld/schMan_v7_east_africa_ld_dist.tbl': No such file or directory\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_autosomes.vcf\n",
      "\t--chr SM_V7_1\n",
      "\t--keep results/ld/east_africa.list\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 58 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 162219 out of a possible 475081 Sites\n",
      "Run Time = 91.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_autosomes.vcf\n",
      "\t--chr SM_V7_2\n",
      "\t--keep results/ld/east_africa.list\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 58 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 78902 out of a possible 475081 Sites\n",
      "Run Time = 78.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_autosomes.vcf\n",
      "\t--chr SM_V7_3\n",
      "\t--keep results/ld/east_africa.list\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 58 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 80481 out of a possible 475081 Sites\n",
      "Run Time = 66.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_autosomes.vcf\n",
      "\t--chr SM_V7_4\n",
      "\t--keep results/ld/east_africa.list\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 58 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 75581 out of a possible 475081 Sites\n",
      "Run Time = 62.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_autosomes.vcf\n",
      "\t--chr SM_V7_5\n",
      "\t--keep results/ld/east_africa.list\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 58 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 24681 out of a possible 475081 Sites\n",
      "Run Time = 39.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_autosomes.vcf\n",
      "\t--chr SM_V7_6\n",
      "\t--keep results/ld/east_africa.list\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 58 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 31877 out of a possible 475081 Sites\n",
      "Run Time = 23.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_autosomes.vcf\n",
      "\t--chr SM_V7_7\n",
      "\t--keep results/ld/east_africa.list\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 58 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 21340 out of a possible 475081 Sites\n",
      "Run Time = 47.00 seconds\n",
      "sed: can't read results/ld/smv7_ex_autosomes_east_africa_ld_dist.tbl: No such file or directory\n",
      "sed: can't read results/ld/smv7_ex_autosomes_east_africa_ld_dist.tbl: No such file or directory\n",
      "awk: cmd. line:1: fatal: cannot open file `results/ld/smv7_ex_autosomes_east_africa_ld_dist.tbl' for reading (No such file or directory)\n",
      "rm: cannot remove `results/ld/schMan_v7_west_africa_ld_dist.tbl': No such file or directory\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_autosomes.vcf\n",
      "\t--chr SM_V7_1\n",
      "\t--keep results/ld/west_africa.list\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 37 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 162219 out of a possible 475081 Sites\n",
      "Run Time = 90.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_autosomes.vcf\n",
      "\t--chr SM_V7_2\n",
      "\t--keep results/ld/west_africa.list\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 37 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 78902 out of a possible 475081 Sites\n",
      "Run Time = 54.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_autosomes.vcf\n",
      "\t--chr SM_V7_3\n",
      "\t--keep results/ld/west_africa.list\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 37 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 80481 out of a possible 475081 Sites\n",
      "Run Time = 36.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_autosomes.vcf\n",
      "\t--chr SM_V7_4\n",
      "\t--keep results/ld/west_africa.list\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 37 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 75581 out of a possible 475081 Sites\n",
      "Run Time = 55.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_autosomes.vcf\n",
      "\t--chr SM_V7_5\n",
      "\t--keep results/ld/west_africa.list\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 37 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 24681 out of a possible 475081 Sites\n",
      "Run Time = 14.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_autosomes.vcf\n",
      "\t--chr SM_V7_6\n",
      "\t--keep results/ld/west_africa.list\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 37 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 31877 out of a possible 475081 Sites\n",
      "Run Time = 39.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_autosomes.vcf\n",
      "\t--chr SM_V7_7\n",
      "\t--keep results/ld/west_africa.list\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 37 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 21340 out of a possible 475081 Sites\n",
      "Run Time = 15.00 seconds\n",
      "sed: can't read results/ld/smv7_ex_autosomes_west_africa_ld_dist.tbl: No such file or directory\n",
      "sed: can't read results/ld/smv7_ex_autosomes_west_africa_ld_dist.tbl: No such file or directory\n",
      "awk: cmd. line:1: fatal: cannot open file `results/ld/smv7_ex_autosomes_west_africa_ld_dist.tbl' for reading (No such file or directory)\n",
      "rm: cannot remove `results/ld/schMan_v7_new_world_ld_dist.tbl': No such file or directory\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_autosomes.vcf\n",
      "\t--chr SM_V7_1\n",
      "\t--keep results/ld/new_world.list\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 51 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 162219 out of a possible 475081 Sites\n",
      "Run Time = 80.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_autosomes.vcf\n",
      "\t--chr SM_V7_2\n",
      "\t--keep results/ld/new_world.list\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 51 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 78902 out of a possible 475081 Sites\n",
      "Run Time = 62.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_autosomes.vcf\n",
      "\t--chr SM_V7_3\n",
      "\t--keep results/ld/new_world.list\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 51 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 80481 out of a possible 475081 Sites\n",
      "Run Time = 69.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_autosomes.vcf\n",
      "\t--chr SM_V7_4\n",
      "\t--keep results/ld/new_world.list\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 51 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 75581 out of a possible 475081 Sites\n",
      "Run Time = 61.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_autosomes.vcf\n",
      "\t--chr SM_V7_5\n",
      "\t--keep results/ld/new_world.list\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 51 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 24681 out of a possible 475081 Sites\n",
      "Run Time = 38.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_autosomes.vcf\n",
      "\t--chr SM_V7_6\n",
      "\t--keep results/ld/new_world.list\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 51 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 31877 out of a possible 475081 Sites\n",
      "Run Time = 39.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_autosomes.vcf\n",
      "\t--chr SM_V7_7\n",
      "\t--keep results/ld/new_world.list\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 51 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 21340 out of a possible 475081 Sites\n",
      "Run Time = 36.00 seconds\n",
      "sed: can't read results/ld/smv7_ex_autosomes_new_world_ld_dist.tbl: No such file or directory\n",
      "sed: can't read results/ld/smv7_ex_autosomes_new_world_ld_dist.tbl: No such file or directory\n",
      "awk: cmd. line:1: fatal: cannot open file `results/ld/smv7_ex_autosomes_new_world_ld_dist.tbl' for reading (No such file or directory)\n"
     ]
    },
    {
     "ename": "CalledProcessError",
     "evalue": "Command 'b'\\n#get pop specific VCF files\\nfor POP in east_africa west_africa new_world; do\\n    if [ ! -f results/ld/schMan_v7_\"$POP\"_ld_dist.tbl ]; then\\n        rm results/ld/schMan_v7_\"$POP\"_ld_dist.tbl\\n    fi\\n    \\n    for CHR in \"1\" \"2\" \"3\" \"4\" \"5\" \"6\" \"7\"; do\\n    \\n        CHR=\"SM_V7_$CHR\"\\n    \\n        #get pop specific chr vcf\\n   \\n        vcftools \\\\\\n            --vcf results/variant_filtration/smv7_ex_autosomes.vcf \\\\\\n            --chr $CHR \\\\\\n            --keep results/ld/$POP.list \\\\\\n            --recode \\\\\\n            --recode-INFO-all \\\\\\n            --stdout \\\\\\n            >results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".vcf\\n    \\n        plink \\\\\\n            --threads 6 \\\\\\n            --vcf results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".vcf \\\\\\n            --out results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\" \\\\\\n            --double-id\\\\\\n            --recode12 \\\\\\n            --allow-extra-chr\\n\\n        #calculate R2 between all snps on a chr\\n        plink \\\\\\n            --threads 6 \\\\\\n            --r2 \\\\\\n            --file results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\" \\\\\\n            --out results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\" \\\\\\n            --double-id \\\\\\n            --allow-extra-chr \\\\\\n            --ld-window-r2 0.0 \\\\\\n            --ld-window 1000000 \\\\\\n            --ld-window-kb 90000\\n            \\n        #calc distance and generate a single table\\n        awk \\'{print $0\"\\\\t\"$5-$2}\\' \\\\\\n            results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".ld \\\\\\n            >>results/ld/smv7_ex_autosomes__\"$POP\"_ld_dist.tbl\\n                \\n        #clean up\\n        rm results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".ld\\n        rm results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".map\\n        rm results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".ped\\n        rm results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".log\\n        rm results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".nosex\\n        rm results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".vcf\\n\\n    done\\n    \\n        #remove the header line from the table (duplicated from each chrom)\\n        sed -i -e \\'1p\\' -e \\'/CHR/d\\' results/ld/smv7_ex_autosomes_\"$POP\"_ld_dist.tbl \\n        sed -i -e \\'1s/0/BP_DISTANCE/\\' results/ld/smv7_ex_autosomes_\"$POP\"_ld_dist.tbl\\n        \\n        #create a file with distances lt 500kb\\n        awk \\'{if ($8 <= 500000) print $0}\\' results/ld/smv7_ex_autosomes_\"$POP\"_ld_dist.tbl \\\\\\n            >results/ld/smv7_ex_autosomes_\"$POP\"_ld_dist_lt500kb.tbl\\ndone\\n'' returned non-zero exit status 2.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mCalledProcessError\u001b[0m                        Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-155-ce836a0f9fee>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mget_ipython\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_cell_magic\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'bash'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m''\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'\\n#get pop specific VCF files\\nfor POP in east_africa west_africa new_world; do\\n    if [ ! -f results/ld/schMan_v7_\"$POP\"_ld_dist.tbl ]; then\\n        rm results/ld/schMan_v7_\"$POP\"_ld_dist.tbl\\n    fi\\n    \\n    for CHR in \"1\" \"2\" \"3\" \"4\" \"5\" \"6\" \"7\"; do\\n    \\n        CHR=\"SM_V7_$CHR\"\\n    \\n        #get pop specific chr vcf\\n   \\n        vcftools \\\\\\n            --vcf results/variant_filtration/smv7_ex_autosomes.vcf \\\\\\n            --chr $CHR \\\\\\n            --keep results/ld/$POP.list \\\\\\n            --recode \\\\\\n            --recode-INFO-all \\\\\\n            --stdout \\\\\\n            >results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".vcf\\n    \\n        plink \\\\\\n            --threads 6 \\\\\\n            --vcf results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".vcf \\\\\\n            --out results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\" \\\\\\n            --double-id\\\\\\n            --recode12 \\\\\\n            --allow-extra-chr\\n\\n        #calculate R2 between all snps on a chr\\n        plink \\\\\\n            --threads 6 \\\\\\n            --r2 \\\\\\n            --file results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\" \\\\\\n            --out results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\" \\\\\\n            --double-id \\\\\\n            --allow-extra-chr \\\\\\n            --ld-window-r2 0.0 \\\\\\n            --ld-window 1000000 \\\\\\n            --ld-window-kb 90000\\n            \\n        #calc distance and generate a single table\\n        awk \\'{print $0\"\\\\t\"$5-$2}\\' \\\\\\n            results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".ld \\\\\\n            >>results/ld/smv7_ex_autosomes__\"$POP\"_ld_dist.tbl\\n                \\n        #clean up\\n        rm results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".ld\\n        rm results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".map\\n        rm results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".ped\\n        rm results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".log\\n        rm results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".nosex\\n        rm results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".vcf\\n\\n    done\\n    \\n        #remove the header line from the table (duplicated from each chrom)\\n        sed -i -e \\'1p\\' -e \\'/CHR/d\\' results/ld/smv7_ex_autosomes_\"$POP\"_ld_dist.tbl \\n        sed -i -e \\'1s/0/BP_DISTANCE/\\' results/ld/smv7_ex_autosomes_\"$POP\"_ld_dist.tbl\\n        \\n        #create a file with distances lt 500kb\\n        awk \\'{if ($8 <= 500000) print $0}\\' results/ld/smv7_ex_autosomes_\"$POP\"_ld_dist.tbl \\\\\\n            >results/ld/smv7_ex_autosomes_\"$POP\"_ld_dist_lt500kb.tbl\\ndone\\n'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/anaconda3/envs/sch_man_nwinvasion-nbanalyses/lib/python3.7/site-packages/IPython/core/interactiveshell.py\u001b[0m in \u001b[0;36mrun_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m   2357\u001b[0m             \u001b[0;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuiltin_trap\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2358\u001b[0m                 \u001b[0margs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mmagic_arg_s\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2359\u001b[0;31m                 \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2360\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2361\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/sch_man_nwinvasion-nbanalyses/lib/python3.7/site-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mnamed_script_magic\u001b[0;34m(line, cell)\u001b[0m\n\u001b[1;32m    140\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    141\u001b[0m                 \u001b[0mline\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mscript\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 142\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshebang\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    143\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    144\u001b[0m         \u001b[0;31m# write a basic docstring:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m</master/nplatt/anaconda3/envs/sch_man_nwinvasion-nbanalyses/lib/python3.7/site-packages/decorator.py:decorator-gen-110>\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/sch_man_nwinvasion-nbanalyses/lib/python3.7/site-packages/IPython/core/magic.py\u001b[0m in \u001b[0;36m<lambda>\u001b[0;34m(f, *a, **k)\u001b[0m\n\u001b[1;32m    185\u001b[0m     \u001b[0;31m# but it's overkill for just that one bit of state.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    186\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mmagic_deco\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 187\u001b[0;31m         \u001b[0mcall\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mlambda\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    188\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    189\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcallable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/sch_man_nwinvasion-nbanalyses/lib/python3.7/site-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n\u001b[1;32m    243\u001b[0m             \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstderr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mflush\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    244\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mraise_error\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m!=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 245\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mCalledProcessError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutput\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0merr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    246\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    247\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_run_script\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mto_close\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mCalledProcessError\u001b[0m: Command 'b'\\n#get pop specific VCF files\\nfor POP in east_africa west_africa new_world; do\\n    if [ ! -f results/ld/schMan_v7_\"$POP\"_ld_dist.tbl ]; then\\n        rm results/ld/schMan_v7_\"$POP\"_ld_dist.tbl\\n    fi\\n    \\n    for CHR in \"1\" \"2\" \"3\" \"4\" \"5\" \"6\" \"7\"; do\\n    \\n        CHR=\"SM_V7_$CHR\"\\n    \\n        #get pop specific chr vcf\\n   \\n        vcftools \\\\\\n            --vcf results/variant_filtration/smv7_ex_autosomes.vcf \\\\\\n            --chr $CHR \\\\\\n            --keep results/ld/$POP.list \\\\\\n            --recode \\\\\\n            --recode-INFO-all \\\\\\n            --stdout \\\\\\n            >results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".vcf\\n    \\n        plink \\\\\\n            --threads 6 \\\\\\n            --vcf results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".vcf \\\\\\n            --out results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\" \\\\\\n            --double-id\\\\\\n            --recode12 \\\\\\n            --allow-extra-chr\\n\\n        #calculate R2 between all snps on a chr\\n        plink \\\\\\n            --threads 6 \\\\\\n            --r2 \\\\\\n            --file results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\" \\\\\\n            --out results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\" \\\\\\n            --double-id \\\\\\n            --allow-extra-chr \\\\\\n            --ld-window-r2 0.0 \\\\\\n            --ld-window 1000000 \\\\\\n            --ld-window-kb 90000\\n            \\n        #calc distance and generate a single table\\n        awk \\'{print $0\"\\\\t\"$5-$2}\\' \\\\\\n            results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".ld \\\\\\n            >>results/ld/smv7_ex_autosomes__\"$POP\"_ld_dist.tbl\\n                \\n        #clean up\\n        rm results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".ld\\n        rm results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".map\\n        rm results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".ped\\n        rm results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".log\\n        rm results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".nosex\\n        rm results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".vcf\\n\\n    done\\n    \\n        #remove the header line from the table (duplicated from each chrom)\\n        sed -i -e \\'1p\\' -e \\'/CHR/d\\' results/ld/smv7_ex_autosomes_\"$POP\"_ld_dist.tbl \\n        sed -i -e \\'1s/0/BP_DISTANCE/\\' results/ld/smv7_ex_autosomes_\"$POP\"_ld_dist.tbl\\n        \\n        #create a file with distances lt 500kb\\n        awk \\'{if ($8 <= 500000) print $0}\\' results/ld/smv7_ex_autosomes_\"$POP\"_ld_dist.tbl \\\\\\n            >results/ld/smv7_ex_autosomes_\"$POP\"_ld_dist_lt500kb.tbl\\ndone\\n'' returned non-zero exit status 2."
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "#get pop specific VCF files\n",
    "for POP in east_africa west_africa new_world; do\n",
    "    if [ ! -f results/ld/schMan_v7_\"$POP\"_ld_dist.tbl ]; then\n",
    "        rm results/ld/schMan_v7_\"$POP\"_ld_dist.tbl\n",
    "    fi\n",
    "    \n",
    "    for CHR in \"1\" \"2\" \"3\" \"4\" \"5\" \"6\" \"7\"; do\n",
    "    \n",
    "        CHR=\"SM_V7_$CHR\"\n",
    "    \n",
    "        #get pop specific chr vcf\n",
    "   \n",
    "        vcftools \\\n",
    "            --vcf results/variant_filtration/smv7_ex_autosomes.vcf \\\n",
    "            --chr $CHR \\\n",
    "            --keep results/ld/$POP.list \\\n",
    "            --recode \\\n",
    "            --recode-INFO-all \\\n",
    "            --stdout \\\n",
    "            >results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".vcf\n",
    "    \n",
    "        plink \\\n",
    "            --threads 6 \\\n",
    "            --vcf results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".vcf \\\n",
    "            --out results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\" \\\n",
    "            --double-id\\\n",
    "            --recode12 \\\n",
    "            --allow-extra-chr\n",
    "\n",
    "        #calculate R2 between all snps on a chr\n",
    "        plink \\\n",
    "            --threads 6 \\\n",
    "            --r2 \\\n",
    "            --file results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\" \\\n",
    "            --out results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\" \\\n",
    "            --double-id \\\n",
    "            --allow-extra-chr \\\n",
    "            --ld-window-r2 0.0 \\\n",
    "            --ld-window 1000000 \\\n",
    "            --ld-window-kb 90000\n",
    "            \n",
    "        #calc distance and generate a single table\n",
    "        awk '{print $0\"\\t\"$5-$2}' \\\n",
    "            results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".ld \\\n",
    "            >>results/ld/smv7_ex_autosomes__\"$POP\"_ld_dist.tbl\n",
    "                \n",
    "        #clean up\n",
    "        rm results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".ld\n",
    "        rm results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".map\n",
    "        rm results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".ped\n",
    "        rm results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".log\n",
    "        rm results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".nosex\n",
    "        rm results/ld/smv7_ex_autosomes_\"$POP\"_\"$CHR\".vcf\n",
    "\n",
    "    done\n",
    "    \n",
    "        #remove the header line from the table (duplicated from each chrom)\n",
    "        sed -i -e '1p' -e '/CHR/d' results/ld/smv7_ex_autosomes_\"$POP\"_ld_dist.tbl \n",
    "        sed -i -e '1s/0/BP_DISTANCE/' results/ld/smv7_ex_autosomes_\"$POP\"_ld_dist.tbl\n",
    "        \n",
    "        #create a file with distances lt 500kb\n",
    "        awk '{if ($8 <= 500000) print $0}' results/ld/smv7_ex_autosomes_\"$POP\"_ld_dist.tbl \\\n",
    "            >results/ld/smv7_ex_autosomes_\"$POP\"_ld_dist_lt500kb.tbl\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "#in R generate summary stats for bins and plot\n",
    "\n",
    "#500bp bins for 500kb\n",
    "breaks <- seq( 0, 5e5, 500)\n",
    "\n",
    "centers   <- vector()\n",
    "means     <- vector()\n",
    "pops      <- vector()\n",
    "smootheds <- vector()\n",
    "\n",
    "for (pop in c(\"east_africa\", \"west_africa\", \"new_world\",)) {\n",
    "    \n",
    "    #read in lt 500kb r2 table from vcftools\n",
    "    ld_table <- read.table(paste(\"results/ld/schMan_v7_maf05_\", pop, \"_ld_dist_lt500kb.tbl\", sep=\"\"), \n",
    "                           header=FALSE)\n",
    "    \n",
    "    #bin r2 values and calculate stats\n",
    "    ld_binned <- stats.bin(ld_table$V8, ld_table$V7, breaks = breaks)\n",
    "    \n",
    "    #created regression line\n",
    "    loessMod  <- loess(ld_binned$stats[\"mean\",] ~ ld_binned$centers, span=0.50)\n",
    "    smoothed  <- predict(loessMod)\n",
    "    \n",
    "    #append all data to vectors\n",
    "    centers   <- append(centers, ld_binned$centers)\n",
    "    means     <- append(means, ld_binned$stats[\"mean\",])\n",
    "    pops      <- append(pops, rep(pop, length(ld_binned$centers)))\n",
    "    smootheds <- append(smootheds, smoothed)\n",
    "}    \n",
    "\n",
    "#build the dataframe and save to csv\n",
    "r2_df <- data.frame(centers, means, pops, smootheds)\n",
    "write.csv(r2_df, file = paste(\"results/ld/ld_dist_lt500kb_binned_smoothed.csv\", sep=\"\") ,row.names=FALSE)\n",
    "\n",
    "#subset desired populations\n",
    "major_groups <- subset(r2_df, pops == \"east_africa\" | \n",
    "                              pops == \"west_africa\" |\n",
    "                              pops == \"new_world\" ) \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "pop_colors <- c(\"eafrica\"   = \"green\",\n",
    "                \"oman\"      = \"yellow\", \n",
    "                \"wafrica\"   = \"red\",\n",
    "                \"brazil_x\"  = \"purple\",\n",
    "                \"caribbean\" = \"blue\")\n",
    "\n",
    "#start plotting\n",
    "p <- ggplot(major_groups, aes(x     = centers, \n",
    "                              y     = means, \n",
    "                              color = pops))\n",
    "\n",
    "#adjust colors\n",
    "p <- p + scale_colour_manual(values = pop_colors)\n",
    "\n",
    "#plot data\n",
    "p <- p + geom_point(alpha = 0.0)\n",
    "\n",
    "#smooth data with loess\n",
    "p <- p + geom_smooth(span   = 0.75, \n",
    "                     method = \"loess\", \n",
    "                     lwd    = 1, \n",
    "                     se     = FALSE)\n",
    "\n",
    "#modify x and y axis\n",
    "p <- p + scale_x_continuous(name   = \"Distance between SNPs (Kb)\", \n",
    "                            breaks = seq(0,500000,100000), \n",
    "                            labels = c(\"0\", \"100\", \"200\", \"300\", \"400\", \"500\"), \n",
    "                            expand = c(0,0),\n",
    "                            )\n",
    "p <- p + scale_y_continuous(name   = \"Mean R2\", \n",
    "                            expand = c(0,0), \n",
    "                            limits = c(0,1))\n",
    "\n",
    "#change theme\n",
    "p <- p + theme_bw()\n",
    "\n",
    "#change fonts on axis elements and titles\n",
    "p <- p + theme(axis.text  = element_text(size = 12),\n",
    "               axis.title = element_text(size = 14,\n",
    "                                         face = \"bold\"))\n",
    "\n",
    "#removing gridlines\n",
    "p <- p + theme(panel.grid.major = element_blank(), \n",
    "               panel.grid.minor = element_blank())\n",
    "\n",
    "#set up plot title etc.\n",
    "p <- p + ggtitle(\"LD Decay\")\n",
    "p <- p + theme(plot.title = element_text(hjust = 0.5, \n",
    "                                         vjust = 0.5, \n",
    "                                         face  = 'bold', \n",
    "                                          size  = 18))\n",
    "\n",
    "#modify legend\n",
    "p <- p + theme(legend.title         = element_text(size = 14, \n",
    "                                                   face = \"bold\"),\n",
    "               legend.text           = element_text(size = 12),\n",
    "               legend.position       = c(0.85, 0.85),\n",
    "               legend.box.background = element_rect(colour = \"black\"))\n",
    "p <- p + labs(col = \"Population\")\n",
    "p <- p + scale_color_manual(labels = c(\"Brazil\",\n",
    "                                       \"Caribbean\",\n",
    "                                       \"E. Africa\",\n",
    "                                       \"Oman\",\n",
    "                                       \"W. Africa\" ), \n",
    "                            values = c(pop_colors[\"brazil_x\"], \n",
    "                                       pop_colors[\"caribbean\"], \n",
    "                                       pop_colors[\"eafrica\"],\n",
    "                                       pop_colors[\"oman\"],\n",
    "                                       pop_colors[\"wafrica\"] ))\n",
    "\n",
    "#save the figure\n",
    "svg_img <- \"results/ld/ld_decay.svg\"\n",
    "png_img <- \"results/ld/ld_decay.png\"\n",
    "ggsave(png_img, plot = p, dpi = 600)\n",
    "ggsave(svg_img, plot = p)\n",
    "\n",
    "\n",
    "#display in notebook\n",
    "print(p)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "#### Bayescan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "import os\n",
    "from os import path\n",
    "import shutil\n",
    "\n",
    "#make a clean dir\n",
    "#if os.path.isdir(\"results/bayescan\"):\n",
    "#    shutil.rmtree(\"results/bayescan\")\n",
    "\n",
    "#os.mkdir(\"results/bayescan\")\n",
    "\n",
    "bayescan_pop_assignments = {\n",
    "    \"Sm.BR_PdV.0447.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.1039.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.1079.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.1094.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.1103.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.1127.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.1278.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.1340.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.1340.2\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.1371.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.1404.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.1409.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.1418.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.1475.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.1489.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2039.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2039.2\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2072.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2074.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2076.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2133.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2147.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2189.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2196.2\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2225.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2227.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2265.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2290.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2300.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2334.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2368.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2406.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2422.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2450.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2456.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2481.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2489.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2508.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2516.2\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2530.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2538.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2546.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2556.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.2577.1\"        : \"new_world\",\n",
    "    \"Sm.BR_PdV.4293.2\"        : \"new_world\",\n",
    "    \"Sm.NE_Di158.1\"           : \"west_africa\",\n",
    "    \"Sm.NE_Di186.1\"           : \"west_africa\",\n",
    "    \"Sm.NE_Di238.1\"           : \"west_africa\",\n",
    "    \"Sm.NE_Di297.1\"           : \"west_africa\",\n",
    "    \"Sm.NE_Di297.2\"           : \"west_africa\",\n",
    "    \"Sm.NE_Di68.2\"            : \"west_africa\",\n",
    "    \"Sm.NE_Na376.2\"           : \"west_africa\",\n",
    "    \"Sm.NE_Na381.1\"           : \"west_africa\",\n",
    "    \"Sm.NE_Na39.1\"            : \"west_africa\",\n",
    "    \"Sm.NE_Na40.1\"            : \"west_africa\",\n",
    "    \"Sm.SN_Nd103.1\"           : \"west_africa\",\n",
    "    \"Sm.SN_Nd109.1\"           : \"west_africa\",\n",
    "    \"Sm.SN_Nd114.1\"           : \"west_africa\",\n",
    "    \"Sm.SN_Nd115.1\"           : \"west_africa\",\n",
    "    \"Sm.SN_Nd18.1\"            : \"west_africa\",\n",
    "    \"Sm.SN_Nd22.1\"            : \"west_africa\",\n",
    "    \"Sm.SN_Nd24.1\"            : \"west_africa\",\n",
    "    \"Sm.SN_Nd25.1\"            : \"west_africa\",\n",
    "    \"Sm.SN_Nd34.1\"            : \"west_africa\",\n",
    "    \"Sm.SN_Nd43.1\"            : \"west_africa\",\n",
    "    \"Sm.SN_Nd47.1\"            : \"west_africa\",\n",
    "    \"Sm.SN_Nd5.1\"             : \"west_africa\",\n",
    "    \"Sm.SN_Nd5.2\"             : \"west_africa\",\n",
    "    \"Sm.SN_Nd50.1\"            : \"west_africa\",\n",
    "    \"Sm.SN_Nd54.1\"            : \"west_africa\",\n",
    "    \"Sm.SN_Nd56.1\"            : \"west_africa\",\n",
    "    \"Sm.SN_Nd77.1\"            : \"west_africa\",\n",
    "    \"Sm.SN_Nd79.1\"            : \"west_africa\",\n",
    "    \"Sm.SN_Nd9.1\"             : \"west_africa\",\n",
    "    \"Sm.SN_Nd90.1\"            : \"west_africa\",\n",
    "    \"Sm.SN_Te26.1\"            : \"west_africa\",\n",
    "    \"Sm.SN_Te3.1\"             : \"west_africa\",\n",
    "    \"Sm.SN_Te49.1\"            : \"west_africa\",\n",
    "    \"Sm.SN_Te55.1\"            : \"west_africa\",\n",
    "    \"Sm.SN_Te68.1\"            : \"west_africa\",\n",
    "    \"Sm.TZ_009.1.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_009.10.1\"          : \"east_africa\",\n",
    "    \"Sm.TZ_009.2.2\"           : \"east_africa\",\n",
    "    \"Sm.TZ_009.3.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_009.4.2\"           : \"east_africa\",\n",
    "    \"Sm.TZ_009.5.2\"           : \"east_africa\",\n",
    "    \"Sm.TZ_009.6.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_009.7.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_009.8.2\"           : \"east_africa\",\n",
    "    \"Sm.TZ_009.9.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_055.1.3\"           : \"east_africa\",\n",
    "    \"Sm.TZ_055.10.1\"          : \"east_africa\",\n",
    "    \"Sm.TZ_055.2.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_055.3.2\"           : \"east_africa\",\n",
    "    \"Sm.TZ_055.5.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_055.6.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_055.7.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_055.8.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_074N.1.2\"          : \"east_africa\",\n",
    "    \"Sm.TZ_074N.10.2\"         : \"east_africa\",\n",
    "    \"Sm.TZ_074N.2.2\"          : \"east_africa\",\n",
    "    \"Sm.TZ_074N.3.2\"          : \"east_africa\",\n",
    "    \"Sm.TZ_074N.4.1\"          : \"east_africa\",\n",
    "    \"Sm.TZ_074N.6.3\"          : \"east_africa\",\n",
    "    \"Sm.TZ_074N.7.2\"          : \"east_africa\",\n",
    "    \"Sm.TZ_074N.8.1\"          : \"east_africa\",\n",
    "    \"Sm.TZ_074N.9.1\"          : \"east_africa\",\n",
    "    \"Sm.TZ_077.2.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_077.3.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_077.4.2\"           : \"east_africa\",\n",
    "    \"Sm.TZ_077.5.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_077.6.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_077.7.3\"           : \"east_africa\",\n",
    "    \"Sm.TZ_077.8.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_077.9.2\"           : \"east_africa\",\n",
    "    \"Sm.TZ_086.1.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_086.2.3\"           : \"east_africa\",\n",
    "    \"Sm.TZ_086.3.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_086.4.2\"           : \"east_africa\",\n",
    "    \"Sm.TZ_086.5.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_086.6.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_086.7.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_086.8.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_134.1.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_134.2.2\"           : \"east_africa\",\n",
    "    \"Sm.TZ_134.4.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_134.5.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_134.6.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_141.1.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_141.3.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_141.4.2\"           : \"east_africa\",\n",
    "    \"Sm.TZ_141.5.3\"           : \"east_africa\",\n",
    "    \"Sm.TZ_141.6.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_141.7.1\"           : \"east_africa\",\n",
    "    \"Sm.TZ_141.8.1\"           : \"east_africa\" }\n",
    "\n",
    "\n",
    "#create files with lists of samples per population\n",
    "for sample in bayescan_pop_assignments:\n",
    "    with open(\"results/bayescan/\" + bayescan_pop_assignments[sample] + \".list\", 'a') as list_file:\n",
    "        list_file.write(sample + \"\\n\")\n",
    "    with open(\"results/bayescan/\" + bayescan_pop_assignments[sample] + \".pop\", 'a') as list_file:\n",
    "        list_file.write(sample + \"\\t\" + bayescan_pop_assignments[sample] + \"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%bash \n",
    "\n",
    "#get pgdspider to convert formats for bayescan\n",
    "if [ ! -f bin/PGDSpider_2.1.1.5/PGDSpider2-cli.jar ]; then\n",
    "    wget -P bin/ http://www.cmpg.unibe.ch/software/PGDSpider/PGDSpider_2.1.1.5.zip\n",
    "    unzip bin/PGDSpider_2.1.1.5.zip -d bin\n",
    "fi\n",
    "\n",
    "if [ ! -f bin/BayeScan2.1/binaries/BayeScan2.1_linux64bits ]; then\n",
    "    wget -P bin/ http://cmpg.unibe.ch/software/BayeScan/files/BayeScan2.1.zip\n",
    "    unzip bin/BayeScan2.1.zip -d bin\n",
    "    chmod u+x bin/BayeScan2.1/binaries/BayeScan2.1_linux64bits\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#populations comparisons to make\n",
    "bayes_comparisons=[  [\"new_world\",   \"west_africa\" ],\n",
    "                     [\"new_world\",   \"east_africa\" ],\n",
    "                     [\"west_africa\", \"east_africa\" ]  ]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#prep input files and submit bayescan runs\n",
    "for comparison in bayes_comparisons:\n",
    "    pop1 = comparison[0]\n",
    "    pop2 = comparison[1]\n",
    "\n",
    "    #get vcf\n",
    "    get_vcf = \"vcftools \\\n",
    "    --vcf results/variant_filtration/smv7_ex_autosomes.vcf \\\n",
    "    --keep results/bayescan/\" + pop1 + \".list \\\n",
    "    --keep results/bayescan/\" + pop2 + \".list \\\n",
    "    --mac 2 \\\n",
    "    --min-alleles 2 \\\n",
    "    --max-alleles 2 \\\n",
    "    --stdout \\\n",
    "    --recode \\\n",
    "    >results/bayescan/\" + pop1 + \"_\" + pop2 + \".vcf\"\n",
    "    os.system(get_vcf)\n",
    "\n",
    "    #get list of sites (for plotting downstream)\n",
    "    with open(\"results/bayescan/\" + pop1 + \"_\" + pop2 + \".sites\", 'w') as sites_file:\n",
    "        outline = \"chrom\\tpos\\n\".format(\"chrom\", \"pos\")\n",
    "        sites_file.write(outline)\n",
    "\n",
    "        vcf   = allel.read_vcf('results/bayescan/' + pop1 + '_' + pop2 + '.vcf')\n",
    "        chrom = list(vcf['variants/CHROM'])\n",
    "        pos   = list(vcf['variants/POS'])\n",
    "\n",
    "        for i in list(range(0, len(chrom))):\n",
    "            outline = \"{0}\\t{1}\\n\".format(chrom[i], pos[i])\n",
    "            sites_file.write(outline)    \n",
    "    #-----------------------------------------------------------------------------\n",
    "    #convert with spider\n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#prep input files and submit bayescan runs\n",
    "for comparison in bayes_comparisons:\n",
    "    pop1 = comparison[0]\n",
    "    pop2 = comparison[1]\n",
    "    \n",
    "    #create pop file\n",
    "    !cat results/bayescan/{pop1}.pop results/bayescan/{pop2}.pop >results/bayescan/{pop1}_{pop2}.pop\n",
    "    \n",
    "    #create spid for conversion from vcf to pgd\n",
    "    vcf_pgd_spid ='''# spid-file generated: NA\n",
    "\n",
    "# VCF Parser questions\n",
    "PARSER_FORMAT=VCF\n",
    "\n",
    "# Only output SNPs with a phred-scaled quality of at least:\n",
    "VCF_PARSER_QUAL_QUESTION=\n",
    "# Select population definition file:\n",
    "VCF_PARSER_POP_FILE_QUESTION=./results/bayescan/{pop1}_{pop2}.pop\n",
    "# What is the ploidy of the data?\n",
    "VCF_PARSER_PLOIDY_QUESTION=DIPLOID\n",
    "# Do you want to include a file with population definitions?\n",
    "VCF_PARSER_POP_QUESTION=true\n",
    "# Output genotypes as missing if the phred-scale genotype quality is below:\n",
    "VCF_PARSER_GTQUAL_QUESTION=\n",
    "# Do you want to include non-polymorphic SNPs?\n",
    "VCF_PARSER_MONOMORPHIC_QUESTION=false\n",
    "# Only output following individuals (ind1, ind2, ind4, ...):\n",
    "VCF_PARSER_IND_QUESTION=\n",
    "# Only input following regions (refSeqName:start:end, multiple regions: whitespace separated):\n",
    "VCF_PARSER_REGION_QUESTION=\n",
    "# Output genotypes as missing if the read depth of a position for the sample is below:\n",
    "VCF_PARSER_READ_QUESTION=\n",
    "# Take most likely genotype if \"PL\" or \"GL\" is given in the genotype field?\n",
    "VCF_PARSER_PL_QUESTION=false\n",
    "# Do you want to exclude loci with only missing data?\n",
    "VCF_PARSER_EXC_MISSING_LOCI_QUESTION=false\n",
    "\n",
    "# PGD Writer questions\n",
    "WRITER_FORMAT=PGD\n",
    "'''.format(pop1 = pop1, \n",
    "           pop2 = pop2)\n",
    "    \n",
    "    #write to file\n",
    "    with open(\"results/bayescan/\" + pop1 + \"_\" + pop2 + \"_vcf_pgd.spid\", 'w') as vcf_pgd_spid_file:\n",
    "       vcf_pgd_spid_file.write(vcf_pgd_spid)\n",
    "\n",
    "    pgd_bayescan_spid = '''# spid-file generated: NA\n",
    "    \n",
    "# VCF Parser questions\n",
    "PARSER_FORMAT=PGD\n",
    "\n",
    "# Only output SNPs with a phred-scaled quality of at least:\n",
    "VCF_PARSER_QUAL_QUESTION=\n",
    "# Select population definition file:\n",
    "VCF_PARSER_POP_FILE_QUESTION=./results/bayescan/{pop1}_{pop2}.pop\n",
    "# What is the ploidy of the data?\n",
    "VCF_PARSER_PLOIDY_QUESTION=DIPLOID\n",
    "# Do you want to include a file with population definitions?\n",
    "VCF_PARSER_POP_QUESTION=true\n",
    "# Output genotypes as missing if the phred-scale genotype quality is below:\n",
    "VCF_PARSER_GTQUAL_QUESTION=\n",
    "# Do you want to include non-polymorphic SNPs?\n",
    "VCF_PARSER_MONOMORPHIC_QUESTION=false\n",
    "# Only output following individuals (ind1, ind2, ind4, ...):\n",
    "VCF_PARSER_IND_QUESTION=\n",
    "# Only input following regions (refSeqName:start:end, multiple regions: whitespace separated):\n",
    "VCF_PARSER_REGION_QUESTION=\n",
    "# Output genotypes as missing if the read depth of a position for the sample is below:\n",
    "VCF_PARSER_READ_QUESTION=\n",
    "# Take most likely genotype if \"PL\" or \"GL\" is given in the genotype field?\n",
    "VCF_PARSER_PL_QUESTION=false\n",
    "# Do you want to exclude loci with only missing data?\n",
    "VCF_PARSER_EXC_MISSING_LOCI_QUESTION=false\n",
    "\n",
    "# PGD Writer questions\n",
    "WRITER_FORMAT=BAYESCAN\n",
    "'''.format(pop1 = pop1, \n",
    "           pop2 = pop2)\n",
    "    \n",
    "    with open(\"results/bayescan/\" + pop1 + \"_\" + pop2 + \"_pgd_bayescan.spid\", 'w') as pgd_bayescan_spid_file:\n",
    "        pgd_bayescan_spid_file.write(pgd_bayescan_spid)\n",
    "        \n",
    "    vcf_to_pgd = \"java -jar bin/PGDSpider_2.1.1.5/PGDSpider2-cli.jar \\\n",
    "        -inputfile results/bayescan/{pop1}_{pop2}.vcf \\\n",
    "        -inputformat VCF \\\n",
    "        -outputfile results/bayescan/{pop1}_{pop2}.pgd \\\n",
    "        -outputformat PGD \\\n",
    "        -spid results/bayescan/{pop1}_{pop2}_vcf_pgd.spid\".format(pop1=pop1, pop2=pop2)\n",
    "\n",
    "    pgd_to_bayescan = \"java -jar bin/PGDSpider_2.1.1.5/PGDSpider2-cli.jar \\\n",
    "        -inputfile results/bayescan/{pop1}_{pop2}.pgd \\\n",
    "        -inputformat PGD \\\n",
    "        -outputfile results/bayescan/{pop1}_{pop2}.bayescan \\\n",
    "        -outputformat GESTE_BAYE_SCAN \\\n",
    "        -spid results/bayescan/{pop1}_{pop2}_pgd_bayescan.spid\".format(pop1=pop1, pop2=pop2)\n",
    "    \n",
    "    os.system(vcf_to_pgd)\n",
    "    os.system(pgd_to_bayescan)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[1]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(range(1,2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "for comparison in bayes_comparisons:\n",
    "    pop1 = comparison[0]\n",
    "    pop2 = comparison[1]\n",
    "    \n",
    "    for chain in list(range(1, 2)):\n",
    "\n",
    "        chain=\"{chain:02d}\".format(chain=chain)\n",
    "        #create a CLEAN chain specific dir\n",
    "        chain_dir=\"results/bayescan/{pop1}_{pop2}/chain_{chain}\".format(pop1=pop1, pop2=pop2, chain=chain)\n",
    "        if os.path.isdir(chain_dir):\n",
    "            shutil.rmtree(chain_dir)\n",
    "        os.makedirs(chain_dir, exist_ok=True)\n",
    "    \n",
    "        #set bayescan paramters \n",
    "        job_id=\"{pop1}_{pop2}_chain{chain}\".format(pop1=pop1, pop2=pop2, chain=chain)\n",
    "        job_log=\"{chain_dir}/{job_id}.log\".format(chain_dir=chain_dir, job_id=job_id)\n",
    "        infile=\"results/bayescan/{pop1}_{pop2}.bayescan\".format(pop1=pop1, pop2=pop2)\n",
    "        threads=12\n",
    "        \n",
    "        #build bayescan code\n",
    "        cmd=\"conda activate sch_man_nwinvasion; \\\n",
    "            bin/BayeScan2.1/binaries/BayeScan2.1_linux64bits  \\\n",
    "                {infile} \\\n",
    "                -threads {threads} \\\n",
    "                -od {chain_dir}/ \\\n",
    "                -o {out_prefix} \\\n",
    "                -pr_odds {prior_odds} \\\n",
    "                -pilot {pilot} \\\n",
    "                -burn {burnin} \\\n",
    "                -nbp {nbp} \\\n",
    "                -n {gens} \\\n",
    "                -thin {thin}\".format(infile=infile,\n",
    "                                     pop1=pop1, \n",
    "                                     pop2=pop2,\n",
    "                                     threads=threads,\n",
    "                                     chain_dir=chain_dir,\n",
    "                                     out_prefix=\"bayescan\",\n",
    "                                     prior_odds=10,\n",
    "                                     pilot=5_000,\n",
    "                                     burnin=5_000,\n",
    "                                     nbp=20,\n",
    "                                     gens=50_000,\n",
    "                                     thin=20)\n",
    "\n",
    "        #print parameters for logging purposes\n",
    "        with open(chain_dir + \"/run.sh\", 'w') as run_sh:\n",
    "            run_sh.write(cmd)\n",
    "        \n",
    "        #build command to submit to scheduler\n",
    "        qsub=\"echo \\\"{cmd}\\\" | qsub -V -cwd -S /bin/bash -q all.q -j y -N {job_id} -o {job_log} -pe smp {threads}\".format(cmd     = cmd,\n",
    "                                                                                                                          job_id  = job_id,\n",
    "                                                                                                                          job_log = job_log,\n",
    "                                                                                                                          threads = threads)\n",
    "        \n",
    "        #submit\n",
    "        os.system(qsub)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "echo \"conda activate sch_man_nwinvasion;             bin/BayeScan2.1/binaries/BayeScan2.1_linux64bits                  results/bayescan/west_africa_east_africa_maf05.bayescan                 -threads 12                 -od results/bayescan/west_africa_east_africa/chain_01/                 -o bayescan                 -pr_odds 10                 -pilot 5000                 -burn 5000                 -nbp 20                 -n 50000                 -thin 20\" | qsub -V -cwd -S /bin/bash -q all.q -j y -N chain01 -o results/bayescan/west_africa_east_africa/chain_01/chain01.log -pe smp 12\n"
     ]
    }
   ],
   "source": [
    "print(qsub)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "# #check to ID min qvalues (see if any sig sel...quick and dirty)\n",
    "# fst_files<-list.files(\"results/bayescan\", pattern=\"*fst.txt\", recursive=TRUE, full.names=TRUE)\n",
    "\n",
    "# min_qs<-vector()\n",
    "\n",
    "# for ( fst_file in fst_files){\n",
    "\n",
    "\n",
    "#     fst_tbl<-read.table(fst_file, header=TRUE)\n",
    "    \n",
    "#     min_q <- min(fst_tbl$qval)\n",
    "    \n",
    "#     min_qs <- append(min_qs, min_q)\n",
    "# }\n",
    "\n",
    "#print(fst_files)\n",
    "# svg(\"results/bayescan/min_qs.svg\")\n",
    "#     plot(min_qs, ylim=c(0, 0.5), pch=21, col=\"black\", cex=1.5, bg=\"grey\", ylab=\"Min. q-value\", xlab=\"\")\n",
    "#     abline(h=min(min_qs), lwd=1, col=\"black\", lty=2)\n",
    "#     abline(h=0.05, lwd=1, col=\"red\", lty=2)\n",
    "#     text(x=0, y = 0.17, labels = \"Brazil v Africa\", col=\"black\", cex=1, adj=0)\n",
    "#     text(x=10, y = 0.425, labels = \"Brazil v Niger\", col=\"black\", cex=1, adj=0)\n",
    "#     text(x=20, y = 0.115, labels = \"Brazil v Senegal\", col=\"black\", cex=1, adj=0)\n",
    "#     text(x=29, y = 0.25, labels = \"Brazil v Tanzania\", col=\"black\", cex=1, adj=0)\n",
    "#     text(x=39, y = 0.18, labels = \"Brazil v wAfrica\", col=\"black\", cex=1, adj=0)\n",
    "#     text(x=43, y = 0.45, labels = \"wAfrica v Tanzania\", col=\"black\", cex=1, adj=0)\n",
    "# dev.off()\n",
    "\n",
    "source(\"\")\n",
    "plot_bayescan(\"results/bayescan/brazil_africa/chain_00/bayescan_fst.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "paste ../../brazil_africa.sites bayescan_fst.txt | sed 's/ \\+/,/gi' | sed 's/\\t/,/gi' | sed '2,$s/.$//' | cut --complement -f3 -d\",\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "bs<-read.csv(\"results/bayescan/brazil_africa/chain_00/bayescan_fst.csv\", header=TRUE, sep=\",\")\n",
    "\n",
    "#head(bs)\n",
    "cumul_center_pos<-get_cumul_pos(bs$chrom, bs$pos)\n",
    "\n",
    "#build dataframe\n",
    "bs_df<-data.frame(bs, cumul_center_pos)\n",
    "\n",
    "#start plotting\n",
    "p <- ggplot(bs_df, \n",
    "            aes(x     = cumul_center_pos, \n",
    "                y     = alpha, \n",
    "                color = chrom))\n",
    "\n",
    "#adjust colors\n",
    "p <- p + scale_colour_manual(values = chr_colors)\n",
    "\n",
    "#plot data\n",
    "p <- p + geom_point(alpha = 1)\n",
    "\n",
    "#add horizontal line at 99.9th percintile\n",
    "# p <- p + geom_hline(yintercept = quantile(pbs_df$PBS, 0.999), \n",
    "#                     linetype   = \"dashed\", \n",
    "#                     color      = \"red\")\n",
    "# p <- p + geom_hline(yintercept = quantile(pbs_df$PBS, 0.001), \n",
    "#                     linetype   = \"dashed\", \n",
    "#                     color      = \"red\")\n",
    "\n",
    "#modify x and y axis\n",
    "p <- p + scale_x_continuous(name   = \"\", \n",
    "                            breaks = cumul_starts, \n",
    "                            expand = c(0,0))\n",
    "p <- p + scale_y_continuous(name   = \"Alpha\", \n",
    "                            expand = c(0,0),\n",
    "                            limits = c(min(bs$alpha * 1.15), \n",
    "                                       max(bs$alpha * 1.15)),\n",
    "                            #breaks = c(-0.2, 0, 0.2, 0.4, 0.6, 0.8)\n",
    "                           )\n",
    "\n",
    "#change theme\n",
    "p <- p + theme_bw()\n",
    "\n",
    "#change fonts on axis elements and titles\n",
    "p <- p + theme(axis.text   = element_text(size = 12),\n",
    "               axis.title  = element_text(size = 12,\n",
    "                                          face = \"bold\"),\n",
    "               axis.text.x = element_blank())\n",
    "\n",
    "#removing gridlines\n",
    "p <- p + theme(panel.grid.major  = element_blank(), \n",
    "                panel.grid.minor = element_blank())\n",
    "\n",
    "#set up plot title etc.\n",
    "#p <- p + ggtitle(\"PBS (Brazil; WAfrica; EAfrica)\")\n",
    "# p <- p + theme(plot.title = element_text(hjust = 0.5, \n",
    "#                                          vjust = 0.5, \n",
    "#                                          face  = 'bold', \n",
    "#                                          size  = 18))\n",
    "\n",
    "# #modify legend\n",
    "p <- p + theme(legend.position = \"none\")\n",
    "\n",
    "#save the figure\n",
    "svg_img <- \"results/bayescan/b_v_a.svg\"\n",
    "png_img <- \"results/bayescan/b_v_a.png\"\n",
    "ggsave(png_img, plot = p, dpi = 600)\n",
    "ggsave(svg_img, plot = p)\n",
    "\n",
    "ggsave(\"results/bayescan/b_v_a.COMPOSITE.png\", plot = p, width = 10, height = 1.5, units = \"in\", dpi = 300)\n",
    "# #display in notebook\n",
    "print(p)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
