{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-23T01:40:26.654860Z",
     "start_time": "2020-06-23T01:40:26.647342Z"
    }
   },
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "os.chdir(\"/master/nplatt/sch_man_nwinvasion\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-29T20:41:30.092973Z",
     "start_time": "2020-06-29T20:41:30.086176Z"
    },
    "heading_collapsed": true
   },
   "source": [
    "# Neutrally evolving loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## msPrime\n",
    "Simulate neutral VCFS for each population"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "I am using the probed MSprime runs from the full dataset but instead randomly downsampling each replicate to 10 indiviudals.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-23T15:52:56.288307Z",
     "start_time": "2020-06-23T15:51:11.241046Z"
    },
    "collapsed": true,
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5672368 (\"tanzania_1\") has been submitted\n",
      "Your job 5672369 (\"tanzania_2\") has been submitted\n",
      "Your job 5672370 (\"tanzania_3\") has been submitted\n",
      "Your job 5672371 (\"tanzania_4\") has been submitted\n",
      "Your job 5672372 (\"tanzania_5\") has been submitted\n",
      "Your job 5672373 (\"tanzania_6\") has been submitted\n",
      "Your job 5672374 (\"tanzania_7\") has been submitted\n",
      "Your job 5672375 (\"tanzania_8\") has been submitted\n",
      "Your job 5672376 (\"tanzania_9\") has been submitted\n",
      "Your job 5672377 (\"tanzania_10\") has been submitted\n",
      "Your job 5672378 (\"tanzania_11\") has been submitted\n",
      "Your job 5672379 (\"tanzania_12\") has been submitted\n",
      "Your job 5672380 (\"tanzania_13\") has been submitted\n",
      "Your job 5672381 (\"tanzania_14\") has been submitted\n",
      "Your job 5672382 (\"tanzania_15\") has been submitted\n",
      "Your job 5672383 (\"tanzania_16\") has been submitted\n",
      "Your job 5672384 (\"tanzania_17\") has been submitted\n",
      "Your job 5672385 (\"tanzania_18\") has been submitted\n",
      "Your job 5672386 (\"tanzania_19\") has been submitted\n",
      "Your job 5672387 (\"tanzania_20\") has been submitted\n",
      "Your job 5672388 (\"tanzania_21\") has been submitted\n",
      "Your job 5672389 (\"tanzania_22\") has been submitted\n",
      "Your job 5672390 (\"tanzania_23\") has been submitted\n",
      "Your job 5672391 (\"tanzania_24\") has been submitted\n",
      "Your job 5672392 (\"tanzania_25\") has been submitted\n",
      "Your job 5672393 (\"tanzania_26\") has been submitted\n",
      "Your job 5672394 (\"tanzania_27\") has been submitted\n",
      "Your job 5672395 (\"tanzania_28\") has been submitted\n",
      "Your job 5672396 (\"tanzania_29\") has been submitted\n",
      "Your job 5672397 (\"tanzania_30\") has been submitted\n",
      "Your job 5672398 (\"tanzania_31\") has been submitted\n",
      "Your job 5672399 (\"tanzania_32\") has been submitted\n",
      "Your job 5672400 (\"tanzania_33\") has been submitted\n",
      "Your job 5672401 (\"tanzania_34\") has been submitted\n",
      "Your job 5672402 (\"tanzania_35\") has been submitted\n",
      "Your job 5672403 (\"tanzania_36\") has been submitted\n",
      "Your job 5672404 (\"tanzania_37\") has been submitted\n",
      "Your job 5672405 (\"tanzania_38\") has been submitted\n",
      "Your job 5672406 (\"tanzania_39\") has been submitted\n",
      "Your job 5672407 (\"tanzania_40\") has been submitted\n",
      "Your job 5672408 (\"tanzania_41\") has been submitted\n",
      "Your job 5672409 (\"tanzania_42\") has been submitted\n",
      "Your job 5672410 (\"tanzania_43\") has been submitted\n",
      "Your job 5672411 (\"tanzania_44\") has been submitted\n",
      "Your job 5672412 (\"tanzania_45\") has been submitted\n",
      "Your job 5672413 (\"tanzania_46\") has been submitted\n",
      "Your job 5672414 (\"tanzania_47\") has been submitted\n",
      "Your job 5672415 (\"tanzania_48\") has been submitted\n",
      "Your job 5672416 (\"tanzania_49\") has been submitted\n",
      "Your job 5672417 (\"tanzania_50\") has been submitted\n",
      "Your job 5672418 (\"tanzania_51\") has been submitted\n",
      "Your job 5672419 (\"tanzania_52\") has been submitted\n",
      "Your job 5672420 (\"tanzania_53\") has been submitted\n",
      "Your job 5672421 (\"tanzania_54\") has been submitted\n",
      "Your job 5672422 (\"tanzania_55\") has been submitted\n",
      "Your job 5672423 (\"tanzania_56\") has been submitted\n",
      "Your job 5672424 (\"tanzania_57\") has been submitted\n",
      "Your job 5672425 (\"tanzania_58\") has been submitted\n",
      "Your job 5672426 (\"tanzania_59\") has been submitted\n",
      "Your job 5672427 (\"tanzania_60\") has been submitted\n",
      "Your job 5672428 (\"tanzania_61\") has been submitted\n",
      "Your job 5672429 (\"tanzania_62\") has been submitted\n",
      "Your job 5672430 (\"tanzania_63\") has been submitted\n",
      "Your job 5672431 (\"tanzania_64\") has been submitted\n",
      "Your job 5672432 (\"tanzania_65\") has been submitted\n",
      "Your job 5672433 (\"tanzania_66\") has been submitted\n",
      "Your job 5672434 (\"tanzania_67\") has been submitted\n",
      "Your job 5672435 (\"tanzania_68\") has been submitted\n",
      "Your job 5672436 (\"tanzania_69\") has been submitted\n",
      "Your job 5672437 (\"tanzania_70\") has been submitted\n",
      "Your job 5672438 (\"tanzania_71\") has been submitted\n",
      "Your job 5672439 (\"tanzania_72\") has been submitted\n",
      "Your job 5672440 (\"tanzania_73\") has been submitted\n",
      "Your job 5672441 (\"tanzania_74\") has been submitted\n",
      "Your job 5672442 (\"tanzania_75\") has been submitted\n",
      "Your job 5672443 (\"tanzania_76\") has been submitted\n",
      "Your job 5672444 (\"tanzania_77\") has been submitted\n",
      "Your job 5672445 (\"tanzania_78\") has been submitted\n",
      "Your job 5672446 (\"tanzania_79\") has been submitted\n",
      "Your job 5672447 (\"tanzania_80\") has been submitted\n",
      "Your job 5672448 (\"tanzania_81\") has been submitted\n",
      "Your job 5672449 (\"tanzania_82\") has been submitted\n",
      "Your job 5672450 (\"tanzania_83\") has been submitted\n",
      "Your job 5672451 (\"tanzania_84\") has been submitted\n",
      "Your job 5672452 (\"tanzania_85\") has been submitted\n",
      "Your job 5672453 (\"tanzania_86\") has been submitted\n",
      "Your job 5672454 (\"tanzania_87\") has been submitted\n",
      "Your job 5672455 (\"tanzania_88\") has been submitted\n",
      "Your job 5672456 (\"tanzania_89\") has been submitted\n",
      "Your job 5672457 (\"tanzania_90\") has been submitted\n",
      "Your job 5672458 (\"tanzania_91\") has been submitted\n",
      "Your job 5672459 (\"tanzania_92\") has been submitted\n",
      "Your job 5672460 (\"tanzania_93\") has been submitted\n",
      "Your job 5672461 (\"tanzania_94\") has been submitted\n",
      "Your job 5672462 (\"tanzania_95\") has been submitted\n",
      "Your job 5672463 (\"tanzania_96\") has been submitted\n",
      "Your job 5672464 (\"tanzania_97\") has been submitted\n",
      "Your job 5672465 (\"tanzania_98\") has been submitted\n",
      "Your job 5672466 (\"tanzania_99\") has been submitted\n",
      "Your job 5672467 (\"tanzania_100\") has been submitted\n",
      "Your job 5672468 (\"tanzania_101\") has been submitted\n",
      "Your job 5672469 (\"tanzania_102\") has been submitted\n",
      "Your job 5672470 (\"tanzania_103\") has been submitted\n",
      "Your job 5672471 (\"tanzania_104\") has been submitted\n",
      "Your job 5672472 (\"tanzania_105\") has been submitted\n",
      "Your job 5672473 (\"tanzania_106\") has been submitted\n",
      "Your job 5672474 (\"tanzania_107\") has been submitted\n",
      "Your job 5672475 (\"tanzania_108\") has been submitted\n",
      "Your job 5672476 (\"tanzania_109\") has been submitted\n",
      "Your job 5672477 (\"tanzania_110\") has been submitted\n",
      "Your job 5672478 (\"tanzania_111\") has been submitted\n",
      "Your job 5672479 (\"tanzania_112\") has been submitted\n",
      "Your job 5672480 (\"tanzania_113\") has been submitted\n",
      "Your job 5672481 (\"tanzania_114\") has been submitted\n",
      "Your job 5672482 (\"tanzania_115\") has been submitted\n",
      "Your job 5672483 (\"tanzania_116\") has been submitted\n",
      "Your job 5672484 (\"tanzania_117\") has been submitted\n",
      "Your job 5672485 (\"tanzania_118\") has been submitted\n",
      "Your job 5672486 (\"tanzania_119\") has been submitted\n",
      "Your job 5672487 (\"tanzania_120\") has been submitted\n",
      "Your job 5672488 (\"tanzania_121\") has been submitted\n",
      "Your job 5672489 (\"tanzania_122\") has been submitted\n",
      "Your job 5672490 (\"tanzania_123\") has been submitted\n",
      "Your job 5672491 (\"tanzania_124\") has been submitted\n",
      "Your job 5672492 (\"tanzania_125\") has been submitted\n",
      "Your job 5672493 (\"tanzania_126\") has been submitted\n",
      "Your job 5672494 (\"tanzania_127\") has been submitted\n",
      "Your job 5672495 (\"tanzania_128\") has been submitted\n",
      "Your job 5672496 (\"tanzania_129\") has been submitted\n",
      "Your job 5672497 (\"tanzania_130\") has been submitted\n",
      "Your job 5672498 (\"tanzania_131\") has been submitted\n",
      "Your job 5672499 (\"tanzania_132\") has been submitted\n",
      "Your job 5672500 (\"tanzania_133\") has been submitted\n",
      "Your job 5672501 (\"tanzania_134\") has been submitted\n",
      "Your job 5672502 (\"tanzania_135\") has been submitted\n",
      "Your job 5672503 (\"tanzania_136\") has been submitted\n",
      "Your job 5672504 (\"tanzania_137\") has been submitted\n",
      "Your job 5672505 (\"tanzania_138\") has been submitted\n",
      "Your job 5672506 (\"tanzania_139\") has been submitted\n",
      "Your job 5672507 (\"tanzania_140\") has been submitted\n",
      "Your job 5672508 (\"tanzania_141\") has been submitted\n",
      "Your job 5672509 (\"tanzania_142\") has been submitted\n",
      "Your job 5672510 (\"tanzania_143\") has been submitted\n",
      "Your job 5672511 (\"tanzania_144\") has been submitted\n",
      "Your job 5672512 (\"tanzania_145\") has been submitted\n",
      "Your job 5672513 (\"tanzania_146\") has been submitted\n",
      "Your job 5672514 (\"tanzania_147\") has been submitted\n",
      "Your job 5672515 (\"tanzania_148\") has been submitted\n",
      "Your job 5672516 (\"tanzania_149\") has been submitted\n",
      "Your job 5672517 (\"tanzania_150\") has been submitted\n",
      "Your job 5672518 (\"tanzania_151\") has been submitted\n",
      "Your job 5672519 (\"tanzania_152\") has been submitted\n",
      "Your job 5672520 (\"tanzania_153\") has been submitted\n",
      "Your job 5672521 (\"tanzania_154\") has been submitted\n",
      "Your job 5672522 (\"tanzania_155\") has been submitted\n",
      "Your job 5672523 (\"tanzania_156\") has been submitted\n",
      "Your job 5672524 (\"tanzania_157\") has been submitted\n",
      "Your job 5672525 (\"tanzania_158\") has been submitted\n",
      "Your job 5672526 (\"tanzania_159\") has been submitted\n",
      "Your job 5672527 (\"tanzania_160\") has been submitted\n",
      "Your job 5672528 (\"tanzania_161\") has been submitted\n",
      "Your job 5672529 (\"tanzania_162\") has been submitted\n",
      "Your job 5672530 (\"tanzania_163\") has been submitted\n",
      "Your job 5672531 (\"tanzania_164\") has been submitted\n",
      "Your job 5672532 (\"tanzania_165\") has been submitted\n",
      "Your job 5672533 (\"tanzania_166\") has been submitted\n",
      "Your job 5672534 (\"tanzania_167\") has been submitted\n",
      "Your job 5672535 (\"tanzania_168\") has been submitted\n",
      "Your job 5672536 (\"tanzania_169\") has been submitted\n",
      "Your job 5672537 (\"tanzania_170\") has been submitted\n",
      "Your job 5672538 (\"tanzania_171\") has been submitted\n",
      "Your job 5672539 (\"tanzania_172\") has been submitted\n",
      "Your job 5672540 (\"tanzania_173\") has been submitted\n",
      "Your job 5672541 (\"tanzania_174\") has been submitted\n",
      "Your job 5672542 (\"tanzania_175\") has been submitted\n",
      "Your job 5672543 (\"tanzania_176\") has been submitted\n",
      "Your job 5672544 (\"tanzania_177\") has been submitted\n",
      "Your job 5672545 (\"tanzania_178\") has been submitted\n",
      "Your job 5672546 (\"tanzania_179\") has been submitted\n",
      "Your job 5672547 (\"tanzania_180\") has been submitted\n",
      "Your job 5672548 (\"tanzania_181\") has been submitted\n",
      "Your job 5672549 (\"tanzania_182\") has been submitted\n",
      "Your job 5672550 (\"tanzania_183\") has been submitted\n",
      "Your job 5672551 (\"tanzania_184\") has been submitted\n",
      "Your job 5672552 (\"tanzania_185\") has been submitted\n",
      "Your job 5672553 (\"tanzania_186\") has been submitted\n",
      "Your job 5672554 (\"tanzania_187\") has been submitted\n",
      "Your job 5672555 (\"tanzania_188\") has been submitted\n",
      "Your job 5672556 (\"tanzania_189\") has been submitted\n",
      "Your job 5672557 (\"tanzania_190\") has been submitted\n",
      "Your job 5672558 (\"tanzania_191\") has been submitted\n",
      "Your job 5672559 (\"tanzania_192\") has been submitted\n",
      "Your job 5672560 (\"tanzania_193\") has been submitted\n",
      "Your job 5672561 (\"tanzania_194\") has been submitted\n",
      "Your job 5672562 (\"tanzania_195\") has been submitted\n",
      "Your job 5672563 (\"tanzania_196\") has been submitted\n",
      "Your job 5672564 (\"tanzania_197\") has been submitted\n",
      "Your job 5672565 (\"tanzania_198\") has been submitted\n",
      "Your job 5672566 (\"tanzania_199\") has been submitted\n",
      "Your job 5672567 (\"tanzania_200\") has been submitted\n",
      "Your job 5672568 (\"tanzania_201\") has been submitted\n",
      "Your job 5672569 (\"tanzania_202\") has been submitted\n",
      "Your job 5672570 (\"tanzania_203\") has been submitted\n",
      "Your job 5672571 (\"tanzania_204\") has been submitted\n",
      "Your job 5672572 (\"tanzania_205\") has been submitted\n",
      "Your job 5672573 (\"tanzania_206\") has been submitted\n",
      "Your job 5672574 (\"tanzania_207\") has been submitted\n",
      "Your job 5672575 (\"tanzania_208\") has been submitted\n",
      "Your job 5672576 (\"tanzania_209\") has been submitted\n",
      "Your job 5672577 (\"tanzania_210\") has been submitted\n",
      "Your job 5672578 (\"tanzania_211\") has been submitted\n",
      "Your job 5672579 (\"tanzania_212\") has been submitted\n",
      "Your job 5672580 (\"tanzania_213\") has been submitted\n",
      "Your job 5672581 (\"tanzania_214\") has been submitted\n",
      "Your job 5672582 (\"tanzania_215\") has been submitted\n",
      "Your job 5672583 (\"tanzania_216\") has been submitted\n",
      "Your job 5672584 (\"tanzania_217\") has been submitted\n",
      "Your job 5672585 (\"tanzania_218\") has been submitted\n",
      "Your job 5672586 (\"tanzania_219\") has been submitted\n",
      "Your job 5672587 (\"tanzania_220\") has been submitted\n",
      "Your job 5672588 (\"tanzania_221\") has been submitted\n",
      "Your job 5672589 (\"tanzania_222\") has been submitted\n",
      "Your job 5672590 (\"tanzania_223\") has been submitted\n",
      "Your job 5672591 (\"tanzania_224\") has been submitted\n",
      "Your job 5672592 (\"tanzania_225\") has been submitted\n",
      "Your job 5672593 (\"tanzania_226\") has been submitted\n",
      "Your job 5672594 (\"tanzania_227\") has been submitted\n",
      "Your job 5672595 (\"tanzania_228\") has been submitted\n",
      "Your job 5672596 (\"tanzania_229\") has been submitted\n",
      "Your job 5672597 (\"tanzania_230\") has been submitted\n",
      "Your job 5672598 (\"tanzania_231\") has been submitted\n",
      "Your job 5672599 (\"tanzania_232\") has been submitted\n",
      "Your job 5672600 (\"tanzania_233\") has been submitted\n",
      "Your job 5672601 (\"tanzania_234\") has been submitted\n",
      "Your job 5672602 (\"tanzania_235\") has been submitted\n",
      "Your job 5672603 (\"tanzania_236\") has been submitted\n",
      "Your job 5672604 (\"tanzania_237\") has been submitted\n",
      "Your job 5672605 (\"tanzania_238\") has been submitted\n",
      "Your job 5672606 (\"tanzania_239\") has been submitted\n",
      "Your job 5672607 (\"tanzania_240\") has been submitted\n",
      "Your job 5672608 (\"tanzania_241\") has been submitted\n",
      "Your job 5672609 (\"tanzania_242\") has been submitted\n",
      "Your job 5672610 (\"tanzania_243\") has been submitted\n",
      "Your job 5672611 (\"tanzania_244\") has been submitted\n",
      "Your job 5672612 (\"tanzania_245\") has been submitted\n",
      "Your job 5672613 (\"tanzania_246\") has been submitted\n",
      "Your job 5672614 (\"tanzania_247\") has been submitted\n",
      "Your job 5672615 (\"tanzania_248\") has been submitted\n",
      "Your job 5672616 (\"tanzania_249\") has been submitted\n",
      "Your job 5672617 (\"tanzania_250\") has been submitted\n",
      "Your job 5672618 (\"tanzania_251\") has been submitted\n",
      "Your job 5672619 (\"tanzania_252\") has been submitted\n",
      "Your job 5672620 (\"tanzania_253\") has been submitted\n",
      "Your job 5672621 (\"tanzania_254\") has been submitted\n",
      "Your job 5672622 (\"tanzania_255\") has been submitted\n",
      "Your job 5672623 (\"tanzania_256\") has been submitted\n",
      "Your job 5672624 (\"tanzania_257\") has been submitted\n",
      "Your job 5672625 (\"tanzania_258\") has been submitted\n",
      "Your job 5672626 (\"tanzania_259\") has been submitted\n",
      "Your job 5672627 (\"tanzania_260\") has been submitted\n",
      "Your job 5672628 (\"tanzania_261\") has been submitted\n",
      "Your job 5672629 (\"tanzania_262\") has been submitted\n",
      "Your job 5672630 (\"tanzania_263\") has been submitted\n",
      "Your job 5672631 (\"tanzania_264\") has been submitted\n",
      "Your job 5672632 (\"tanzania_265\") has been submitted\n",
      "Your job 5672633 (\"tanzania_266\") has been submitted\n",
      "Your job 5672634 (\"tanzania_267\") has been submitted\n",
      "Your job 5672635 (\"tanzania_268\") has been submitted\n",
      "Your job 5672636 (\"tanzania_269\") has been submitted\n",
      "Your job 5672637 (\"tanzania_270\") has been submitted\n",
      "Your job 5672638 (\"tanzania_271\") has been submitted\n",
      "Your job 5672639 (\"tanzania_272\") has been submitted\n",
      "Your job 5672640 (\"tanzania_273\") has been submitted\n",
      "Your job 5672641 (\"tanzania_274\") has been submitted\n",
      "Your job 5672642 (\"tanzania_275\") has been submitted\n",
      "Your job 5672643 (\"tanzania_276\") has been submitted\n",
      "Your job 5672644 (\"tanzania_277\") has been submitted\n",
      "Your job 5672645 (\"tanzania_278\") has been submitted\n",
      "Your job 5672646 (\"tanzania_279\") has been submitted\n",
      "Your job 5672647 (\"tanzania_280\") has been submitted\n",
      "Your job 5672648 (\"tanzania_281\") has been submitted\n",
      "Your job 5672649 (\"tanzania_282\") has been submitted\n",
      "Your job 5672650 (\"tanzania_283\") has been submitted\n",
      "Your job 5672651 (\"tanzania_284\") has been submitted\n",
      "Your job 5672652 (\"tanzania_285\") has been submitted\n",
      "Your job 5672653 (\"tanzania_286\") has been submitted\n",
      "Your job 5672654 (\"tanzania_287\") has been submitted\n",
      "Your job 5672655 (\"tanzania_288\") has been submitted\n",
      "Your job 5672656 (\"tanzania_289\") has been submitted\n",
      "Your job 5672657 (\"tanzania_290\") has been submitted\n",
      "Your job 5672658 (\"tanzania_291\") has been submitted\n",
      "Your job 5672659 (\"tanzania_292\") has been submitted\n",
      "Your job 5672660 (\"tanzania_293\") has been submitted\n",
      "Your job 5672661 (\"tanzania_294\") has been submitted\n",
      "Your job 5672662 (\"tanzania_295\") has been submitted\n",
      "Your job 5672663 (\"tanzania_296\") has been submitted\n",
      "Your job 5672664 (\"tanzania_297\") has been submitted\n",
      "Your job 5672665 (\"tanzania_298\") has been submitted\n",
      "Your job 5672666 (\"tanzania_299\") has been submitted\n",
      "Your job 5672667 (\"tanzania_300\") has been submitted\n",
      "Your job 5672668 (\"tanzania_301\") has been submitted\n",
      "Your job 5672669 (\"tanzania_302\") has been submitted\n",
      "Your job 5672670 (\"tanzania_303\") has been submitted\n",
      "Your job 5672671 (\"tanzania_304\") has been submitted\n",
      "Your job 5672672 (\"tanzania_305\") has been submitted\n",
      "Your job 5672673 (\"tanzania_306\") has been submitted\n",
      "Your job 5672674 (\"tanzania_307\") has been submitted\n",
      "Your job 5672675 (\"tanzania_308\") has been submitted\n",
      "Your job 5672676 (\"tanzania_309\") has been submitted\n",
      "Your job 5672677 (\"tanzania_310\") has been submitted\n",
      "Your job 5672678 (\"tanzania_311\") has been submitted\n",
      "Your job 5672679 (\"tanzania_312\") has been submitted\n",
      "Your job 5672680 (\"tanzania_313\") has been submitted\n",
      "Your job 5672681 (\"tanzania_314\") has been submitted\n",
      "Your job 5672682 (\"tanzania_315\") has been submitted\n",
      "Your job 5672683 (\"tanzania_316\") has been submitted\n",
      "Your job 5672684 (\"tanzania_317\") has been submitted\n",
      "Your job 5672685 (\"tanzania_318\") has been submitted\n",
      "Your job 5672686 (\"tanzania_319\") has been submitted\n",
      "Your job 5672687 (\"tanzania_320\") has been submitted\n",
      "Your job 5672688 (\"tanzania_321\") has been submitted\n",
      "Your job 5672689 (\"tanzania_322\") has been submitted\n",
      "Your job 5672690 (\"tanzania_323\") has been submitted\n",
      "Your job 5672691 (\"tanzania_324\") has been submitted\n",
      "Your job 5672692 (\"tanzania_325\") has been submitted\n",
      "Your job 5672693 (\"tanzania_326\") has been submitted\n",
      "Your job 5672694 (\"tanzania_327\") has been submitted\n",
      "Your job 5672695 (\"tanzania_328\") has been submitted\n",
      "Your job 5672696 (\"tanzania_329\") has been submitted\n",
      "Your job 5672697 (\"tanzania_330\") has been submitted\n",
      "Your job 5672698 (\"tanzania_331\") has been submitted\n",
      "Your job 5672699 (\"tanzania_332\") has been submitted\n",
      "Your job 5672700 (\"tanzania_333\") has been submitted\n",
      "Your job 5672701 (\"tanzania_334\") has been submitted\n",
      "Your job 5672702 (\"tanzania_335\") has been submitted\n",
      "Your job 5672703 (\"tanzania_336\") has been submitted\n",
      "Your job 5672704 (\"tanzania_337\") has been submitted\n",
      "Your job 5672705 (\"tanzania_338\") has been submitted\n",
      "Your job 5672706 (\"tanzania_339\") has been submitted\n",
      "Your job 5672707 (\"tanzania_340\") has been submitted\n",
      "Your job 5672708 (\"tanzania_341\") has been submitted\n",
      "Your job 5672709 (\"tanzania_342\") has been submitted\n",
      "Your job 5672710 (\"senegal_1\") has been submitted\n",
      "Your job 5672711 (\"senegal_2\") has been submitted\n",
      "Your job 5672712 (\"senegal_3\") has been submitted\n",
      "Your job 5672713 (\"senegal_4\") has been submitted\n",
      "Your job 5672714 (\"senegal_5\") has been submitted\n",
      "Your job 5672715 (\"senegal_6\") has been submitted\n",
      "Your job 5672716 (\"senegal_7\") has been submitted\n",
      "Your job 5672717 (\"senegal_8\") has been submitted\n",
      "Your job 5672718 (\"senegal_9\") has been submitted\n",
      "Your job 5672719 (\"senegal_10\") has been submitted\n",
      "Your job 5672720 (\"senegal_11\") has been submitted\n",
      "Your job 5672721 (\"senegal_12\") has been submitted\n",
      "Your job 5672722 (\"senegal_13\") has been submitted\n",
      "Your job 5672723 (\"senegal_14\") has been submitted\n",
      "Your job 5672724 (\"senegal_15\") has been submitted\n",
      "Your job 5672725 (\"senegal_16\") has been submitted\n",
      "Your job 5672726 (\"senegal_17\") has been submitted\n",
      "Your job 5672727 (\"senegal_18\") has been submitted\n",
      "Your job 5672728 (\"senegal_19\") has been submitted\n",
      "Your job 5672729 (\"senegal_20\") has been submitted\n",
      "Your job 5672730 (\"senegal_21\") has been submitted\n",
      "Your job 5672731 (\"senegal_22\") has been submitted\n",
      "Your job 5672732 (\"senegal_23\") has been submitted\n",
      "Your job 5672733 (\"senegal_24\") has been submitted\n",
      "Your job 5672734 (\"senegal_25\") has been submitted\n",
      "Your job 5672735 (\"senegal_26\") has been submitted\n",
      "Your job 5672736 (\"senegal_27\") has been submitted\n",
      "Your job 5672737 (\"senegal_28\") has been submitted\n",
      "Your job 5672738 (\"senegal_29\") has been submitted\n",
      "Your job 5672739 (\"senegal_30\") has been submitted\n",
      "Your job 5672740 (\"senegal_31\") has been submitted\n",
      "Your job 5672741 (\"senegal_32\") has been submitted\n",
      "Your job 5672742 (\"senegal_33\") has been submitted\n",
      "Your job 5672743 (\"senegal_34\") has been submitted\n",
      "Your job 5672744 (\"senegal_35\") has been submitted\n",
      "Your job 5672745 (\"senegal_36\") has been submitted\n",
      "Your job 5672746 (\"senegal_37\") has been submitted\n",
      "Your job 5672747 (\"senegal_38\") has been submitted\n",
      "Your job 5672748 (\"senegal_39\") has been submitted\n",
      "Your job 5672749 (\"senegal_40\") has been submitted\n",
      "Your job 5672750 (\"senegal_41\") has been submitted\n",
      "Your job 5672751 (\"senegal_42\") has been submitted\n",
      "Your job 5672752 (\"senegal_43\") has been submitted\n",
      "Your job 5672753 (\"senegal_44\") has been submitted\n",
      "Your job 5672754 (\"senegal_45\") has been submitted\n",
      "Your job 5672755 (\"senegal_46\") has been submitted\n",
      "Your job 5672756 (\"senegal_47\") has been submitted\n",
      "Your job 5672757 (\"senegal_48\") has been submitted\n",
      "Your job 5672758 (\"senegal_49\") has been submitted\n",
      "Your job 5672759 (\"senegal_50\") has been submitted\n",
      "Your job 5672760 (\"senegal_51\") has been submitted\n",
      "Your job 5672761 (\"senegal_52\") has been submitted\n",
      "Your job 5672762 (\"senegal_53\") has been submitted\n",
      "Your job 5672763 (\"senegal_54\") has been submitted\n",
      "Your job 5672764 (\"senegal_55\") has been submitted\n",
      "Your job 5672765 (\"senegal_56\") has been submitted\n",
      "Your job 5672766 (\"senegal_57\") has been submitted\n",
      "Your job 5672767 (\"senegal_58\") has been submitted\n",
      "Your job 5672768 (\"senegal_59\") has been submitted\n",
      "Your job 5672769 (\"senegal_60\") has been submitted\n",
      "Your job 5672770 (\"senegal_61\") has been submitted\n",
      "Your job 5672771 (\"senegal_62\") has been submitted\n",
      "Your job 5672772 (\"senegal_63\") has been submitted\n",
      "Your job 5672773 (\"senegal_64\") has been submitted\n",
      "Your job 5672774 (\"senegal_65\") has been submitted\n",
      "Your job 5672775 (\"senegal_66\") has been submitted\n",
      "Your job 5672776 (\"senegal_67\") has been submitted\n",
      "Your job 5672777 (\"senegal_68\") has been submitted\n",
      "Your job 5672778 (\"senegal_69\") has been submitted\n",
      "Your job 5672779 (\"senegal_70\") has been submitted\n",
      "Your job 5672780 (\"senegal_71\") has been submitted\n",
      "Your job 5672781 (\"senegal_72\") has been submitted\n",
      "Your job 5672782 (\"senegal_73\") has been submitted\n",
      "Your job 5672783 (\"senegal_74\") has been submitted\n",
      "Your job 5672784 (\"senegal_75\") has been submitted\n",
      "Your job 5672785 (\"senegal_76\") has been submitted\n",
      "Your job 5672786 (\"senegal_77\") has been submitted\n",
      "Your job 5672787 (\"senegal_78\") has been submitted\n",
      "Your job 5672788 (\"senegal_79\") has been submitted\n",
      "Your job 5672789 (\"senegal_80\") has been submitted\n",
      "Your job 5672790 (\"senegal_81\") has been submitted\n",
      "Your job 5672791 (\"senegal_82\") has been submitted\n",
      "Your job 5672792 (\"senegal_83\") has been submitted\n",
      "Your job 5672793 (\"senegal_84\") has been submitted\n",
      "Your job 5672794 (\"senegal_85\") has been submitted\n",
      "Your job 5672795 (\"senegal_86\") has been submitted\n",
      "Your job 5672796 (\"senegal_87\") has been submitted\n",
      "Your job 5672797 (\"senegal_88\") has been submitted\n",
      "Your job 5672798 (\"senegal_89\") has been submitted\n",
      "Your job 5672799 (\"senegal_90\") has been submitted\n",
      "Your job 5672800 (\"senegal_91\") has been submitted\n",
      "Your job 5672801 (\"senegal_92\") has been submitted\n",
      "Your job 5672802 (\"senegal_93\") has been submitted\n",
      "Your job 5672803 (\"senegal_94\") has been submitted\n",
      "Your job 5672804 (\"senegal_95\") has been submitted\n",
      "Your job 5672805 (\"senegal_96\") has been submitted\n",
      "Your job 5672806 (\"senegal_97\") has been submitted\n",
      "Your job 5672807 (\"senegal_98\") has been submitted\n",
      "Your job 5672808 (\"senegal_99\") has been submitted\n",
      "Your job 5672809 (\"senegal_100\") has been submitted\n",
      "Your job 5672810 (\"senegal_101\") has been submitted\n",
      "Your job 5672811 (\"senegal_102\") has been submitted\n",
      "Your job 5672812 (\"senegal_103\") has been submitted\n",
      "Your job 5672813 (\"senegal_104\") has been submitted\n",
      "Your job 5672814 (\"senegal_105\") has been submitted\n",
      "Your job 5672815 (\"senegal_106\") has been submitted\n",
      "Your job 5672816 (\"senegal_107\") has been submitted\n",
      "Your job 5672817 (\"senegal_108\") has been submitted\n",
      "Your job 5672818 (\"senegal_109\") has been submitted\n",
      "Your job 5672819 (\"senegal_110\") has been submitted\n",
      "Your job 5672820 (\"senegal_111\") has been submitted\n",
      "Your job 5672821 (\"senegal_112\") has been submitted\n",
      "Your job 5672822 (\"senegal_113\") has been submitted\n",
      "Your job 5672823 (\"senegal_114\") has been submitted\n",
      "Your job 5672824 (\"senegal_115\") has been submitted\n",
      "Your job 5672825 (\"senegal_116\") has been submitted\n",
      "Your job 5672826 (\"senegal_117\") has been submitted\n",
      "Your job 5672827 (\"senegal_118\") has been submitted\n",
      "Your job 5672828 (\"senegal_119\") has been submitted\n",
      "Your job 5672829 (\"senegal_120\") has been submitted\n",
      "Your job 5672830 (\"senegal_121\") has been submitted\n",
      "Your job 5672831 (\"senegal_122\") has been submitted\n",
      "Your job 5672832 (\"senegal_123\") has been submitted\n",
      "Your job 5672833 (\"senegal_124\") has been submitted\n",
      "Your job 5672834 (\"senegal_125\") has been submitted\n",
      "Your job 5672835 (\"senegal_126\") has been submitted\n",
      "Your job 5672836 (\"senegal_127\") has been submitted\n",
      "Your job 5672837 (\"senegal_128\") has been submitted\n",
      "Your job 5672838 (\"senegal_129\") has been submitted\n",
      "Your job 5672839 (\"senegal_130\") has been submitted\n",
      "Your job 5672840 (\"senegal_131\") has been submitted\n",
      "Your job 5672841 (\"senegal_132\") has been submitted\n",
      "Your job 5672842 (\"senegal_133\") has been submitted\n",
      "Your job 5672843 (\"senegal_134\") has been submitted\n",
      "Your job 5672844 (\"senegal_135\") has been submitted\n",
      "Your job 5672845 (\"senegal_136\") has been submitted\n",
      "Your job 5672846 (\"senegal_137\") has been submitted\n",
      "Your job 5672847 (\"senegal_138\") has been submitted\n",
      "Your job 5672848 (\"senegal_139\") has been submitted\n",
      "Your job 5672849 (\"senegal_140\") has been submitted\n",
      "Your job 5672850 (\"senegal_141\") has been submitted\n",
      "Your job 5672851 (\"senegal_142\") has been submitted\n",
      "Your job 5672852 (\"senegal_143\") has been submitted\n",
      "Your job 5672853 (\"senegal_144\") has been submitted\n",
      "Your job 5672854 (\"senegal_145\") has been submitted\n",
      "Your job 5672855 (\"senegal_146\") has been submitted\n",
      "Your job 5672856 (\"senegal_147\") has been submitted\n",
      "Your job 5672857 (\"senegal_148\") has been submitted\n",
      "Your job 5672858 (\"senegal_149\") has been submitted\n",
      "Your job 5672859 (\"senegal_150\") has been submitted\n",
      "Your job 5672860 (\"senegal_151\") has been submitted\n",
      "Your job 5672861 (\"senegal_152\") has been submitted\n",
      "Your job 5672862 (\"senegal_153\") has been submitted\n",
      "Your job 5672863 (\"senegal_154\") has been submitted\n",
      "Your job 5672864 (\"senegal_155\") has been submitted\n",
      "Your job 5672865 (\"senegal_156\") has been submitted\n",
      "Your job 5672866 (\"senegal_157\") has been submitted\n",
      "Your job 5672867 (\"senegal_158\") has been submitted\n",
      "Your job 5672868 (\"senegal_159\") has been submitted\n",
      "Your job 5672869 (\"senegal_160\") has been submitted\n",
      "Your job 5672870 (\"senegal_161\") has been submitted\n",
      "Your job 5672871 (\"senegal_162\") has been submitted\n",
      "Your job 5672872 (\"senegal_163\") has been submitted\n",
      "Your job 5672873 (\"senegal_164\") has been submitted\n",
      "Your job 5672874 (\"senegal_165\") has been submitted\n",
      "Your job 5672875 (\"senegal_166\") has been submitted\n",
      "Your job 5672876 (\"senegal_167\") has been submitted\n",
      "Your job 5672877 (\"senegal_168\") has been submitted\n",
      "Your job 5672878 (\"senegal_169\") has been submitted\n",
      "Your job 5672879 (\"senegal_170\") has been submitted\n",
      "Your job 5672880 (\"senegal_171\") has been submitted\n",
      "Your job 5672881 (\"senegal_172\") has been submitted\n",
      "Your job 5672882 (\"senegal_173\") has been submitted\n",
      "Your job 5672883 (\"senegal_174\") has been submitted\n",
      "Your job 5672884 (\"senegal_175\") has been submitted\n",
      "Your job 5672885 (\"senegal_176\") has been submitted\n",
      "Your job 5672886 (\"senegal_177\") has been submitted\n",
      "Your job 5672887 (\"senegal_178\") has been submitted\n",
      "Your job 5672888 (\"senegal_179\") has been submitted\n",
      "Your job 5672889 (\"senegal_180\") has been submitted\n",
      "Your job 5672890 (\"senegal_181\") has been submitted\n",
      "Your job 5672891 (\"senegal_182\") has been submitted\n",
      "Your job 5672892 (\"senegal_183\") has been submitted\n",
      "Your job 5672893 (\"senegal_184\") has been submitted\n",
      "Your job 5672894 (\"senegal_185\") has been submitted\n",
      "Your job 5672895 (\"senegal_186\") has been submitted\n",
      "Your job 5672896 (\"senegal_187\") has been submitted\n",
      "Your job 5672897 (\"senegal_188\") has been submitted\n",
      "Your job 5672898 (\"senegal_189\") has been submitted\n",
      "Your job 5672899 (\"senegal_190\") has been submitted\n",
      "Your job 5672900 (\"senegal_191\") has been submitted\n",
      "Your job 5672901 (\"senegal_192\") has been submitted\n",
      "Your job 5672902 (\"senegal_193\") has been submitted\n",
      "Your job 5672903 (\"senegal_194\") has been submitted\n",
      "Your job 5672904 (\"senegal_195\") has been submitted\n",
      "Your job 5672905 (\"senegal_196\") has been submitted\n",
      "Your job 5672906 (\"senegal_197\") has been submitted\n",
      "Your job 5672907 (\"senegal_198\") has been submitted\n",
      "Your job 5672908 (\"senegal_199\") has been submitted\n",
      "Your job 5672909 (\"senegal_200\") has been submitted\n",
      "Your job 5672910 (\"senegal_201\") has been submitted\n",
      "Your job 5672911 (\"senegal_202\") has been submitted\n",
      "Your job 5672912 (\"senegal_203\") has been submitted\n",
      "Your job 5672913 (\"senegal_204\") has been submitted\n",
      "Your job 5672914 (\"senegal_205\") has been submitted\n",
      "Your job 5672915 (\"senegal_206\") has been submitted\n",
      "Your job 5672916 (\"senegal_207\") has been submitted\n",
      "Your job 5672917 (\"senegal_208\") has been submitted\n",
      "Your job 5672918 (\"senegal_209\") has been submitted\n",
      "Your job 5672919 (\"senegal_210\") has been submitted\n",
      "Your job 5672920 (\"senegal_211\") has been submitted\n",
      "Your job 5672921 (\"senegal_212\") has been submitted\n",
      "Your job 5672922 (\"senegal_213\") has been submitted\n",
      "Your job 5672923 (\"senegal_214\") has been submitted\n",
      "Your job 5672924 (\"senegal_215\") has been submitted\n",
      "Your job 5672925 (\"senegal_216\") has been submitted\n",
      "Your job 5672926 (\"senegal_217\") has been submitted\n",
      "Your job 5672927 (\"senegal_218\") has been submitted\n",
      "Your job 5672928 (\"senegal_219\") has been submitted\n",
      "Your job 5672929 (\"senegal_220\") has been submitted\n",
      "Your job 5672930 (\"senegal_221\") has been submitted\n",
      "Your job 5672931 (\"senegal_222\") has been submitted\n",
      "Your job 5672932 (\"senegal_223\") has been submitted\n",
      "Your job 5672933 (\"senegal_224\") has been submitted\n",
      "Your job 5672934 (\"senegal_225\") has been submitted\n",
      "Your job 5672935 (\"senegal_226\") has been submitted\n",
      "Your job 5672936 (\"senegal_227\") has been submitted\n",
      "Your job 5672937 (\"senegal_228\") has been submitted\n",
      "Your job 5672938 (\"senegal_229\") has been submitted\n",
      "Your job 5672939 (\"senegal_230\") has been submitted\n",
      "Your job 5672940 (\"senegal_231\") has been submitted\n",
      "Your job 5672941 (\"senegal_232\") has been submitted\n",
      "Your job 5672942 (\"senegal_233\") has been submitted\n",
      "Your job 5672943 (\"senegal_234\") has been submitted\n",
      "Your job 5672944 (\"senegal_235\") has been submitted\n",
      "Your job 5672945 (\"senegal_236\") has been submitted\n",
      "Your job 5672946 (\"senegal_237\") has been submitted\n",
      "Your job 5672947 (\"senegal_238\") has been submitted\n",
      "Your job 5672948 (\"senegal_239\") has been submitted\n",
      "Your job 5672949 (\"senegal_240\") has been submitted\n",
      "Your job 5672950 (\"senegal_241\") has been submitted\n",
      "Your job 5672951 (\"senegal_242\") has been submitted\n",
      "Your job 5672952 (\"senegal_243\") has been submitted\n",
      "Your job 5672953 (\"senegal_244\") has been submitted\n",
      "Your job 5672954 (\"senegal_245\") has been submitted\n",
      "Your job 5672955 (\"senegal_246\") has been submitted\n",
      "Your job 5672956 (\"senegal_247\") has been submitted\n",
      "Your job 5672957 (\"senegal_248\") has been submitted\n",
      "Your job 5672958 (\"senegal_249\") has been submitted\n",
      "Your job 5672959 (\"senegal_250\") has been submitted\n",
      "Your job 5672960 (\"senegal_251\") has been submitted\n",
      "Your job 5672961 (\"senegal_252\") has been submitted\n",
      "Your job 5672962 (\"senegal_253\") has been submitted\n",
      "Your job 5672963 (\"senegal_254\") has been submitted\n",
      "Your job 5672964 (\"senegal_255\") has been submitted\n",
      "Your job 5672965 (\"senegal_256\") has been submitted\n",
      "Your job 5672966 (\"senegal_257\") has been submitted\n",
      "Your job 5672967 (\"senegal_258\") has been submitted\n",
      "Your job 5672968 (\"senegal_259\") has been submitted\n",
      "Your job 5672969 (\"senegal_260\") has been submitted\n",
      "Your job 5672970 (\"senegal_261\") has been submitted\n",
      "Your job 5672971 (\"senegal_262\") has been submitted\n",
      "Your job 5672972 (\"senegal_263\") has been submitted\n",
      "Your job 5672973 (\"senegal_264\") has been submitted\n",
      "Your job 5672974 (\"senegal_265\") has been submitted\n",
      "Your job 5672975 (\"senegal_266\") has been submitted\n",
      "Your job 5672976 (\"senegal_267\") has been submitted\n",
      "Your job 5672977 (\"senegal_268\") has been submitted\n",
      "Your job 5672978 (\"senegal_269\") has been submitted\n",
      "Your job 5672979 (\"senegal_270\") has been submitted\n",
      "Your job 5672980 (\"senegal_271\") has been submitted\n",
      "Your job 5672981 (\"senegal_272\") has been submitted\n",
      "Your job 5672982 (\"senegal_273\") has been submitted\n",
      "Your job 5672983 (\"senegal_274\") has been submitted\n",
      "Your job 5672984 (\"senegal_275\") has been submitted\n",
      "Your job 5672985 (\"senegal_276\") has been submitted\n",
      "Your job 5672986 (\"senegal_277\") has been submitted\n",
      "Your job 5672987 (\"senegal_278\") has been submitted\n",
      "Your job 5672988 (\"senegal_279\") has been submitted\n",
      "Your job 5672989 (\"senegal_280\") has been submitted\n",
      "Your job 5672990 (\"senegal_281\") has been submitted\n",
      "Your job 5672991 (\"senegal_282\") has been submitted\n",
      "Your job 5672992 (\"senegal_283\") has been submitted\n",
      "Your job 5672993 (\"senegal_284\") has been submitted\n",
      "Your job 5672994 (\"senegal_285\") has been submitted\n",
      "Your job 5672995 (\"senegal_286\") has been submitted\n",
      "Your job 5672996 (\"senegal_287\") has been submitted\n",
      "Your job 5672997 (\"senegal_288\") has been submitted\n",
      "Your job 5672998 (\"senegal_289\") has been submitted\n",
      "Your job 5672999 (\"senegal_290\") has been submitted\n",
      "Your job 5673000 (\"senegal_291\") has been submitted\n",
      "Your job 5673001 (\"senegal_292\") has been submitted\n",
      "Your job 5673002 (\"senegal_293\") has been submitted\n",
      "Your job 5673003 (\"senegal_294\") has been submitted\n",
      "Your job 5673004 (\"senegal_295\") has been submitted\n",
      "Your job 5673005 (\"senegal_296\") has been submitted\n",
      "Your job 5673006 (\"senegal_297\") has been submitted\n",
      "Your job 5673007 (\"senegal_298\") has been submitted\n",
      "Your job 5673008 (\"senegal_299\") has been submitted\n",
      "Your job 5673009 (\"senegal_300\") has been submitted\n",
      "Your job 5673010 (\"senegal_301\") has been submitted\n",
      "Your job 5673011 (\"senegal_302\") has been submitted\n",
      "Your job 5673012 (\"senegal_303\") has been submitted\n",
      "Your job 5673013 (\"senegal_304\") has been submitted\n",
      "Your job 5673014 (\"senegal_305\") has been submitted\n",
      "Your job 5673015 (\"senegal_306\") has been submitted\n",
      "Your job 5673016 (\"senegal_307\") has been submitted\n",
      "Your job 5673017 (\"senegal_308\") has been submitted\n",
      "Your job 5673018 (\"senegal_309\") has been submitted\n",
      "Your job 5673019 (\"senegal_310\") has been submitted\n",
      "Your job 5673020 (\"senegal_311\") has been submitted\n",
      "Your job 5673021 (\"senegal_312\") has been submitted\n",
      "Your job 5673022 (\"senegal_313\") has been submitted\n",
      "Your job 5673023 (\"senegal_314\") has been submitted\n",
      "Your job 5673024 (\"senegal_315\") has been submitted\n",
      "Your job 5673025 (\"senegal_316\") has been submitted\n",
      "Your job 5673026 (\"senegal_317\") has been submitted\n",
      "Your job 5673027 (\"senegal_318\") has been submitted\n",
      "Your job 5673028 (\"senegal_319\") has been submitted\n",
      "Your job 5673029 (\"senegal_320\") has been submitted\n",
      "Your job 5673030 (\"senegal_321\") has been submitted\n",
      "Your job 5673031 (\"senegal_322\") has been submitted\n",
      "Your job 5673032 (\"senegal_323\") has been submitted\n",
      "Your job 5673033 (\"senegal_324\") has been submitted\n",
      "Your job 5673034 (\"senegal_325\") has been submitted\n",
      "Your job 5673035 (\"senegal_326\") has been submitted\n",
      "Your job 5673036 (\"senegal_327\") has been submitted\n",
      "Your job 5673037 (\"senegal_328\") has been submitted\n",
      "Your job 5673038 (\"senegal_329\") has been submitted\n",
      "Your job 5673039 (\"senegal_330\") has been submitted\n",
      "Your job 5673040 (\"senegal_331\") has been submitted\n",
      "Your job 5673041 (\"senegal_332\") has been submitted\n",
      "Your job 5673042 (\"senegal_333\") has been submitted\n",
      "Your job 5673043 (\"senegal_334\") has been submitted\n",
      "Your job 5673044 (\"senegal_335\") has been submitted\n",
      "Your job 5673045 (\"senegal_336\") has been submitted\n",
      "Your job 5673046 (\"senegal_337\") has been submitted\n",
      "Your job 5673047 (\"senegal_338\") has been submitted\n",
      "Your job 5673048 (\"senegal_339\") has been submitted\n",
      "Your job 5673049 (\"senegal_340\") has been submitted\n",
      "Your job 5673050 (\"senegal_341\") has been submitted\n",
      "Your job 5673051 (\"senegal_342\") has been submitted\n",
      "Your job 5673052 (\"niger_1\") has been submitted\n",
      "Your job 5673053 (\"niger_2\") has been submitted\n",
      "Your job 5673054 (\"niger_3\") has been submitted\n",
      "Your job 5673055 (\"niger_4\") has been submitted\n",
      "Your job 5673056 (\"niger_5\") has been submitted\n",
      "Your job 5673057 (\"niger_6\") has been submitted\n",
      "Your job 5673058 (\"niger_7\") has been submitted\n",
      "Your job 5673059 (\"niger_8\") has been submitted\n",
      "Your job 5673060 (\"niger_9\") has been submitted\n",
      "Your job 5673061 (\"niger_10\") has been submitted\n",
      "Your job 5673062 (\"niger_11\") has been submitted\n",
      "Your job 5673063 (\"niger_12\") has been submitted\n",
      "Your job 5673064 (\"niger_13\") has been submitted\n",
      "Your job 5673065 (\"niger_14\") has been submitted\n",
      "Your job 5673066 (\"niger_15\") has been submitted\n",
      "Your job 5673067 (\"niger_16\") has been submitted\n",
      "Your job 5673068 (\"niger_17\") has been submitted\n",
      "Your job 5673069 (\"niger_18\") has been submitted\n",
      "Your job 5673070 (\"niger_19\") has been submitted\n",
      "Your job 5673071 (\"niger_20\") has been submitted\n",
      "Your job 5673072 (\"niger_21\") has been submitted\n",
      "Your job 5673073 (\"niger_22\") has been submitted\n",
      "Your job 5673074 (\"niger_23\") has been submitted\n",
      "Your job 5673075 (\"niger_24\") has been submitted\n",
      "Your job 5673076 (\"niger_25\") has been submitted\n",
      "Your job 5673077 (\"niger_26\") has been submitted\n",
      "Your job 5673078 (\"niger_27\") has been submitted\n",
      "Your job 5673079 (\"niger_28\") has been submitted\n",
      "Your job 5673080 (\"niger_29\") has been submitted\n",
      "Your job 5673081 (\"niger_30\") has been submitted\n",
      "Your job 5673082 (\"niger_31\") has been submitted\n",
      "Your job 5673083 (\"niger_32\") has been submitted\n",
      "Your job 5673084 (\"niger_33\") has been submitted\n",
      "Your job 5673085 (\"niger_34\") has been submitted\n",
      "Your job 5673086 (\"niger_35\") has been submitted\n",
      "Your job 5673087 (\"niger_36\") has been submitted\n",
      "Your job 5673088 (\"niger_37\") has been submitted\n",
      "Your job 5673089 (\"niger_38\") has been submitted\n",
      "Your job 5673090 (\"niger_39\") has been submitted\n",
      "Your job 5673091 (\"niger_40\") has been submitted\n",
      "Your job 5673092 (\"niger_41\") has been submitted\n",
      "Your job 5673093 (\"niger_42\") has been submitted\n",
      "Your job 5673094 (\"niger_43\") has been submitted\n",
      "Your job 5673095 (\"niger_44\") has been submitted\n",
      "Your job 5673096 (\"niger_45\") has been submitted\n",
      "Your job 5673097 (\"niger_46\") has been submitted\n",
      "Your job 5673098 (\"niger_47\") has been submitted\n",
      "Your job 5673099 (\"niger_48\") has been submitted\n",
      "Your job 5673100 (\"niger_49\") has been submitted\n",
      "Your job 5673101 (\"niger_50\") has been submitted\n",
      "Your job 5673102 (\"niger_51\") has been submitted\n",
      "Your job 5673103 (\"niger_52\") has been submitted\n",
      "Your job 5673104 (\"niger_53\") has been submitted\n",
      "Your job 5673105 (\"niger_54\") has been submitted\n",
      "Your job 5673106 (\"niger_55\") has been submitted\n",
      "Your job 5673107 (\"niger_56\") has been submitted\n",
      "Your job 5673108 (\"niger_57\") has been submitted\n",
      "Your job 5673109 (\"niger_58\") has been submitted\n",
      "Your job 5673110 (\"niger_59\") has been submitted\n",
      "Your job 5673111 (\"niger_60\") has been submitted\n",
      "Your job 5673112 (\"niger_61\") has been submitted\n",
      "Your job 5673113 (\"niger_62\") has been submitted\n",
      "Your job 5673114 (\"niger_63\") has been submitted\n",
      "Your job 5673115 (\"niger_64\") has been submitted\n",
      "Your job 5673116 (\"niger_65\") has been submitted\n",
      "Your job 5673117 (\"niger_66\") has been submitted\n",
      "Your job 5673118 (\"niger_67\") has been submitted\n",
      "Your job 5673119 (\"niger_68\") has been submitted\n",
      "Your job 5673120 (\"niger_69\") has been submitted\n",
      "Your job 5673121 (\"niger_70\") has been submitted\n",
      "Your job 5673122 (\"niger_71\") has been submitted\n",
      "Your job 5673123 (\"niger_72\") has been submitted\n",
      "Your job 5673124 (\"niger_73\") has been submitted\n",
      "Your job 5673125 (\"niger_74\") has been submitted\n",
      "Your job 5673126 (\"niger_75\") has been submitted\n",
      "Your job 5673127 (\"niger_76\") has been submitted\n",
      "Your job 5673128 (\"niger_77\") has been submitted\n",
      "Your job 5673129 (\"niger_78\") has been submitted\n",
      "Your job 5673130 (\"niger_79\") has been submitted\n",
      "Your job 5673131 (\"niger_80\") has been submitted\n",
      "Your job 5673132 (\"niger_81\") has been submitted\n",
      "Your job 5673133 (\"niger_82\") has been submitted\n",
      "Your job 5673134 (\"niger_83\") has been submitted\n",
      "Your job 5673135 (\"niger_84\") has been submitted\n",
      "Your job 5673136 (\"niger_85\") has been submitted\n",
      "Your job 5673137 (\"niger_86\") has been submitted\n",
      "Your job 5673138 (\"niger_87\") has been submitted\n",
      "Your job 5673139 (\"niger_88\") has been submitted\n",
      "Your job 5673140 (\"niger_89\") has been submitted\n",
      "Your job 5673141 (\"niger_90\") has been submitted\n",
      "Your job 5673142 (\"niger_91\") has been submitted\n",
      "Your job 5673143 (\"niger_92\") has been submitted\n",
      "Your job 5673144 (\"niger_93\") has been submitted\n",
      "Your job 5673145 (\"niger_94\") has been submitted\n",
      "Your job 5673146 (\"niger_95\") has been submitted\n",
      "Your job 5673147 (\"niger_96\") has been submitted\n",
      "Your job 5673148 (\"niger_97\") has been submitted\n",
      "Your job 5673149 (\"niger_98\") has been submitted\n",
      "Your job 5673150 (\"niger_99\") has been submitted\n",
      "Your job 5673151 (\"niger_100\") has been submitted\n",
      "Your job 5673152 (\"niger_101\") has been submitted\n",
      "Your job 5673153 (\"niger_102\") has been submitted\n",
      "Your job 5673154 (\"niger_103\") has been submitted\n",
      "Your job 5673155 (\"niger_104\") has been submitted\n",
      "Your job 5673156 (\"niger_105\") has been submitted\n",
      "Your job 5673157 (\"niger_106\") has been submitted\n",
      "Your job 5673158 (\"niger_107\") has been submitted\n",
      "Your job 5673159 (\"niger_108\") has been submitted\n",
      "Your job 5673160 (\"niger_109\") has been submitted\n",
      "Your job 5673161 (\"niger_110\") has been submitted\n",
      "Your job 5673162 (\"niger_111\") has been submitted\n",
      "Your job 5673163 (\"niger_112\") has been submitted\n",
      "Your job 5673164 (\"niger_113\") has been submitted\n",
      "Your job 5673165 (\"niger_114\") has been submitted\n",
      "Your job 5673166 (\"niger_115\") has been submitted\n",
      "Your job 5673167 (\"niger_116\") has been submitted\n",
      "Your job 5673168 (\"niger_117\") has been submitted\n",
      "Your job 5673169 (\"niger_118\") has been submitted\n",
      "Your job 5673170 (\"niger_119\") has been submitted\n",
      "Your job 5673171 (\"niger_120\") has been submitted\n",
      "Your job 5673172 (\"niger_121\") has been submitted\n",
      "Your job 5673173 (\"niger_122\") has been submitted\n",
      "Your job 5673174 (\"niger_123\") has been submitted\n",
      "Your job 5673175 (\"niger_124\") has been submitted\n",
      "Your job 5673176 (\"niger_125\") has been submitted\n",
      "Your job 5673177 (\"niger_126\") has been submitted\n",
      "Your job 5673178 (\"niger_127\") has been submitted\n",
      "Your job 5673179 (\"niger_128\") has been submitted\n",
      "Your job 5673180 (\"niger_129\") has been submitted\n",
      "Your job 5673181 (\"niger_130\") has been submitted\n",
      "Your job 5673182 (\"niger_131\") has been submitted\n",
      "Your job 5673183 (\"niger_132\") has been submitted\n",
      "Your job 5673184 (\"niger_133\") has been submitted\n",
      "Your job 5673185 (\"niger_134\") has been submitted\n",
      "Your job 5673186 (\"niger_135\") has been submitted\n",
      "Your job 5673187 (\"niger_136\") has been submitted\n",
      "Your job 5673188 (\"niger_137\") has been submitted\n",
      "Your job 5673189 (\"niger_138\") has been submitted\n",
      "Your job 5673190 (\"niger_139\") has been submitted\n",
      "Your job 5673191 (\"niger_140\") has been submitted\n",
      "Your job 5673192 (\"niger_141\") has been submitted\n",
      "Your job 5673193 (\"niger_142\") has been submitted\n",
      "Your job 5673194 (\"niger_143\") has been submitted\n",
      "Your job 5673195 (\"niger_144\") has been submitted\n",
      "Your job 5673196 (\"niger_145\") has been submitted\n",
      "Your job 5673197 (\"niger_146\") has been submitted\n",
      "Your job 5673198 (\"niger_147\") has been submitted\n",
      "Your job 5673199 (\"niger_148\") has been submitted\n",
      "Your job 5673200 (\"niger_149\") has been submitted\n",
      "Your job 5673201 (\"niger_150\") has been submitted\n",
      "Your job 5673202 (\"niger_151\") has been submitted\n",
      "Your job 5673203 (\"niger_152\") has been submitted\n",
      "Your job 5673204 (\"niger_153\") has been submitted\n",
      "Your job 5673205 (\"niger_154\") has been submitted\n",
      "Your job 5673206 (\"niger_155\") has been submitted\n",
      "Your job 5673207 (\"niger_156\") has been submitted\n",
      "Your job 5673208 (\"niger_157\") has been submitted\n",
      "Your job 5673209 (\"niger_158\") has been submitted\n",
      "Your job 5673210 (\"niger_159\") has been submitted\n",
      "Your job 5673211 (\"niger_160\") has been submitted\n",
      "Your job 5673212 (\"niger_161\") has been submitted\n",
      "Your job 5673213 (\"niger_162\") has been submitted\n",
      "Your job 5673214 (\"niger_163\") has been submitted\n",
      "Your job 5673215 (\"niger_164\") has been submitted\n",
      "Your job 5673216 (\"niger_165\") has been submitted\n",
      "Your job 5673217 (\"niger_166\") has been submitted\n",
      "Your job 5673218 (\"niger_167\") has been submitted\n",
      "Your job 5673219 (\"niger_168\") has been submitted\n",
      "Your job 5673220 (\"niger_169\") has been submitted\n",
      "Your job 5673221 (\"niger_170\") has been submitted\n",
      "Your job 5673222 (\"niger_171\") has been submitted\n",
      "Your job 5673223 (\"niger_172\") has been submitted\n",
      "Your job 5673224 (\"niger_173\") has been submitted\n",
      "Your job 5673225 (\"niger_174\") has been submitted\n",
      "Your job 5673226 (\"niger_175\") has been submitted\n",
      "Your job 5673227 (\"niger_176\") has been submitted\n",
      "Your job 5673228 (\"niger_177\") has been submitted\n",
      "Your job 5673229 (\"niger_178\") has been submitted\n",
      "Your job 5673230 (\"niger_179\") has been submitted\n",
      "Your job 5673231 (\"niger_180\") has been submitted\n",
      "Your job 5673232 (\"niger_181\") has been submitted\n",
      "Your job 5673233 (\"niger_182\") has been submitted\n",
      "Your job 5673234 (\"niger_183\") has been submitted\n",
      "Your job 5673235 (\"niger_184\") has been submitted\n",
      "Your job 5673236 (\"niger_185\") has been submitted\n",
      "Your job 5673237 (\"niger_186\") has been submitted\n",
      "Your job 5673238 (\"niger_187\") has been submitted\n",
      "Your job 5673239 (\"niger_188\") has been submitted\n",
      "Your job 5673240 (\"niger_189\") has been submitted\n",
      "Your job 5673241 (\"niger_190\") has been submitted\n",
      "Your job 5673242 (\"niger_191\") has been submitted\n",
      "Your job 5673243 (\"niger_192\") has been submitted\n",
      "Your job 5673244 (\"niger_193\") has been submitted\n",
      "Your job 5673245 (\"niger_194\") has been submitted\n",
      "Your job 5673246 (\"niger_195\") has been submitted\n",
      "Your job 5673247 (\"niger_196\") has been submitted\n",
      "Your job 5673248 (\"niger_197\") has been submitted\n",
      "Your job 5673249 (\"niger_198\") has been submitted\n",
      "Your job 5673250 (\"niger_199\") has been submitted\n",
      "Your job 5673251 (\"niger_200\") has been submitted\n",
      "Your job 5673252 (\"niger_201\") has been submitted\n",
      "Your job 5673253 (\"niger_202\") has been submitted\n",
      "Your job 5673254 (\"niger_203\") has been submitted\n",
      "Your job 5673255 (\"niger_204\") has been submitted\n",
      "Your job 5673256 (\"niger_205\") has been submitted\n",
      "Your job 5673257 (\"niger_206\") has been submitted\n",
      "Your job 5673258 (\"niger_207\") has been submitted\n",
      "Your job 5673259 (\"niger_208\") has been submitted\n",
      "Your job 5673260 (\"niger_209\") has been submitted\n",
      "Your job 5673261 (\"niger_210\") has been submitted\n",
      "Your job 5673262 (\"niger_211\") has been submitted\n",
      "Your job 5673263 (\"niger_212\") has been submitted\n",
      "Your job 5673264 (\"niger_213\") has been submitted\n",
      "Your job 5673265 (\"niger_214\") has been submitted\n",
      "Your job 5673266 (\"niger_215\") has been submitted\n",
      "Your job 5673267 (\"niger_216\") has been submitted\n",
      "Your job 5673268 (\"niger_217\") has been submitted\n",
      "Your job 5673269 (\"niger_218\") has been submitted\n",
      "Your job 5673270 (\"niger_219\") has been submitted\n",
      "Your job 5673271 (\"niger_220\") has been submitted\n",
      "Your job 5673272 (\"niger_221\") has been submitted\n",
      "Your job 5673273 (\"niger_222\") has been submitted\n",
      "Your job 5673274 (\"niger_223\") has been submitted\n",
      "Your job 5673275 (\"niger_224\") has been submitted\n",
      "Your job 5673276 (\"niger_225\") has been submitted\n",
      "Your job 5673277 (\"niger_226\") has been submitted\n",
      "Your job 5673278 (\"niger_227\") has been submitted\n",
      "Your job 5673279 (\"niger_228\") has been submitted\n",
      "Your job 5673280 (\"niger_229\") has been submitted\n",
      "Your job 5673281 (\"niger_230\") has been submitted\n",
      "Your job 5673282 (\"niger_231\") has been submitted\n",
      "Your job 5673283 (\"niger_232\") has been submitted\n",
      "Your job 5673284 (\"niger_233\") has been submitted\n",
      "Your job 5673285 (\"niger_234\") has been submitted\n",
      "Your job 5673286 (\"niger_235\") has been submitted\n",
      "Your job 5673287 (\"niger_236\") has been submitted\n",
      "Your job 5673288 (\"niger_237\") has been submitted\n",
      "Your job 5673289 (\"niger_238\") has been submitted\n",
      "Your job 5673290 (\"niger_239\") has been submitted\n",
      "Your job 5673291 (\"niger_240\") has been submitted\n",
      "Your job 5673292 (\"niger_241\") has been submitted\n",
      "Your job 5673293 (\"niger_242\") has been submitted\n",
      "Your job 5673294 (\"niger_243\") has been submitted\n",
      "Your job 5673295 (\"niger_244\") has been submitted\n",
      "Your job 5673296 (\"niger_245\") has been submitted\n",
      "Your job 5673297 (\"niger_246\") has been submitted\n",
      "Your job 5673298 (\"niger_247\") has been submitted\n",
      "Your job 5673299 (\"niger_248\") has been submitted\n",
      "Your job 5673300 (\"niger_249\") has been submitted\n",
      "Your job 5673301 (\"niger_250\") has been submitted\n",
      "Your job 5673302 (\"niger_251\") has been submitted\n",
      "Your job 5673303 (\"niger_252\") has been submitted\n",
      "Your job 5673304 (\"niger_253\") has been submitted\n",
      "Your job 5673305 (\"niger_254\") has been submitted\n",
      "Your job 5673306 (\"niger_255\") has been submitted\n",
      "Your job 5673307 (\"niger_256\") has been submitted\n",
      "Your job 5673308 (\"niger_257\") has been submitted\n",
      "Your job 5673309 (\"niger_258\") has been submitted\n",
      "Your job 5673310 (\"niger_259\") has been submitted\n",
      "Your job 5673311 (\"niger_260\") has been submitted\n",
      "Your job 5673312 (\"niger_261\") has been submitted\n",
      "Your job 5673313 (\"niger_262\") has been submitted\n",
      "Your job 5673314 (\"niger_263\") has been submitted\n",
      "Your job 5673315 (\"niger_264\") has been submitted\n",
      "Your job 5673316 (\"niger_265\") has been submitted\n",
      "Your job 5673317 (\"niger_266\") has been submitted\n",
      "Your job 5673318 (\"niger_267\") has been submitted\n",
      "Your job 5673319 (\"niger_268\") has been submitted\n",
      "Your job 5673320 (\"niger_269\") has been submitted\n",
      "Your job 5673321 (\"niger_270\") has been submitted\n",
      "Your job 5673322 (\"niger_271\") has been submitted\n",
      "Your job 5673323 (\"niger_272\") has been submitted\n",
      "Your job 5673324 (\"niger_273\") has been submitted\n",
      "Your job 5673325 (\"niger_274\") has been submitted\n",
      "Your job 5673326 (\"niger_275\") has been submitted\n",
      "Your job 5673327 (\"niger_276\") has been submitted\n",
      "Your job 5673328 (\"niger_277\") has been submitted\n",
      "Your job 5673329 (\"niger_278\") has been submitted\n",
      "Your job 5673330 (\"niger_279\") has been submitted\n",
      "Your job 5673331 (\"niger_280\") has been submitted\n",
      "Your job 5673332 (\"niger_281\") has been submitted\n",
      "Your job 5673333 (\"niger_282\") has been submitted\n",
      "Your job 5673334 (\"niger_283\") has been submitted\n",
      "Your job 5673335 (\"niger_284\") has been submitted\n",
      "Your job 5673336 (\"niger_285\") has been submitted\n",
      "Your job 5673337 (\"niger_286\") has been submitted\n",
      "Your job 5673338 (\"niger_287\") has been submitted\n",
      "Your job 5673339 (\"niger_288\") has been submitted\n",
      "Your job 5673340 (\"niger_289\") has been submitted\n",
      "Your job 5673341 (\"niger_290\") has been submitted\n",
      "Your job 5673342 (\"niger_291\") has been submitted\n",
      "Your job 5673343 (\"niger_292\") has been submitted\n",
      "Your job 5673344 (\"niger_293\") has been submitted\n",
      "Your job 5673345 (\"niger_294\") has been submitted\n",
      "Your job 5673346 (\"niger_295\") has been submitted\n",
      "Your job 5673347 (\"niger_296\") has been submitted\n",
      "Your job 5673348 (\"niger_297\") has been submitted\n",
      "Your job 5673349 (\"niger_298\") has been submitted\n",
      "Your job 5673350 (\"niger_299\") has been submitted\n",
      "Your job 5673351 (\"niger_300\") has been submitted\n",
      "Your job 5673352 (\"niger_301\") has been submitted\n",
      "Your job 5673353 (\"niger_302\") has been submitted\n",
      "Your job 5673354 (\"niger_303\") has been submitted\n",
      "Your job 5673355 (\"niger_304\") has been submitted\n",
      "Your job 5673356 (\"niger_305\") has been submitted\n",
      "Your job 5673357 (\"niger_306\") has been submitted\n",
      "Your job 5673358 (\"niger_307\") has been submitted\n",
      "Your job 5673359 (\"niger_308\") has been submitted\n",
      "Your job 5673360 (\"niger_309\") has been submitted\n",
      "Your job 5673361 (\"niger_310\") has been submitted\n",
      "Your job 5673362 (\"niger_311\") has been submitted\n",
      "Your job 5673363 (\"niger_312\") has been submitted\n",
      "Your job 5673364 (\"niger_313\") has been submitted\n",
      "Your job 5673365 (\"niger_314\") has been submitted\n",
      "Your job 5673366 (\"niger_315\") has been submitted\n",
      "Your job 5673367 (\"niger_316\") has been submitted\n",
      "Your job 5673368 (\"niger_317\") has been submitted\n",
      "Your job 5673369 (\"niger_318\") has been submitted\n",
      "Your job 5673370 (\"niger_319\") has been submitted\n",
      "Your job 5673371 (\"niger_320\") has been submitted\n",
      "Your job 5673372 (\"niger_321\") has been submitted\n",
      "Your job 5673373 (\"niger_322\") has been submitted\n",
      "Your job 5673374 (\"niger_323\") has been submitted\n",
      "Your job 5673375 (\"niger_324\") has been submitted\n",
      "Your job 5673376 (\"niger_325\") has been submitted\n",
      "Your job 5673377 (\"niger_326\") has been submitted\n",
      "Your job 5673378 (\"niger_327\") has been submitted\n",
      "Your job 5673379 (\"niger_328\") has been submitted\n",
      "Your job 5673380 (\"niger_329\") has been submitted\n",
      "Your job 5673381 (\"niger_330\") has been submitted\n",
      "Your job 5673382 (\"niger_331\") has been submitted\n",
      "Your job 5673383 (\"niger_332\") has been submitted\n",
      "Your job 5673384 (\"niger_333\") has been submitted\n",
      "Your job 5673385 (\"niger_334\") has been submitted\n",
      "Your job 5673386 (\"niger_335\") has been submitted\n",
      "Your job 5673387 (\"niger_336\") has been submitted\n",
      "Your job 5673388 (\"niger_337\") has been submitted\n",
      "Your job 5673389 (\"niger_338\") has been submitted\n",
      "Your job 5673390 (\"niger_339\") has been submitted\n",
      "Your job 5673391 (\"niger_340\") has been submitted\n",
      "Your job 5673392 (\"niger_341\") has been submitted\n",
      "Your job 5673393 (\"niger_342\") has been submitted\n",
      "Your job 5673394 (\"brazil_1\") has been submitted\n",
      "Your job 5673395 (\"brazil_2\") has been submitted\n",
      "Your job 5673396 (\"brazil_3\") has been submitted\n",
      "Your job 5673397 (\"brazil_4\") has been submitted\n",
      "Your job 5673398 (\"brazil_5\") has been submitted\n",
      "Your job 5673399 (\"brazil_6\") has been submitted\n",
      "Your job 5673400 (\"brazil_7\") has been submitted\n",
      "Your job 5673401 (\"brazil_8\") has been submitted\n",
      "Your job 5673402 (\"brazil_9\") has been submitted\n",
      "Your job 5673403 (\"brazil_10\") has been submitted\n",
      "Your job 5673404 (\"brazil_11\") has been submitted\n",
      "Your job 5673405 (\"brazil_12\") has been submitted\n",
      "Your job 5673406 (\"brazil_13\") has been submitted\n",
      "Your job 5673407 (\"brazil_14\") has been submitted\n",
      "Your job 5673408 (\"brazil_15\") has been submitted\n",
      "Your job 5673409 (\"brazil_16\") has been submitted\n",
      "Your job 5673410 (\"brazil_17\") has been submitted\n",
      "Your job 5673411 (\"brazil_18\") has been submitted\n",
      "Your job 5673412 (\"brazil_19\") has been submitted\n",
      "Your job 5673413 (\"brazil_20\") has been submitted\n",
      "Your job 5673414 (\"brazil_21\") has been submitted\n",
      "Your job 5673415 (\"brazil_22\") has been submitted\n",
      "Your job 5673416 (\"brazil_23\") has been submitted\n",
      "Your job 5673417 (\"brazil_24\") has been submitted\n",
      "Your job 5673418 (\"brazil_25\") has been submitted\n",
      "Your job 5673419 (\"brazil_26\") has been submitted\n",
      "Your job 5673420 (\"brazil_27\") has been submitted\n",
      "Your job 5673421 (\"brazil_28\") has been submitted\n",
      "Your job 5673422 (\"brazil_29\") has been submitted\n",
      "Your job 5673423 (\"brazil_30\") has been submitted\n",
      "Your job 5673424 (\"brazil_31\") has been submitted\n",
      "Your job 5673425 (\"brazil_32\") has been submitted\n",
      "Your job 5673426 (\"brazil_33\") has been submitted\n",
      "Your job 5673427 (\"brazil_34\") has been submitted\n",
      "Your job 5673428 (\"brazil_35\") has been submitted\n",
      "Your job 5673429 (\"brazil_36\") has been submitted\n",
      "Your job 5673430 (\"brazil_37\") has been submitted\n",
      "Your job 5673431 (\"brazil_38\") has been submitted\n",
      "Your job 5673432 (\"brazil_39\") has been submitted\n",
      "Your job 5673433 (\"brazil_40\") has been submitted\n",
      "Your job 5673434 (\"brazil_41\") has been submitted\n",
      "Your job 5673435 (\"brazil_42\") has been submitted\n",
      "Your job 5673436 (\"brazil_43\") has been submitted\n",
      "Your job 5673437 (\"brazil_44\") has been submitted\n",
      "Your job 5673438 (\"brazil_45\") has been submitted\n",
      "Your job 5673439 (\"brazil_46\") has been submitted\n",
      "Your job 5673440 (\"brazil_47\") has been submitted\n",
      "Your job 5673441 (\"brazil_48\") has been submitted\n",
      "Your job 5673442 (\"brazil_49\") has been submitted\n",
      "Your job 5673443 (\"brazil_50\") has been submitted\n",
      "Your job 5673444 (\"brazil_51\") has been submitted\n",
      "Your job 5673445 (\"brazil_52\") has been submitted\n",
      "Your job 5673446 (\"brazil_53\") has been submitted\n",
      "Your job 5673447 (\"brazil_54\") has been submitted\n",
      "Your job 5673448 (\"brazil_55\") has been submitted\n",
      "Your job 5673449 (\"brazil_56\") has been submitted\n",
      "Your job 5673450 (\"brazil_57\") has been submitted\n",
      "Your job 5673451 (\"brazil_58\") has been submitted\n",
      "Your job 5673452 (\"brazil_59\") has been submitted\n",
      "Your job 5673453 (\"brazil_60\") has been submitted\n",
      "Your job 5673454 (\"brazil_61\") has been submitted\n",
      "Your job 5673455 (\"brazil_62\") has been submitted\n",
      "Your job 5673456 (\"brazil_63\") has been submitted\n",
      "Your job 5673457 (\"brazil_64\") has been submitted\n",
      "Your job 5673458 (\"brazil_65\") has been submitted\n",
      "Your job 5673459 (\"brazil_66\") has been submitted\n",
      "Your job 5673460 (\"brazil_67\") has been submitted\n",
      "Your job 5673461 (\"brazil_68\") has been submitted\n",
      "Your job 5673462 (\"brazil_69\") has been submitted\n",
      "Your job 5673463 (\"brazil_70\") has been submitted\n",
      "Your job 5673464 (\"brazil_71\") has been submitted\n",
      "Your job 5673465 (\"brazil_72\") has been submitted\n",
      "Your job 5673466 (\"brazil_73\") has been submitted\n",
      "Your job 5673467 (\"brazil_74\") has been submitted\n",
      "Your job 5673468 (\"brazil_75\") has been submitted\n",
      "Your job 5673469 (\"brazil_76\") has been submitted\n",
      "Your job 5673470 (\"brazil_77\") has been submitted\n",
      "Your job 5673471 (\"brazil_78\") has been submitted\n",
      "Your job 5673472 (\"brazil_79\") has been submitted\n",
      "Your job 5673473 (\"brazil_80\") has been submitted\n",
      "Your job 5673474 (\"brazil_81\") has been submitted\n",
      "Your job 5673475 (\"brazil_82\") has been submitted\n",
      "Your job 5673476 (\"brazil_83\") has been submitted\n",
      "Your job 5673477 (\"brazil_84\") has been submitted\n",
      "Your job 5673478 (\"brazil_85\") has been submitted\n",
      "Your job 5673479 (\"brazil_86\") has been submitted\n",
      "Your job 5673480 (\"brazil_87\") has been submitted\n",
      "Your job 5673481 (\"brazil_88\") has been submitted\n",
      "Your job 5673482 (\"brazil_89\") has been submitted\n",
      "Your job 5673483 (\"brazil_90\") has been submitted\n",
      "Your job 5673484 (\"brazil_91\") has been submitted\n",
      "Your job 5673485 (\"brazil_92\") has been submitted\n",
      "Your job 5673486 (\"brazil_93\") has been submitted\n",
      "Your job 5673487 (\"brazil_94\") has been submitted\n",
      "Your job 5673488 (\"brazil_95\") has been submitted\n",
      "Your job 5673489 (\"brazil_96\") has been submitted\n",
      "Your job 5673490 (\"brazil_97\") has been submitted\n",
      "Your job 5673491 (\"brazil_98\") has been submitted\n",
      "Your job 5673492 (\"brazil_99\") has been submitted\n",
      "Your job 5673493 (\"brazil_100\") has been submitted\n",
      "Your job 5673494 (\"brazil_101\") has been submitted\n",
      "Your job 5673495 (\"brazil_102\") has been submitted\n",
      "Your job 5673496 (\"brazil_103\") has been submitted\n",
      "Your job 5673497 (\"brazil_104\") has been submitted\n",
      "Your job 5673498 (\"brazil_105\") has been submitted\n",
      "Your job 5673499 (\"brazil_106\") has been submitted\n",
      "Your job 5673500 (\"brazil_107\") has been submitted\n",
      "Your job 5673501 (\"brazil_108\") has been submitted\n",
      "Your job 5673502 (\"brazil_109\") has been submitted\n",
      "Your job 5673503 (\"brazil_110\") has been submitted\n",
      "Your job 5673504 (\"brazil_111\") has been submitted\n",
      "Your job 5673505 (\"brazil_112\") has been submitted\n",
      "Your job 5673506 (\"brazil_113\") has been submitted\n",
      "Your job 5673507 (\"brazil_114\") has been submitted\n",
      "Your job 5673508 (\"brazil_115\") has been submitted\n",
      "Your job 5673509 (\"brazil_116\") has been submitted\n",
      "Your job 5673510 (\"brazil_117\") has been submitted\n",
      "Your job 5673511 (\"brazil_118\") has been submitted\n",
      "Your job 5673512 (\"brazil_119\") has been submitted\n",
      "Your job 5673513 (\"brazil_120\") has been submitted\n",
      "Your job 5673514 (\"brazil_121\") has been submitted\n",
      "Your job 5673515 (\"brazil_122\") has been submitted\n",
      "Your job 5673516 (\"brazil_123\") has been submitted\n",
      "Your job 5673517 (\"brazil_124\") has been submitted\n",
      "Your job 5673518 (\"brazil_125\") has been submitted\n",
      "Your job 5673519 (\"brazil_126\") has been submitted\n",
      "Your job 5673520 (\"brazil_127\") has been submitted\n",
      "Your job 5673521 (\"brazil_128\") has been submitted\n",
      "Your job 5673522 (\"brazil_129\") has been submitted\n",
      "Your job 5673523 (\"brazil_130\") has been submitted\n",
      "Your job 5673524 (\"brazil_131\") has been submitted\n",
      "Your job 5673525 (\"brazil_132\") has been submitted\n",
      "Your job 5673526 (\"brazil_133\") has been submitted\n",
      "Your job 5673527 (\"brazil_134\") has been submitted\n",
      "Your job 5673528 (\"brazil_135\") has been submitted\n",
      "Your job 5673529 (\"brazil_136\") has been submitted\n",
      "Your job 5673530 (\"brazil_137\") has been submitted\n",
      "Your job 5673531 (\"brazil_138\") has been submitted\n",
      "Your job 5673532 (\"brazil_139\") has been submitted\n",
      "Your job 5673533 (\"brazil_140\") has been submitted\n",
      "Your job 5673534 (\"brazil_141\") has been submitted\n",
      "Your job 5673535 (\"brazil_142\") has been submitted\n",
      "Your job 5673536 (\"brazil_143\") has been submitted\n",
      "Your job 5673537 (\"brazil_144\") has been submitted\n",
      "Your job 5673538 (\"brazil_145\") has been submitted\n",
      "Your job 5673539 (\"brazil_146\") has been submitted\n",
      "Your job 5673540 (\"brazil_147\") has been submitted\n",
      "Your job 5673541 (\"brazil_148\") has been submitted\n",
      "Your job 5673542 (\"brazil_149\") has been submitted\n",
      "Your job 5673543 (\"brazil_150\") has been submitted\n",
      "Your job 5673544 (\"brazil_151\") has been submitted\n",
      "Your job 5673545 (\"brazil_152\") has been submitted\n",
      "Your job 5673546 (\"brazil_153\") has been submitted\n",
      "Your job 5673547 (\"brazil_154\") has been submitted\n",
      "Your job 5673548 (\"brazil_155\") has been submitted\n",
      "Your job 5673549 (\"brazil_156\") has been submitted\n",
      "Your job 5673550 (\"brazil_157\") has been submitted\n",
      "Your job 5673551 (\"brazil_158\") has been submitted\n",
      "Your job 5673552 (\"brazil_159\") has been submitted\n",
      "Your job 5673553 (\"brazil_160\") has been submitted\n",
      "Your job 5673554 (\"brazil_161\") has been submitted\n",
      "Your job 5673555 (\"brazil_162\") has been submitted\n",
      "Your job 5673556 (\"brazil_163\") has been submitted\n",
      "Your job 5673557 (\"brazil_164\") has been submitted\n",
      "Your job 5673558 (\"brazil_165\") has been submitted\n",
      "Your job 5673559 (\"brazil_166\") has been submitted\n",
      "Your job 5673560 (\"brazil_167\") has been submitted\n",
      "Your job 5673561 (\"brazil_168\") has been submitted\n",
      "Your job 5673562 (\"brazil_169\") has been submitted\n",
      "Your job 5673563 (\"brazil_170\") has been submitted\n",
      "Your job 5673564 (\"brazil_171\") has been submitted\n",
      "Your job 5673565 (\"brazil_172\") has been submitted\n",
      "Your job 5673566 (\"brazil_173\") has been submitted\n",
      "Your job 5673567 (\"brazil_174\") has been submitted\n",
      "Your job 5673568 (\"brazil_175\") has been submitted\n",
      "Your job 5673569 (\"brazil_176\") has been submitted\n",
      "Your job 5673570 (\"brazil_177\") has been submitted\n",
      "Your job 5673571 (\"brazil_178\") has been submitted\n",
      "Your job 5673572 (\"brazil_179\") has been submitted\n",
      "Your job 5673573 (\"brazil_180\") has been submitted\n",
      "Your job 5673574 (\"brazil_181\") has been submitted\n",
      "Your job 5673575 (\"brazil_182\") has been submitted\n",
      "Your job 5673576 (\"brazil_183\") has been submitted\n",
      "Your job 5673577 (\"brazil_184\") has been submitted\n",
      "Your job 5673578 (\"brazil_185\") has been submitted\n",
      "Your job 5673579 (\"brazil_186\") has been submitted\n",
      "Your job 5673580 (\"brazil_187\") has been submitted\n",
      "Your job 5673581 (\"brazil_188\") has been submitted\n",
      "Your job 5673582 (\"brazil_189\") has been submitted\n",
      "Your job 5673583 (\"brazil_190\") has been submitted\n",
      "Your job 5673584 (\"brazil_191\") has been submitted\n",
      "Your job 5673585 (\"brazil_192\") has been submitted\n",
      "Your job 5673586 (\"brazil_193\") has been submitted\n",
      "Your job 5673587 (\"brazil_194\") has been submitted\n",
      "Your job 5673588 (\"brazil_195\") has been submitted\n",
      "Your job 5673589 (\"brazil_196\") has been submitted\n",
      "Your job 5673590 (\"brazil_197\") has been submitted\n",
      "Your job 5673591 (\"brazil_198\") has been submitted\n",
      "Your job 5673592 (\"brazil_199\") has been submitted\n",
      "Your job 5673593 (\"brazil_200\") has been submitted\n",
      "Your job 5673594 (\"brazil_201\") has been submitted\n",
      "Your job 5673595 (\"brazil_202\") has been submitted\n",
      "Your job 5673596 (\"brazil_203\") has been submitted\n",
      "Your job 5673597 (\"brazil_204\") has been submitted\n",
      "Your job 5673598 (\"brazil_205\") has been submitted\n",
      "Your job 5673599 (\"brazil_206\") has been submitted\n",
      "Your job 5673600 (\"brazil_207\") has been submitted\n",
      "Your job 5673601 (\"brazil_208\") has been submitted\n",
      "Your job 5673602 (\"brazil_209\") has been submitted\n",
      "Your job 5673603 (\"brazil_210\") has been submitted\n",
      "Your job 5673604 (\"brazil_211\") has been submitted\n",
      "Your job 5673605 (\"brazil_212\") has been submitted\n",
      "Your job 5673606 (\"brazil_213\") has been submitted\n",
      "Your job 5673607 (\"brazil_214\") has been submitted\n",
      "Your job 5673608 (\"brazil_215\") has been submitted\n",
      "Your job 5673609 (\"brazil_216\") has been submitted\n",
      "Your job 5673610 (\"brazil_217\") has been submitted\n",
      "Your job 5673611 (\"brazil_218\") has been submitted\n",
      "Your job 5673612 (\"brazil_219\") has been submitted\n",
      "Your job 5673613 (\"brazil_220\") has been submitted\n",
      "Your job 5673614 (\"brazil_221\") has been submitted\n",
      "Your job 5673615 (\"brazil_222\") has been submitted\n",
      "Your job 5673616 (\"brazil_223\") has been submitted\n",
      "Your job 5673617 (\"brazil_224\") has been submitted\n",
      "Your job 5673618 (\"brazil_225\") has been submitted\n",
      "Your job 5673619 (\"brazil_226\") has been submitted\n",
      "Your job 5673620 (\"brazil_227\") has been submitted\n",
      "Your job 5673621 (\"brazil_228\") has been submitted\n",
      "Your job 5673622 (\"brazil_229\") has been submitted\n",
      "Your job 5673623 (\"brazil_230\") has been submitted\n",
      "Your job 5673624 (\"brazil_231\") has been submitted\n",
      "Your job 5673625 (\"brazil_232\") has been submitted\n",
      "Your job 5673626 (\"brazil_233\") has been submitted\n",
      "Your job 5673627 (\"brazil_234\") has been submitted\n",
      "Your job 5673628 (\"brazil_235\") has been submitted\n",
      "Your job 5673629 (\"brazil_236\") has been submitted\n",
      "Your job 5673630 (\"brazil_237\") has been submitted\n",
      "Your job 5673631 (\"brazil_238\") has been submitted\n",
      "Your job 5673632 (\"brazil_239\") has been submitted\n",
      "Your job 5673633 (\"brazil_240\") has been submitted\n",
      "Your job 5673634 (\"brazil_241\") has been submitted\n",
      "Your job 5673635 (\"brazil_242\") has been submitted\n",
      "Your job 5673636 (\"brazil_243\") has been submitted\n",
      "Your job 5673637 (\"brazil_244\") has been submitted\n",
      "Your job 5673638 (\"brazil_245\") has been submitted\n",
      "Your job 5673639 (\"brazil_246\") has been submitted\n",
      "Your job 5673640 (\"brazil_247\") has been submitted\n",
      "Your job 5673641 (\"brazil_248\") has been submitted\n",
      "Your job 5673642 (\"brazil_249\") has been submitted\n",
      "Your job 5673643 (\"brazil_250\") has been submitted\n",
      "Your job 5673644 (\"brazil_251\") has been submitted\n",
      "Your job 5673645 (\"brazil_252\") has been submitted\n",
      "Your job 5673646 (\"brazil_253\") has been submitted\n",
      "Your job 5673647 (\"brazil_254\") has been submitted\n",
      "Your job 5673648 (\"brazil_255\") has been submitted\n",
      "Your job 5673649 (\"brazil_256\") has been submitted\n",
      "Your job 5673650 (\"brazil_257\") has been submitted\n",
      "Your job 5673651 (\"brazil_258\") has been submitted\n",
      "Your job 5673652 (\"brazil_259\") has been submitted\n",
      "Your job 5673653 (\"brazil_260\") has been submitted\n",
      "Your job 5673654 (\"brazil_261\") has been submitted\n",
      "Your job 5673655 (\"brazil_262\") has been submitted\n",
      "Your job 5673656 (\"brazil_263\") has been submitted\n",
      "Your job 5673657 (\"brazil_264\") has been submitted\n",
      "Your job 5673658 (\"brazil_265\") has been submitted\n",
      "Your job 5673659 (\"brazil_266\") has been submitted\n",
      "Your job 5673660 (\"brazil_267\") has been submitted\n",
      "Your job 5673661 (\"brazil_268\") has been submitted\n",
      "Your job 5673662 (\"brazil_269\") has been submitted\n",
      "Your job 5673663 (\"brazil_270\") has been submitted\n",
      "Your job 5673664 (\"brazil_271\") has been submitted\n",
      "Your job 5673665 (\"brazil_272\") has been submitted\n",
      "Your job 5673666 (\"brazil_273\") has been submitted\n",
      "Your job 5673667 (\"brazil_274\") has been submitted\n",
      "Your job 5673668 (\"brazil_275\") has been submitted\n",
      "Your job 5673669 (\"brazil_276\") has been submitted\n",
      "Your job 5673670 (\"brazil_277\") has been submitted\n",
      "Your job 5673671 (\"brazil_278\") has been submitted\n",
      "Your job 5673672 (\"brazil_279\") has been submitted\n",
      "Your job 5673673 (\"brazil_280\") has been submitted\n",
      "Your job 5673674 (\"brazil_281\") has been submitted\n",
      "Your job 5673675 (\"brazil_282\") has been submitted\n",
      "Your job 5673676 (\"brazil_283\") has been submitted\n",
      "Your job 5673677 (\"brazil_284\") has been submitted\n",
      "Your job 5673678 (\"brazil_285\") has been submitted\n",
      "Your job 5673679 (\"brazil_286\") has been submitted\n",
      "Your job 5673680 (\"brazil_287\") has been submitted\n",
      "Your job 5673681 (\"brazil_288\") has been submitted\n",
      "Your job 5673682 (\"brazil_289\") has been submitted\n",
      "Your job 5673683 (\"brazil_290\") has been submitted\n",
      "Your job 5673684 (\"brazil_291\") has been submitted\n",
      "Your job 5673685 (\"brazil_292\") has been submitted\n",
      "Your job 5673686 (\"brazil_293\") has been submitted\n",
      "Your job 5673687 (\"brazil_294\") has been submitted\n",
      "Your job 5673688 (\"brazil_295\") has been submitted\n",
      "Your job 5673689 (\"brazil_296\") has been submitted\n",
      "Your job 5673690 (\"brazil_297\") has been submitted\n",
      "Your job 5673691 (\"brazil_298\") has been submitted\n",
      "Your job 5673692 (\"brazil_299\") has been submitted\n",
      "Your job 5673693 (\"brazil_300\") has been submitted\n",
      "Your job 5673694 (\"brazil_301\") has been submitted\n",
      "Your job 5673695 (\"brazil_302\") has been submitted\n",
      "Your job 5673696 (\"brazil_303\") has been submitted\n",
      "Your job 5673697 (\"brazil_304\") has been submitted\n",
      "Your job 5673698 (\"brazil_305\") has been submitted\n",
      "Your job 5673699 (\"brazil_306\") has been submitted\n",
      "Your job 5673700 (\"brazil_307\") has been submitted\n",
      "Your job 5673701 (\"brazil_308\") has been submitted\n",
      "Your job 5673702 (\"brazil_309\") has been submitted\n",
      "Your job 5673703 (\"brazil_310\") has been submitted\n",
      "Your job 5673704 (\"brazil_311\") has been submitted\n",
      "Your job 5673705 (\"brazil_312\") has been submitted\n",
      "Your job 5673706 (\"brazil_313\") has been submitted\n",
      "Your job 5673707 (\"brazil_314\") has been submitted\n",
      "Your job 5673708 (\"brazil_315\") has been submitted\n",
      "Your job 5673709 (\"brazil_316\") has been submitted\n",
      "Your job 5673710 (\"brazil_317\") has been submitted\n",
      "Your job 5673711 (\"brazil_318\") has been submitted\n",
      "Your job 5673712 (\"brazil_319\") has been submitted\n",
      "Your job 5673713 (\"brazil_320\") has been submitted\n",
      "Your job 5673714 (\"brazil_321\") has been submitted\n",
      "Your job 5673715 (\"brazil_322\") has been submitted\n",
      "Your job 5673716 (\"brazil_323\") has been submitted\n",
      "Your job 5673717 (\"brazil_324\") has been submitted\n",
      "Your job 5673718 (\"brazil_325\") has been submitted\n",
      "Your job 5673719 (\"brazil_326\") has been submitted\n",
      "Your job 5673720 (\"brazil_327\") has been submitted\n",
      "Your job 5673721 (\"brazil_328\") has been submitted\n",
      "Your job 5673722 (\"brazil_329\") has been submitted\n",
      "Your job 5673723 (\"brazil_330\") has been submitted\n",
      "Your job 5673724 (\"brazil_331\") has been submitted\n",
      "Your job 5673725 (\"brazil_332\") has been submitted\n",
      "Your job 5673726 (\"brazil_333\") has been submitted\n",
      "Your job 5673727 (\"brazil_334\") has been submitted\n",
      "Your job 5673728 (\"brazil_335\") has been submitted\n",
      "Your job 5673729 (\"brazil_336\") has been submitted\n",
      "Your job 5673730 (\"brazil_337\") has been submitted\n",
      "Your job 5673731 (\"brazil_338\") has been submitted\n",
      "Your job 5673732 (\"brazil_339\") has been submitted\n",
      "Your job 5673733 (\"brazil_340\") has been submitted\n",
      "Your job 5673734 (\"brazil_341\") has been submitted\n",
      "Your job 5673735 (\"brazil_342\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p results/msprime/logs\n",
    "\n",
    "CONDA=\"conda activate sch_man_nwinvasion-msprime;\"\n",
    "QSUB=\"qsub -V -cwd -S /bin/bash -q all.q -j y -pe smp 1 \"    \n",
    "\n",
    "for POP in tanzania senegal niger brazil; do\n",
    "    mkdir results/msprime/$POP\n",
    "    for I in $(seq 1 100); do\n",
    "        MSPRIME_CMD=\"python code/msprime-qsub.py $POP $I\"       \n",
    "        echo \"$CONDA $MSPRIME_CMD\" | $QSUB -N $POP\"_\"$I -o results/msprime/logs/$POP\"_\"$I.log\n",
    "    done\n",
    "done\n",
    "\n",
    "\n",
    "#--------------------------------------------------------------------------------------------------------\n",
    "# msprime-qsub.py\n",
    "#--------------------------------------------------------------------------------------------------------\n",
    "\n",
    "# import msprime\n",
    "# import os\n",
    "# import subprocess\n",
    "# import sys\n",
    "# from random import randint\n",
    "\n",
    "# pop = sys.argv[1]\n",
    "# iteration = sys.argv[2]\n",
    "\n",
    "# #################################\n",
    "# def sim_tree(iteration, pop, length, recomb_rate, mut_rate, seed):\n",
    "    \n",
    "#     tree_sequence = msprime.simulate( sample_size=n_samples[pop], \n",
    "#         Ne=ne[pop], \n",
    "#         length=88.9e6, \n",
    "#         recombination_rate=3.4e-8,\n",
    "#         mutation_rate=8.1e-9,\n",
    "#         random_seed=seed)\n",
    "\n",
    "#     print(\"iter {}: {} completed\".format(iteration, pop))\n",
    "#     #print(tree.draw(format=\"unicode\"))\n",
    "\n",
    "#     with open(\"results/msprime/{}/chr1_{}_rep_{}_seed_{}.vcf\".format(pop, pop, iteration, seed), \"w\") as vcf_file:\n",
    "#         tree_sequence.write_vcf(vcf_file, ploidy=2)\n",
    "# ########################################\n",
    "\n",
    "# #calcualted by est_pop_params.py\n",
    "# ne = { \"brazil\"      : 18293,\n",
    "#        \"niger\"       : 19971,\n",
    "#        \"senegal\"     : 22421,\n",
    "#        \"tanzania\"    : 52234 }\n",
    "\n",
    "# n_samples = { \"brazil\"   : 48*2,\n",
    "#               \"senegal\"  : 25*2,\n",
    "#               \"niger\"    : 10*2,\n",
    "#               \"tanzania\" : 58*2  }\n",
    "\n",
    "# seed = randint(0,1e6)\n",
    "# sim_tree(iteration, pop, 88e-9, 3.4e-8, 8.1e-9, seed)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Now after all the jobs have finished the through the scheudler take each simulated chrom and extract only the snps that fall within exome-probed regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#--------------------------------------------------------------------------------------------------------\n",
    "# msprime-probe_snps_from_vcf.py\n",
    "#--------------------------------------------------------------------------------------------------------\n",
    "\n",
    "import msprime\n",
    "import os\n",
    "import subprocess\n",
    "import glob\n",
    "\n",
    "\n",
    "#now create bed for the new \"sim\" chr1\n",
    "with open('data/renamed-sma_agilent_baits.v7.0.chr_reorderd.bed', 'r') as in_bed:\n",
    "    with open('results/msprime/sim_probes.bed', 'w') as out_bed:\n",
    "        for bed_entry in in_bed:\n",
    "            chrom, start, stop = bed_entry.rstrip().split(\"\\t\")\n",
    "            if chrom == \"SM_V7_1\":\n",
    "                out_bed.write(\"1\\t{}\\t{}\\n\".format(start, stop))\n",
    "\n",
    "bed = 'results/msprime/sim_probes.bed'\n",
    "#now loop through all of the sim vcf files to get snps at probed regions\n",
    "for pop in [\"niger\", \"senegal\", \"tanzania\", \"brazil\"]:\n",
    "    out_dir = \"results/msprime/{}\".format(pop)\n",
    "    \n",
    "    sim_vcfs = glob.glob(\"{}/chr1_*_rep_*.vcf\".format(out_dir))\n",
    "    for sim_vcf in sim_vcfs:\n",
    "\n",
    "        probed_vcf = sim_vcf.replace(\".vcf\", \"_probed.vcf\")         \n",
    "        jid = \"probe_{}\".format(probed_vcf.split(\"/\")[-1])\n",
    "        log = \"{}/logs\".format(out_dir)\n",
    "\n",
    "        vcf_cmd = \"vcftools --vcf {} --bed {} --recode --recode-INFO-all --stdout >{}\".format(sim_vcf, bed, probed_vcf)\n",
    "        qsub_cmd =  \"qsub -V -cwd -S /bin/bash -q all.q -j y -pe smp 3 -N {} -o {}\".format(jid, log)\n",
    "        conda_cmd = \"conda activate sch_man_nwinvasion-msprime\"\n",
    "\n",
    "        cmd =\"echo \\\"{}; {}\\\" | {}\".format(conda_cmd, vcf_cmd, qsub_cmd)\n",
    "\n",
    "        #run vcf cmd\n",
    "        #process = subprocess.Popen(cmd.split(\"\"),\n",
    "        #                     stdout=subprocess.PIPE, \n",
    "        #                     stderr=subprocess.PIPE)\n",
    "        !{cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## HSCAN"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Sweepfinder2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Actual data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## HSCAN\n",
    "\n",
    "HSCAN identifies regions under selection based on the length of haplotypes.  (citation).  We used a custom code to convert HSCAN to VCF input formats."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-23T01:40:44.758168Z",
     "start_time": "2020-06-23T01:40:43.813729Z"
    },
    "collapsed": true,
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5666864 (\"brazil_SM_V7_1\") has been submitted\n",
      "Your job 5666865 (\"brazil_SM_V7_2\") has been submitted\n",
      "Your job 5666866 (\"brazil_SM_V7_3\") has been submitted\n",
      "Your job 5666867 (\"brazil_SM_V7_4\") has been submitted\n",
      "Your job 5666868 (\"brazil_SM_V7_5\") has been submitted\n",
      "Your job 5666869 (\"brazil_SM_V7_6\") has been submitted\n",
      "Your job 5666870 (\"brazil_SM_V7_7\") has been submitted\n",
      "Your job 5666871 (\"tanzania_SM_V7_1\") has been submitted\n",
      "Your job 5666872 (\"tanzania_SM_V7_2\") has been submitted\n",
      "Your job 5666873 (\"tanzania_SM_V7_3\") has been submitted\n",
      "Your job 5666874 (\"tanzania_SM_V7_4\") has been submitted\n",
      "Your job 5666875 (\"tanzania_SM_V7_5\") has been submitted\n",
      "Your job 5666876 (\"tanzania_SM_V7_6\") has been submitted\n",
      "Your job 5666877 (\"tanzania_SM_V7_7\") has been submitted\n",
      "Your job 5666878 (\"niger_SM_V7_1\") has been submitted\n",
      "Your job 5666879 (\"niger_SM_V7_2\") has been submitted\n",
      "Your job 5666880 (\"niger_SM_V7_3\") has been submitted\n",
      "Your job 5666881 (\"niger_SM_V7_4\") has been submitted\n",
      "Your job 5666882 (\"niger_SM_V7_5\") has been submitted\n",
      "Your job 5666883 (\"niger_SM_V7_6\") has been submitted\n",
      "Your job 5666884 (\"niger_SM_V7_7\") has been submitted\n",
      "Your job 5666885 (\"senegal_SM_V7_1\") has been submitted\n",
      "Your job 5666886 (\"senegal_SM_V7_2\") has been submitted\n",
      "Your job 5666887 (\"senegal_SM_V7_3\") has been submitted\n",
      "Your job 5666888 (\"senegal_SM_V7_4\") has been submitted\n",
      "Your job 5666889 (\"senegal_SM_V7_5\") has been submitted\n",
      "Your job 5666890 (\"senegal_SM_V7_6\") has been submitted\n",
      "Your job 5666891 (\"senegal_SM_V7_7\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd ~/sch_man_nwinvasion/\n",
    "\n",
    "CONDA=\"conda activate sch_man_nwinvasion-hscan\"\n",
    "QSUB=\"qsub -V -cwd -S /bin/bash -q all.q -j y -pe smp 1 \"    \n",
    "OUTDIR=\"results/hscan\"\n",
    "\n",
    "if [ ! -d $OUTDIR ] \n",
    "then\n",
    "    mkdir $OUTDIR\n",
    "fi\n",
    "\n",
    "mkdir -p $OUTDIR/logs\n",
    "\n",
    "#REAL DATA\n",
    "for POP in brazil tanzania niger senegal; do\n",
    "    mkdir $OUTDIR/$POP\n",
    "        for i in $(seq 1 7 ); do\n",
    "            CHROM=\"SM_V7_\"$i\n",
    "    \n",
    "            #get chrom vcf\n",
    "            VCF_CMD=\"vcftools \\\n",
    "                --vcf results/smv7_ex_autosomes.vcf \\\n",
    "                --keep results/\"$POP\".list \\\n",
    "                --chr $CHROM \\\n",
    "                --recode \\\n",
    "                --recode-INFO-all \\\n",
    "                --stdout \\\n",
    "                >$OUTDIR/$POP/\"$CHROM\"_\"$POP\".vcf\"\n",
    "\n",
    "            #convert vcf to hscan\n",
    "            CONV_CMD=\"python code/vcf2hscan.py \\\n",
    "                $OUTDIR/$POP/\"$CHROM\"_\"$POP\".vcf \\\n",
    "                $OUTDIR/$POP/\"$CHROM\"_\"$POP\".hscan-in\"\n",
    "\n",
    "            #convert to hscan\n",
    "            HSCAN_CMD=\"bin/H-scan \\\n",
    "                -i $OUTDIR/$POP/\"$CHROM\"_\"$POP\".hscan-in \\\n",
    "                -g 10000 \\\n",
    "                >$OUTDIR/$POP/\"$CHROM\"_\"$POP\".hscan-out\"\n",
    "\n",
    "        echo \"$CONDA; $VCF_CMD; $CONV_CMD; $HSCAN_CMD\" | $QSUB -N $POP\"_\"$CHROM -o $OUTDIR/logs/$POP\"_\"$CHROM.log\n",
    "\n",
    "    done\n",
    "done\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "### Calculate significance from Hscan results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-23T16:19:43.995732Z",
     "start_time": "2020-06-23T16:07:28.438665Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "niger\n",
      "senegal\n",
      "brazil\n",
      "tanzania\n"
     ]
    }
   ],
   "source": [
    "#calculate p-values\n",
    "import scipy.stats\n",
    "import statsmodels.api as sm\n",
    "import statsmodels as sm\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import glob\n",
    "\n",
    "#make sure that all stops are not gt chrom length\n",
    "chr_length = {}\n",
    "#genome_size = 0\n",
    "with open('/master/nplatt/sch_man_nwinvasion/data/genomes/Smansoni_v7.fa.fai', 'r') as fai:\n",
    "    for entry in fai:\n",
    "        chrom, length, *offset = entry.rstrip().split(\"\\t\")\n",
    "        chr_length[chrom]=int(length)\n",
    "\n",
    "    cumul_start={}\n",
    "    cumul_start['SM_V7_1']=0\n",
    "    cumul_start['SM_V7_2']= cumul_start['SM_V7_1'] + chr_length['SM_V7_1']\n",
    "    cumul_start['SM_V7_3']= cumul_start['SM_V7_2'] + chr_length['SM_V7_2']\n",
    "    cumul_start['SM_V7_4']= cumul_start['SM_V7_3'] + chr_length['SM_V7_3']\n",
    "    cumul_start['SM_V7_5']= cumul_start['SM_V7_4'] + chr_length['SM_V7_4']\n",
    "    cumul_start['SM_V7_6']= cumul_start['SM_V7_5'] + chr_length['SM_V7_5']\n",
    "    cumul_start['SM_V7_7']= cumul_start['SM_V7_6'] + chr_length['SM_V7_6']\n",
    "    scanned_size = cumul_start['SM_V7_7'] + chr_length['SM_V7_7']\n",
    "\n",
    "#cycle through each population to calculate generate csv file with H scores, pvalues etc. \n",
    "for pop in [\"niger\", \"senegal\", \"brazil\", \"tanzania\"]:\n",
    "    print(pop)\n",
    "    \n",
    "    #make sure lists are clean\n",
    "    chrom_s=[]\n",
    "    pos_s=[]\n",
    "    h_s=[]\n",
    "\n",
    "    #read all the hscan output for the population into lists for chrom, h, and pos\n",
    "    for i in range(1,8):\n",
    "        chrom = \"SM_V7_{}\".format(i)\n",
    "        \n",
    "        with open(\"results/hscan/{}/{}_{}.hscan-out\".format(pop, chrom, pop), 'r') as hscan_file:\n",
    "            #skip the header line\n",
    "            next(hscan_file)\n",
    "            \n",
    "            for calc in hscan_file:\n",
    "                \n",
    "                pos, h = calc.split(\"\\t\")\n",
    "                \n",
    "                chrom_s.append(chrom)\n",
    "                h_s.append(h)\n",
    "                pos_s.append(pos)\n",
    "\n",
    "    #convert the lists to numpy arrays\n",
    "    chrom_s=np.array(chrom_s)\n",
    "    pos_s=np.array(pos_s)\n",
    "    h_s=np.array(h_s).astype(np.float)\n",
    "    \n",
    "    #calculate z score, p and q values. Here i used an alpha of 0.01 and the Benjamini/Hochberg (non-negative) correction for multiple test\n",
    "    z_s=scipy.stats.zscore(h_s)\n",
    "    p_s=scipy.special.ndtr(-z_s)   \n",
    "    reject_s, q_s, sidak, bonf =sm.stats.multitest.multipletests(p_s, alpha=0.01, method=\"fdr_bh\")\n",
    "    logq_s=-np.log10(q_s)\n",
    "\n",
    "    #now add all info to a data table\n",
    "    columns = [\"chrom\", \"pos\", \"h\", \"z\", \"p\", \"q\", \"-log10(q)\"]\n",
    "    df = pd.DataFrame(data = [chrom_s, pos_s, h_s, z_s, p_s, q_s, logq_s]).T\n",
    "    df.columns=columns\n",
    "\n",
    "    #get cumulative positions along the genome (additive positions; used to easily organize data in chr order)\n",
    "    fig_x_pos_s=[]\n",
    "    for index, row in df.iterrows(): \n",
    "        fig_x_pos_s.append(int(row[\"pos\"]) + int(cumul_start[row['chrom']]))\n",
    "\n",
    "    df['fig_x_pos']=fig_x_pos_s\n",
    "    df = df.sort_values([\"fig_x_pos\"], ascending = True)\n",
    "\n",
    "    #save data to csv file\n",
    "    csv_file =\"results/hscan/{}_hscan.csv\".format(pop)\n",
    "    df.to_csv(csv_file, index=False, header=True, mode='w')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## SWEEPFINDER2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "### Calc SFS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-23T01:17:36.033333Z",
     "start_time": "2020-06-23T01:17:07.321087Z"
    },
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "done readsnps datasize=475081 nmax=80 nmin=24 xmax=81 invar=2\n",
      "getfreq rdatasize=1327\n",
      "calling findmax numpar=80\n",
      "RUNNING THE L-BFGS-B CODE\n",
      "\n",
      "\t* * *\n",
      "\n",
      "Machine precision=2.220446e-16\n",
      "N = 80    M = 12\n",
      "ITERATION 0\n",
      "At iterate    0 f=449028.786508\t|proj g| = 1.732743e+01\n",
      "\n",
      "\t* * *\n",
      "\n",
      "Tit   = total number of iterations\n",
      "Tnf   = total number of function evaluations\n",
      "Tnint = total number of segments explored during Cauchy searches\n",
      "Skip  = number of BFGS updates skipped\n",
      "Nact  = number of active bounds at final generalized Cuachy point\n",
      "Projg = norm of the final projected gradient\n",
      "F     = final function value\n",
      "\n",
      "\t* * *\n",
      "N\tTit\tTnf\tTnint\tSkip\tNact\tProjg\tF\n",
      "80\t15\t19\t41\t0\t0\t1.222590e+01\t4.488109e+05\n",
      " F = 448810.933613\n",
      "CONVERGENCE: REL_REDUCTION_OF_F <= FACTR*EPSMCH            \n",
      "Cauchy time 0 seconds.\n",
      "Subspace minimization time 0 seconds.\n",
      "Line search time 13 seconds.\n",
      "Total User time 14 seconds.\n",
      "\n",
      "frequency spectrum written to smv7_ex_autosomes_n10.SpectFile\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--counts2\n",
      "\t--out smv7_ex_autosomes_n10_tmp\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "After filtering, kept 40 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 475081 out of a possible 475081 Sites\n",
      "Run Time = 11.00 seconds\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "if [ ! -d \"results/sweepfinder\" ] \n",
    "then\n",
    "    mkdir -p results/sweepfinder\n",
    "fi\n",
    "\n",
    "cd results/sweepfinder\n",
    "\n",
    "#convert to sweefinder2 format\n",
    "vcftools \\\n",
    "    --vcf ../smv7_ex_autosomes.vcf \\\n",
    "    --counts2 \\\n",
    "    --out smv7_ex_autosomes_tmp\n",
    "\n",
    "tail -n+2 smv7_ex_autosomes_tmp.frq.count \\\n",
    "    | awk -v OFS=\"\\t\" '{print $2,$6,$4,\"1\"}' \\\n",
    "    >smv7_ex_autosomes.in\n",
    "\n",
    "echo -e 'position\\tx\\tn\\tfolded' \\\n",
    "    | cat - smv7_ex_autosomes.in \\\n",
    "    > temp && mv temp smv7_ex_autosomes.in\n",
    "\n",
    "#calculate genome-wide sfs\n",
    "SweepFinder2 -f smv7_ex_autosomes.in smv7_ex_autosomes.SpectFile"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "### run sweepfinder on the actual data\n",
    "\n",
    "code/sweepfinder2-run_with_real_data.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-23T01:32:36.495826Z",
     "start_time": "2020-06-23T01:30:47.729264Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5666836 (\"sf2r1_brazil\") has been submitted\n",
      "Your job 5666837 (\"sf2r2_brazil\") has been submitted\n",
      "Your job 5666838 (\"sf2r3_brazil\") has been submitted\n",
      "Your job 5666839 (\"sf2r4_brazil\") has been submitted\n",
      "Your job 5666840 (\"sf2r5_brazil\") has been submitted\n",
      "Your job 5666841 (\"sf2r6_brazil\") has been submitted\n",
      "Your job 5666842 (\"sf2r7_brazil\") has been submitted\n",
      "Your job 5666843 (\"sf2r1_senegal\") has been submitted\n",
      "Your job 5666844 (\"sf2r2_senegal\") has been submitted\n",
      "Your job 5666845 (\"sf2r3_senegal\") has been submitted\n",
      "Your job 5666846 (\"sf2r4_senegal\") has been submitted\n",
      "Your job 5666847 (\"sf2r5_senegal\") has been submitted\n",
      "Your job 5666848 (\"sf2r6_senegal\") has been submitted\n",
      "Your job 5666849 (\"sf2r7_senegal\") has been submitted\n",
      "Your job 5666850 (\"sf2r1_niger\") has been submitted\n",
      "Your job 5666851 (\"sf2r2_niger\") has been submitted\n",
      "Your job 5666852 (\"sf2r3_niger\") has been submitted\n",
      "Your job 5666853 (\"sf2r4_niger\") has been submitted\n",
      "Your job 5666854 (\"sf2r5_niger\") has been submitted\n",
      "Your job 5666855 (\"sf2r6_niger\") has been submitted\n",
      "Your job 5666856 (\"sf2r7_niger\") has been submitted\n",
      "Your job 5666857 (\"sf2r1_tanzania\") has been submitted\n",
      "Your job 5666858 (\"sf2r2_tanzania\") has been submitted\n",
      "Your job 5666859 (\"sf2r3_tanzania\") has been submitted\n",
      "Your job 5666860 (\"sf2r4_tanzania\") has been submitted\n",
      "Your job 5666861 (\"sf2r5_tanzania\") has been submitted\n",
      "Your job 5666862 (\"sf2r6_tanzania\") has been submitted\n",
      "Your job 5666863 (\"sf2r7_tanzania\") has been submitted\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory `brazil': File exists\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_1\n",
      "\t--keep ../brazil_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 162219 out of a possible 475081 Sites\n",
      "Run Time = 6.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_2\n",
      "\t--keep ../brazil_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 78902 out of a possible 475081 Sites\n",
      "Run Time = 4.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_3\n",
      "\t--keep ../brazil_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 80481 out of a possible 475081 Sites\n",
      "Run Time = 4.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_4\n",
      "\t--keep ../brazil_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 75581 out of a possible 475081 Sites\n",
      "Run Time = 4.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_5\n",
      "\t--keep ../brazil_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 24681 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_6\n",
      "\t--keep ../brazil_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 31877 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_7\n",
      "\t--keep ../brazil_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 21340 out of a possible 475081 Sites\n",
      "Run Time = 2.00 seconds\n",
      "mkdir: cannot create directory `senegal': File exists\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_1\n",
      "\t--keep ../senegal_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 162219 out of a possible 475081 Sites\n",
      "Run Time = 6.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_2\n",
      "\t--keep ../senegal_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 78902 out of a possible 475081 Sites\n",
      "Run Time = 4.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_3\n",
      "\t--keep ../senegal_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 80481 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_4\n",
      "\t--keep ../senegal_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 75581 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_5\n",
      "\t--keep ../senegal_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 24681 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_6\n",
      "\t--keep ../senegal_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 31877 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_7\n",
      "\t--keep ../senegal_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 21340 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "mkdir: cannot create directory `niger': File exists\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_1\n",
      "\t--keep ../niger_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 162219 out of a possible 475081 Sites\n",
      "Run Time = 5.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_2\n",
      "\t--keep ../niger_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 78902 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_3\n",
      "\t--keep ../niger_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 80481 out of a possible 475081 Sites\n",
      "Run Time = 4.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_4\n",
      "\t--keep ../niger_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 75581 out of a possible 475081 Sites\n",
      "Run Time = 4.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_5\n",
      "\t--keep ../niger_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 24681 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_6\n",
      "\t--keep ../niger_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 31877 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_7\n",
      "\t--keep ../niger_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 21340 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "mkdir: cannot create directory `tanzania': File exists\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_1\n",
      "\t--keep ../tanzania_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 162219 out of a possible 475081 Sites\n",
      "Run Time = 5.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_2\n",
      "\t--keep ../tanzania_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 78902 out of a possible 475081 Sites\n",
      "Run Time = 4.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_3\n",
      "\t--keep ../tanzania_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 80481 out of a possible 475081 Sites\n",
      "Run Time = 4.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_4\n",
      "\t--keep ../tanzania_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 75581 out of a possible 475081 Sites\n",
      "Run Time = 4.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_5\n",
      "\t--keep ../tanzania_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 24681 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_6\n",
      "\t--keep ../tanzania_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 31877 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_7\n",
      "\t--keep ../tanzania_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 21340 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd results/sweepfinder\n",
    "\n",
    "\n",
    "for POP in brazil senegal niger tanzania; do\n",
    "    mkdir $POP\n",
    "    \n",
    "    #get the pop specific sfs\n",
    "    vcftools \\\n",
    "        --keep ../\"$POP\".list \\\n",
    "        --vcf ../smv7_ex_autosomes.vcf  \\\n",
    "        --counts2 \\\n",
    "        --stdout \\\n",
    "        >$POP/$POP.freq\n",
    "\n",
    "    tail -n+2 $POP/$POP.freq  | awk -v OFS=\"\\t\" '{print $2,$6,$4,\"1\"}' >$POP/$POP.in\n",
    "\n",
    "    echo -e 'position\\tx\\tn\\tfolded' | cat - $POP/$POP.in > temp && mv temp $POP/$POP.in\n",
    "\n",
    "    SweepFinder2 -f $POP/$POP.in $POP/$POP.sfs\n",
    " \n",
    "    #run sweepfinder for each chrom\n",
    "    for I in $(seq 1 7); do\n",
    "        CHR=SM_V7_$I\n",
    "\n",
    "        #convert vcf to sw input format\n",
    "        vcftools \\\n",
    "            --keep ../\"$POP\".list \\\n",
    "            --vcf ../smv7_ex_autosomes.vcf \\\n",
    "            --chr $CHR \\\n",
    "            --counts2 \\\n",
    "            --stdout \\\n",
    "            >$POP/$CHR\"_\"$POP.freq\n",
    "\n",
    "        tail -n+2 $POP/$CHR\"_\"$POP.freq  | awk -v OFS=\"\\t\" '{print $2,$6,$4,\"1\"}' >$POP/$CHR\"_\"$POP.in\n",
    "\n",
    "        echo -e 'position\\tx\\tn\\tfolded' | cat - $POP/$CHR\"_\"$POP.in > temp && mv temp $POP/$CHR\"_\"$POP.in\n",
    "\n",
    "        #submit sweepfinder        \n",
    "        CMD=\"conda activate sch_man_nwinvasion-sweepfinder; SweepFinder2 -lg 1000 $POP/$CHR\"_\"$POP.in $POP/$POP.sfs $POP/$CHR\"_\"$POP.sw2out\"\n",
    "\n",
    "        echo $CMD | qsub -V -cwd -S /bin/bash -q all.q -j y -N sf2r\"$I\"_\"$POP\" -o $POP/real_$CHR.log -pe smp 2\n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "cd results/sweepfinder\n",
    "\n",
    "CONDA=\"conda activate sch_man_nwinvasion-sweepfinder\"\n",
    "QSUB=\"qsub -V -cwd -S /bin/bash -q all.q -j y -pe smp 2 \"    \n",
    "\n",
    "mkdir logs\n",
    "\n",
    "for POP in senegal tanzania niger brazil; do\n",
    "    mkdir $POP-sim\n",
    "\n",
    "    for PROBED_VCF in $(ls ~/sch_man_nwinvasion/results/msprime/$POP/chr1_\"$POP\"_rep_*seed_*_probed.vcf); do\n",
    "        SAMPLE_NAME=$(basename $PROBED_VCF .vcf)\n",
    "        \n",
    "        #convert to sweep format\n",
    "        vcftools \\\n",
    "            --vcf $PROBED_VCF  \\\n",
    "            --counts2 \\\n",
    "            --stdout \\\n",
    "            >$POP-sim/$SAMPLE_NAME.freq\n",
    "\n",
    "        tail -n+2 $POP-sim/$SAMPLE_NAME.freq  | awk -v OFS=\"\\t\" '{print $2,$6,$4,\"1\"}' >$POP-sim/$SAMPLE_NAME.in\n",
    "\n",
    "        echo -e 'position\\tx\\tn\\tfolded' | cat - $POP-sim/$SAMPLE_NAME.in > temp && mv temp $POP-sim/$SAMPLE_NAME.in\n",
    "\n",
    "        #generate sfs\n",
    "        SFS_CMD=\"SweepFinder2 -f $POP-sim/$SAMPLE_NAME.in $POP-sim/$SAMPLE_NAME.sfs >logs/\"$SAMPLE_NAME\"_sfs.log\"\n",
    "\n",
    "        #gen sweepfinder2\n",
    "        SF2_CMD=\"SweepFinder2 -lg 1000 $POP-sim/$SAMPLE_NAME.in $POP-sim/$SAMPLE_NAME.sfs $POP-sim/$SAMPLE_NAME.sw2out\"\n",
    "        \n",
    "        #submit to the queue\n",
    "        CMD=\"$CONDA; $SFS_CMD; $SF2_CMD\"\n",
    "        echo $CMD | $QSUB -N sf2s_$SAMPLE_NAME -o logs/sf2_sim_$SAMPLE_NAME.log\n",
    "\n",
    "    done\n",
    "done\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### generate csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-25T17:40:59.468884Z",
     "start_time": "2020-06-25T17:33:07.818458Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "niger\n",
      "senegal\n",
      "brazil\n",
      "tanzania\n"
     ]
    }
   ],
   "source": [
    "#calculate p-values\n",
    "import scipy.stats\n",
    "import statsmodels.api as sm\n",
    "import statsmodels as sm\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import glob\n",
    "\n",
    "#make sure that all stops are not gt chrom length\n",
    "chr_length = {}\n",
    "\n",
    "with open('/master/nplatt/sch_man_nwinvasion/data/genomes/Smansoni_v7.fa.fai', 'r') as fai:\n",
    "    for entry in fai:\n",
    "        chrom, length, *offset = entry.rstrip().split(\"\\t\")\n",
    "        chr_length[chrom]=int(length)\n",
    "\n",
    "    cumul_start={}\n",
    "    cumul_start['SM_V7_1']=0\n",
    "    cumul_start['SM_V7_2']= cumul_start['SM_V7_1'] + chr_length['SM_V7_1']\n",
    "    cumul_start['SM_V7_3']= cumul_start['SM_V7_2'] + chr_length['SM_V7_2']\n",
    "    cumul_start['SM_V7_4']= cumul_start['SM_V7_3'] + chr_length['SM_V7_3']\n",
    "    cumul_start['SM_V7_5']= cumul_start['SM_V7_4'] + chr_length['SM_V7_4']\n",
    "    cumul_start['SM_V7_6']= cumul_start['SM_V7_5'] + chr_length['SM_V7_5']\n",
    "    cumul_start['SM_V7_7']= cumul_start['SM_V7_6'] + chr_length['SM_V7_6']\n",
    "    scanned_size = cumul_start['SM_V7_7'] + chr_length['SM_V7_7']\n",
    "\n",
    "#cycle through each population to calculate generate csv file with H scores, pvalues etc. \n",
    "for pop in [\"niger\", \"senegal\", \"brazil\", \"tanzania\"]:\n",
    "    print(pop)\n",
    "    \n",
    "    #make sure lists are clean\n",
    "    chrom_s=[]\n",
    "    pos_s=[]\n",
    "    lr_s=[]\n",
    "\n",
    "    #read all the sweepfinder output for the population into lists for chrom, lr, and pos\n",
    "    for i in range(1,8):\n",
    "        chrom = \"SM_V7_{}\".format(i)\n",
    "        \n",
    "        with open(\"results/n10/sweepfinder/{}/{}_{}.sw2out\".format(pop, chrom, pop), 'r') as sf2_file:\n",
    "            #skip header line\n",
    "            next(sf2_file)\n",
    "            for calc in sf2_file:\n",
    "                \n",
    "                pos, lr, alpha = calc.rstrip().split(\"\\t\")\n",
    "                \n",
    "                chrom_s.append(chrom)\n",
    "                lr_s.append(lr)\n",
    "                pos_s.append(pos)\n",
    "\n",
    "    #convert the lists to numpy arrays\n",
    "    chrom_s=np.array(chrom_s)\n",
    "    pos_s=np.array(pos_s)\n",
    "    lr_s=np.array(lr_s).astype(np.float)\n",
    "    \n",
    "    #calculate z score, p and q values. Here i used an alpha of 0.01 and the Benjamini/Hochberg (non-negative) correction for multiple test\n",
    "    z_s=scipy.stats.zscore(lr_s)\n",
    "    p_s=scipy.special.ndtr(-z_s)   \n",
    "    reject_s, q_s, sidak, bonf =sm.stats.multitest.multipletests(p_s, alpha=0.01, method=\"fdr_bh\")\n",
    "    logq_s=-np.log10(q_s)\n",
    "\n",
    "    #now add all info to a data table\n",
    "    columns = [\"chrom\", \"pos\", \"lr\", \"z\", \"p\", \"q\", \"-log10(q)\"]\n",
    "    df = pd.DataFrame(data = [chrom_s, pos_s, lr_s, z_s, p_s, q_s, logq_s]).T\n",
    "    df.columns=columns\n",
    "\n",
    "    #get cumulative positions along the genome (additive positions; used to easily organize data in chr order)\n",
    "    fig_x_pos_s=[]\n",
    "    for index, row in df.iterrows(): \n",
    "        fig_x_pos_s.append(int(float(row[\"pos\"])) + int(cumul_start[row['chrom']]))\n",
    "\n",
    "    df['fig_x_pos']=fig_x_pos_s\n",
    "    df = df.sort_values([\"fig_x_pos\"], ascending = True)\n",
    "\n",
    "    #save data to csv file\n",
    "    csv_file =\"results/n10/sweepfinder/{}_n10_sw2.csv\".format(pop)\n",
    "    df.to_csv(csv_file, index=False, header=True, mode='w')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## FST"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-29T21:06:39.615431Z",
     "start_time": "2020-06-29T21:05:28.739670Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import shutil\n",
    "import allel\n",
    "import math\n",
    "import yaml\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "from collections import defaultdict\n",
    "from scipy import stats\n",
    "from scipy import signal\n",
    "\n",
    "os.chdir(\"/master/nplatt/sch_man_nwinvasion\")\n",
    "\n",
    "if not os.path.exists(\"results/fst_per_window\"):\n",
    "    os.mkdir(\"results/fst_per_window\")\n",
    "    \n",
    "with open('data/pop_assign.yml') as yaml_file:\n",
    "    pop_assign = yaml.load(yaml_file, Loader=yaml.FullLoader)\n",
    "\n",
    "#-----------------------------------\n",
    "# get lengths from cumul positions\n",
    "#make sure that all stops are not gt chrom length\n",
    "chr_length = {}\n",
    "#genome_size = 0\n",
    "with open('/master/nplatt/sch_man_nwinvasion/data/genomes/Smansoni_v7.fa.fai', 'r') as fai:\n",
    "    for entry in fai:\n",
    "        chrom, length, *offset = entry.rstrip().split(\"\\t\")\n",
    "        chr_length[chrom]=int(length)\n",
    "\n",
    "    cumul_start={}\n",
    "    cumul_start['SM_V7_1']=0\n",
    "    cumul_start['SM_V7_2']= cumul_start['SM_V7_1'] + chr_length['SM_V7_1']\n",
    "    cumul_start['SM_V7_3']= cumul_start['SM_V7_2'] + chr_length['SM_V7_2']\n",
    "    cumul_start['SM_V7_4']= cumul_start['SM_V7_3'] + chr_length['SM_V7_3']\n",
    "    cumul_start['SM_V7_5']= cumul_start['SM_V7_4'] + chr_length['SM_V7_4']\n",
    "    cumul_start['SM_V7_6']= cumul_start['SM_V7_5'] + chr_length['SM_V7_5']\n",
    "    cumul_start['SM_V7_7']= cumul_start['SM_V7_6'] + chr_length['SM_V7_6']\n",
    "    scanned_size = cumul_start['SM_V7_7'] + chr_length['SM_V7_7']\n",
    "\n",
    "#-------------------------------------------------------------------------------\n",
    "# get genotype info per population\n",
    "\n",
    "#read in vcf\n",
    "callset=allel.read_vcf('results/variant_filtration/smv7_ex_autosomes.vcf')\n",
    "\n",
    "#now get an index for each sample/population\n",
    "samples = callset[\"samples\"]\n",
    "\n",
    "i=0 \n",
    "pop_idxs = defaultdict(list)   \n",
    "for sample in samples:  \n",
    "     pop_idxs[pop_assign[sample]].append(i) \n",
    "     i=i+1 \n",
    "\n",
    "pops= list(pop_idxs.keys()) \n",
    "\n",
    "#get genotypes\n",
    "gt=allel.GenotypeArray(callset['calldata/GT'])\n",
    "\n",
    "#now get allele count per population\n",
    "ac=gt.count_alleles()\n",
    "\n",
    "#for simplicity add maf info to callset data\n",
    "maf=ac[:, :2].min(axis=1)/ac[:, :2].sum(axis=1)\n",
    "callset['maf']=maf \n",
    "\n",
    "pop_ac={}\n",
    "for pop in pops:\n",
    "    pop_ac[pop] = gt.count_alleles(subpop=pop_idxs[pop])\n",
    "    \n",
    "#-------------------------------------------------------------------------------\n",
    "#generate windows\n",
    "window=100_000\n",
    "\n",
    "#define an array of window start and stops\n",
    "window_starts = [int(x - (window/2)) for x in callset['variants/POS']]\n",
    "window_stops  = [int(x + (window/2)) for x in callset['variants/POS']]\n",
    "\n",
    "#make sure that window starts are all gt 1\n",
    "window_starts = [1 if i < 1 else i for i in window_starts]\n",
    "\n",
    "\n",
    "#make sure that all stops are not gt chrom length\n",
    "chr_length = {}\n",
    "#genome_size = 0\n",
    "with open('/master/nplatt/sch_man_nwinvasion/data/genomes/Smansoni_v7.fa.fai', 'r') as fai:\n",
    "    for entry in fai:\n",
    "        chrom, length, *offset = entry.rstrip().split(\"\\t\")\n",
    "        chr_length[chrom]=int(length)\n",
    "        #genome_size = genome_size + chr_length[chrom]\n",
    "    \n",
    "i=0\n",
    "for stop in window_stops:\n",
    "    chrom = callset['variants/CHROM'][i]\n",
    "    \n",
    "    if stop > chr_length[chrom]:\n",
    "        window_stops[i]=chr_length[chrom]\n",
    "    i=i+1\n",
    "    \n",
    "windows = np.column_stack((np.array(window_starts), \n",
    "                           np.array(window_stops)))\n",
    "\n",
    "callset['windows']=windows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-29T21:50:54.571489Z",
     "start_time": "2020-06-29T21:30:21.821049Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "brazil\n",
      "tanzania\n",
      "niger\n",
      "senegal\n"
     ]
    }
   ],
   "source": [
    "#-------------------------------------------------------------------------------\n",
    "# fst calculations\n",
    "\n",
    "pops = [\"brazil\", \"tanzania\", \"niger\", \"senegal\" ]\n",
    "\n",
    "idx_comps = {\"brazil\":   [pop_idxs[\"brazil\"],   pop_idxs[\"tanzania\"] + pop_idxs[\"niger\"]  + pop_idxs[\"senegal\"] ],\n",
    "             \"tanzania\": [pop_idxs[\"tanzania\"], pop_idxs[\"brazil\"]   + pop_idxs[\"niger\"]  + pop_idxs[\"senegal\"] ],\n",
    "             \"niger\":    [pop_idxs[\"niger\"],    pop_idxs[\"tanzania\"] + pop_idxs[\"brazil\"] + pop_idxs[\"senegal\"] ],\n",
    "             \"senegal\":  [pop_idxs[\"senegal\"],  pop_idxs[\"tanzania\"] + pop_idxs[\"niger\"]  + pop_idxs[\"brazil\"] ]}\n",
    "\n",
    "#make comparisons between population\n",
    "for pop in idx_comps.keys():\n",
    "    print(pop)\n",
    "    pop1_idx = idx_comps[pop][0]\n",
    "    pop2_idx = idx_comps[pop][1]\n",
    "\n",
    "    fst_s             = []\n",
    "    fst_calc_window_s = []\n",
    "    fst_count_s       = []\n",
    "\n",
    "    #create empty dataframe to store data    \n",
    "    headers = [\"chrom\", \"pos\", \"fst\", \"smoothed_fst\", \"window\", \"num_snps\", \"zscore\", \"pvalue\", \"-log10(p)\"]\n",
    "    df=pd.DataFrame(columns=headers) \n",
    "\n",
    "    #now loop through each chromosome\n",
    "    for chrom in list(set(callset['variants/CHROM'])) :\n",
    "        target_sites = np.logical_and( callset['maf'] < 0.05, \n",
    "                                       callset['variants/CHROM'] == chrom )  \n",
    "\n",
    "        chr_gts  = gt[target_sites]\n",
    "        chr_poss = callset['variants/POS'][target_sites]\n",
    "        chr_wins = callset['windows'][target_sites]\n",
    "\n",
    "        \n",
    "        fsts, fst_calc_windows, fst_counts =allel.windowed_weir_cockerham_fst(chr_poss, chr_gts, subpops=[pop1_idx, pop2_idx], windows=chr_wins )\n",
    "\n",
    "        #get rid of nan values\n",
    "        useful_values = np.logical_and( np.isfinite(fsts), fst_counts>=10) \n",
    "\n",
    "        fsts = fsts[useful_values]\n",
    "        fst_calc_windows = fst_calc_windows[useful_values]\n",
    "        fst_counts = fst_counts[useful_values]\n",
    "        chr_poss = chr_poss[useful_values]\n",
    "\n",
    "        #set negative fst values to 0\n",
    "        i=0\n",
    "        for fst in fsts:\n",
    "            if fst <0:\n",
    "                fsts[i]=0\n",
    "            i=i+1        \n",
    "        \n",
    "        #smooth\n",
    "        smoothed_fsts=signal.medfilt(fsts, kernel_size = 101)\n",
    "\n",
    "        #calculate pvalue\n",
    "        zs = stats.zscore(fsts)\n",
    "        ps = stats.norm.cdf(zs)\n",
    "        logps = [-1*np.log10(p) for p in ps] \n",
    "\n",
    "        #add data to dataframe/table\n",
    "        data = list(zip([chrom]*len(fsts), chr_poss, fsts, smoothed_fsts, fst_calc_windows, fst_counts, zs, ps, logps))\n",
    "        chr_df=pd.DataFrame(data, columns=headers)\n",
    "        df = df.append(chr_df)\n",
    "\n",
    "    #add cumul positions\n",
    "    fig_x_pos_s=[]\n",
    "    for index, row in df.iterrows(): \n",
    "        fig_x_pos_s.append(int(row[\"pos\"]) + int(cumul_start[row['chrom']]))\n",
    "\n",
    "    df['fig_x_pos']=fig_x_pos_s\n",
    "\n",
    "    #save data to csv file\n",
    "    csv_file = \"./results/fst_per_window/{}_vs_all_windowed_fst.csv\".format(pop)\n",
    "    df = df.sort_values([\"fig_x_pos\"], ascending = True)\n",
    "    df.to_csv(csv_file, index=False, header=True, mode='w')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Reconcile stats per window/site"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-30T17:07:20.744762Z",
     "start_time": "2020-06-30T16:39:21.703481Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "senegal\n",
      "niger\n",
      "brazil\n",
      "tanzania\n"
     ]
    }
   ],
   "source": [
    "from collections import defaultdict\n",
    "from intervaltree import IntervalTree\n",
    "import pandas\n",
    "import os\n",
    "\n",
    "if not os.path.exists(\"results/selection_peaks\"):\n",
    "    os.mkdir(\"results/selection_peaks\")\n",
    "  \n",
    "for pop in [\"senegal\", \"niger\", \"brazil\", \"tanzania\"]:\n",
    "    print(pop)\n",
    "\n",
    "    #designate files\n",
    "    hscan_file = \"results/hscan/{}/{}_hscan.csv\".format(pop, pop)\n",
    "    sweep_file = \"results/sweepfinder/{}/{}_sw2.csv\".format(pop, pop)\n",
    "    fst_file   = \"results/fst_per_window/{}_vs_all_windowed_fst.csv\".format(pop)\n",
    "\n",
    "    #read in to dfs    \n",
    "    hscan_df = pd.read_csv(hscan_file, sep=\",\")\n",
    "    sweep_df = pd.read_csv(sweep_file, sep=\",\")\n",
    "    fst_df   = pd.read_csv(fst_file,   sep=\",\")\n",
    "\n",
    "    #merge data frames\n",
    "    df = hscan_df\n",
    "    df = df.merge(fst_df, how=\"outer\", left_on=['chrom', 'pos'], right_on=['chrom', 'pos'])\n",
    "\n",
    "    for chrom in range(1,8):\n",
    "        chrom=\"SM_V7_{}\".format(chrom)\n",
    "\n",
    "        chrom_df       = df.loc[df['chrom'] == chrom]     \n",
    "        chrom_sweep_df = sweep_df.loc[sweep_df['chrom'] == chrom]     \n",
    "       \n",
    "    #build interval tree of all sweepfinder values\n",
    "    sweep_trees=defaultdict(lambda: IntervalTree()) \n",
    "    for index, row in sweep_df.iterrows():\n",
    "        sweep_trees[row.chrom][row.pos:row.pos+1001] = row.lr\n",
    "\n",
    "    lrs=[]\n",
    "    for index, row in df.iterrows():\n",
    "        lr = sorted(sweep_trees[row.chrom].at(row.pos))[0].data   \n",
    "        lrs.append(lr)\n",
    "    \n",
    "    #add sweepfinder values to dataframe\n",
    "    df[\"lr\"]=lrs\n",
    "    \n",
    "    #GET PERCENTILES XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX\n",
    "    df[\"lr_perc_rank\"]  = df.lr.rank(pct=True)\n",
    "    df[\"fst_perc_rank\"] = df.fst.rank(pct=True)\n",
    "    df[\"h_perc_rank\"]   = df.h.rank(pct=True)\n",
    "\n",
    "    \n",
    "    #subsample dataframe for desired values\n",
    "    sub_df = df[[\"chrom\", \"pos\", \"h\", \"h_perc_rank\", \"fst\", \"fst_perc_rank\", \"lr\", \"lr_perc_rank\", \"fig_x_pos_x\"]]\n",
    "\n",
    "    #get complete records\n",
    "    sub_df = sub_df.dropna()\n",
    "\n",
    "    #save as new csv for dcms\n",
    "    csv_file = \"./results/selection_peaks/{}_raw_data.csv\".format(pop)\n",
    "    sub_df = sub_df.sort_values([\"fig_x_pos_x\"], ascending = True)\n",
    "    sub_df.to_csv(csv_file, index=False, header=True, mode='w')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Find \"PEAKS OF INTEREST\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-30T17:16:52.266843Z",
     "start_time": "2020-06-30T17:10:56.495775Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "senegal h_max=38426.4 lr_max=17.397502\n",
      "niger h_max=27040.7 lr_max=15.94272\n",
      "tanzania h_max=45532.9 lr_max=16.871399\n",
      "brazil h_max=43924.1 lr_max=15.387932\n"
     ]
    }
   ],
   "source": [
    "import glob\n",
    "from collections import defaultdict\n",
    "\n",
    "#get highest simluated score for each pop/test\n",
    "\n",
    "max_h = defaultdict(lambda: float(0))\n",
    "max_lr = defaultdict(lambda: float(0))\n",
    "    \n",
    "for pop in [\"senegal\", \"niger\", \"tanzania\", \"brazil\"]:\n",
    "    \n",
    "    sim_hscan_files = glob.glob(\"results/hscan/{}-msprime/chr1_{}_rep_*_seed_*_probed.vcf.hscan-out\".format(pop, pop))\n",
    "    sim_sf2_files = glob.glob(\"results/sweepfinder/{}-sim/chr1_{}_rep_*_seed_*_probed.sw2out\".format(pop, pop))\n",
    "\n",
    "    for hscan_file in sim_hscan_files:\n",
    "        with open(hscan_file, 'r') as infile:\n",
    "            next(infile)\n",
    "            for entry in infile:\n",
    "                x, h = entry.rstrip().split(\"\\t\")\n",
    "                if float(h) > max_h[pop]:\n",
    "                    max_h[pop]=float(h)\n",
    "\n",
    "    for sf2_file in sim_sf2_files:\n",
    "        with open(sf2_file, 'r') as infile:\n",
    "            next(infile)\n",
    "            for entry in infile:\n",
    "                location, lr, alpha = entry.rstrip().split(\"\\t\")\n",
    "                if float(lr) > max_lr[pop]:\n",
    "                    max_lr[pop]=float(lr)\n",
    "    \n",
    "    outline=\"{} h_max={} lr_max={}\".format(pop, max_h[pop], max_lr[pop])\n",
    "    print(outline)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-30T19:40:20.233003Z",
     "start_time": "2020-06-30T19:40:15.101684Z"
    }
   },
   "outputs": [],
   "source": [
    "#now find sites with hscan, sweepfinder, and fst in the top 5% (as long as all scores gt sim)\n",
    "\n",
    "for pop in [\"senegal\", \"niger\", \"brazil\", \"tanzania\"]:\n",
    "    #print(pop)\n",
    "    sig_hits=0\n",
    "    highest_h=0\n",
    "    highest_lr=0\n",
    "    csv_file = \"./results/selection_peaks/{}_raw_data.csv\".format(pop)\n",
    "\n",
    "    with open(csv_file, 'r') as infile:\n",
    "        \n",
    "        bed_file = \"./results/selection_peaks/{}_sig_snps.bed\".format(pop)\n",
    "        with open(bed_file, 'w') as outfile:\n",
    "            next(infile)\n",
    "            for entry in infile:\n",
    "                chrom, pos, h, h_perc_rank, fst, fst_perc_rank, lr, lr_perc_rank, fig_x_pos_x = entry.rstrip().split(\",\")\n",
    "\n",
    "                if ((float(h_perc_rank)   >0.99) and \n",
    "                    (float(lr_perc_rank)  >0.99) and\n",
    "                    (float(h) > max_h[pop])     and \n",
    "                    (float(lr) > max_lr[pop] )):\n",
    "\n",
    "                    sig_bed_entry=\"{}\\t{}\\t{}\\n\".format(chrom, pos, str(int(pos)+1))\n",
    "                    outfile.write(sig_bed_entry)\n",
    "\n",
    "#bedtools sort -i brazil_sig_snps.bed | bedtools merge -d 100000"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Plot Hscan and Sweepfinder data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-30T21:26:47.486200Z",
     "start_time": "2020-06-30T21:26:47.441483Z"
    }
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "from collections import defaultdict\n",
    "from intervaltree import IntervalTree\n",
    "\n",
    "#make sure that all stops are not gt chrom length\n",
    "chr_length = {}\n",
    "#genome_size = 0\n",
    "with open('data/genomes/Smansoni_v7.fa.fai', 'r') as fai:\n",
    "    for entry in fai:\n",
    "        chrom, length, *offset = entry.rstrip().split(\"\\t\")\n",
    "        chr_length[chrom]=int(length)\n",
    "\n",
    "cumul_start={}\n",
    "cumul_start['SM_V7_1']=0\n",
    "cumul_start['SM_V7_2']= cumul_start['SM_V7_1'] + chr_length['SM_V7_1']\n",
    "cumul_start['SM_V7_3']= cumul_start['SM_V7_2'] + chr_length['SM_V7_2']\n",
    "cumul_start['SM_V7_4']= cumul_start['SM_V7_3'] + chr_length['SM_V7_3']\n",
    "cumul_start['SM_V7_5']= cumul_start['SM_V7_4'] + chr_length['SM_V7_4']\n",
    "cumul_start['SM_V7_6']= cumul_start['SM_V7_5'] + chr_length['SM_V7_5']\n",
    "cumul_start['SM_V7_7']= cumul_start['SM_V7_6'] + chr_length['SM_V7_6']\n",
    "scanned_size = cumul_start['SM_V7_7'] + chr_length['SM_V7_7']\n",
    "\n",
    "def get_x_pos(chrom, pos):\n",
    "    x_pos=pos+cumul_start[chrom]\n",
    "    return x_pos\n",
    "\n",
    "#set plot colors\n",
    "chrom_colors={}\n",
    "chrom_colors[\"SM_V7_1\"] = \"gainsboro\"\n",
    "chrom_colors[\"SM_V7_2\"] = \"grey\"\n",
    "chrom_colors[\"SM_V7_3\"] = \"gainsboro\"\n",
    "chrom_colors[\"SM_V7_4\"] = \"grey\"\n",
    "chrom_colors[\"SM_V7_5\"] = \"gainsboro\"\n",
    "chrom_colors[\"SM_V7_6\"] = \"grey\"\n",
    "chrom_colors[\"SM_V7_7\"] = \"gainsboro\"\n",
    "sig_color = \"magenta\"\n",
    "\n",
    "ticks = [ (cumul_start['SM_V7_1'] + cumul_start['SM_V7_2'] )/2,\n",
    "          (cumul_start['SM_V7_2'] + cumul_start['SM_V7_3'] )/2,\n",
    "          (cumul_start['SM_V7_3'] + cumul_start['SM_V7_4'] )/2,\n",
    "          (cumul_start['SM_V7_4'] + cumul_start['SM_V7_5'] )/2,\n",
    "          (cumul_start['SM_V7_5'] + cumul_start['SM_V7_6'] )/2,\n",
    "          (cumul_start['SM_V7_6'] + cumul_start['SM_V7_7'] )/2,\n",
    "          (cumul_start['SM_V7_7'] + scanned_size )/2 ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2020-06-30T21:50:20.672Z"
    }
   },
   "outputs": [],
   "source": [
    "for pop in [\"senegal\", \"niger\", \"brazil\", \"tanzania\"]:\n",
    "#for pop in [\"niger\"]:\n",
    "\n",
    "    csv_file = \"./results/selection_peaks/{}_raw_data.csv\".format(pop)\n",
    "\n",
    "    df=pd.read_csv(csv_file, sep=\",\")\n",
    "    \n",
    "    #update colors of significant peaks\n",
    "    fig_chrom_colors=[]\n",
    "    fig_x_pos=[]\n",
    "    \n",
    "    for index, row in df.iterrows(): \n",
    "        fig_x_pos.append(get_x_pos(row.chrom, row.pos))\n",
    "\n",
    "        if ((float(row.h_perc_rank)   >0.99) and \n",
    "            (float(row.lr_perc_rank)  >0.99) and\n",
    "            (float(row.h) > max_h[pop])      and \n",
    "            (float(row.lr) > max_lr[pop] )):\n",
    "            fig_chrom_colors.append(sig_color)\n",
    "        else:\n",
    "            fig_chrom_colors.append(chrom_colors[row.chrom])\n",
    "\n",
    "    #plot log data for H\n",
    "    tick_lbls = [ \"1\", \"2\", \"3\", \"4\", \"5\", \"6\" ,\"7\"]\n",
    "\n",
    "    fig, axs = plt.subplots(2, sharex=True, gridspec_kw={'hspace': 0})\n",
    "    \n",
    "    axs[0].scatter(fig_x_pos, df['h'], marker =\".\", s = 1, c = fig_chrom_colors) \n",
    "    axs[1].scatter(fig_x_pos, df['lr'], marker =\".\", s = 1, c = fig_chrom_colors) \n",
    "\n",
    "    #axs[0].xticks(ticks, tick_lbls)\n",
    "    #axs[1].xticks(ticks, tick_lbls) \n",
    "\n",
    "    #axs[0].ylabel(\"H\")\n",
    "    #axs[1].ylabel(\"LR\")\n",
    "\n",
    "    axs[0].axhline(y=max_h[pop], color=\"red\", linestyle=\":\", linewidth=1)\n",
    "    axs[1].axhline(y=max_lr[pop], color=\"red\", linestyle=\":\", linewidth=1)\n",
    "\n",
    "    #for ax in axs:\n",
    "    #    ax.label_outer()\n",
    "        \n",
    "    title=\"{} selection\".format(pop)\n",
    "    fig.suptitle(title)\n",
    "\n",
    "    #plt.text(df['fig_x_pos'][-1:], -np.log10(2.056684e-05), 'FDR=0.01', horizontalalignment='right')\n",
    "\n",
    "    fig.savefig(\"results/selection_peaks/{}_selection.png\".format(pop), dpi=300) \n",
    "    plt.close()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
