{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-23T01:40:26.654860Z",
     "start_time": "2020-06-23T01:40:26.647342Z"
    }
   },
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "os.chdir(\"/master/nplatt/sch_man_nwinvasion\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Get random samples\n",
    "Randomly select 10 individuals from each population. This allows us see if/how sample size influences our results."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true,
    "hidden": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory `results/n10': File exists\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf results/variant_filtration/smv7_ex_autosomes.vcf\n",
      "\t--keep results/n10/n10.list\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 40 out of 156 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 475081 out of a possible 475081 Sites\n",
      "Run Time = 198.00 seconds\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "\n",
    "mkdir results/n10\n",
    "\n",
    "#get lists of each population and random subset (of 10)\n",
    "for POP in brazil tanzania senegal niger; do\n",
    "    cat results/lists/$POP.list | shuf | shuf | shuf | shuf | head -n 10 >results/n10/\"$POP\"_n10.list\n",
    "done\n",
    "\n",
    "cat results/n10/*_n10.list >results/n10/n10.list\n",
    "\n",
    "vcftools \\\n",
    "    --vcf results/variant_filtration/smv7_ex_autosomes.vcf \\\n",
    "    --keep results/n10/n10.list \\\n",
    "    --recode \\\n",
    "    --recode-INFO-all \\\n",
    "    --stdout \\\n",
    "    >results/n10/smv7_ex_autosomes_n10.vcf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## MSPRIME\n",
    "Simulate neutral VCFS for each population from 10 individuals"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": true,
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5665356 (\"tanzania_1\") has been submitted\n",
      "Your job 5665357 (\"tanzania_2\") has been submitted\n",
      "Your job 5665358 (\"tanzania_3\") has been submitted\n",
      "Your job 5665359 (\"tanzania_4\") has been submitted\n",
      "Your job 5665360 (\"tanzania_5\") has been submitted\n",
      "Your job 5665361 (\"tanzania_6\") has been submitted\n",
      "Your job 5665362 (\"tanzania_7\") has been submitted\n",
      "Your job 5665363 (\"tanzania_8\") has been submitted\n",
      "Your job 5665364 (\"tanzania_9\") has been submitted\n",
      "Your job 5665365 (\"tanzania_10\") has been submitted\n",
      "Your job 5665366 (\"tanzania_11\") has been submitted\n",
      "Your job 5665367 (\"tanzania_12\") has been submitted\n",
      "Your job 5665368 (\"tanzania_13\") has been submitted\n",
      "Your job 5665369 (\"tanzania_14\") has been submitted\n",
      "Your job 5665370 (\"tanzania_15\") has been submitted\n",
      "Your job 5665371 (\"tanzania_16\") has been submitted\n",
      "Your job 5665372 (\"tanzania_17\") has been submitted\n",
      "Your job 5665373 (\"tanzania_18\") has been submitted\n",
      "Your job 5665374 (\"tanzania_19\") has been submitted\n",
      "Your job 5665375 (\"tanzania_20\") has been submitted\n",
      "Your job 5665376 (\"tanzania_21\") has been submitted\n",
      "Your job 5665377 (\"tanzania_22\") has been submitted\n",
      "Your job 5665378 (\"tanzania_23\") has been submitted\n",
      "Your job 5665379 (\"tanzania_24\") has been submitted\n",
      "Your job 5665380 (\"tanzania_25\") has been submitted\n",
      "Your job 5665381 (\"tanzania_26\") has been submitted\n",
      "Your job 5665382 (\"tanzania_27\") has been submitted\n",
      "Your job 5665383 (\"tanzania_28\") has been submitted\n",
      "Your job 5665384 (\"tanzania_29\") has been submitted\n",
      "Your job 5665385 (\"tanzania_30\") has been submitted\n",
      "Your job 5665386 (\"tanzania_31\") has been submitted\n",
      "Your job 5665387 (\"tanzania_32\") has been submitted\n",
      "Your job 5665388 (\"tanzania_33\") has been submitted\n",
      "Your job 5665389 (\"tanzania_34\") has been submitted\n",
      "Your job 5665390 (\"tanzania_35\") has been submitted\n",
      "Your job 5665391 (\"tanzania_36\") has been submitted\n",
      "Your job 5665392 (\"tanzania_37\") has been submitted\n",
      "Your job 5665393 (\"tanzania_38\") has been submitted\n",
      "Your job 5665394 (\"tanzania_39\") has been submitted\n",
      "Your job 5665395 (\"tanzania_40\") has been submitted\n",
      "Your job 5665396 (\"tanzania_41\") has been submitted\n",
      "Your job 5665397 (\"tanzania_42\") has been submitted\n",
      "Your job 5665398 (\"tanzania_43\") has been submitted\n",
      "Your job 5665399 (\"tanzania_44\") has been submitted\n",
      "Your job 5665400 (\"tanzania_45\") has been submitted\n",
      "Your job 5665401 (\"tanzania_46\") has been submitted\n",
      "Your job 5665402 (\"tanzania_47\") has been submitted\n",
      "Your job 5665403 (\"tanzania_48\") has been submitted\n",
      "Your job 5665404 (\"tanzania_49\") has been submitted\n",
      "Your job 5665405 (\"tanzania_50\") has been submitted\n",
      "Your job 5665406 (\"tanzania_51\") has been submitted\n",
      "Your job 5665407 (\"tanzania_52\") has been submitted\n",
      "Your job 5665408 (\"tanzania_53\") has been submitted\n",
      "Your job 5665409 (\"tanzania_54\") has been submitted\n",
      "Your job 5665410 (\"tanzania_55\") has been submitted\n",
      "Your job 5665411 (\"tanzania_56\") has been submitted\n",
      "Your job 5665412 (\"tanzania_57\") has been submitted\n",
      "Your job 5665413 (\"tanzania_58\") has been submitted\n",
      "Your job 5665414 (\"tanzania_59\") has been submitted\n",
      "Your job 5665415 (\"tanzania_60\") has been submitted\n",
      "Your job 5665416 (\"tanzania_61\") has been submitted\n",
      "Your job 5665417 (\"tanzania_62\") has been submitted\n",
      "Your job 5665418 (\"tanzania_63\") has been submitted\n",
      "Your job 5665419 (\"tanzania_64\") has been submitted\n",
      "Your job 5665420 (\"tanzania_65\") has been submitted\n",
      "Your job 5665421 (\"tanzania_66\") has been submitted\n",
      "Your job 5665422 (\"tanzania_67\") has been submitted\n",
      "Your job 5665423 (\"tanzania_68\") has been submitted\n",
      "Your job 5665424 (\"tanzania_69\") has been submitted\n",
      "Your job 5665425 (\"tanzania_70\") has been submitted\n",
      "Your job 5665426 (\"tanzania_71\") has been submitted\n",
      "Your job 5665427 (\"tanzania_72\") has been submitted\n",
      "Your job 5665428 (\"tanzania_73\") has been submitted\n",
      "Your job 5665429 (\"tanzania_74\") has been submitted\n",
      "Your job 5665430 (\"tanzania_75\") has been submitted\n",
      "Your job 5665431 (\"tanzania_76\") has been submitted\n",
      "Your job 5665432 (\"tanzania_77\") has been submitted\n",
      "Your job 5665433 (\"tanzania_78\") has been submitted\n",
      "Your job 5665434 (\"tanzania_79\") has been submitted\n",
      "Your job 5665435 (\"tanzania_80\") has been submitted\n",
      "Your job 5665436 (\"tanzania_81\") has been submitted\n",
      "Your job 5665437 (\"tanzania_82\") has been submitted\n",
      "Your job 5665438 (\"tanzania_83\") has been submitted\n",
      "Your job 5665439 (\"tanzania_84\") has been submitted\n",
      "Your job 5665440 (\"tanzania_85\") has been submitted\n",
      "Your job 5665441 (\"tanzania_86\") has been submitted\n",
      "Your job 5665442 (\"tanzania_87\") has been submitted\n",
      "Your job 5665443 (\"tanzania_88\") has been submitted\n",
      "Your job 5665444 (\"tanzania_89\") has been submitted\n",
      "Your job 5665445 (\"tanzania_90\") has been submitted\n",
      "Your job 5665446 (\"tanzania_91\") has been submitted\n",
      "Your job 5665447 (\"tanzania_92\") has been submitted\n",
      "Your job 5665448 (\"tanzania_93\") has been submitted\n",
      "Your job 5665449 (\"tanzania_94\") has been submitted\n",
      "Your job 5665450 (\"tanzania_95\") has been submitted\n",
      "Your job 5665451 (\"tanzania_96\") has been submitted\n",
      "Your job 5665452 (\"tanzania_97\") has been submitted\n",
      "Your job 5665453 (\"tanzania_98\") has been submitted\n",
      "Your job 5665454 (\"tanzania_99\") has been submitted\n",
      "Your job 5665455 (\"tanzania_100\") has been submitted\n",
      "Your job 5665456 (\"tanzania_101\") has been submitted\n",
      "Your job 5665457 (\"tanzania_102\") has been submitted\n",
      "Your job 5665458 (\"tanzania_103\") has been submitted\n",
      "Your job 5665459 (\"tanzania_104\") has been submitted\n",
      "Your job 5665460 (\"tanzania_105\") has been submitted\n",
      "Your job 5665461 (\"tanzania_106\") has been submitted\n",
      "Your job 5665462 (\"tanzania_107\") has been submitted\n",
      "Your job 5665463 (\"tanzania_108\") has been submitted\n",
      "Your job 5665464 (\"tanzania_109\") has been submitted\n",
      "Your job 5665465 (\"tanzania_110\") has been submitted\n",
      "Your job 5665466 (\"tanzania_111\") has been submitted\n",
      "Your job 5665467 (\"tanzania_112\") has been submitted\n",
      "Your job 5665468 (\"tanzania_113\") has been submitted\n",
      "Your job 5665469 (\"tanzania_114\") has been submitted\n",
      "Your job 5665470 (\"tanzania_115\") has been submitted\n",
      "Your job 5665471 (\"tanzania_116\") has been submitted\n",
      "Your job 5665472 (\"tanzania_117\") has been submitted\n",
      "Your job 5665473 (\"tanzania_118\") has been submitted\n",
      "Your job 5665474 (\"tanzania_119\") has been submitted\n",
      "Your job 5665475 (\"tanzania_120\") has been submitted\n",
      "Your job 5665476 (\"tanzania_121\") has been submitted\n",
      "Your job 5665477 (\"tanzania_122\") has been submitted\n",
      "Your job 5665478 (\"tanzania_123\") has been submitted\n",
      "Your job 5665479 (\"tanzania_124\") has been submitted\n",
      "Your job 5665480 (\"tanzania_125\") has been submitted\n",
      "Your job 5665481 (\"tanzania_126\") has been submitted\n",
      "Your job 5665482 (\"tanzania_127\") has been submitted\n",
      "Your job 5665483 (\"tanzania_128\") has been submitted\n",
      "Your job 5665484 (\"tanzania_129\") has been submitted\n",
      "Your job 5665485 (\"tanzania_130\") has been submitted\n",
      "Your job 5665486 (\"tanzania_131\") has been submitted\n",
      "Your job 5665487 (\"tanzania_132\") has been submitted\n",
      "Your job 5665488 (\"tanzania_133\") has been submitted\n",
      "Your job 5665489 (\"tanzania_134\") has been submitted\n",
      "Your job 5665490 (\"tanzania_135\") has been submitted\n",
      "Your job 5665491 (\"tanzania_136\") has been submitted\n",
      "Your job 5665492 (\"tanzania_137\") has been submitted\n",
      "Your job 5665493 (\"tanzania_138\") has been submitted\n",
      "Your job 5665494 (\"tanzania_139\") has been submitted\n",
      "Your job 5665495 (\"tanzania_140\") has been submitted\n",
      "Your job 5665496 (\"tanzania_141\") has been submitted\n",
      "Your job 5665497 (\"tanzania_142\") has been submitted\n",
      "Your job 5665498 (\"tanzania_143\") has been submitted\n",
      "Your job 5665499 (\"tanzania_144\") has been submitted\n",
      "Your job 5665500 (\"tanzania_145\") has been submitted\n",
      "Your job 5665501 (\"tanzania_146\") has been submitted\n",
      "Your job 5665502 (\"tanzania_147\") has been submitted\n",
      "Your job 5665503 (\"tanzania_148\") has been submitted\n",
      "Your job 5665504 (\"tanzania_149\") has been submitted\n",
      "Your job 5665505 (\"tanzania_150\") has been submitted\n",
      "Your job 5665506 (\"tanzania_151\") has been submitted\n",
      "Your job 5665507 (\"tanzania_152\") has been submitted\n",
      "Your job 5665508 (\"tanzania_153\") has been submitted\n",
      "Your job 5665509 (\"tanzania_154\") has been submitted\n",
      "Your job 5665510 (\"tanzania_155\") has been submitted\n",
      "Your job 5665511 (\"tanzania_156\") has been submitted\n",
      "Your job 5665512 (\"tanzania_157\") has been submitted\n",
      "Your job 5665513 (\"tanzania_158\") has been submitted\n",
      "Your job 5665514 (\"tanzania_159\") has been submitted\n",
      "Your job 5665515 (\"tanzania_160\") has been submitted\n",
      "Your job 5665516 (\"tanzania_161\") has been submitted\n",
      "Your job 5665517 (\"tanzania_162\") has been submitted\n",
      "Your job 5665518 (\"tanzania_163\") has been submitted\n",
      "Your job 5665519 (\"tanzania_164\") has been submitted\n",
      "Your job 5665520 (\"tanzania_165\") has been submitted\n",
      "Your job 5665521 (\"tanzania_166\") has been submitted\n",
      "Your job 5665522 (\"tanzania_167\") has been submitted\n",
      "Your job 5665523 (\"tanzania_168\") has been submitted\n",
      "Your job 5665524 (\"tanzania_169\") has been submitted\n",
      "Your job 5665525 (\"tanzania_170\") has been submitted\n",
      "Your job 5665526 (\"tanzania_171\") has been submitted\n",
      "Your job 5665527 (\"tanzania_172\") has been submitted\n",
      "Your job 5665528 (\"tanzania_173\") has been submitted\n",
      "Your job 5665529 (\"tanzania_174\") has been submitted\n",
      "Your job 5665530 (\"tanzania_175\") has been submitted\n",
      "Your job 5665531 (\"tanzania_176\") has been submitted\n",
      "Your job 5665532 (\"tanzania_177\") has been submitted\n",
      "Your job 5665533 (\"tanzania_178\") has been submitted\n",
      "Your job 5665534 (\"tanzania_179\") has been submitted\n",
      "Your job 5665535 (\"tanzania_180\") has been submitted\n",
      "Your job 5665536 (\"tanzania_181\") has been submitted\n",
      "Your job 5665537 (\"tanzania_182\") has been submitted\n",
      "Your job 5665538 (\"tanzania_183\") has been submitted\n",
      "Your job 5665539 (\"tanzania_184\") has been submitted\n",
      "Your job 5665540 (\"tanzania_185\") has been submitted\n",
      "Your job 5665541 (\"tanzania_186\") has been submitted\n",
      "Your job 5665542 (\"tanzania_187\") has been submitted\n",
      "Your job 5665543 (\"tanzania_188\") has been submitted\n",
      "Your job 5665544 (\"tanzania_189\") has been submitted\n",
      "Your job 5665545 (\"tanzania_190\") has been submitted\n",
      "Your job 5665546 (\"tanzania_191\") has been submitted\n",
      "Your job 5665547 (\"tanzania_192\") has been submitted\n",
      "Your job 5665548 (\"tanzania_193\") has been submitted\n",
      "Your job 5665549 (\"tanzania_194\") has been submitted\n",
      "Your job 5665550 (\"tanzania_195\") has been submitted\n",
      "Your job 5665551 (\"tanzania_196\") has been submitted\n",
      "Your job 5665552 (\"tanzania_197\") has been submitted\n",
      "Your job 5665553 (\"tanzania_198\") has been submitted\n",
      "Your job 5665554 (\"tanzania_199\") has been submitted\n",
      "Your job 5665555 (\"tanzania_200\") has been submitted\n",
      "Your job 5665556 (\"tanzania_201\") has been submitted\n",
      "Your job 5665557 (\"tanzania_202\") has been submitted\n",
      "Your job 5665558 (\"tanzania_203\") has been submitted\n",
      "Your job 5665559 (\"tanzania_204\") has been submitted\n",
      "Your job 5665560 (\"tanzania_205\") has been submitted\n",
      "Your job 5665561 (\"tanzania_206\") has been submitted\n",
      "Your job 5665562 (\"tanzania_207\") has been submitted\n",
      "Your job 5665563 (\"tanzania_208\") has been submitted\n",
      "Your job 5665564 (\"tanzania_209\") has been submitted\n",
      "Your job 5665565 (\"tanzania_210\") has been submitted\n",
      "Your job 5665566 (\"tanzania_211\") has been submitted\n",
      "Your job 5665567 (\"tanzania_212\") has been submitted\n",
      "Your job 5665568 (\"tanzania_213\") has been submitted\n",
      "Your job 5665569 (\"tanzania_214\") has been submitted\n",
      "Your job 5665570 (\"tanzania_215\") has been submitted\n",
      "Your job 5665571 (\"tanzania_216\") has been submitted\n",
      "Your job 5665572 (\"tanzania_217\") has been submitted\n",
      "Your job 5665573 (\"tanzania_218\") has been submitted\n",
      "Your job 5665574 (\"tanzania_219\") has been submitted\n",
      "Your job 5665575 (\"tanzania_220\") has been submitted\n",
      "Your job 5665576 (\"tanzania_221\") has been submitted\n",
      "Your job 5665577 (\"tanzania_222\") has been submitted\n",
      "Your job 5665578 (\"tanzania_223\") has been submitted\n",
      "Your job 5665579 (\"tanzania_224\") has been submitted\n",
      "Your job 5665580 (\"tanzania_225\") has been submitted\n",
      "Your job 5665581 (\"tanzania_226\") has been submitted\n",
      "Your job 5665582 (\"tanzania_227\") has been submitted\n",
      "Your job 5665583 (\"tanzania_228\") has been submitted\n",
      "Your job 5665584 (\"tanzania_229\") has been submitted\n",
      "Your job 5665585 (\"tanzania_230\") has been submitted\n",
      "Your job 5665586 (\"tanzania_231\") has been submitted\n",
      "Your job 5665587 (\"tanzania_232\") has been submitted\n",
      "Your job 5665588 (\"tanzania_233\") has been submitted\n",
      "Your job 5665589 (\"tanzania_234\") has been submitted\n",
      "Your job 5665590 (\"tanzania_235\") has been submitted\n",
      "Your job 5665591 (\"tanzania_236\") has been submitted\n",
      "Your job 5665592 (\"tanzania_237\") has been submitted\n",
      "Your job 5665593 (\"tanzania_238\") has been submitted\n",
      "Your job 5665594 (\"tanzania_239\") has been submitted\n",
      "Your job 5665595 (\"tanzania_240\") has been submitted\n",
      "Your job 5665596 (\"tanzania_241\") has been submitted\n",
      "Your job 5665597 (\"tanzania_242\") has been submitted\n",
      "Your job 5665598 (\"tanzania_243\") has been submitted\n",
      "Your job 5665599 (\"tanzania_244\") has been submitted\n",
      "Your job 5665600 (\"tanzania_245\") has been submitted\n",
      "Your job 5665601 (\"tanzania_246\") has been submitted\n",
      "Your job 5665602 (\"tanzania_247\") has been submitted\n",
      "Your job 5665603 (\"tanzania_248\") has been submitted\n",
      "Your job 5665604 (\"tanzania_249\") has been submitted\n",
      "Your job 5665605 (\"tanzania_250\") has been submitted\n",
      "Your job 5665606 (\"tanzania_251\") has been submitted\n",
      "Your job 5665607 (\"tanzania_252\") has been submitted\n",
      "Your job 5665608 (\"tanzania_253\") has been submitted\n",
      "Your job 5665609 (\"tanzania_254\") has been submitted\n",
      "Your job 5665610 (\"tanzania_255\") has been submitted\n",
      "Your job 5665611 (\"tanzania_256\") has been submitted\n",
      "Your job 5665612 (\"tanzania_257\") has been submitted\n",
      "Your job 5665613 (\"tanzania_258\") has been submitted\n",
      "Your job 5665614 (\"tanzania_259\") has been submitted\n",
      "Your job 5665615 (\"tanzania_260\") has been submitted\n",
      "Your job 5665616 (\"tanzania_261\") has been submitted\n",
      "Your job 5665617 (\"tanzania_262\") has been submitted\n",
      "Your job 5665618 (\"tanzania_263\") has been submitted\n",
      "Your job 5665619 (\"tanzania_264\") has been submitted\n",
      "Your job 5665620 (\"tanzania_265\") has been submitted\n",
      "Your job 5665621 (\"tanzania_266\") has been submitted\n",
      "Your job 5665622 (\"tanzania_267\") has been submitted\n",
      "Your job 5665623 (\"tanzania_268\") has been submitted\n",
      "Your job 5665624 (\"tanzania_269\") has been submitted\n",
      "Your job 5665625 (\"tanzania_270\") has been submitted\n",
      "Your job 5665626 (\"tanzania_271\") has been submitted\n",
      "Your job 5665627 (\"tanzania_272\") has been submitted\n",
      "Your job 5665628 (\"tanzania_273\") has been submitted\n",
      "Your job 5665629 (\"tanzania_274\") has been submitted\n",
      "Your job 5665630 (\"tanzania_275\") has been submitted\n",
      "Your job 5665631 (\"tanzania_276\") has been submitted\n",
      "Your job 5665632 (\"tanzania_277\") has been submitted\n",
      "Your job 5665633 (\"tanzania_278\") has been submitted\n",
      "Your job 5665634 (\"tanzania_279\") has been submitted\n",
      "Your job 5665635 (\"tanzania_280\") has been submitted\n",
      "Your job 5665636 (\"tanzania_281\") has been submitted\n",
      "Your job 5665637 (\"tanzania_282\") has been submitted\n",
      "Your job 5665638 (\"tanzania_283\") has been submitted\n",
      "Your job 5665639 (\"tanzania_284\") has been submitted\n",
      "Your job 5665640 (\"tanzania_285\") has been submitted\n",
      "Your job 5665641 (\"tanzania_286\") has been submitted\n",
      "Your job 5665642 (\"tanzania_287\") has been submitted\n",
      "Your job 5665643 (\"tanzania_288\") has been submitted\n",
      "Your job 5665644 (\"tanzania_289\") has been submitted\n",
      "Your job 5665645 (\"tanzania_290\") has been submitted\n",
      "Your job 5665646 (\"tanzania_291\") has been submitted\n",
      "Your job 5665647 (\"tanzania_292\") has been submitted\n",
      "Your job 5665648 (\"tanzania_293\") has been submitted\n",
      "Your job 5665649 (\"tanzania_294\") has been submitted\n",
      "Your job 5665650 (\"tanzania_295\") has been submitted\n",
      "Your job 5665651 (\"tanzania_296\") has been submitted\n",
      "Your job 5665652 (\"tanzania_297\") has been submitted\n",
      "Your job 5665653 (\"tanzania_298\") has been submitted\n",
      "Your job 5665654 (\"tanzania_299\") has been submitted\n",
      "Your job 5665655 (\"tanzania_300\") has been submitted\n",
      "Your job 5665656 (\"tanzania_301\") has been submitted\n",
      "Your job 5665657 (\"tanzania_302\") has been submitted\n",
      "Your job 5665658 (\"tanzania_303\") has been submitted\n",
      "Your job 5665659 (\"tanzania_304\") has been submitted\n",
      "Your job 5665660 (\"tanzania_305\") has been submitted\n",
      "Your job 5665661 (\"tanzania_306\") has been submitted\n",
      "Your job 5665662 (\"tanzania_307\") has been submitted\n",
      "Your job 5665663 (\"tanzania_308\") has been submitted\n",
      "Your job 5665664 (\"tanzania_309\") has been submitted\n",
      "Your job 5665665 (\"tanzania_310\") has been submitted\n",
      "Your job 5665666 (\"tanzania_311\") has been submitted\n",
      "Your job 5665667 (\"tanzania_312\") has been submitted\n",
      "Your job 5665668 (\"tanzania_313\") has been submitted\n",
      "Your job 5665669 (\"tanzania_314\") has been submitted\n",
      "Your job 5665670 (\"tanzania_315\") has been submitted\n",
      "Your job 5665671 (\"tanzania_316\") has been submitted\n",
      "Your job 5665672 (\"tanzania_317\") has been submitted\n",
      "Your job 5665673 (\"tanzania_318\") has been submitted\n",
      "Your job 5665674 (\"tanzania_319\") has been submitted\n",
      "Your job 5665675 (\"tanzania_320\") has been submitted\n",
      "Your job 5665676 (\"tanzania_321\") has been submitted\n",
      "Your job 5665677 (\"tanzania_322\") has been submitted\n",
      "Your job 5665678 (\"tanzania_323\") has been submitted\n",
      "Your job 5665679 (\"tanzania_324\") has been submitted\n",
      "Your job 5665680 (\"tanzania_325\") has been submitted\n",
      "Your job 5665681 (\"tanzania_326\") has been submitted\n",
      "Your job 5665682 (\"tanzania_327\") has been submitted\n",
      "Your job 5665683 (\"tanzania_328\") has been submitted\n",
      "Your job 5665684 (\"tanzania_329\") has been submitted\n",
      "Your job 5665685 (\"tanzania_330\") has been submitted\n",
      "Your job 5665686 (\"tanzania_331\") has been submitted\n",
      "Your job 5665687 (\"tanzania_332\") has been submitted\n",
      "Your job 5665688 (\"tanzania_333\") has been submitted\n",
      "Your job 5665689 (\"tanzania_334\") has been submitted\n",
      "Your job 5665690 (\"tanzania_335\") has been submitted\n",
      "Your job 5665691 (\"tanzania_336\") has been submitted\n",
      "Your job 5665692 (\"tanzania_337\") has been submitted\n",
      "Your job 5665693 (\"tanzania_338\") has been submitted\n",
      "Your job 5665694 (\"tanzania_339\") has been submitted\n",
      "Your job 5665695 (\"tanzania_340\") has been submitted\n",
      "Your job 5665696 (\"tanzania_341\") has been submitted\n",
      "Your job 5665697 (\"tanzania_342\") has been submitted\n",
      "Your job 5665698 (\"senegal_1\") has been submitted\n",
      "Your job 5665699 (\"senegal_2\") has been submitted\n",
      "Your job 5665700 (\"senegal_3\") has been submitted\n",
      "Your job 5665701 (\"senegal_4\") has been submitted\n",
      "Your job 5665702 (\"senegal_5\") has been submitted\n",
      "Your job 5665703 (\"senegal_6\") has been submitted\n",
      "Your job 5665704 (\"senegal_7\") has been submitted\n",
      "Your job 5665705 (\"senegal_8\") has been submitted\n",
      "Your job 5665706 (\"senegal_9\") has been submitted\n",
      "Your job 5665707 (\"senegal_10\") has been submitted\n",
      "Your job 5665708 (\"senegal_11\") has been submitted\n",
      "Your job 5665709 (\"senegal_12\") has been submitted\n",
      "Your job 5665710 (\"senegal_13\") has been submitted\n",
      "Your job 5665711 (\"senegal_14\") has been submitted\n",
      "Your job 5665712 (\"senegal_15\") has been submitted\n",
      "Your job 5665713 (\"senegal_16\") has been submitted\n",
      "Your job 5665714 (\"senegal_17\") has been submitted\n",
      "Your job 5665715 (\"senegal_18\") has been submitted\n",
      "Your job 5665716 (\"senegal_19\") has been submitted\n",
      "Your job 5665717 (\"senegal_20\") has been submitted\n",
      "Your job 5665718 (\"senegal_21\") has been submitted\n",
      "Your job 5665719 (\"senegal_22\") has been submitted\n",
      "Your job 5665720 (\"senegal_23\") has been submitted\n",
      "Your job 5665721 (\"senegal_24\") has been submitted\n",
      "Your job 5665722 (\"senegal_25\") has been submitted\n",
      "Your job 5665723 (\"senegal_26\") has been submitted\n",
      "Your job 5665724 (\"senegal_27\") has been submitted\n",
      "Your job 5665725 (\"senegal_28\") has been submitted\n",
      "Your job 5665726 (\"senegal_29\") has been submitted\n",
      "Your job 5665727 (\"senegal_30\") has been submitted\n",
      "Your job 5665728 (\"senegal_31\") has been submitted\n",
      "Your job 5665729 (\"senegal_32\") has been submitted\n",
      "Your job 5665730 (\"senegal_33\") has been submitted\n",
      "Your job 5665731 (\"senegal_34\") has been submitted\n",
      "Your job 5665732 (\"senegal_35\") has been submitted\n",
      "Your job 5665733 (\"senegal_36\") has been submitted\n",
      "Your job 5665734 (\"senegal_37\") has been submitted\n",
      "Your job 5665735 (\"senegal_38\") has been submitted\n",
      "Your job 5665736 (\"senegal_39\") has been submitted\n",
      "Your job 5665737 (\"senegal_40\") has been submitted\n",
      "Your job 5665738 (\"senegal_41\") has been submitted\n",
      "Your job 5665739 (\"senegal_42\") has been submitted\n",
      "Your job 5665740 (\"senegal_43\") has been submitted\n",
      "Your job 5665741 (\"senegal_44\") has been submitted\n",
      "Your job 5665742 (\"senegal_45\") has been submitted\n",
      "Your job 5665743 (\"senegal_46\") has been submitted\n",
      "Your job 5665744 (\"senegal_47\") has been submitted\n",
      "Your job 5665745 (\"senegal_48\") has been submitted\n",
      "Your job 5665746 (\"senegal_49\") has been submitted\n",
      "Your job 5665747 (\"senegal_50\") has been submitted\n",
      "Your job 5665748 (\"senegal_51\") has been submitted\n",
      "Your job 5665749 (\"senegal_52\") has been submitted\n",
      "Your job 5665750 (\"senegal_53\") has been submitted\n",
      "Your job 5665751 (\"senegal_54\") has been submitted\n",
      "Your job 5665752 (\"senegal_55\") has been submitted\n",
      "Your job 5665753 (\"senegal_56\") has been submitted\n",
      "Your job 5665754 (\"senegal_57\") has been submitted\n",
      "Your job 5665755 (\"senegal_58\") has been submitted\n",
      "Your job 5665756 (\"senegal_59\") has been submitted\n",
      "Your job 5665757 (\"senegal_60\") has been submitted\n",
      "Your job 5665758 (\"senegal_61\") has been submitted\n",
      "Your job 5665759 (\"senegal_62\") has been submitted\n",
      "Your job 5665760 (\"senegal_63\") has been submitted\n",
      "Your job 5665761 (\"senegal_64\") has been submitted\n",
      "Your job 5665762 (\"senegal_65\") has been submitted\n",
      "Your job 5665763 (\"senegal_66\") has been submitted\n",
      "Your job 5665764 (\"senegal_67\") has been submitted\n",
      "Your job 5665765 (\"senegal_68\") has been submitted\n",
      "Your job 5665766 (\"senegal_69\") has been submitted\n",
      "Your job 5665767 (\"senegal_70\") has been submitted\n",
      "Your job 5665768 (\"senegal_71\") has been submitted\n",
      "Your job 5665769 (\"senegal_72\") has been submitted\n",
      "Your job 5665770 (\"senegal_73\") has been submitted\n",
      "Your job 5665771 (\"senegal_74\") has been submitted\n",
      "Your job 5665772 (\"senegal_75\") has been submitted\n",
      "Your job 5665773 (\"senegal_76\") has been submitted\n",
      "Your job 5665774 (\"senegal_77\") has been submitted\n",
      "Your job 5665775 (\"senegal_78\") has been submitted\n",
      "Your job 5665776 (\"senegal_79\") has been submitted\n",
      "Your job 5665777 (\"senegal_80\") has been submitted\n",
      "Your job 5665778 (\"senegal_81\") has been submitted\n",
      "Your job 5665779 (\"senegal_82\") has been submitted\n",
      "Your job 5665780 (\"senegal_83\") has been submitted\n",
      "Your job 5665781 (\"senegal_84\") has been submitted\n",
      "Your job 5665782 (\"senegal_85\") has been submitted\n",
      "Your job 5665783 (\"senegal_86\") has been submitted\n",
      "Your job 5665784 (\"senegal_87\") has been submitted\n",
      "Your job 5665785 (\"senegal_88\") has been submitted\n",
      "Your job 5665786 (\"senegal_89\") has been submitted\n",
      "Your job 5665787 (\"senegal_90\") has been submitted\n",
      "Your job 5665788 (\"senegal_91\") has been submitted\n",
      "Your job 5665789 (\"senegal_92\") has been submitted\n",
      "Your job 5665790 (\"senegal_93\") has been submitted\n",
      "Your job 5665791 (\"senegal_94\") has been submitted\n",
      "Your job 5665792 (\"senegal_95\") has been submitted\n",
      "Your job 5665793 (\"senegal_96\") has been submitted\n",
      "Your job 5665794 (\"senegal_97\") has been submitted\n",
      "Your job 5665795 (\"senegal_98\") has been submitted\n",
      "Your job 5665796 (\"senegal_99\") has been submitted\n",
      "Your job 5665797 (\"senegal_100\") has been submitted\n",
      "Your job 5665798 (\"senegal_101\") has been submitted\n",
      "Your job 5665799 (\"senegal_102\") has been submitted\n",
      "Your job 5665800 (\"senegal_103\") has been submitted\n",
      "Your job 5665801 (\"senegal_104\") has been submitted\n",
      "Your job 5665802 (\"senegal_105\") has been submitted\n",
      "Your job 5665803 (\"senegal_106\") has been submitted\n",
      "Your job 5665804 (\"senegal_107\") has been submitted\n",
      "Your job 5665805 (\"senegal_108\") has been submitted\n",
      "Your job 5665806 (\"senegal_109\") has been submitted\n",
      "Your job 5665807 (\"senegal_110\") has been submitted\n",
      "Your job 5665808 (\"senegal_111\") has been submitted\n",
      "Your job 5665809 (\"senegal_112\") has been submitted\n",
      "Your job 5665810 (\"senegal_113\") has been submitted\n",
      "Your job 5665811 (\"senegal_114\") has been submitted\n",
      "Your job 5665812 (\"senegal_115\") has been submitted\n",
      "Your job 5665813 (\"senegal_116\") has been submitted\n",
      "Your job 5665814 (\"senegal_117\") has been submitted\n",
      "Your job 5665815 (\"senegal_118\") has been submitted\n",
      "Your job 5665816 (\"senegal_119\") has been submitted\n",
      "Your job 5665817 (\"senegal_120\") has been submitted\n",
      "Your job 5665818 (\"senegal_121\") has been submitted\n",
      "Your job 5665819 (\"senegal_122\") has been submitted\n",
      "Your job 5665820 (\"senegal_123\") has been submitted\n",
      "Your job 5665821 (\"senegal_124\") has been submitted\n",
      "Your job 5665822 (\"senegal_125\") has been submitted\n",
      "Your job 5665823 (\"senegal_126\") has been submitted\n",
      "Your job 5665824 (\"senegal_127\") has been submitted\n",
      "Your job 5665825 (\"senegal_128\") has been submitted\n",
      "Your job 5665826 (\"senegal_129\") has been submitted\n",
      "Your job 5665827 (\"senegal_130\") has been submitted\n",
      "Your job 5665828 (\"senegal_131\") has been submitted\n",
      "Your job 5665829 (\"senegal_132\") has been submitted\n",
      "Your job 5665830 (\"senegal_133\") has been submitted\n",
      "Your job 5665831 (\"senegal_134\") has been submitted\n",
      "Your job 5665832 (\"senegal_135\") has been submitted\n",
      "Your job 5665833 (\"senegal_136\") has been submitted\n",
      "Your job 5665834 (\"senegal_137\") has been submitted\n",
      "Your job 5665835 (\"senegal_138\") has been submitted\n",
      "Your job 5665836 (\"senegal_139\") has been submitted\n",
      "Your job 5665837 (\"senegal_140\") has been submitted\n",
      "Your job 5665838 (\"senegal_141\") has been submitted\n",
      "Your job 5665839 (\"senegal_142\") has been submitted\n",
      "Your job 5665840 (\"senegal_143\") has been submitted\n",
      "Your job 5665841 (\"senegal_144\") has been submitted\n",
      "Your job 5665842 (\"senegal_145\") has been submitted\n",
      "Your job 5665843 (\"senegal_146\") has been submitted\n",
      "Your job 5665844 (\"senegal_147\") has been submitted\n",
      "Your job 5665845 (\"senegal_148\") has been submitted\n",
      "Your job 5665846 (\"senegal_149\") has been submitted\n",
      "Your job 5665847 (\"senegal_150\") has been submitted\n",
      "Your job 5665848 (\"senegal_151\") has been submitted\n",
      "Your job 5665849 (\"senegal_152\") has been submitted\n",
      "Your job 5665850 (\"senegal_153\") has been submitted\n",
      "Your job 5665851 (\"senegal_154\") has been submitted\n",
      "Your job 5665852 (\"senegal_155\") has been submitted\n",
      "Your job 5665853 (\"senegal_156\") has been submitted\n",
      "Your job 5665854 (\"senegal_157\") has been submitted\n",
      "Your job 5665855 (\"senegal_158\") has been submitted\n",
      "Your job 5665856 (\"senegal_159\") has been submitted\n",
      "Your job 5665857 (\"senegal_160\") has been submitted\n",
      "Your job 5665858 (\"senegal_161\") has been submitted\n",
      "Your job 5665859 (\"senegal_162\") has been submitted\n",
      "Your job 5665860 (\"senegal_163\") has been submitted\n",
      "Your job 5665861 (\"senegal_164\") has been submitted\n",
      "Your job 5665862 (\"senegal_165\") has been submitted\n",
      "Your job 5665863 (\"senegal_166\") has been submitted\n",
      "Your job 5665864 (\"senegal_167\") has been submitted\n",
      "Your job 5665865 (\"senegal_168\") has been submitted\n",
      "Your job 5665866 (\"senegal_169\") has been submitted\n",
      "Your job 5665867 (\"senegal_170\") has been submitted\n",
      "Your job 5665868 (\"senegal_171\") has been submitted\n",
      "Your job 5665869 (\"senegal_172\") has been submitted\n",
      "Your job 5665870 (\"senegal_173\") has been submitted\n",
      "Your job 5665871 (\"senegal_174\") has been submitted\n",
      "Your job 5665872 (\"senegal_175\") has been submitted\n",
      "Your job 5665873 (\"senegal_176\") has been submitted\n",
      "Your job 5665874 (\"senegal_177\") has been submitted\n",
      "Your job 5665875 (\"senegal_178\") has been submitted\n",
      "Your job 5665876 (\"senegal_179\") has been submitted\n",
      "Your job 5665877 (\"senegal_180\") has been submitted\n",
      "Your job 5665878 (\"senegal_181\") has been submitted\n",
      "Your job 5665879 (\"senegal_182\") has been submitted\n",
      "Your job 5665880 (\"senegal_183\") has been submitted\n",
      "Your job 5665881 (\"senegal_184\") has been submitted\n",
      "Your job 5665882 (\"senegal_185\") has been submitted\n",
      "Your job 5665883 (\"senegal_186\") has been submitted\n",
      "Your job 5665884 (\"senegal_187\") has been submitted\n",
      "Your job 5665885 (\"senegal_188\") has been submitted\n",
      "Your job 5665886 (\"senegal_189\") has been submitted\n",
      "Your job 5665887 (\"senegal_190\") has been submitted\n",
      "Your job 5665888 (\"senegal_191\") has been submitted\n",
      "Your job 5665889 (\"senegal_192\") has been submitted\n",
      "Your job 5665890 (\"senegal_193\") has been submitted\n",
      "Your job 5665891 (\"senegal_194\") has been submitted\n",
      "Your job 5665892 (\"senegal_195\") has been submitted\n",
      "Your job 5665893 (\"senegal_196\") has been submitted\n",
      "Your job 5665894 (\"senegal_197\") has been submitted\n",
      "Your job 5665895 (\"senegal_198\") has been submitted\n",
      "Your job 5665896 (\"senegal_199\") has been submitted\n",
      "Your job 5665897 (\"senegal_200\") has been submitted\n",
      "Your job 5665898 (\"senegal_201\") has been submitted\n",
      "Your job 5665899 (\"senegal_202\") has been submitted\n",
      "Your job 5665900 (\"senegal_203\") has been submitted\n",
      "Your job 5665901 (\"senegal_204\") has been submitted\n",
      "Your job 5665902 (\"senegal_205\") has been submitted\n",
      "Your job 5665903 (\"senegal_206\") has been submitted\n",
      "Your job 5665904 (\"senegal_207\") has been submitted\n",
      "Your job 5665905 (\"senegal_208\") has been submitted\n",
      "Your job 5665906 (\"senegal_209\") has been submitted\n",
      "Your job 5665907 (\"senegal_210\") has been submitted\n",
      "Your job 5665908 (\"senegal_211\") has been submitted\n",
      "Your job 5665909 (\"senegal_212\") has been submitted\n",
      "Your job 5665910 (\"senegal_213\") has been submitted\n",
      "Your job 5665911 (\"senegal_214\") has been submitted\n",
      "Your job 5665912 (\"senegal_215\") has been submitted\n",
      "Your job 5665913 (\"senegal_216\") has been submitted\n",
      "Your job 5665914 (\"senegal_217\") has been submitted\n",
      "Your job 5665915 (\"senegal_218\") has been submitted\n",
      "Your job 5665916 (\"senegal_219\") has been submitted\n",
      "Your job 5665917 (\"senegal_220\") has been submitted\n",
      "Your job 5665918 (\"senegal_221\") has been submitted\n",
      "Your job 5665919 (\"senegal_222\") has been submitted\n",
      "Your job 5665920 (\"senegal_223\") has been submitted\n",
      "Your job 5665921 (\"senegal_224\") has been submitted\n",
      "Your job 5665922 (\"senegal_225\") has been submitted\n",
      "Your job 5665923 (\"senegal_226\") has been submitted\n",
      "Your job 5665924 (\"senegal_227\") has been submitted\n",
      "Your job 5665925 (\"senegal_228\") has been submitted\n",
      "Your job 5665926 (\"senegal_229\") has been submitted\n",
      "Your job 5665927 (\"senegal_230\") has been submitted\n",
      "Your job 5665928 (\"senegal_231\") has been submitted\n",
      "Your job 5665929 (\"senegal_232\") has been submitted\n",
      "Your job 5665930 (\"senegal_233\") has been submitted\n",
      "Your job 5665931 (\"senegal_234\") has been submitted\n",
      "Your job 5665932 (\"senegal_235\") has been submitted\n",
      "Your job 5665933 (\"senegal_236\") has been submitted\n",
      "Your job 5665934 (\"senegal_237\") has been submitted\n",
      "Your job 5665935 (\"senegal_238\") has been submitted\n",
      "Your job 5665936 (\"senegal_239\") has been submitted\n",
      "Your job 5665937 (\"senegal_240\") has been submitted\n",
      "Your job 5665938 (\"senegal_241\") has been submitted\n",
      "Your job 5665939 (\"senegal_242\") has been submitted\n",
      "Your job 5665940 (\"senegal_243\") has been submitted\n",
      "Your job 5665941 (\"senegal_244\") has been submitted\n",
      "Your job 5665942 (\"senegal_245\") has been submitted\n",
      "Your job 5665943 (\"senegal_246\") has been submitted\n",
      "Your job 5665944 (\"senegal_247\") has been submitted\n",
      "Your job 5665945 (\"senegal_248\") has been submitted\n",
      "Your job 5665946 (\"senegal_249\") has been submitted\n",
      "Your job 5665947 (\"senegal_250\") has been submitted\n",
      "Your job 5665948 (\"senegal_251\") has been submitted\n",
      "Your job 5665949 (\"senegal_252\") has been submitted\n",
      "Your job 5665950 (\"senegal_253\") has been submitted\n",
      "Your job 5665951 (\"senegal_254\") has been submitted\n",
      "Your job 5665952 (\"senegal_255\") has been submitted\n",
      "Your job 5665953 (\"senegal_256\") has been submitted\n",
      "Your job 5665954 (\"senegal_257\") has been submitted\n",
      "Your job 5665955 (\"senegal_258\") has been submitted\n",
      "Your job 5665956 (\"senegal_259\") has been submitted\n",
      "Your job 5665957 (\"senegal_260\") has been submitted\n",
      "Your job 5665958 (\"senegal_261\") has been submitted\n",
      "Your job 5665959 (\"senegal_262\") has been submitted\n",
      "Your job 5665960 (\"senegal_263\") has been submitted\n",
      "Your job 5665961 (\"senegal_264\") has been submitted\n",
      "Your job 5665962 (\"senegal_265\") has been submitted\n",
      "Your job 5665963 (\"senegal_266\") has been submitted\n",
      "Your job 5665964 (\"senegal_267\") has been submitted\n",
      "Your job 5665965 (\"senegal_268\") has been submitted\n",
      "Your job 5665966 (\"senegal_269\") has been submitted\n",
      "Your job 5665967 (\"senegal_270\") has been submitted\n",
      "Your job 5665968 (\"senegal_271\") has been submitted\n",
      "Your job 5665969 (\"senegal_272\") has been submitted\n",
      "Your job 5665970 (\"senegal_273\") has been submitted\n",
      "Your job 5665971 (\"senegal_274\") has been submitted\n",
      "Your job 5665972 (\"senegal_275\") has been submitted\n",
      "Your job 5665973 (\"senegal_276\") has been submitted\n",
      "Your job 5665974 (\"senegal_277\") has been submitted\n",
      "Your job 5665975 (\"senegal_278\") has been submitted\n",
      "Your job 5665976 (\"senegal_279\") has been submitted\n",
      "Your job 5665977 (\"senegal_280\") has been submitted\n",
      "Your job 5665978 (\"senegal_281\") has been submitted\n",
      "Your job 5665979 (\"senegal_282\") has been submitted\n",
      "Your job 5665980 (\"senegal_283\") has been submitted\n",
      "Your job 5665981 (\"senegal_284\") has been submitted\n",
      "Your job 5665982 (\"senegal_285\") has been submitted\n",
      "Your job 5665983 (\"senegal_286\") has been submitted\n",
      "Your job 5665984 (\"senegal_287\") has been submitted\n",
      "Your job 5665985 (\"senegal_288\") has been submitted\n",
      "Your job 5665986 (\"senegal_289\") has been submitted\n",
      "Your job 5665987 (\"senegal_290\") has been submitted\n",
      "Your job 5665988 (\"senegal_291\") has been submitted\n",
      "Your job 5665989 (\"senegal_292\") has been submitted\n",
      "Your job 5665990 (\"senegal_293\") has been submitted\n",
      "Your job 5665991 (\"senegal_294\") has been submitted\n",
      "Your job 5665992 (\"senegal_295\") has been submitted\n",
      "Your job 5665993 (\"senegal_296\") has been submitted\n",
      "Your job 5665994 (\"senegal_297\") has been submitted\n",
      "Your job 5665995 (\"senegal_298\") has been submitted\n",
      "Your job 5665996 (\"senegal_299\") has been submitted\n",
      "Your job 5665997 (\"senegal_300\") has been submitted\n",
      "Your job 5665998 (\"senegal_301\") has been submitted\n",
      "Your job 5665999 (\"senegal_302\") has been submitted\n",
      "Your job 5666000 (\"senegal_303\") has been submitted\n",
      "Your job 5666001 (\"senegal_304\") has been submitted\n",
      "Your job 5666002 (\"senegal_305\") has been submitted\n",
      "Your job 5666003 (\"senegal_306\") has been submitted\n",
      "Your job 5666004 (\"senegal_307\") has been submitted\n",
      "Your job 5666005 (\"senegal_308\") has been submitted\n",
      "Your job 5666006 (\"senegal_309\") has been submitted\n",
      "Your job 5666007 (\"senegal_310\") has been submitted\n",
      "Your job 5666008 (\"senegal_311\") has been submitted\n",
      "Your job 5666009 (\"senegal_312\") has been submitted\n",
      "Your job 5666010 (\"senegal_313\") has been submitted\n",
      "Your job 5666011 (\"senegal_314\") has been submitted\n",
      "Your job 5666012 (\"senegal_315\") has been submitted\n",
      "Your job 5666013 (\"senegal_316\") has been submitted\n",
      "Your job 5666014 (\"senegal_317\") has been submitted\n",
      "Your job 5666015 (\"senegal_318\") has been submitted\n",
      "Your job 5666016 (\"senegal_319\") has been submitted\n",
      "Your job 5666017 (\"senegal_320\") has been submitted\n",
      "Your job 5666018 (\"senegal_321\") has been submitted\n",
      "Your job 5666019 (\"senegal_322\") has been submitted\n",
      "Your job 5666020 (\"senegal_323\") has been submitted\n",
      "Your job 5666021 (\"senegal_324\") has been submitted\n",
      "Your job 5666022 (\"senegal_325\") has been submitted\n",
      "Your job 5666023 (\"senegal_326\") has been submitted\n",
      "Your job 5666024 (\"senegal_327\") has been submitted\n",
      "Your job 5666025 (\"senegal_328\") has been submitted\n",
      "Your job 5666026 (\"senegal_329\") has been submitted\n",
      "Your job 5666027 (\"senegal_330\") has been submitted\n",
      "Your job 5666028 (\"senegal_331\") has been submitted\n",
      "Your job 5666029 (\"senegal_332\") has been submitted\n",
      "Your job 5666030 (\"senegal_333\") has been submitted\n",
      "Your job 5666031 (\"senegal_334\") has been submitted\n",
      "Your job 5666032 (\"senegal_335\") has been submitted\n",
      "Your job 5666033 (\"senegal_336\") has been submitted\n",
      "Your job 5666034 (\"senegal_337\") has been submitted\n",
      "Your job 5666035 (\"senegal_338\") has been submitted\n",
      "Your job 5666036 (\"senegal_339\") has been submitted\n",
      "Your job 5666037 (\"senegal_340\") has been submitted\n",
      "Your job 5666038 (\"senegal_341\") has been submitted\n",
      "Your job 5666039 (\"senegal_342\") has been submitted\n",
      "Your job 5666040 (\"niger_1\") has been submitted\n",
      "Your job 5666041 (\"niger_2\") has been submitted\n",
      "Your job 5666042 (\"niger_3\") has been submitted\n",
      "Your job 5666043 (\"niger_4\") has been submitted\n",
      "Your job 5666044 (\"niger_5\") has been submitted\n",
      "Your job 5666045 (\"niger_6\") has been submitted\n",
      "Your job 5666046 (\"niger_7\") has been submitted\n",
      "Your job 5666047 (\"niger_8\") has been submitted\n",
      "Your job 5666048 (\"niger_9\") has been submitted\n",
      "Your job 5666049 (\"niger_10\") has been submitted\n",
      "Your job 5666050 (\"niger_11\") has been submitted\n",
      "Your job 5666051 (\"niger_12\") has been submitted\n",
      "Your job 5666052 (\"niger_13\") has been submitted\n",
      "Your job 5666053 (\"niger_14\") has been submitted\n",
      "Your job 5666054 (\"niger_15\") has been submitted\n",
      "Your job 5666055 (\"niger_16\") has been submitted\n",
      "Your job 5666056 (\"niger_17\") has been submitted\n",
      "Your job 5666057 (\"niger_18\") has been submitted\n",
      "Your job 5666058 (\"niger_19\") has been submitted\n",
      "Your job 5666059 (\"niger_20\") has been submitted\n",
      "Your job 5666060 (\"niger_21\") has been submitted\n",
      "Your job 5666061 (\"niger_22\") has been submitted\n",
      "Your job 5666062 (\"niger_23\") has been submitted\n",
      "Your job 5666063 (\"niger_24\") has been submitted\n",
      "Your job 5666064 (\"niger_25\") has been submitted\n",
      "Your job 5666065 (\"niger_26\") has been submitted\n",
      "Your job 5666066 (\"niger_27\") has been submitted\n",
      "Your job 5666067 (\"niger_28\") has been submitted\n",
      "Your job 5666068 (\"niger_29\") has been submitted\n",
      "Your job 5666069 (\"niger_30\") has been submitted\n",
      "Your job 5666070 (\"niger_31\") has been submitted\n",
      "Your job 5666071 (\"niger_32\") has been submitted\n",
      "Your job 5666072 (\"niger_33\") has been submitted\n",
      "Your job 5666073 (\"niger_34\") has been submitted\n",
      "Your job 5666074 (\"niger_35\") has been submitted\n",
      "Your job 5666075 (\"niger_36\") has been submitted\n",
      "Your job 5666076 (\"niger_37\") has been submitted\n",
      "Your job 5666077 (\"niger_38\") has been submitted\n",
      "Your job 5666078 (\"niger_39\") has been submitted\n",
      "Your job 5666079 (\"niger_40\") has been submitted\n",
      "Your job 5666080 (\"niger_41\") has been submitted\n",
      "Your job 5666081 (\"niger_42\") has been submitted\n",
      "Your job 5666082 (\"niger_43\") has been submitted\n",
      "Your job 5666083 (\"niger_44\") has been submitted\n",
      "Your job 5666084 (\"niger_45\") has been submitted\n",
      "Your job 5666085 (\"niger_46\") has been submitted\n",
      "Your job 5666086 (\"niger_47\") has been submitted\n",
      "Your job 5666087 (\"niger_48\") has been submitted\n",
      "Your job 5666088 (\"niger_49\") has been submitted\n",
      "Your job 5666089 (\"niger_50\") has been submitted\n",
      "Your job 5666090 (\"niger_51\") has been submitted\n",
      "Your job 5666091 (\"niger_52\") has been submitted\n",
      "Your job 5666092 (\"niger_53\") has been submitted\n",
      "Your job 5666093 (\"niger_54\") has been submitted\n",
      "Your job 5666094 (\"niger_55\") has been submitted\n",
      "Your job 5666095 (\"niger_56\") has been submitted\n",
      "Your job 5666096 (\"niger_57\") has been submitted\n",
      "Your job 5666097 (\"niger_58\") has been submitted\n",
      "Your job 5666098 (\"niger_59\") has been submitted\n",
      "Your job 5666099 (\"niger_60\") has been submitted\n",
      "Your job 5666100 (\"niger_61\") has been submitted\n",
      "Your job 5666101 (\"niger_62\") has been submitted\n",
      "Your job 5666102 (\"niger_63\") has been submitted\n",
      "Your job 5666103 (\"niger_64\") has been submitted\n",
      "Your job 5666104 (\"niger_65\") has been submitted\n",
      "Your job 5666105 (\"niger_66\") has been submitted\n",
      "Your job 5666106 (\"niger_67\") has been submitted\n",
      "Your job 5666107 (\"niger_68\") has been submitted\n",
      "Your job 5666108 (\"niger_69\") has been submitted\n",
      "Your job 5666109 (\"niger_70\") has been submitted\n",
      "Your job 5666110 (\"niger_71\") has been submitted\n",
      "Your job 5666111 (\"niger_72\") has been submitted\n",
      "Your job 5666112 (\"niger_73\") has been submitted\n",
      "Your job 5666113 (\"niger_74\") has been submitted\n",
      "Your job 5666114 (\"niger_75\") has been submitted\n",
      "Your job 5666115 (\"niger_76\") has been submitted\n",
      "Your job 5666116 (\"niger_77\") has been submitted\n",
      "Your job 5666117 (\"niger_78\") has been submitted\n",
      "Your job 5666118 (\"niger_79\") has been submitted\n",
      "Your job 5666119 (\"niger_80\") has been submitted\n",
      "Your job 5666120 (\"niger_81\") has been submitted\n",
      "Your job 5666121 (\"niger_82\") has been submitted\n",
      "Your job 5666122 (\"niger_83\") has been submitted\n",
      "Your job 5666123 (\"niger_84\") has been submitted\n",
      "Your job 5666124 (\"niger_85\") has been submitted\n",
      "Your job 5666125 (\"niger_86\") has been submitted\n",
      "Your job 5666126 (\"niger_87\") has been submitted\n",
      "Your job 5666127 (\"niger_88\") has been submitted\n",
      "Your job 5666128 (\"niger_89\") has been submitted\n",
      "Your job 5666129 (\"niger_90\") has been submitted\n",
      "Your job 5666130 (\"niger_91\") has been submitted\n",
      "Your job 5666131 (\"niger_92\") has been submitted\n",
      "Your job 5666132 (\"niger_93\") has been submitted\n",
      "Your job 5666133 (\"niger_94\") has been submitted\n",
      "Your job 5666134 (\"niger_95\") has been submitted\n",
      "Your job 5666135 (\"niger_96\") has been submitted\n",
      "Your job 5666136 (\"niger_97\") has been submitted\n",
      "Your job 5666137 (\"niger_98\") has been submitted\n",
      "Your job 5666138 (\"niger_99\") has been submitted\n",
      "Your job 5666139 (\"niger_100\") has been submitted\n",
      "Your job 5666140 (\"niger_101\") has been submitted\n",
      "Your job 5666141 (\"niger_102\") has been submitted\n",
      "Your job 5666142 (\"niger_103\") has been submitted\n",
      "Your job 5666143 (\"niger_104\") has been submitted\n",
      "Your job 5666144 (\"niger_105\") has been submitted\n",
      "Your job 5666145 (\"niger_106\") has been submitted\n",
      "Your job 5666146 (\"niger_107\") has been submitted\n",
      "Your job 5666147 (\"niger_108\") has been submitted\n",
      "Your job 5666148 (\"niger_109\") has been submitted\n",
      "Your job 5666149 (\"niger_110\") has been submitted\n",
      "Your job 5666150 (\"niger_111\") has been submitted\n",
      "Your job 5666151 (\"niger_112\") has been submitted\n",
      "Your job 5666152 (\"niger_113\") has been submitted\n",
      "Your job 5666153 (\"niger_114\") has been submitted\n",
      "Your job 5666154 (\"niger_115\") has been submitted\n",
      "Your job 5666155 (\"niger_116\") has been submitted\n",
      "Your job 5666156 (\"niger_117\") has been submitted\n",
      "Your job 5666157 (\"niger_118\") has been submitted\n",
      "Your job 5666158 (\"niger_119\") has been submitted\n",
      "Your job 5666159 (\"niger_120\") has been submitted\n",
      "Your job 5666160 (\"niger_121\") has been submitted\n",
      "Your job 5666161 (\"niger_122\") has been submitted\n",
      "Your job 5666162 (\"niger_123\") has been submitted\n",
      "Your job 5666163 (\"niger_124\") has been submitted\n",
      "Your job 5666164 (\"niger_125\") has been submitted\n",
      "Your job 5666165 (\"niger_126\") has been submitted\n",
      "Your job 5666166 (\"niger_127\") has been submitted\n",
      "Your job 5666167 (\"niger_128\") has been submitted\n",
      "Your job 5666168 (\"niger_129\") has been submitted\n",
      "Your job 5666169 (\"niger_130\") has been submitted\n",
      "Your job 5666170 (\"niger_131\") has been submitted\n",
      "Your job 5666171 (\"niger_132\") has been submitted\n",
      "Your job 5666172 (\"niger_133\") has been submitted\n",
      "Your job 5666173 (\"niger_134\") has been submitted\n",
      "Your job 5666174 (\"niger_135\") has been submitted\n",
      "Your job 5666175 (\"niger_136\") has been submitted\n",
      "Your job 5666176 (\"niger_137\") has been submitted\n",
      "Your job 5666177 (\"niger_138\") has been submitted\n",
      "Your job 5666178 (\"niger_139\") has been submitted\n",
      "Your job 5666179 (\"niger_140\") has been submitted\n",
      "Your job 5666180 (\"niger_141\") has been submitted\n",
      "Your job 5666181 (\"niger_142\") has been submitted\n",
      "Your job 5666182 (\"niger_143\") has been submitted\n",
      "Your job 5666183 (\"niger_144\") has been submitted\n",
      "Your job 5666184 (\"niger_145\") has been submitted\n",
      "Your job 5666185 (\"niger_146\") has been submitted\n",
      "Your job 5666186 (\"niger_147\") has been submitted\n",
      "Your job 5666187 (\"niger_148\") has been submitted\n",
      "Your job 5666188 (\"niger_149\") has been submitted\n",
      "Your job 5666189 (\"niger_150\") has been submitted\n",
      "Your job 5666190 (\"niger_151\") has been submitted\n",
      "Your job 5666191 (\"niger_152\") has been submitted\n",
      "Your job 5666192 (\"niger_153\") has been submitted\n",
      "Your job 5666193 (\"niger_154\") has been submitted\n",
      "Your job 5666194 (\"niger_155\") has been submitted\n",
      "Your job 5666195 (\"niger_156\") has been submitted\n",
      "Your job 5666196 (\"niger_157\") has been submitted\n",
      "Your job 5666197 (\"niger_158\") has been submitted\n",
      "Your job 5666198 (\"niger_159\") has been submitted\n",
      "Your job 5666199 (\"niger_160\") has been submitted\n",
      "Your job 5666200 (\"niger_161\") has been submitted\n",
      "Your job 5666201 (\"niger_162\") has been submitted\n",
      "Your job 5666202 (\"niger_163\") has been submitted\n",
      "Your job 5666203 (\"niger_164\") has been submitted\n",
      "Your job 5666204 (\"niger_165\") has been submitted\n",
      "Your job 5666205 (\"niger_166\") has been submitted\n",
      "Your job 5666206 (\"niger_167\") has been submitted\n",
      "Your job 5666207 (\"niger_168\") has been submitted\n",
      "Your job 5666208 (\"niger_169\") has been submitted\n",
      "Your job 5666209 (\"niger_170\") has been submitted\n",
      "Your job 5666210 (\"niger_171\") has been submitted\n",
      "Your job 5666211 (\"niger_172\") has been submitted\n",
      "Your job 5666212 (\"niger_173\") has been submitted\n",
      "Your job 5666213 (\"niger_174\") has been submitted\n",
      "Your job 5666214 (\"niger_175\") has been submitted\n",
      "Your job 5666215 (\"niger_176\") has been submitted\n",
      "Your job 5666216 (\"niger_177\") has been submitted\n",
      "Your job 5666217 (\"niger_178\") has been submitted\n",
      "Your job 5666218 (\"niger_179\") has been submitted\n",
      "Your job 5666219 (\"niger_180\") has been submitted\n",
      "Your job 5666220 (\"niger_181\") has been submitted\n",
      "Your job 5666221 (\"niger_182\") has been submitted\n",
      "Your job 5666222 (\"niger_183\") has been submitted\n",
      "Your job 5666223 (\"niger_184\") has been submitted\n",
      "Your job 5666224 (\"niger_185\") has been submitted\n",
      "Your job 5666225 (\"niger_186\") has been submitted\n",
      "Your job 5666226 (\"niger_187\") has been submitted\n",
      "Your job 5666227 (\"niger_188\") has been submitted\n",
      "Your job 5666228 (\"niger_189\") has been submitted\n",
      "Your job 5666229 (\"niger_190\") has been submitted\n",
      "Your job 5666230 (\"niger_191\") has been submitted\n",
      "Your job 5666231 (\"niger_192\") has been submitted\n",
      "Your job 5666232 (\"niger_193\") has been submitted\n",
      "Your job 5666233 (\"niger_194\") has been submitted\n",
      "Your job 5666234 (\"niger_195\") has been submitted\n",
      "Your job 5666235 (\"niger_196\") has been submitted\n",
      "Your job 5666236 (\"niger_197\") has been submitted\n",
      "Your job 5666237 (\"niger_198\") has been submitted\n",
      "Your job 5666238 (\"niger_199\") has been submitted\n",
      "Your job 5666239 (\"niger_200\") has been submitted\n",
      "Your job 5666240 (\"niger_201\") has been submitted\n",
      "Your job 5666241 (\"niger_202\") has been submitted\n",
      "Your job 5666242 (\"niger_203\") has been submitted\n",
      "Your job 5666243 (\"niger_204\") has been submitted\n",
      "Your job 5666244 (\"niger_205\") has been submitted\n",
      "Your job 5666245 (\"niger_206\") has been submitted\n",
      "Your job 5666246 (\"niger_207\") has been submitted\n",
      "Your job 5666247 (\"niger_208\") has been submitted\n",
      "Your job 5666248 (\"niger_209\") has been submitted\n",
      "Your job 5666249 (\"niger_210\") has been submitted\n",
      "Your job 5666250 (\"niger_211\") has been submitted\n",
      "Your job 5666251 (\"niger_212\") has been submitted\n",
      "Your job 5666252 (\"niger_213\") has been submitted\n",
      "Your job 5666253 (\"niger_214\") has been submitted\n",
      "Your job 5666254 (\"niger_215\") has been submitted\n",
      "Your job 5666255 (\"niger_216\") has been submitted\n",
      "Your job 5666256 (\"niger_217\") has been submitted\n",
      "Your job 5666257 (\"niger_218\") has been submitted\n",
      "Your job 5666258 (\"niger_219\") has been submitted\n",
      "Your job 5666259 (\"niger_220\") has been submitted\n",
      "Your job 5666260 (\"niger_221\") has been submitted\n",
      "Your job 5666261 (\"niger_222\") has been submitted\n",
      "Your job 5666262 (\"niger_223\") has been submitted\n",
      "Your job 5666263 (\"niger_224\") has been submitted\n",
      "Your job 5666264 (\"niger_225\") has been submitted\n",
      "Your job 5666265 (\"niger_226\") has been submitted\n",
      "Your job 5666266 (\"niger_227\") has been submitted\n",
      "Your job 5666267 (\"niger_228\") has been submitted\n",
      "Your job 5666268 (\"niger_229\") has been submitted\n",
      "Your job 5666269 (\"niger_230\") has been submitted\n",
      "Your job 5666270 (\"niger_231\") has been submitted\n",
      "Your job 5666271 (\"niger_232\") has been submitted\n",
      "Your job 5666272 (\"niger_233\") has been submitted\n",
      "Your job 5666273 (\"niger_234\") has been submitted\n",
      "Your job 5666274 (\"niger_235\") has been submitted\n",
      "Your job 5666275 (\"niger_236\") has been submitted\n",
      "Your job 5666276 (\"niger_237\") has been submitted\n",
      "Your job 5666277 (\"niger_238\") has been submitted\n",
      "Your job 5666278 (\"niger_239\") has been submitted\n",
      "Your job 5666279 (\"niger_240\") has been submitted\n",
      "Your job 5666280 (\"niger_241\") has been submitted\n",
      "Your job 5666281 (\"niger_242\") has been submitted\n",
      "Your job 5666282 (\"niger_243\") has been submitted\n",
      "Your job 5666283 (\"niger_244\") has been submitted\n",
      "Your job 5666284 (\"niger_245\") has been submitted\n",
      "Your job 5666285 (\"niger_246\") has been submitted\n",
      "Your job 5666286 (\"niger_247\") has been submitted\n",
      "Your job 5666287 (\"niger_248\") has been submitted\n",
      "Your job 5666288 (\"niger_249\") has been submitted\n",
      "Your job 5666289 (\"niger_250\") has been submitted\n",
      "Your job 5666290 (\"niger_251\") has been submitted\n",
      "Your job 5666291 (\"niger_252\") has been submitted\n",
      "Your job 5666292 (\"niger_253\") has been submitted\n",
      "Your job 5666293 (\"niger_254\") has been submitted\n",
      "Your job 5666294 (\"niger_255\") has been submitted\n",
      "Your job 5666295 (\"niger_256\") has been submitted\n",
      "Your job 5666296 (\"niger_257\") has been submitted\n",
      "Your job 5666297 (\"niger_258\") has been submitted\n",
      "Your job 5666298 (\"niger_259\") has been submitted\n",
      "Your job 5666299 (\"niger_260\") has been submitted\n",
      "Your job 5666300 (\"niger_261\") has been submitted\n",
      "Your job 5666301 (\"niger_262\") has been submitted\n",
      "Your job 5666302 (\"niger_263\") has been submitted\n",
      "Your job 5666303 (\"niger_264\") has been submitted\n",
      "Your job 5666304 (\"niger_265\") has been submitted\n",
      "Your job 5666305 (\"niger_266\") has been submitted\n",
      "Your job 5666306 (\"niger_267\") has been submitted\n",
      "Your job 5666307 (\"niger_268\") has been submitted\n",
      "Your job 5666308 (\"niger_269\") has been submitted\n",
      "Your job 5666309 (\"niger_270\") has been submitted\n",
      "Your job 5666310 (\"niger_271\") has been submitted\n",
      "Your job 5666311 (\"niger_272\") has been submitted\n",
      "Your job 5666312 (\"niger_273\") has been submitted\n",
      "Your job 5666313 (\"niger_274\") has been submitted\n",
      "Your job 5666314 (\"niger_275\") has been submitted\n",
      "Your job 5666315 (\"niger_276\") has been submitted\n",
      "Your job 5666316 (\"niger_277\") has been submitted\n",
      "Your job 5666317 (\"niger_278\") has been submitted\n",
      "Your job 5666318 (\"niger_279\") has been submitted\n",
      "Your job 5666319 (\"niger_280\") has been submitted\n",
      "Your job 5666320 (\"niger_281\") has been submitted\n",
      "Your job 5666321 (\"niger_282\") has been submitted\n",
      "Your job 5666322 (\"niger_283\") has been submitted\n",
      "Your job 5666323 (\"niger_284\") has been submitted\n",
      "Your job 5666324 (\"niger_285\") has been submitted\n",
      "Your job 5666325 (\"niger_286\") has been submitted\n",
      "Your job 5666326 (\"niger_287\") has been submitted\n",
      "Your job 5666327 (\"niger_288\") has been submitted\n",
      "Your job 5666328 (\"niger_289\") has been submitted\n",
      "Your job 5666329 (\"niger_290\") has been submitted\n",
      "Your job 5666330 (\"niger_291\") has been submitted\n",
      "Your job 5666331 (\"niger_292\") has been submitted\n",
      "Your job 5666332 (\"niger_293\") has been submitted\n",
      "Your job 5666333 (\"niger_294\") has been submitted\n",
      "Your job 5666334 (\"niger_295\") has been submitted\n",
      "Your job 5666335 (\"niger_296\") has been submitted\n",
      "Your job 5666336 (\"niger_297\") has been submitted\n",
      "Your job 5666337 (\"niger_298\") has been submitted\n",
      "Your job 5666338 (\"niger_299\") has been submitted\n",
      "Your job 5666339 (\"niger_300\") has been submitted\n",
      "Your job 5666340 (\"niger_301\") has been submitted\n",
      "Your job 5666341 (\"niger_302\") has been submitted\n",
      "Your job 5666342 (\"niger_303\") has been submitted\n",
      "Your job 5666343 (\"niger_304\") has been submitted\n",
      "Your job 5666344 (\"niger_305\") has been submitted\n",
      "Your job 5666345 (\"niger_306\") has been submitted\n",
      "Your job 5666346 (\"niger_307\") has been submitted\n",
      "Your job 5666347 (\"niger_308\") has been submitted\n",
      "Your job 5666348 (\"niger_309\") has been submitted\n",
      "Your job 5666349 (\"niger_310\") has been submitted\n",
      "Your job 5666350 (\"niger_311\") has been submitted\n",
      "Your job 5666351 (\"niger_312\") has been submitted\n",
      "Your job 5666352 (\"niger_313\") has been submitted\n",
      "Your job 5666353 (\"niger_314\") has been submitted\n",
      "Your job 5666354 (\"niger_315\") has been submitted\n",
      "Your job 5666355 (\"niger_316\") has been submitted\n",
      "Your job 5666356 (\"niger_317\") has been submitted\n",
      "Your job 5666357 (\"niger_318\") has been submitted\n",
      "Your job 5666358 (\"niger_319\") has been submitted\n",
      "Your job 5666359 (\"niger_320\") has been submitted\n",
      "Your job 5666360 (\"niger_321\") has been submitted\n",
      "Your job 5666361 (\"niger_322\") has been submitted\n",
      "Your job 5666362 (\"niger_323\") has been submitted\n",
      "Your job 5666363 (\"niger_324\") has been submitted\n",
      "Your job 5666364 (\"niger_325\") has been submitted\n",
      "Your job 5666365 (\"niger_326\") has been submitted\n",
      "Your job 5666366 (\"niger_327\") has been submitted\n",
      "Your job 5666367 (\"niger_328\") has been submitted\n",
      "Your job 5666368 (\"niger_329\") has been submitted\n",
      "Your job 5666369 (\"niger_330\") has been submitted\n",
      "Your job 5666370 (\"niger_331\") has been submitted\n",
      "Your job 5666371 (\"niger_332\") has been submitted\n",
      "Your job 5666372 (\"niger_333\") has been submitted\n",
      "Your job 5666373 (\"niger_334\") has been submitted\n",
      "Your job 5666374 (\"niger_335\") has been submitted\n",
      "Your job 5666375 (\"niger_336\") has been submitted\n",
      "Your job 5666376 (\"niger_337\") has been submitted\n",
      "Your job 5666377 (\"niger_338\") has been submitted\n",
      "Your job 5666378 (\"niger_339\") has been submitted\n",
      "Your job 5666379 (\"niger_340\") has been submitted\n",
      "Your job 5666380 (\"niger_341\") has been submitted\n",
      "Your job 5666381 (\"niger_342\") has been submitted\n",
      "Your job 5666382 (\"brazil_1\") has been submitted\n",
      "Your job 5666383 (\"brazil_2\") has been submitted\n",
      "Your job 5666384 (\"brazil_3\") has been submitted\n",
      "Your job 5666385 (\"brazil_4\") has been submitted\n",
      "Your job 5666386 (\"brazil_5\") has been submitted\n",
      "Your job 5666387 (\"brazil_6\") has been submitted\n",
      "Your job 5666388 (\"brazil_7\") has been submitted\n",
      "Your job 5666389 (\"brazil_8\") has been submitted\n",
      "Your job 5666390 (\"brazil_9\") has been submitted\n",
      "Your job 5666391 (\"brazil_10\") has been submitted\n",
      "Your job 5666392 (\"brazil_11\") has been submitted\n",
      "Your job 5666393 (\"brazil_12\") has been submitted\n",
      "Your job 5666394 (\"brazil_13\") has been submitted\n",
      "Your job 5666395 (\"brazil_14\") has been submitted\n",
      "Your job 5666396 (\"brazil_15\") has been submitted\n",
      "Your job 5666397 (\"brazil_16\") has been submitted\n",
      "Your job 5666398 (\"brazil_17\") has been submitted\n",
      "Your job 5666399 (\"brazil_18\") has been submitted\n",
      "Your job 5666400 (\"brazil_19\") has been submitted\n",
      "Your job 5666401 (\"brazil_20\") has been submitted\n",
      "Your job 5666402 (\"brazil_21\") has been submitted\n",
      "Your job 5666403 (\"brazil_22\") has been submitted\n",
      "Your job 5666404 (\"brazil_23\") has been submitted\n",
      "Your job 5666405 (\"brazil_24\") has been submitted\n",
      "Your job 5666406 (\"brazil_25\") has been submitted\n",
      "Your job 5666407 (\"brazil_26\") has been submitted\n",
      "Your job 5666408 (\"brazil_27\") has been submitted\n",
      "Your job 5666409 (\"brazil_28\") has been submitted\n",
      "Your job 5666410 (\"brazil_29\") has been submitted\n",
      "Your job 5666411 (\"brazil_30\") has been submitted\n",
      "Your job 5666412 (\"brazil_31\") has been submitted\n",
      "Your job 5666413 (\"brazil_32\") has been submitted\n",
      "Your job 5666414 (\"brazil_33\") has been submitted\n",
      "Your job 5666415 (\"brazil_34\") has been submitted\n",
      "Your job 5666416 (\"brazil_35\") has been submitted\n",
      "Your job 5666417 (\"brazil_36\") has been submitted\n",
      "Your job 5666418 (\"brazil_37\") has been submitted\n",
      "Your job 5666419 (\"brazil_38\") has been submitted\n",
      "Your job 5666420 (\"brazil_39\") has been submitted\n",
      "Your job 5666421 (\"brazil_40\") has been submitted\n",
      "Your job 5666422 (\"brazil_41\") has been submitted\n",
      "Your job 5666423 (\"brazil_42\") has been submitted\n",
      "Your job 5666424 (\"brazil_43\") has been submitted\n",
      "Your job 5666425 (\"brazil_44\") has been submitted\n",
      "Your job 5666426 (\"brazil_45\") has been submitted\n",
      "Your job 5666427 (\"brazil_46\") has been submitted\n",
      "Your job 5666428 (\"brazil_47\") has been submitted\n",
      "Your job 5666429 (\"brazil_48\") has been submitted\n",
      "Your job 5666430 (\"brazil_49\") has been submitted\n",
      "Your job 5666431 (\"brazil_50\") has been submitted\n",
      "Your job 5666432 (\"brazil_51\") has been submitted\n",
      "Your job 5666433 (\"brazil_52\") has been submitted\n",
      "Your job 5666434 (\"brazil_53\") has been submitted\n",
      "Your job 5666435 (\"brazil_54\") has been submitted\n",
      "Your job 5666436 (\"brazil_55\") has been submitted\n",
      "Your job 5666437 (\"brazil_56\") has been submitted\n",
      "Your job 5666438 (\"brazil_57\") has been submitted\n",
      "Your job 5666439 (\"brazil_58\") has been submitted\n",
      "Your job 5666440 (\"brazil_59\") has been submitted\n",
      "Your job 5666441 (\"brazil_60\") has been submitted\n",
      "Your job 5666442 (\"brazil_61\") has been submitted\n",
      "Your job 5666443 (\"brazil_62\") has been submitted\n",
      "Your job 5666444 (\"brazil_63\") has been submitted\n",
      "Your job 5666445 (\"brazil_64\") has been submitted\n",
      "Your job 5666446 (\"brazil_65\") has been submitted\n",
      "Your job 5666447 (\"brazil_66\") has been submitted\n",
      "Your job 5666448 (\"brazil_67\") has been submitted\n",
      "Your job 5666449 (\"brazil_68\") has been submitted\n",
      "Your job 5666450 (\"brazil_69\") has been submitted\n",
      "Your job 5666451 (\"brazil_70\") has been submitted\n",
      "Your job 5666452 (\"brazil_71\") has been submitted\n",
      "Your job 5666453 (\"brazil_72\") has been submitted\n",
      "Your job 5666454 (\"brazil_73\") has been submitted\n",
      "Your job 5666455 (\"brazil_74\") has been submitted\n",
      "Your job 5666456 (\"brazil_75\") has been submitted\n",
      "Your job 5666457 (\"brazil_76\") has been submitted\n",
      "Your job 5666458 (\"brazil_77\") has been submitted\n",
      "Your job 5666459 (\"brazil_78\") has been submitted\n",
      "Your job 5666460 (\"brazil_79\") has been submitted\n",
      "Your job 5666461 (\"brazil_80\") has been submitted\n",
      "Your job 5666462 (\"brazil_81\") has been submitted\n",
      "Your job 5666463 (\"brazil_82\") has been submitted\n",
      "Your job 5666464 (\"brazil_83\") has been submitted\n",
      "Your job 5666465 (\"brazil_84\") has been submitted\n",
      "Your job 5666466 (\"brazil_85\") has been submitted\n",
      "Your job 5666467 (\"brazil_86\") has been submitted\n",
      "Your job 5666468 (\"brazil_87\") has been submitted\n",
      "Your job 5666469 (\"brazil_88\") has been submitted\n",
      "Your job 5666470 (\"brazil_89\") has been submitted\n",
      "Your job 5666471 (\"brazil_90\") has been submitted\n",
      "Your job 5666472 (\"brazil_91\") has been submitted\n",
      "Your job 5666473 (\"brazil_92\") has been submitted\n",
      "Your job 5666474 (\"brazil_93\") has been submitted\n",
      "Your job 5666475 (\"brazil_94\") has been submitted\n",
      "Your job 5666476 (\"brazil_95\") has been submitted\n",
      "Your job 5666477 (\"brazil_96\") has been submitted\n",
      "Your job 5666478 (\"brazil_97\") has been submitted\n",
      "Your job 5666479 (\"brazil_98\") has been submitted\n",
      "Your job 5666480 (\"brazil_99\") has been submitted\n",
      "Your job 5666481 (\"brazil_100\") has been submitted\n",
      "Your job 5666482 (\"brazil_101\") has been submitted\n",
      "Your job 5666483 (\"brazil_102\") has been submitted\n",
      "Your job 5666484 (\"brazil_103\") has been submitted\n",
      "Your job 5666485 (\"brazil_104\") has been submitted\n",
      "Your job 5666486 (\"brazil_105\") has been submitted\n",
      "Your job 5666487 (\"brazil_106\") has been submitted\n",
      "Your job 5666488 (\"brazil_107\") has been submitted\n",
      "Your job 5666489 (\"brazil_108\") has been submitted\n",
      "Your job 5666490 (\"brazil_109\") has been submitted\n",
      "Your job 5666491 (\"brazil_110\") has been submitted\n",
      "Your job 5666492 (\"brazil_111\") has been submitted\n",
      "Your job 5666493 (\"brazil_112\") has been submitted\n",
      "Your job 5666494 (\"brazil_113\") has been submitted\n",
      "Your job 5666495 (\"brazil_114\") has been submitted\n",
      "Your job 5666496 (\"brazil_115\") has been submitted\n",
      "Your job 5666497 (\"brazil_116\") has been submitted\n",
      "Your job 5666498 (\"brazil_117\") has been submitted\n",
      "Your job 5666499 (\"brazil_118\") has been submitted\n",
      "Your job 5666500 (\"brazil_119\") has been submitted\n",
      "Your job 5666501 (\"brazil_120\") has been submitted\n",
      "Your job 5666502 (\"brazil_121\") has been submitted\n",
      "Your job 5666503 (\"brazil_122\") has been submitted\n",
      "Your job 5666504 (\"brazil_123\") has been submitted\n",
      "Your job 5666505 (\"brazil_124\") has been submitted\n",
      "Your job 5666506 (\"brazil_125\") has been submitted\n",
      "Your job 5666507 (\"brazil_126\") has been submitted\n",
      "Your job 5666508 (\"brazil_127\") has been submitted\n",
      "Your job 5666509 (\"brazil_128\") has been submitted\n",
      "Your job 5666510 (\"brazil_129\") has been submitted\n",
      "Your job 5666511 (\"brazil_130\") has been submitted\n",
      "Your job 5666512 (\"brazil_131\") has been submitted\n",
      "Your job 5666513 (\"brazil_132\") has been submitted\n",
      "Your job 5666514 (\"brazil_133\") has been submitted\n",
      "Your job 5666515 (\"brazil_134\") has been submitted\n",
      "Your job 5666516 (\"brazil_135\") has been submitted\n",
      "Your job 5666517 (\"brazil_136\") has been submitted\n",
      "Your job 5666518 (\"brazil_137\") has been submitted\n",
      "Your job 5666519 (\"brazil_138\") has been submitted\n",
      "Your job 5666520 (\"brazil_139\") has been submitted\n",
      "Your job 5666521 (\"brazil_140\") has been submitted\n",
      "Your job 5666522 (\"brazil_141\") has been submitted\n",
      "Your job 5666523 (\"brazil_142\") has been submitted\n",
      "Your job 5666524 (\"brazil_143\") has been submitted\n",
      "Your job 5666525 (\"brazil_144\") has been submitted\n",
      "Your job 5666526 (\"brazil_145\") has been submitted\n",
      "Your job 5666527 (\"brazil_146\") has been submitted\n",
      "Your job 5666528 (\"brazil_147\") has been submitted\n",
      "Your job 5666529 (\"brazil_148\") has been submitted\n",
      "Your job 5666530 (\"brazil_149\") has been submitted\n",
      "Your job 5666531 (\"brazil_150\") has been submitted\n",
      "Your job 5666532 (\"brazil_151\") has been submitted\n",
      "Your job 5666533 (\"brazil_152\") has been submitted\n",
      "Your job 5666534 (\"brazil_153\") has been submitted\n",
      "Your job 5666535 (\"brazil_154\") has been submitted\n",
      "Your job 5666536 (\"brazil_155\") has been submitted\n",
      "Your job 5666537 (\"brazil_156\") has been submitted\n",
      "Your job 5666538 (\"brazil_157\") has been submitted\n",
      "Your job 5666539 (\"brazil_158\") has been submitted\n",
      "Your job 5666540 (\"brazil_159\") has been submitted\n",
      "Your job 5666541 (\"brazil_160\") has been submitted\n",
      "Your job 5666542 (\"brazil_161\") has been submitted\n",
      "Your job 5666543 (\"brazil_162\") has been submitted\n",
      "Your job 5666544 (\"brazil_163\") has been submitted\n",
      "Your job 5666545 (\"brazil_164\") has been submitted\n",
      "Your job 5666546 (\"brazil_165\") has been submitted\n",
      "Your job 5666547 (\"brazil_166\") has been submitted\n",
      "Your job 5666548 (\"brazil_167\") has been submitted\n",
      "Your job 5666549 (\"brazil_168\") has been submitted\n",
      "Your job 5666550 (\"brazil_169\") has been submitted\n",
      "Your job 5666551 (\"brazil_170\") has been submitted\n",
      "Your job 5666552 (\"brazil_171\") has been submitted\n",
      "Your job 5666553 (\"brazil_172\") has been submitted\n",
      "Your job 5666554 (\"brazil_173\") has been submitted\n",
      "Your job 5666555 (\"brazil_174\") has been submitted\n",
      "Your job 5666556 (\"brazil_175\") has been submitted\n",
      "Your job 5666557 (\"brazil_176\") has been submitted\n",
      "Your job 5666558 (\"brazil_177\") has been submitted\n",
      "Your job 5666559 (\"brazil_178\") has been submitted\n",
      "Your job 5666560 (\"brazil_179\") has been submitted\n",
      "Your job 5666561 (\"brazil_180\") has been submitted\n",
      "Your job 5666562 (\"brazil_181\") has been submitted\n",
      "Your job 5666563 (\"brazil_182\") has been submitted\n",
      "Your job 5666564 (\"brazil_183\") has been submitted\n",
      "Your job 5666565 (\"brazil_184\") has been submitted\n",
      "Your job 5666566 (\"brazil_185\") has been submitted\n",
      "Your job 5666567 (\"brazil_186\") has been submitted\n",
      "Your job 5666568 (\"brazil_187\") has been submitted\n",
      "Your job 5666569 (\"brazil_188\") has been submitted\n",
      "Your job 5666570 (\"brazil_189\") has been submitted\n",
      "Your job 5666571 (\"brazil_190\") has been submitted\n",
      "Your job 5666572 (\"brazil_191\") has been submitted\n",
      "Your job 5666573 (\"brazil_192\") has been submitted\n",
      "Your job 5666574 (\"brazil_193\") has been submitted\n",
      "Your job 5666575 (\"brazil_194\") has been submitted\n",
      "Your job 5666576 (\"brazil_195\") has been submitted\n",
      "Your job 5666577 (\"brazil_196\") has been submitted\n",
      "Your job 5666578 (\"brazil_197\") has been submitted\n",
      "Your job 5666579 (\"brazil_198\") has been submitted\n",
      "Your job 5666580 (\"brazil_199\") has been submitted\n",
      "Your job 5666581 (\"brazil_200\") has been submitted\n",
      "Your job 5666582 (\"brazil_201\") has been submitted\n",
      "Your job 5666583 (\"brazil_202\") has been submitted\n",
      "Your job 5666584 (\"brazil_203\") has been submitted\n",
      "Your job 5666585 (\"brazil_204\") has been submitted\n",
      "Your job 5666586 (\"brazil_205\") has been submitted\n",
      "Your job 5666587 (\"brazil_206\") has been submitted\n",
      "Your job 5666588 (\"brazil_207\") has been submitted\n",
      "Your job 5666589 (\"brazil_208\") has been submitted\n",
      "Your job 5666590 (\"brazil_209\") has been submitted\n",
      "Your job 5666591 (\"brazil_210\") has been submitted\n",
      "Your job 5666592 (\"brazil_211\") has been submitted\n",
      "Your job 5666593 (\"brazil_212\") has been submitted\n",
      "Your job 5666594 (\"brazil_213\") has been submitted\n",
      "Your job 5666595 (\"brazil_214\") has been submitted\n",
      "Your job 5666596 (\"brazil_215\") has been submitted\n",
      "Your job 5666597 (\"brazil_216\") has been submitted\n",
      "Your job 5666598 (\"brazil_217\") has been submitted\n",
      "Your job 5666599 (\"brazil_218\") has been submitted\n",
      "Your job 5666600 (\"brazil_219\") has been submitted\n",
      "Your job 5666601 (\"brazil_220\") has been submitted\n",
      "Your job 5666602 (\"brazil_221\") has been submitted\n",
      "Your job 5666603 (\"brazil_222\") has been submitted\n",
      "Your job 5666604 (\"brazil_223\") has been submitted\n",
      "Your job 5666605 (\"brazil_224\") has been submitted\n",
      "Your job 5666606 (\"brazil_225\") has been submitted\n",
      "Your job 5666607 (\"brazil_226\") has been submitted\n",
      "Your job 5666608 (\"brazil_227\") has been submitted\n",
      "Your job 5666609 (\"brazil_228\") has been submitted\n",
      "Your job 5666610 (\"brazil_229\") has been submitted\n",
      "Your job 5666611 (\"brazil_230\") has been submitted\n",
      "Your job 5666612 (\"brazil_231\") has been submitted\n",
      "Your job 5666613 (\"brazil_232\") has been submitted\n",
      "Your job 5666614 (\"brazil_233\") has been submitted\n",
      "Your job 5666615 (\"brazil_234\") has been submitted\n",
      "Your job 5666616 (\"brazil_235\") has been submitted\n",
      "Your job 5666617 (\"brazil_236\") has been submitted\n",
      "Your job 5666618 (\"brazil_237\") has been submitted\n",
      "Your job 5666619 (\"brazil_238\") has been submitted\n",
      "Your job 5666620 (\"brazil_239\") has been submitted\n",
      "Your job 5666621 (\"brazil_240\") has been submitted\n",
      "Your job 5666622 (\"brazil_241\") has been submitted\n",
      "Your job 5666623 (\"brazil_242\") has been submitted\n",
      "Your job 5666624 (\"brazil_243\") has been submitted\n",
      "Your job 5666625 (\"brazil_244\") has been submitted\n",
      "Your job 5666626 (\"brazil_245\") has been submitted\n",
      "Your job 5666627 (\"brazil_246\") has been submitted\n",
      "Your job 5666628 (\"brazil_247\") has been submitted\n",
      "Your job 5666629 (\"brazil_248\") has been submitted\n",
      "Your job 5666630 (\"brazil_249\") has been submitted\n",
      "Your job 5666631 (\"brazil_250\") has been submitted\n",
      "Your job 5666632 (\"brazil_251\") has been submitted\n",
      "Your job 5666633 (\"brazil_252\") has been submitted\n",
      "Your job 5666634 (\"brazil_253\") has been submitted\n",
      "Your job 5666635 (\"brazil_254\") has been submitted\n",
      "Your job 5666636 (\"brazil_255\") has been submitted\n",
      "Your job 5666637 (\"brazil_256\") has been submitted\n",
      "Your job 5666638 (\"brazil_257\") has been submitted\n",
      "Your job 5666639 (\"brazil_258\") has been submitted\n",
      "Your job 5666640 (\"brazil_259\") has been submitted\n",
      "Your job 5666641 (\"brazil_260\") has been submitted\n",
      "Your job 5666642 (\"brazil_261\") has been submitted\n",
      "Your job 5666643 (\"brazil_262\") has been submitted\n",
      "Your job 5666644 (\"brazil_263\") has been submitted\n",
      "Your job 5666645 (\"brazil_264\") has been submitted\n",
      "Your job 5666646 (\"brazil_265\") has been submitted\n",
      "Your job 5666647 (\"brazil_266\") has been submitted\n",
      "Your job 5666648 (\"brazil_267\") has been submitted\n",
      "Your job 5666649 (\"brazil_268\") has been submitted\n",
      "Your job 5666650 (\"brazil_269\") has been submitted\n",
      "Your job 5666651 (\"brazil_270\") has been submitted\n",
      "Your job 5666652 (\"brazil_271\") has been submitted\n",
      "Your job 5666653 (\"brazil_272\") has been submitted\n",
      "Your job 5666654 (\"brazil_273\") has been submitted\n",
      "Your job 5666655 (\"brazil_274\") has been submitted\n",
      "Your job 5666656 (\"brazil_275\") has been submitted\n",
      "Your job 5666657 (\"brazil_276\") has been submitted\n",
      "Your job 5666658 (\"brazil_277\") has been submitted\n",
      "Your job 5666659 (\"brazil_278\") has been submitted\n",
      "Your job 5666660 (\"brazil_279\") has been submitted\n",
      "Your job 5666661 (\"brazil_280\") has been submitted\n",
      "Your job 5666662 (\"brazil_281\") has been submitted\n",
      "Your job 5666663 (\"brazil_282\") has been submitted\n",
      "Your job 5666664 (\"brazil_283\") has been submitted\n",
      "Your job 5666665 (\"brazil_284\") has been submitted\n",
      "Your job 5666666 (\"brazil_285\") has been submitted\n",
      "Your job 5666667 (\"brazil_286\") has been submitted\n",
      "Your job 5666668 (\"brazil_287\") has been submitted\n",
      "Your job 5666669 (\"brazil_288\") has been submitted\n",
      "Your job 5666670 (\"brazil_289\") has been submitted\n",
      "Your job 5666671 (\"brazil_290\") has been submitted\n",
      "Your job 5666672 (\"brazil_291\") has been submitted\n",
      "Your job 5666673 (\"brazil_292\") has been submitted\n",
      "Your job 5666674 (\"brazil_293\") has been submitted\n",
      "Your job 5666675 (\"brazil_294\") has been submitted\n",
      "Your job 5666676 (\"brazil_295\") has been submitted\n",
      "Your job 5666677 (\"brazil_296\") has been submitted\n",
      "Your job 5666678 (\"brazil_297\") has been submitted\n",
      "Your job 5666679 (\"brazil_298\") has been submitted\n",
      "Your job 5666680 (\"brazil_299\") has been submitted\n",
      "Your job 5666681 (\"brazil_300\") has been submitted\n",
      "Your job 5666682 (\"brazil_301\") has been submitted\n",
      "Your job 5666683 (\"brazil_302\") has been submitted\n",
      "Your job 5666684 (\"brazil_303\") has been submitted\n",
      "Your job 5666685 (\"brazil_304\") has been submitted\n",
      "Your job 5666686 (\"brazil_305\") has been submitted\n",
      "Your job 5666687 (\"brazil_306\") has been submitted\n",
      "Your job 5666688 (\"brazil_307\") has been submitted\n",
      "Your job 5666689 (\"brazil_308\") has been submitted\n",
      "Your job 5666690 (\"brazil_309\") has been submitted\n",
      "Your job 5666691 (\"brazil_310\") has been submitted\n",
      "Your job 5666692 (\"brazil_311\") has been submitted\n",
      "Your job 5666693 (\"brazil_312\") has been submitted\n",
      "Your job 5666694 (\"brazil_313\") has been submitted\n",
      "Your job 5666695 (\"brazil_314\") has been submitted\n",
      "Your job 5666696 (\"brazil_315\") has been submitted\n",
      "Your job 5666697 (\"brazil_316\") has been submitted\n",
      "Your job 5666698 (\"brazil_317\") has been submitted\n",
      "Your job 5666699 (\"brazil_318\") has been submitted\n",
      "Your job 5666700 (\"brazil_319\") has been submitted\n",
      "Your job 5666701 (\"brazil_320\") has been submitted\n",
      "Your job 5666702 (\"brazil_321\") has been submitted\n",
      "Your job 5666703 (\"brazil_322\") has been submitted\n",
      "Your job 5666704 (\"brazil_323\") has been submitted\n",
      "Your job 5666705 (\"brazil_324\") has been submitted\n",
      "Your job 5666706 (\"brazil_325\") has been submitted\n",
      "Your job 5666707 (\"brazil_326\") has been submitted\n",
      "Your job 5666708 (\"brazil_327\") has been submitted\n",
      "Your job 5666709 (\"brazil_328\") has been submitted\n",
      "Your job 5666710 (\"brazil_329\") has been submitted\n",
      "Your job 5666711 (\"brazil_330\") has been submitted\n",
      "Your job 5666712 (\"brazil_331\") has been submitted\n",
      "Your job 5666713 (\"brazil_332\") has been submitted\n",
      "Your job 5666714 (\"brazil_333\") has been submitted\n",
      "Your job 5666715 (\"brazil_334\") has been submitted\n",
      "Your job 5666716 (\"brazil_335\") has been submitted\n",
      "Your job 5666717 (\"brazil_336\") has been submitted\n",
      "Your job 5666718 (\"brazil_337\") has been submitted\n",
      "Your job 5666719 (\"brazil_338\") has been submitted\n",
      "Your job 5666720 (\"brazil_339\") has been submitted\n",
      "Your job 5666721 (\"brazil_340\") has been submitted\n",
      "Your job 5666722 (\"brazil_341\") has been submitted\n",
      "Your job 5666723 (\"brazil_342\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p results/n10/msprime/logs\n",
    "\n",
    "CONDA=\"conda activate sch_man_nwinvasion-msprime;\"\n",
    "QSUB=\"qsub -V -cwd -S /bin/bash -q all.q -j y -pe smp 1 \"    \n",
    "\n",
    "for POP in tanzania senegal niger brazil; do\n",
    "    mkdir -p results/n10/msprime/$POP\n",
    "    for I in $(seq 1 342); do\n",
    "        MSPRIME_CMD=\"python code/msprime-qsub_n10.py $POP $I\"       \n",
    "        echo \"$CONDA $MSPRIME_CMD\" | $QSUB -N $POP\"_\"$I -o results/n10/msprime/logs/$POP\"_\"$I.log\n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#now get the vcfs\n",
    "#python code/msprime-probe_snps_from_vcf.py (modified for n10)\n",
    "\n",
    "import msprime\n",
    "import os\n",
    "import subprocess\n",
    "import glob\n",
    "\n",
    "\n",
    "#now create bed for the new \"sim\" chr1\n",
    "with open('data/renamed-sma_agilent_baits.v7.0.chr_reorderd.bed', 'r') as in_bed:\n",
    "    with open('results/msprime/sim_probes.bed', 'w') as out_bed:\n",
    "        for bed_entry in in_bed:\n",
    "            chrom, start, stop = bed_entry.rstrip().split(\"\\t\")\n",
    "            if chrom == \"SM_V7_1\":\n",
    "                out_bed.write(\"1\\t{}\\t{}\\n\".format(start, stop))\n",
    "\n",
    "bed = 'results/msprime/sim_probes.bed'\n",
    "#now loop through all of the sim vcf files to get snps at probed regions\n",
    "for pop in [\"niger\", \"senegal\", \"tanzania\", \"brazil\"]:\n",
    "    out_dir = \"results/n10/msprime/{}\".format(pop)\n",
    "    \n",
    "    sim_vcfs = glob.glob(\"{}/chr1_*_rep_*.vcf\".format(out_dir))\n",
    "    for sim_vcf in sim_vcfs:\n",
    "\n",
    "        probed_vcf = sim_vcf.replace(\".vcf\", \"_probed.vcf\")         \n",
    "        jid = \"probe_{}\".format(probed_vcf.split(\"/\")[-1])\n",
    "        log = \"{}/logs\".format(out_dir)\n",
    "\n",
    "        vcf_cmd = \"vcftools --vcf {} --bed {} --recode --recode-INFO-all --stdout >{}\".format(sim_vcf, bed, probed_vcf)\n",
    "        qsub_cmd =  \"qsub -V -cwd -S /bin/bash -q all.q -j y -pe smp 3 -N {} -o {}\".format(jid, log)\n",
    "        conda_cmd = \"conda activate sch_man_nwinvasion-msprime\"\n",
    "\n",
    "        cmd =\"echo \\\"{}; {}\\\" | {}\".format(conda_cmd, vcf_cmd, qsub_cmd)\n",
    "\n",
    "        #run vcf cmd\n",
    "        #process = subprocess.Popen(cmd.split(\"\"),\n",
    "        #                     stdout=subprocess.PIPE, \n",
    "        #                     stderr=subprocess.PIPE)\n",
    "        !{cmd}\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## HSCAN\n",
    "\n",
    "HSCAN identifies regions under selection based on the length of haplotypes.  (citation).  We used a custom code to convert HSCAN to VCF input formats."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-23T01:40:44.758168Z",
     "start_time": "2020-06-23T01:40:43.813729Z"
    },
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5666864 (\"brazil_SM_V7_1\") has been submitted\n",
      "Your job 5666865 (\"brazil_SM_V7_2\") has been submitted\n",
      "Your job 5666866 (\"brazil_SM_V7_3\") has been submitted\n",
      "Your job 5666867 (\"brazil_SM_V7_4\") has been submitted\n",
      "Your job 5666868 (\"brazil_SM_V7_5\") has been submitted\n",
      "Your job 5666869 (\"brazil_SM_V7_6\") has been submitted\n",
      "Your job 5666870 (\"brazil_SM_V7_7\") has been submitted\n",
      "Your job 5666871 (\"tanzania_SM_V7_1\") has been submitted\n",
      "Your job 5666872 (\"tanzania_SM_V7_2\") has been submitted\n",
      "Your job 5666873 (\"tanzania_SM_V7_3\") has been submitted\n",
      "Your job 5666874 (\"tanzania_SM_V7_4\") has been submitted\n",
      "Your job 5666875 (\"tanzania_SM_V7_5\") has been submitted\n",
      "Your job 5666876 (\"tanzania_SM_V7_6\") has been submitted\n",
      "Your job 5666877 (\"tanzania_SM_V7_7\") has been submitted\n",
      "Your job 5666878 (\"niger_SM_V7_1\") has been submitted\n",
      "Your job 5666879 (\"niger_SM_V7_2\") has been submitted\n",
      "Your job 5666880 (\"niger_SM_V7_3\") has been submitted\n",
      "Your job 5666881 (\"niger_SM_V7_4\") has been submitted\n",
      "Your job 5666882 (\"niger_SM_V7_5\") has been submitted\n",
      "Your job 5666883 (\"niger_SM_V7_6\") has been submitted\n",
      "Your job 5666884 (\"niger_SM_V7_7\") has been submitted\n",
      "Your job 5666885 (\"senegal_SM_V7_1\") has been submitted\n",
      "Your job 5666886 (\"senegal_SM_V7_2\") has been submitted\n",
      "Your job 5666887 (\"senegal_SM_V7_3\") has been submitted\n",
      "Your job 5666888 (\"senegal_SM_V7_4\") has been submitted\n",
      "Your job 5666889 (\"senegal_SM_V7_5\") has been submitted\n",
      "Your job 5666890 (\"senegal_SM_V7_6\") has been submitted\n",
      "Your job 5666891 (\"senegal_SM_V7_7\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd ~/sch_man_nwinvasion/\n",
    "\n",
    "CONDA=\"conda activate sch_man_nwinvasion-hscan\"\n",
    "QSUB=\"qsub -V -cwd -S /bin/bash -q all.q -j y -pe smp 1 \"    \n",
    "OUTDIR=\"results/n10/hscan\"\n",
    "\n",
    "if [ ! -d $OUTDIR ] \n",
    "then\n",
    "    mkdir $OUTDIR\n",
    "fi\n",
    "\n",
    "mkdir -p $OUTDIR/logs\n",
    "\n",
    "#REAL DATA\n",
    "for POP in brazil tanzania niger senegal; do\n",
    "    mkdir $OUTDIR/$POP\n",
    "        for i in $(seq 1 7 ); do\n",
    "            CHROM=\"SM_V7_\"$i\n",
    "    \n",
    "            #get chrom vcf\n",
    "            VCF_CMD=\"vcftools \\\n",
    "                --vcf results/n10/smv7_ex_autosomes_n10.vcf \\\n",
    "                --keep results/n10/\"$POP\"_n10.list \\\n",
    "                --chr $CHROM \\\n",
    "                --recode \\\n",
    "                --recode-INFO-all \\\n",
    "                --stdout \\\n",
    "                >$OUTDIR/$POP/\"$CHROM\"_\"$POP\".vcf\"\n",
    "\n",
    "            #convert vcf to hscan\n",
    "            CONV_CMD=\"python code/vcf2hscan.py \\\n",
    "                $OUTDIR/$POP/\"$CHROM\"_\"$POP\".vcf \\\n",
    "                $OUTDIR/$POP/\"$CHROM\"_\"$POP\".hscan-in\"\n",
    "\n",
    "            #convert to hscan\n",
    "            HSCAN_CMD=\"bin/H-scan \\\n",
    "                -i $OUTDIR/$POP/\"$CHROM\"_\"$POP\".hscan-in \\\n",
    "                -g 10000 \\\n",
    "                >$OUTDIR/$POP/\"$CHROM\"_\"$POP\".hscan-out\"\n",
    "\n",
    "        echo \"$CONDA; $VCF_CMD; $CONV_CMD; $HSCAN_CMD\" | $QSUB -N $POP\"_\"$CHROM -o $OUTDIR/logs/$POP\"_\"$CHROM.log\n",
    "\n",
    "    done\n",
    "done\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### Calculate P value from Hscan and Plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#when these are done running. plot\n",
    "\n",
    "################################################################################\n",
    "#make table and plot \n",
    "\n",
    "#calculate p-values\n",
    "import scipy.stats\n",
    "import statsmodels.api as sm\n",
    "import statsmodels as sm\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import glob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#make sure that all stops are not gt chrom length\n",
    "chr_length = {}\n",
    "#genome_size = 0\n",
    "with open('/master/nplatt/sch_man_nwinvasion/data/genomes/Smansoni_v7.fa.fai', 'r') as fai:\n",
    "    for entry in fai:\n",
    "        chrom, length, *offset = entry.rstrip().split(\"\\t\")\n",
    "        chr_length[chrom]=int(length)\n",
    "\n",
    "    cumul_start={}\n",
    "    cumul_start['SM_V7_1']=0\n",
    "    cumul_start['SM_V7_2']= cumul_start['SM_V7_1'] + chr_length['SM_V7_1']\n",
    "    cumul_start['SM_V7_3']= cumul_start['SM_V7_2'] + chr_length['SM_V7_2']\n",
    "    cumul_start['SM_V7_4']= cumul_start['SM_V7_3'] + chr_length['SM_V7_3']\n",
    "    cumul_start['SM_V7_5']= cumul_start['SM_V7_4'] + chr_length['SM_V7_4']\n",
    "    cumul_start['SM_V7_6']= cumul_start['SM_V7_5'] + chr_length['SM_V7_5']\n",
    "    cumul_start['SM_V7_7']= cumul_start['SM_V7_6'] + chr_length['SM_V7_6']\n",
    "    scanned_size = cumul_start['SM_V7_7'] + chr_length['SM_V7_7']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "x\tH\n",
      "302288\t0\n",
      "705190\t127.556\n",
      "705210\t157.333\n",
      "705322\t137.556\n",
      "705337\t134.756\n",
      "705340\t146.356\n",
      "705343\t145.156\n",
      "705389\t145.156\n",
      "705395\t134.756\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "head results/n10/hscan/niger/SM_V7_1_niger.hscan-out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'results/n10/hscan/niger_n10/SM_V7_1_niger_n10.hscan-out'"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"results/n10/hscan/{}_n10/{}_{}_n10.hscan-out\".format(pop, chrom, pop)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "niger\n"
     ]
    },
    {
     "ename": "NameError",
     "evalue": "name 'h_max' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-22-04444ba1ad73>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     40\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     41\u001b[0m     \u001b[0;31m#calculate qs (with simulated hmax\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 42\u001b[0;31m     \u001b[0mz_s\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mscipy\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstats\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mzscore\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0minsert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mh_s\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mh_max\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     43\u001b[0m     \u001b[0mp_s\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mscipy\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mspecial\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mndtr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0mz_s\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     44\u001b[0m     \u001b[0mreject_s\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mq_s\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msidak\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbonf\u001b[0m \u001b[0;34m=\u001b[0m\u001b[0msm\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstats\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmultitest\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmultipletests\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mp_s\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0malpha\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0.01\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmethod\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"fdr_bh\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'h_max' is not defined"
     ]
    }
   ],
   "source": [
    "for pop in [\"niger\",     \"senegal\",     \"brazil\",     \"tanzania\"]:\n",
    "    print(pop)\n",
    "    chrom_s=[]\n",
    "    pos_s=[]\n",
    "    h_s=[]\n",
    "\n",
    "    for i in range(1,8):\n",
    "        chrom = \"SM_V7_{}\".format(i)\n",
    "        \n",
    "        with open(\"results/n10/hscan/{}/{}_{}.hscan-out\".format(pop, chrom, pop), 'r') as hscan_file:\n",
    "            next(hscan_file)\n",
    "            for calc in hscan_file:\n",
    "                \n",
    "                pos, h = calc.split(\"\\t\")\n",
    "                \n",
    "                chrom_s.append(chrom)\n",
    "                h_s.append(h)\n",
    "                pos_s.append(pos)\n",
    "\n",
    "    chrom_s=np.array(chrom_s)\n",
    "    pos_s=np.array(pos_s)\n",
    "    h_s=np.array(h_s).astype(np.float)\n",
    "    \n",
    "    ##calculate qs    \n",
    "    #z_s=scipy.stats.zscore(h_s)\n",
    "    #p_s=scipy.special.ndtr(-z_s)     \n",
    "    #reject_s, q_s, sidak, bonf =sm.stats.multitest.multipletests(p_s, alpha=0.01, method=\"fdr_bh\")\n",
    "    #logq_s=-np.log10(q_s)\n",
    "\n",
    "    #calculate max neutral h score\n",
    "    h_max = 0.0\n",
    "    sim_files = glob.glob(\"results/hscan/{}-msprime/*hscan-out\".format(pop))     \n",
    "    for sim_file in sim_files:\n",
    "        with open(sim_file, 'r') as in_sim_file:\n",
    "            next(in_sim_file)\n",
    "            for sim_entry in in_sim_file:\n",
    "                 x, h = sim_entry.rstrip().split(\"\\t\")\n",
    "                 if (h != \"H\") and (float(h)>float(h_max)):\n",
    "                    h_max=float(h)\n",
    "\n",
    "    #calculate qs (with simulated hmax\n",
    "    z_s=scipy.stats.zscore(np.insert(h_s, 0, h_max))\n",
    "    p_s=scipy.special.ndtr(-z_s)   \n",
    "    reject_s, q_s, sidak, bonf =sm.stats.multitest.multipletests(p_s, alpha=0.01, method=\"fdr_bh\")\n",
    "    logq_s=-np.log10(q_s)\n",
    "\n",
    "    #now add all info to pop data table\n",
    "    columns = [\"chrom\", \"pos\", \"h\", \"z\", \"p\", \"q\", \"-log10(q)\"]\n",
    "    df = pd.DataFrame(data = [chrom_s, pos_s, h_s, z_s[1:], p_s[1:], q_s[1:], logq_s[1:]]).T\n",
    "    df.columns=columns\n",
    "\n",
    "    #get cumul positions\n",
    "    fig_x_pos_s=[]\n",
    "    for index, row in df.iterrows(): \n",
    "        fig_x_pos_s.append(int(row[\"pos\"]) + int(cumul_start[row['chrom']]))\n",
    "\n",
    "    df['fig_x_pos']=fig_x_pos_s\n",
    "\n",
    "    #save data to csv file\n",
    "    csv_file =\"results/n10/{}_n10_hscan.csv\".format(pop, pop)\n",
    "    df = df.sort_values([\"fig_x_pos\"], ascending = True)\n",
    "    df.to_csv(csv_file, index=False, header=True, mode='w')\n",
    "\n",
    "    #-------------------------------------------------------------------------------\n",
    "    # generate figure\n",
    "   \n",
    "    #iterate over the df and get the chrom, cumul_pos, and log10pvalue\n",
    "    fig_chrom_colors =[]\n",
    "\n",
    "    chrom_colors={}\n",
    "    chrom_colors[\"SM_V7_1\"] = \"black\"\n",
    "    chrom_colors[\"SM_V7_2\"] = \"darkgray\"\n",
    "    chrom_colors[\"SM_V7_3\"] = \"black\"\n",
    "    chrom_colors[\"SM_V7_4\"] = \"darkgray\"\n",
    "    chrom_colors[\"SM_V7_5\"] = \"black\"\n",
    "    chrom_colors[\"SM_V7_6\"] = \"darkgray\"\n",
    "    chrom_colors[\"SM_V7_7\"] = \"black\"\n",
    "    sig_color = \"blue\"\n",
    "\n",
    "    for index, row in df.iterrows(): \n",
    "        fig_chrom_colors.append(chrom_colors[row['chrom']])\n",
    " \n",
    "    #update colors of significant peaks\n",
    "    for index, row in df.iterrows(): \n",
    "        if row.q<sidak and row.h>h_max:\n",
    "            fig_chrom_colors[index]=sig_color\n",
    "    \n",
    "    #plot log data for H\n",
    "    ticks = [ (cumul_start['SM_V7_1'] + cumul_start['SM_V7_2'] )/2,\n",
    "              (cumul_start['SM_V7_2'] + cumul_start['SM_V7_3'] )/2,\n",
    "              (cumul_start['SM_V7_3'] + cumul_start['SM_V7_4'] )/2,\n",
    "              (cumul_start['SM_V7_4'] + cumul_start['SM_V7_5'] )/2,\n",
    "              (cumul_start['SM_V7_5'] + cumul_start['SM_V7_6'] )/2,\n",
    "              (cumul_start['SM_V7_6'] + cumul_start['SM_V7_7'] )/2,\n",
    "              (cumul_start['SM_V7_7'] + scanned_size )/2 ]\n",
    "\n",
    "    tick_lbls = [ \"1\", \"2\", \"3\", \"4\", \"5\", \"6\" ,\"7\"]\n",
    "\n",
    "    plt.scatter(df['fig_x_pos'], df['-log10(q)'], marker =\".\", s = 1, c = fig_chrom_colors) \n",
    "    plt.ylabel(\"-log10(q)\")\n",
    "    #plot FDR line\n",
    "    plt.axhline(y=np.log10(sidak)*-1, color=\"black\", linestyle=\":\", linewidth=1)\n",
    "    #plt.text(df['fig_x_pos'][-1:], np.log10(sidak)*-1, 'FDR=0.01', horizontalalignment='right')\n",
    "    #plot max_h line)\n",
    "    plt.axhline(y=logq_s[0], color=\"black\", linestyle=\"--\", linewidth=1)\n",
    "    #plt.text(df['fig_x_pos'][-1:], logq_s[0], 'max(h) neutral', horizontalalignment='right')\n",
    "    plt.xticks(ticks, tick_lbls) \n",
    "    plt.title(\"{} H\".format(pop))\n",
    "    plt.savefig(\"results/hscan/{}_hscan_q.png\".format(pop, pop), dpi=300) \n",
    "    plt.close()\n",
    "\n",
    "\n",
    "    plt.scatter(df['fig_x_pos'], df['h'], marker =\".\", s = 1, c = fig_chrom_colors) \n",
    "    plt.ylabel(\"h\")\n",
    "    #plt.axhline(y=min(h_s[reject_s[1:]]), color=\"black\", linestyle=\":\", linewidth=1)\n",
    "    #plt.axhline(y=h_max, color=\"black\", linestyle=\"--\", linewidth=1)\n",
    "    plt.xticks(ticks, tick_lbls) \n",
    "    plt.title(\"{} H\".format(pop))\n",
    "    plt.savefig(\"results/n10/{}_n10_hscan_h.png\".format(pop, pop), dpi=300)\n",
    "    plt.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## SWEEPFINDER"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-23T01:17:36.033333Z",
     "start_time": "2020-06-23T01:17:07.321087Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "done readsnps datasize=475081 nmax=80 nmin=24 xmax=81 invar=2\n",
      "getfreq rdatasize=1327\n",
      "calling findmax numpar=80\n",
      "RUNNING THE L-BFGS-B CODE\n",
      "\n",
      "\t* * *\n",
      "\n",
      "Machine precision=2.220446e-16\n",
      "N = 80    M = 12\n",
      "ITERATION 0\n",
      "At iterate    0 f=449028.786508\t|proj g| = 1.732743e+01\n",
      "\n",
      "\t* * *\n",
      "\n",
      "Tit   = total number of iterations\n",
      "Tnf   = total number of function evaluations\n",
      "Tnint = total number of segments explored during Cauchy searches\n",
      "Skip  = number of BFGS updates skipped\n",
      "Nact  = number of active bounds at final generalized Cuachy point\n",
      "Projg = norm of the final projected gradient\n",
      "F     = final function value\n",
      "\n",
      "\t* * *\n",
      "N\tTit\tTnf\tTnint\tSkip\tNact\tProjg\tF\n",
      "80\t15\t19\t41\t0\t0\t1.222590e+01\t4.488109e+05\n",
      " F = 448810.933613\n",
      "CONVERGENCE: REL_REDUCTION_OF_F <= FACTR*EPSMCH            \n",
      "Cauchy time 0 seconds.\n",
      "Subspace minimization time 0 seconds.\n",
      "Line search time 13 seconds.\n",
      "Total User time 14 seconds.\n",
      "\n",
      "frequency spectrum written to smv7_ex_autosomes_n10.SpectFile\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--counts2\n",
      "\t--out smv7_ex_autosomes_n10_tmp\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "After filtering, kept 40 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 475081 out of a possible 475081 Sites\n",
      "Run Time = 11.00 seconds\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "if [ ! -d \"results/n10/sweepfinder\" ] \n",
    "then\n",
    "    mkdir -p results/n10/sweepfinder\n",
    "fi\n",
    "\n",
    "cd results/n10/sweepfinder\n",
    "\n",
    "#convert to sweefinder2 format\n",
    "vcftools \\\n",
    "    --vcf ../smv7_ex_autosomes_n10.vcf \\\n",
    "    --counts2 \\\n",
    "    --out smv7_ex_autosomes_n10_tmp\n",
    "\n",
    "tail -n+2 smv7_ex_autosomes_n10_tmp.frq.count \\\n",
    "    | awk -v OFS=\"\\t\" '{print $2,$6,$4,\"1\"}' \\\n",
    "    >smv7_ex_autosomes_n10.in\n",
    "\n",
    "echo -e 'position\\tx\\tn\\tfolded' \\\n",
    "    | cat - smv7_ex_autosomes_n10.in \\\n",
    "    > temp && mv temp smv7_ex_autosomes_n10.in\n",
    "\n",
    "#calculate genome-wide sfs\n",
    "SweepFinder2 -f smv7_ex_autosomes_n10.in smv7_ex_autosomes_n10.SpectFile"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "### run sweepfinder on the actual data\n",
    "\n",
    "code/sweepfinder2-run_with_real_data.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-23T01:32:36.495826Z",
     "start_time": "2020-06-23T01:30:47.729264Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 5666836 (\"sf2r1_brazil\") has been submitted\n",
      "Your job 5666837 (\"sf2r2_brazil\") has been submitted\n",
      "Your job 5666838 (\"sf2r3_brazil\") has been submitted\n",
      "Your job 5666839 (\"sf2r4_brazil\") has been submitted\n",
      "Your job 5666840 (\"sf2r5_brazil\") has been submitted\n",
      "Your job 5666841 (\"sf2r6_brazil\") has been submitted\n",
      "Your job 5666842 (\"sf2r7_brazil\") has been submitted\n",
      "Your job 5666843 (\"sf2r1_senegal\") has been submitted\n",
      "Your job 5666844 (\"sf2r2_senegal\") has been submitted\n",
      "Your job 5666845 (\"sf2r3_senegal\") has been submitted\n",
      "Your job 5666846 (\"sf2r4_senegal\") has been submitted\n",
      "Your job 5666847 (\"sf2r5_senegal\") has been submitted\n",
      "Your job 5666848 (\"sf2r6_senegal\") has been submitted\n",
      "Your job 5666849 (\"sf2r7_senegal\") has been submitted\n",
      "Your job 5666850 (\"sf2r1_niger\") has been submitted\n",
      "Your job 5666851 (\"sf2r2_niger\") has been submitted\n",
      "Your job 5666852 (\"sf2r3_niger\") has been submitted\n",
      "Your job 5666853 (\"sf2r4_niger\") has been submitted\n",
      "Your job 5666854 (\"sf2r5_niger\") has been submitted\n",
      "Your job 5666855 (\"sf2r6_niger\") has been submitted\n",
      "Your job 5666856 (\"sf2r7_niger\") has been submitted\n",
      "Your job 5666857 (\"sf2r1_tanzania\") has been submitted\n",
      "Your job 5666858 (\"sf2r2_tanzania\") has been submitted\n",
      "Your job 5666859 (\"sf2r3_tanzania\") has been submitted\n",
      "Your job 5666860 (\"sf2r4_tanzania\") has been submitted\n",
      "Your job 5666861 (\"sf2r5_tanzania\") has been submitted\n",
      "Your job 5666862 (\"sf2r6_tanzania\") has been submitted\n",
      "Your job 5666863 (\"sf2r7_tanzania\") has been submitted\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory `brazil': File exists\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_1\n",
      "\t--keep ../brazil_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 162219 out of a possible 475081 Sites\n",
      "Run Time = 6.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_2\n",
      "\t--keep ../brazil_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 78902 out of a possible 475081 Sites\n",
      "Run Time = 4.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_3\n",
      "\t--keep ../brazil_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 80481 out of a possible 475081 Sites\n",
      "Run Time = 4.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_4\n",
      "\t--keep ../brazil_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 75581 out of a possible 475081 Sites\n",
      "Run Time = 4.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_5\n",
      "\t--keep ../brazil_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 24681 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_6\n",
      "\t--keep ../brazil_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 31877 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_7\n",
      "\t--keep ../brazil_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 21340 out of a possible 475081 Sites\n",
      "Run Time = 2.00 seconds\n",
      "mkdir: cannot create directory `senegal': File exists\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_1\n",
      "\t--keep ../senegal_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 162219 out of a possible 475081 Sites\n",
      "Run Time = 6.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_2\n",
      "\t--keep ../senegal_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 78902 out of a possible 475081 Sites\n",
      "Run Time = 4.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_3\n",
      "\t--keep ../senegal_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 80481 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_4\n",
      "\t--keep ../senegal_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 75581 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_5\n",
      "\t--keep ../senegal_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 24681 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_6\n",
      "\t--keep ../senegal_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 31877 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_7\n",
      "\t--keep ../senegal_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 21340 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "mkdir: cannot create directory `niger': File exists\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_1\n",
      "\t--keep ../niger_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 162219 out of a possible 475081 Sites\n",
      "Run Time = 5.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_2\n",
      "\t--keep ../niger_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 78902 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_3\n",
      "\t--keep ../niger_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 80481 out of a possible 475081 Sites\n",
      "Run Time = 4.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_4\n",
      "\t--keep ../niger_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 75581 out of a possible 475081 Sites\n",
      "Run Time = 4.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_5\n",
      "\t--keep ../niger_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 24681 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_6\n",
      "\t--keep ../niger_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 31877 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_7\n",
      "\t--keep ../niger_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 21340 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "mkdir: cannot create directory `tanzania': File exists\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_1\n",
      "\t--keep ../tanzania_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 162219 out of a possible 475081 Sites\n",
      "Run Time = 5.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_2\n",
      "\t--keep ../tanzania_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 78902 out of a possible 475081 Sites\n",
      "Run Time = 4.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_3\n",
      "\t--keep ../tanzania_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 80481 out of a possible 475081 Sites\n",
      "Run Time = 4.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_4\n",
      "\t--keep ../tanzania_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 75581 out of a possible 475081 Sites\n",
      "Run Time = 4.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_5\n",
      "\t--keep ../tanzania_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 24681 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_6\n",
      "\t--keep ../tanzania_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 31877 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf ../smv7_ex_autosomes_n10.vcf\n",
      "\t--chr SM_V7_7\n",
      "\t--keep ../tanzania_n10.list\n",
      "\t--counts2\n",
      "\t--stdout\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PL,Number=G,Type=Integer,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=RGQ,Number=1,Type=Integer,Description=\"Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=culprit,Number=1,Type=String,Description=\"The annotation which was the worst performing in the Gaussian mixture model, likely the reason why the variant was filtered out\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 10 out of 40 Individuals\n",
      "Outputting Frequency Statistics...\n",
      "After filtering, kept 21340 out of a possible 475081 Sites\n",
      "Run Time = 3.00 seconds\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd results/n10/sweepfinder\n",
    "\n",
    "\n",
    "for POP in brazil senegal niger tanzania; do\n",
    "    mkdir $POP\n",
    "    \n",
    "    #get the pop specific sfs\n",
    "    vcftools \\\n",
    "        --keep ../\"$POP\"_n10.list \\\n",
    "        --vcf ../smv7_ex_autosomes_n10.vcf  \\\n",
    "        --counts2 \\\n",
    "        --stdout \\\n",
    "        >$POP/$POP.freq\n",
    "\n",
    "    tail -n+2 $POP/$POP.freq  | awk -v OFS=\"\\t\" '{print $2,$6,$4,\"1\"}' >$POP/$POP.in\n",
    "\n",
    "    echo -e 'position\\tx\\tn\\tfolded' | cat - $POP/$POP.in > temp && mv temp $POP/$POP.in\n",
    "\n",
    "    SweepFinder2 -f $POP/$POP.in $POP/$POP.sfs\n",
    " \n",
    "    #run sweepfinder for each chrom\n",
    "    for I in $(seq 1 7); do\n",
    "        CHR=SM_V7_$I\n",
    "\n",
    "        #convert vcf to sw input format\n",
    "        vcftools \\\n",
    "            --keep ../\"$POP\"_n10.list \\\n",
    "            --vcf ../smv7_ex_autosomes_n10.vcf \\\n",
    "            --chr $CHR \\\n",
    "            --counts2 \\\n",
    "            --stdout \\\n",
    "            >$POP/$CHR\"_\"$POP.freq\n",
    "\n",
    "        tail -n+2 $POP/$CHR\"_\"$POP.freq  | awk -v OFS=\"\\t\" '{print $2,$6,$4,\"1\"}' >$POP/$CHR\"_\"$POP.in\n",
    "\n",
    "        echo -e 'position\\tx\\tn\\tfolded' | cat - $POP/$CHR\"_\"$POP.in > temp && mv temp $POP/$CHR\"_\"$POP.in\n",
    "\n",
    "        #submit sweepfinder        \n",
    "        CMD=\"conda activate sch_man_nwinvasion-sweepfinder; SweepFinder2 -lg 1000 $POP/$CHR\"_\"$POP.in $POP/$POP.sfs $POP/$CHR\"_\"$POP.sw2out\"\n",
    "\n",
    "        echo $CMD | qsub -V -cwd -S /bin/bash -q all.q -j y -N sf2r\"$I\"_\"$POP\" -o $POP/real_$CHR.log -pe smp 2\n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "### run sweepfinder on simulated data\n",
    "code/sweepfinder2-run_with_simulated_data.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "cd results/n10/sweepfinder\n",
    "\n",
    "CONDA=\"conda activate sch_man_nwinvasion-sweepfinder\"\n",
    "QSUB=\"qsub -V -cwd -S /bin/bash -q all.q -j y -pe smp 2 \"    \n",
    "\n",
    "mkdir logs\n",
    "\n",
    "for POP in senegal tanzania niger brazil; do\n",
    "    mkdir $POP-sim\n",
    "\n",
    "    for PROBED_VCF in $(ls ~/sch_man_nwinvasion/results/n10/msprime/$POP/chr1_\"$POP\"_rep_*seed_*_probed.vcf); do\n",
    "        SAMPLE_NAME=$(basename $PROBED_VCF .vcf)\n",
    "        \n",
    "        #convert to sweep format\n",
    "        vcftools \\\n",
    "            --vcf $PROBED_VCF  \\\n",
    "            --counts2 \\\n",
    "            --stdout \\\n",
    "            >$POP-sim/$SAMPLE_NAME.freq\n",
    "\n",
    "        tail -n+2 $POP-sim/$SAMPLE_NAME.freq  | awk -v OFS=\"\\t\" '{print $2,$6,$4,\"1\"}' >$POP-sim/$SAMPLE_NAME.in\n",
    "\n",
    "        echo -e 'position\\tx\\tn\\tfolded' | cat - $POP-sim/$SAMPLE_NAME.in > temp && mv temp $POP-sim/$SAMPLE_NAME.in\n",
    "\n",
    "        #generate sfs\n",
    "        SFS_CMD=\"SweepFinder2 -f $POP-sim/$SAMPLE_NAME.in $POP-sim/$SAMPLE_NAME.sfs >logs/\"$SAMPLE_NAME\"_sfs.log\"\n",
    "\n",
    "        #gen sweepfinder2\n",
    "        SF2_CMD=\"SweepFinder2 -lg 1000 $POP-sim/$SAMPLE_NAME.in $POP-sim/$SAMPLE_NAME.sfs $POP-sim/$SAMPLE_NAME.sw2out\"\n",
    "        \n",
    "        #submit to the queue\n",
    "        CMD=\"$CONDA; $SFS_CMD; $SF2_CMD\"\n",
    "        echo $CMD | $QSUB -N sf2s_$SAMPLE_NAME -o logs/sf2_sim_$SAMPLE_NAME.log\n",
    "\n",
    "    done\n",
    "done\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## FST"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-06-23T01:44:24.882819Z",
     "start_time": "2020-06-23T01:44:03.197190Z"
    }
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import shutil\n",
    "import allel\n",
    "import math\n",
    "import yaml\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "from collections import defaultdict\n",
    "from scipy import stats\n",
    "from scipy import signal\n",
    "\n",
    "os.chdir(\"/master/nplatt/sch_man_nwinvasion\")\n",
    "os.mkdir(\"results/n10/fst\")\n",
    "\n",
    "with open('data/pop_assign.yml') as yaml_file:\n",
    "    pop_assign = yaml.load(yaml_file, Loader=yaml.FullLoader)\n",
    "\n",
    "#-----------------------------------\n",
    "# get lengths from cumul positions\n",
    "#make sure that all stops are not gt chrom length\n",
    "chr_length = {}\n",
    "#genome_size = 0\n",
    "with open('/master/nplatt/sch_man_nwinvasion/data/genomes/Smansoni_v7.fa.fai', 'r') as fai:\n",
    "    for entry in fai:\n",
    "        chrom, length, *offset = entry.rstrip().split(\"\\t\")\n",
    "        chr_length[chrom]=int(length)\n",
    "\n",
    "    cumul_start={}\n",
    "    cumul_start['SM_V7_1']=0\n",
    "    cumul_start['SM_V7_2']= cumul_start['SM_V7_1'] + chr_length['SM_V7_1']\n",
    "    cumul_start['SM_V7_3']= cumul_start['SM_V7_2'] + chr_length['SM_V7_2']\n",
    "    cumul_start['SM_V7_4']= cumul_start['SM_V7_3'] + chr_length['SM_V7_3']\n",
    "    cumul_start['SM_V7_5']= cumul_start['SM_V7_4'] + chr_length['SM_V7_4']\n",
    "    cumul_start['SM_V7_6']= cumul_start['SM_V7_5'] + chr_length['SM_V7_5']\n",
    "    cumul_start['SM_V7_7']= cumul_start['SM_V7_6'] + chr_length['SM_V7_6']\n",
    "    scanned_size = cumul_start['SM_V7_7'] + chr_length['SM_V7_7']\n",
    "\n",
    "#-------------------------------------------------------------------------------\n",
    "# get genotype info per population\n",
    "\n",
    "#read in vcf\n",
    "callset=allel.read_vcf('results/n10/smv7_ex_autosomes_n10.vcf')\n",
    "\n",
    "#now get an index for each sample/population\n",
    "samples = callset[\"samples\"]\n",
    "\n",
    "i=0 \n",
    "pop_idxs = defaultdict(list)   \n",
    "for sample in samples:  \n",
    "     pop_idxs[pop_assign[sample]].append(i) \n",
    "     i=i+1 \n",
    "\n",
    "pops= list(pop_idxs.keys()) \n",
    "\n",
    "#get genotypes\n",
    "gt=allel.GenotypeArray(callset['calldata/GT'])\n",
    "\n",
    "#now get allele count per population\n",
    "ac=gt.count_alleles()\n",
    "\n",
    "#for simplicity add maf info to callset data\n",
    "maf=ac[:, :2].min(axis=1)/ac[:, :2].sum(axis=1)\n",
    "callset['maf']=maf \n",
    "\n",
    "pop_ac={}\n",
    "for pop in pops:\n",
    "    pop_ac[pop] = gt.count_alleles(subpop=pop_idxs[pop])\n",
    "    \n",
    "#-------------------------------------------------------------------------------\n",
    "#generate windows\n",
    "window=100_000\n",
    "\n",
    "#define an array of window start and stops\n",
    "window_starts = [int(x - (window/2)) for x in callset['variants/POS']]\n",
    "window_stops  = [int(x + (window/2)) for x in callset['variants/POS']]\n",
    "\n",
    "#make sure that window starts are all gt 1\n",
    "window_starts = [1 if i < 1 else i for i in window_starts]\n",
    "\n",
    "\n",
    "#make sure that all stops are not gt chrom length\n",
    "chr_length = {}\n",
    "#genome_size = 0\n",
    "with open('/master/nplatt/sch_man_nwinvasion/data/genomes/Smansoni_v7.fa.fai', 'r') as fai:\n",
    "    for entry in fai:\n",
    "        chrom, length, *offset = entry.rstrip().split(\"\\t\")\n",
    "        chr_length[chrom]=int(length)\n",
    "        #genome_size = genome_size + chr_length[chrom]\n",
    "    \n",
    "i=0\n",
    "for stop in window_stops:\n",
    "    chrom = callset['variants/CHROM'][i]\n",
    "    \n",
    "    if stop > chr_length[chrom]:\n",
    "        window_stops[i]=chr_length[chrom]\n",
    "    i=i+1\n",
    "    \n",
    "windows = np.column_stack((np.array(window_starts), \n",
    "                           np.array(window_stops)))\n",
    "\n",
    "callset['windows']=windows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2020-06-23T01:44:32.461Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/master/nplatt/miniconda3/envs/sch_man_nwinvasion-dcms/lib/python3.8/site-packages/allel/stats/fst.py:187: RuntimeWarning: invalid value encountered in true_divide\n",
      "  p = ac / an[:, np.newaxis, :]\n",
      "/master/nplatt/miniconda3/envs/sch_man_nwinvasion-dcms/lib/python3.8/site-packages/allel/stats/fst.py:228: RuntimeWarning: divide by zero encountered in true_divide\n",
      "  a = ((n_bar / n_C) *\n",
      "/master/nplatt/miniconda3/envs/sch_man_nwinvasion-dcms/lib/python3.8/site-packages/allel/stats/fst.py:412: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  return np.nansum(wa) / (np.nansum(wa) + np.nansum(wb) + np.nansum(wc))\n"
     ]
    }
   ],
   "source": [
    "#-------------------------------------------------------------------------------\n",
    "# fst calculations\n",
    "\n",
    "pops = [\"brazil\", \"tanzania\", \"niger\", \"senegal\" ]\n",
    "\n",
    "idx_comps = {\"brazil\":   [pop_idxs[\"brazil\"],   pop_idxs[\"tanzania\"] + pop_idxs[\"niger\"]  + pop_idxs[\"senegal\"] ],\n",
    "             \"tanzania\": [pop_idxs[\"tanzania\"], pop_idxs[\"brazil\"]   + pop_idxs[\"niger\"]  + pop_idxs[\"senegal\"] ],\n",
    "             \"niger\":    [pop_idxs[\"niger\"],    pop_idxs[\"tanzania\"] + pop_idxs[\"brazil\"] + pop_idxs[\"senegal\"] ],\n",
    "             \"senegal\":  [pop_idxs[\"senegal\"],  pop_idxs[\"tanzania\"] + pop_idxs[\"niger\"]  + pop_idxs[\"brazil\"] ]}\n",
    "\n",
    "#make comparisons between population\n",
    "for pop in idx_comps.keys():\n",
    "    pop1_idx = idx_comps[pop][0]\n",
    "    pop2_idx = idx_comps[pop][1]\n",
    "\n",
    "    fst_s             = []\n",
    "    fst_calc_window_s = []\n",
    "    fst_count_s       = []\n",
    "\n",
    "    #create empty dataframe to store data    \n",
    "    headers = [\"chrom\", \"pos\", \"fst\", \"smoothed_fst\", \"window\", \"num_snps\", \"zscore\", \"pvalue\", \"-log10(p)\"]\n",
    "    df=pd.DataFrame(columns=headers) \n",
    "\n",
    "    #now loop through each chromosome\n",
    "    for chrom in list(set(callset['variants/CHROM'])) :\n",
    "        target_sites = np.logical_and( callset['maf'] < 0.05, \n",
    "                                       callset['variants/CHROM'] == chrom )  \n",
    "\n",
    "        chr_gts  = gt[target_sites]\n",
    "        chr_poss = callset['variants/POS'][target_sites]\n",
    "        chr_wins = callset['windows'][target_sites]\n",
    "\n",
    "        \n",
    "        fsts, fst_calc_windows, fst_counts =allel.windowed_weir_cockerham_fst(chr_poss, chr_gts, subpops=[pop1_idx, pop2_idx], windows=chr_wins )\n",
    "\n",
    "        #get rid of nan values\n",
    "        useful_values = np.logical_and( np.isfinite(fsts), fst_counts>=10) \n",
    "\n",
    "        fsts = fsts[useful_values]\n",
    "        fst_calc_windows = fst_calc_windows[useful_values]\n",
    "        fst_counts = fst_counts[useful_values]\n",
    "        chr_poss = chr_poss[useful_values]\n",
    "\n",
    "        #set negative fst values to 0\n",
    "        i=0\n",
    "        for fst in fsts:\n",
    "            if fst <0:\n",
    "                fsts[i]=0\n",
    "            i=i+1        \n",
    "        \n",
    "        #smooth\n",
    "        smoothed_fsts=signal.medfilt(fsts, kernel_size = 101)\n",
    "\n",
    "        #calculate pvalue\n",
    "        zs = stats.zscore(fsts)\n",
    "        ps = stats.norm.cdf(zs)\n",
    "        logps = [-1*np.log10(p) for p in ps] \n",
    "\n",
    "        #add data to dataframe/table\n",
    "        data = list(zip([chrom]*len(fsts), chr_poss, fsts, smoothed_fsts, fst_calc_windows, fst_counts, zs, ps, logps))\n",
    "        chr_df=pd.DataFrame(data, columns=headers)\n",
    "        df = df.append(chr_df)\n",
    "\n",
    "    #add cumul positions\n",
    "    fig_x_pos_s=[]\n",
    "    for index, row in df.iterrows(): \n",
    "        fig_x_pos_s.append(int(row[\"pos\"]) + int(cumul_start[row['chrom']]))\n",
    "\n",
    "    df['fig_x_pos']=fig_x_pos_s\n",
    "\n",
    "    #save data to csv file\n",
    "    csv_file = \"./results/n10/fst/{}_vs_all_windowed_fst.csv\".format(pop)\n",
    "    df = df.sort_values([\"fig_x_pos\"], ascending = True)\n",
    "    df.to_csv(csv_file, index=False, header=True, mode='w')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "#-------------------------------------------------------------------------------\n",
    "# generate figure\n",
    "   \n",
    "for pop in pops:\n",
    "    csv_file = \"results/n10/{}n10_vs_all_windowed_fst.csv\".format(pop)\n",
    "    df = pd.read_csv(csv_file,sep=\",\")\n",
    "\n",
    "    #remove sites lt 10 snps?\n",
    "\n",
    "    #get cumul positions\n",
    "    cumul_start={}\n",
    "    cumul_start['SM_V7_1']=0\n",
    "    cumul_start['SM_V7_2']= cumul_start['SM_V7_1'] + chr_length['SM_V7_1']\n",
    "    cumul_start['SM_V7_3']= cumul_start['SM_V7_2'] + chr_length['SM_V7_2']\n",
    "    cumul_start['SM_V7_4']= cumul_start['SM_V7_3'] + chr_length['SM_V7_3']\n",
    "    cumul_start['SM_V7_5']= cumul_start['SM_V7_4'] + chr_length['SM_V7_4']\n",
    "    cumul_start['SM_V7_6']= cumul_start['SM_V7_5'] + chr_length['SM_V7_5']\n",
    "    cumul_start['SM_V7_7']= cumul_start['SM_V7_6'] + chr_length['SM_V7_6']\n",
    "    scanned_size = cumul_start['SM_V7_7'] + chr_length['SM_V7_7']\n",
    "\n",
    "    #iterate over the df and get the chrom, cumul_pos, and log10pvalue\n",
    "    fig_x_pos=[]\n",
    "    fig_chrom_colors =[]\n",
    "\n",
    "    chrom_colors={}\n",
    "    chrom_colors[\"SM_V7_1\"] = \"black\"\n",
    "    chrom_colors[\"SM_V7_2\"] = \"darkgray\"\n",
    "    chrom_colors[\"SM_V7_3\"] = \"black\"\n",
    "    chrom_colors[\"SM_V7_4\"] = \"darkgray\"\n",
    "    chrom_colors[\"SM_V7_5\"] = \"black\"\n",
    "    chrom_colors[\"SM_V7_6\"] = \"darkgray\"\n",
    "    chrom_colors[\"SM_V7_7\"] = \"black\"\n",
    "    sig_color = \"blue\"\n",
    "\n",
    "    for index, row in df.iterrows(): \n",
    "        fig_x_pos.append(row[\"pos\"] + cumul_start[row['chrom']])\n",
    "        fig_chrom_colors.append(chrom_colors[row['chrom']])\n",
    " \n",
    "    \n",
    "    ticks = [ (cumul_start['SM_V7_1'] + cumul_start['SM_V7_2'] )/2,\n",
    "              (cumul_start['SM_V7_2'] + cumul_start['SM_V7_3'] )/2,\n",
    "              (cumul_start['SM_V7_3'] + cumul_start['SM_V7_4'] )/2,\n",
    "              (cumul_start['SM_V7_4'] + cumul_start['SM_V7_5'] )/2,\n",
    "              (cumul_start['SM_V7_5'] + cumul_start['SM_V7_6'] )/2,\n",
    "              (cumul_start['SM_V7_6'] + cumul_start['SM_V7_7'] )/2,\n",
    "              (cumul_start['SM_V7_7'] + scanned_size )/2 ]\n",
    "\n",
    "    tick_lbls = [ \"1\", \"2\", \"3\", \"4\", \"5\", \"6\" ,\"7\"]\n",
    "\n",
    "    plt.scatter(fig_x_pos, df['smoothed_fst'], marker =\".\", s = 1, c = fig_chrom_colors) \n",
    "    plt.ylabel(\"smoothed_fst\") \n",
    "    plt.xticks(ticks, tick_lbls) \n",
    "    plt.title(\"Fst ({})\".format(pop))\n",
    "    plt.savefig('results/n10/{}n10_vs_all_windowed_fst.png'.format(pop), dpi=300) \n",
    "    plt.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prep DCMS"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## DCMS"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
